{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import pandas as pd\n",
    "from valentine.algorithms import Coma\n",
    "from valentine.algorithms import JaccardDistanceMatcher\n",
    "from valentine.algorithms import SimilarityFlooding\n",
    "from valentine.algorithms import DistributionBased\n",
    "from valentine.algorithms import Cupid\n",
    "from valentine.metrics import F1Score, PrecisionTopNPercent\n",
    "from valentine import valentine_match\n",
    "import pprint\n",
    "import glob\n",
    "from column_representation import ColumnRepresentation, gdc_target_columns, gdc_df\n",
    "\n",
    "from polyfuzz import PolyFuzz\n",
    "from polyfuzz.models import EditDistance\n",
    "from jellyfish import jaro_winkler_similarity\n",
    "from polyfuzz.models import Embeddings\n",
    "from flair.embeddings import TransformerWordEmbeddings, WordEmbeddings\n",
    "\n",
    "from gdc.gdc_api import GDCSchema\n",
    "import utils.file_utils as ut\n",
    "import sys\n",
    "import os\n",
    "sys.path.append(os.path.join(os.path.abspath(''), '..'))\n",
    "\n",
    "\n",
    "sns.set_theme(style=\"white\")\n",
    "\n",
    "\n",
    "pp = pprint.PrettyPrinter(indent=4, sort_dicts=True)\n",
    "\n",
    "ground_truth_path = os.path.join(\n",
    "    '..', 'data', 'table-matching-ground-truth', 'ground-truth')\n",
    "csv_data_path = os.path.join('..', 'data', 'extracted-tables')\n",
    "\n",
    "cat_null_values = set(['n/a', 'na', 'nan', 'null', 'lost to follow-up', 'not available', 'unable to obtain',\n",
    "                       'unknown', 'not reported', 'not allowed to collect',\n",
    "                       'unspecified', 'not specified', 'other, specify','none'])\n",
    "\n",
    "DOMAIN_SIMILARITY_THRESHOLD = 0.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1068, 16)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_submitter_id</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>gender</th>\n",
       "      <th>vital_status</th>\n",
       "      <th>ajcc_pathologic_t</th>\n",
       "      <th>ajcc_pathologic_n</th>\n",
       "      <th>ajcc_pathologic_stage</th>\n",
       "      <th>tumor_grade</th>\n",
       "      <th>tumor_focality</th>\n",
       "      <th>tumor_largest_dimension_diameter</th>\n",
       "      <th>primary_diagnosis</th>\n",
       "      <th>morphology</th>\n",
       "      <th>tissue_or_organ_of_origin</th>\n",
       "      <th>tumor_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>01BR001</td>\n",
       "      <td>20089.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Alive</td>\n",
       "      <td>T2</td>\n",
       "      <td>N1c</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Invasive carcinoma of no special type</td>\n",
       "      <td>8500/3</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01BR008</td>\n",
       "      <td>17532.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>01BR009</td>\n",
       "      <td>23376.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>01BR010</td>\n",
       "      <td>23741.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01BR015</td>\n",
       "      <td>12784.0</td>\n",
       "      <td>white</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Alive</td>\n",
       "      <td>T2</td>\n",
       "      <td>N1</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Invasive carcinoma of no special type</td>\n",
       "      <td>8500/3</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  case_submitter_id  age_at_diagnosis                       race  \\\n",
       "0           01BR001           20089.0  black or african american   \n",
       "1           01BR008           17532.0  black or african american   \n",
       "2           01BR009           23376.0  black or african american   \n",
       "3           01BR010           23741.0  black or african american   \n",
       "4           01BR015           12784.0                      white   \n",
       "\n",
       "                ethnicity  gender  vital_status ajcc_pathologic_t  \\\n",
       "0  not hispanic or latino  female         Alive                T2   \n",
       "1  not hispanic or latino  female  Not Reported      Not Reported   \n",
       "2  not hispanic or latino  female  Not Reported      Not Reported   \n",
       "3  not hispanic or latino  female  Not Reported      Not Reported   \n",
       "4  not hispanic or latino  female         Alive                T2   \n",
       "\n",
       "  ajcc_pathologic_n ajcc_pathologic_stage tumor_grade tumor_focality  \\\n",
       "0               N1c              Stage II          GX   Not Reported   \n",
       "1      Not Reported          Not Reported          GX   Not Reported   \n",
       "2      Not Reported          Not Reported          GX   Not Reported   \n",
       "3      Not Reported          Not Reported          GX   Not Reported   \n",
       "4                N1              Stage II          GX   Not Reported   \n",
       "\n",
       "  tumor_largest_dimension_diameter                      primary_diagnosis  \\\n",
       "0                     Not Reported  Invasive carcinoma of no special type   \n",
       "1                     Not Reported                           Not Reported   \n",
       "2                     Not Reported                           Not Reported   \n",
       "3                     Not Reported                           Not Reported   \n",
       "4                     Not Reported  Invasive carcinoma of no special type   \n",
       "\n",
       "     morphology tissue_or_organ_of_origin tumor_code  \n",
       "0        8500/3               Breast, NOS       BRCA  \n",
       "1  Not Reported               Breast, NOS       BRCA  \n",
       "2  Not Reported               Breast, NOS       BRCA  \n",
       "3  Not Reported               Breast, NOS       BRCA  \n",
       "4        8500/3               Breast, NOS       BRCA  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_target = pd.read_csv('../data/target.csv')\n",
    "df_target.drop('study', axis=1, inplace=True)\n",
    "print(df_target.shape)\n",
    "df_target.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'age_at_diagnosis',\n",
       " 'ajcc_pathologic_n',\n",
       " 'ajcc_pathologic_stage',\n",
       " 'ajcc_pathologic_t',\n",
       " 'ethnicity',\n",
       " 'gender',\n",
       " 'morphology',\n",
       " 'primary_diagnosis',\n",
       " 'race',\n",
       " 'tissue_or_organ_of_origin',\n",
       " 'tumor_code',\n",
       " 'tumor_focality',\n",
       " 'tumor_grade',\n",
       " 'tumor_largest_dimension_diameter',\n",
       " 'vital_status'}"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "interested_cols = set(df_target.columns)\n",
    "interested_cols.remove('case_submitter_id')\n",
    "interested_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['t2d', 't1b', 't2c', 't1b1', 'tx', 't1', 't4a', 't4e', 't0', 't3a',\n",
       "       't1mi', 't2a', 't4c', 't1c2', 'tis', 't2a1', 't3b', 't2',\n",
       "       'tis (dcis)', 't1c', 't1a2', 't3d', 't4b', 't4', 't1a1',\n",
       "       'tis (lcis)', 't2a2', 't3', \"tis (paget's)\", 't1b2', 't3c', 'ta',\n",
       "       't1a', 't4d', 't2b', <NA>], dtype=object)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdc_df\n",
    "gdc_df.ajcc_pathologic_t.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_metrics(groundtruth, matching_pairs):\n",
    "    \n",
    "    true_positives = len(set(groundtruth) & set(matching_pairs))\n",
    "    false_positives = len(set(matching_pairs) - set(groundtruth))\n",
    "    false_negatives = len(set(groundtruth) - set(matching_pairs))\n",
    "    \n",
    "    precision = true_positives / (true_positives + false_positives)\n",
    "    recall = true_positives / (true_positives + false_negatives) \n",
    "    f1_score = 2 * (precision * recall) / (precision + recall)\n",
    "    \n",
    "    return precision, recall, f1_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_domains_tf(colvalues1, colvalues2):\n",
    "    model = PolyFuzz(\"TF-IDF\").match(colvalues1, colvalues2)\n",
    "    value_matches = model.get_matches()\n",
    "    # model.visualize_precision_recall()\n",
    "    # print(value_matches)\n",
    "    value_matches = value_matches.dropna()\n",
    "    # print(value_matches)\n",
    "    n = len(colvalues1)\n",
    "    # get the first elements according to the domain size of the candidate\n",
    "    value_matches = value_matches.head(n)\n",
    "\n",
    "    return value_matches['Similarity'].median()\n",
    "\n",
    "\n",
    "def check_domains_edit(colvalues1, colvalues2):\n",
    "    jellyfish_matcher = EditDistance(n_jobs=1, scorer=jaro_winkler_similarity)\n",
    "    model = PolyFuzz(jellyfish_matcher).match(colvalues1, colvalues2)\n",
    "    value_matches = model.get_matches()\n",
    "    # print(value_matches)\n",
    "    value_matches = value_matches.dropna()\n",
    "\n",
    "    n = len(colvalues1)\n",
    "    # get the first elements according to the domain size of the candidate\n",
    "    value_matches = value_matches.head(n)\n",
    "\n",
    "    return value_matches['Similarity'].median()\n",
    "\n",
    "\n",
    "def check_domains_emb(colvalues1, colvalues2):\n",
    "\n",
    "    bert = TransformerWordEmbeddings('bert-base-multilingual-cased')\n",
    "    bert_matcher = Embeddings(bert, min_similarity=0)\n",
    "\n",
    "    # fasttext = WordEmbeddings('en-crawl')\n",
    "    # fasttext_matcher = Embeddings(fasttext, min_similarity=0)\n",
    "\n",
    "    # matchers = [bert_matcher, fasttext_matcher]\n",
    "\n",
    "    model = PolyFuzz(bert_matcher).match(colvalues1, colvalues2)\n",
    "\n",
    "    value_matches = model.get_matches()\n",
    "\n",
    "    print(value_matches)\n",
    "\n",
    "    # value_matches = value_matches.dropna()\n",
    "\n",
    "    # n = len(colvalues1)\n",
    "    # # get the first elements according to the domain size of the candidate\n",
    "    # value_matches = value_matches.head(n)\n",
    "\n",
    "    # return value_matches['Similarity'].median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ajcc_pathologic_stage 7\n",
      "gender 9\n",
      "ethnicity 6\n",
      "race 4\n",
      "tumor_focality 4\n",
      "primary_diagnosis 4\n",
      "tumor_grade 3\n",
      "ajcc_pathologic_t 5\n",
      "ajcc_pathologic_n 5\n",
      "vital_status 3\n",
      "['Clark', 'Gillette', 'Huang', 'Dou']\n"
     ]
    }
   ],
   "source": [
    "value_ocurrence_map ={}\n",
    "files = []\n",
    "for file in os.listdir(ground_truth_path):\n",
    "    study = file.split('.csv')[0]\n",
    "\n",
    "    groundtruth_path = os.path.join(ground_truth_path, file)\n",
    "    groundtruth = ut.load_table_matching_groundtruth(groundtruth_path)\n",
    "    groundtruth = [\n",
    "        entry for entry in groundtruth if entry[1] in set(gdc_df.columns)]\n",
    "    groundtruth = [(entry[0].lower(), entry[1].lower()) for entry in groundtruth]\n",
    "    \n",
    "    for entry in groundtruth:\n",
    "        value_ocurrence_map[entry[1]] = value_ocurrence_map.get(entry[1], 0) + 1\n",
    "        if entry[1] == 'primary_diagnosis':\n",
    "            files.append(study)\n",
    "\n",
    "for k,v in value_ocurrence_map.items():\n",
    "    if v > 1:\n",
    "        print(k,v)\n",
    "\n",
    "print(files)\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Clark histologic_type primary_diagnosis\n",
      "Gillette dominant.histological.subtype primary_diagnosis\n",
      "Huang histologic_type primary_diagnosis\n",
      "Dou histologic_type primary_diagnosis\n"
     ]
    }
   ],
   "source": [
    "import random\n",
    "\n",
    "dataframe_dict = {}\n",
    "\n",
    "for study in ['Clark', 'Gillette', 'Huang', 'Dou']:\n",
    "    file = study + '.csv'\n",
    "\n",
    "    groundtruth_path = os.path.join(ground_truth_path, file)\n",
    "    groundtruth = ut.load_table_matching_groundtruth(groundtruth_path)\n",
    "    groundtruth = [\n",
    "        entry for entry in groundtruth if entry[1] in set(gdc_df.columns)]\n",
    "    groundtruth = [(entry[0].lower(), entry[1].lower())\n",
    "                   for entry in groundtruth]\n",
    "\n",
    "    interested_entry = None\n",
    "    for entry in groundtruth:\n",
    "        if entry[1] == 'primary_diagnosis':\n",
    "            interested_entry = entry\n",
    "\n",
    "    # print(interested_entry)\n",
    "\n",
    "    if interested_entry == None:\n",
    "        continue\n",
    "\n",
    "    files = glob.glob(os.path.join(csv_data_path, f\"{study}*.csv\"))\n",
    "\n",
    "    \n",
    "\n",
    "    for i, file in enumerate(files):\n",
    "        # print(f\"\\nProcessing {file}\")\n",
    "        candidate_df = pd.read_csv(file)\n",
    "        candidate_df.columns = candidate_df.columns.str.lower()\n",
    "\n",
    "        if interested_entry[0] in candidate_df.columns:\n",
    "\n",
    "            print(study, interested_entry[0], interested_entry[1])\n",
    "\n",
    "            values = candidate_df[interested_entry[0]].unique()\n",
    "            gdc_values = gdc_df[interested_entry[1]].unique()   \n",
    "\n",
    "            # print(len(gdc_values))\n",
    "\n",
    "            subset_size = 8  # Choose the size of the subset\n",
    "            random_subset = set(random.sample(sorted(gdc_values), subset_size))\n",
    "\n",
    "            dataframe_dict['GDC Format'] = list(random_subset)\n",
    "\n",
    "            # print(values, random_subset)\n",
    "\n",
    "            to_complete = subset_size - len(values)\n",
    "\n",
    "            values = list(values)\n",
    "            for i in range(to_complete):\n",
    "                values.append('N/A')\n",
    "\n",
    "            dataframe_dict[study] = values\n",
    "\n",
    "\n",
    "\n",
    "            # common_values = set(values) & set(gdc_values)\n",
    "            # proportion = len(common_values) / len(values)\n",
    "            # print(f\"{study} {file} {proportion}\")\n",
    "\n",
    "            # print(len(values), len(gdc_values), len(common_values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "All arrays must be of the same length",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[49], line 5\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mpd\u001b[39;00m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;66;03m# Create DataFrame\u001b[39;00m\n\u001b[0;32m----> 5\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mDataFrame\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdataframe_dict\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      7\u001b[0m \u001b[38;5;66;03m# Print DataFrame\u001b[39;00m\n\u001b[1;32m      8\u001b[0m \u001b[38;5;66;03m# print(df.head())\u001b[39;00m\n\u001b[1;32m     10\u001b[0m \u001b[38;5;28mprint\u001b[39m(df\u001b[38;5;241m.\u001b[39mto_string(max_rows\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m))\n",
      "File \u001b[0;32m~/miniconda3/envs/arpah/lib/python3.12/site-packages/pandas/core/frame.py:733\u001b[0m, in \u001b[0;36mDataFrame.__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    727\u001b[0m     mgr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_init_mgr(\n\u001b[1;32m    728\u001b[0m         data, axes\u001b[38;5;241m=\u001b[39m{\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mindex\u001b[39m\u001b[38;5;124m\"\u001b[39m: index, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcolumns\u001b[39m\u001b[38;5;124m\"\u001b[39m: columns}, dtype\u001b[38;5;241m=\u001b[39mdtype, copy\u001b[38;5;241m=\u001b[39mcopy\n\u001b[1;32m    729\u001b[0m     )\n\u001b[1;32m    731\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, \u001b[38;5;28mdict\u001b[39m):\n\u001b[1;32m    732\u001b[0m     \u001b[38;5;66;03m# GH#38939 de facto copy defaults to False only in non-dict cases\u001b[39;00m\n\u001b[0;32m--> 733\u001b[0m     mgr \u001b[38;5;241m=\u001b[39m \u001b[43mdict_to_mgr\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcopy\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtyp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmanager\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    734\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(data, ma\u001b[38;5;241m.\u001b[39mMaskedArray):\n\u001b[1;32m    735\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mma\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m mrecords\n",
      "File \u001b[0;32m~/miniconda3/envs/arpah/lib/python3.12/site-packages/pandas/core/internals/construction.py:503\u001b[0m, in \u001b[0;36mdict_to_mgr\u001b[0;34m(data, index, columns, dtype, typ, copy)\u001b[0m\n\u001b[1;32m    499\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    500\u001b[0m         \u001b[38;5;66;03m# dtype check to exclude e.g. range objects, scalars\u001b[39;00m\n\u001b[1;32m    501\u001b[0m         arrays \u001b[38;5;241m=\u001b[39m [x\u001b[38;5;241m.\u001b[39mcopy() \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(x, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdtype\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01melse\u001b[39;00m x \u001b[38;5;28;01mfor\u001b[39;00m x \u001b[38;5;129;01min\u001b[39;00m arrays]\n\u001b[0;32m--> 503\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43marrays_to_mgr\u001b[49m\u001b[43m(\u001b[49m\u001b[43marrays\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolumns\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mindex\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdtype\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtyp\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtyp\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mconsolidate\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcopy\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/miniconda3/envs/arpah/lib/python3.12/site-packages/pandas/core/internals/construction.py:114\u001b[0m, in \u001b[0;36marrays_to_mgr\u001b[0;34m(arrays, columns, index, dtype, verify_integrity, typ, consolidate)\u001b[0m\n\u001b[1;32m    111\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m verify_integrity:\n\u001b[1;32m    112\u001b[0m     \u001b[38;5;66;03m# figure out the index, if necessary\u001b[39;00m\n\u001b[1;32m    113\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m index \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 114\u001b[0m         index \u001b[38;5;241m=\u001b[39m \u001b[43m_extract_index\u001b[49m\u001b[43m(\u001b[49m\u001b[43marrays\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    115\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    116\u001b[0m         index \u001b[38;5;241m=\u001b[39m ensure_index(index)\n",
      "File \u001b[0;32m~/miniconda3/envs/arpah/lib/python3.12/site-packages/pandas/core/internals/construction.py:677\u001b[0m, in \u001b[0;36m_extract_index\u001b[0;34m(data)\u001b[0m\n\u001b[1;32m    675\u001b[0m lengths \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mlist\u001b[39m(\u001b[38;5;28mset\u001b[39m(raw_lengths))\n\u001b[1;32m    676\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(lengths) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[0;32m--> 677\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAll arrays must be of the same length\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    679\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m have_dicts:\n\u001b[1;32m    680\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[1;32m    681\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mMixing dicts with non-Series may lead to ambiguous ordering.\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    682\u001b[0m     )\n",
      "\u001b[0;31mValueError\u001b[0m: All arrays must be of the same length"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "\n",
    "# Create DataFrame\n",
    "df = pd.DataFrame(dataframe_dict)\n",
    "\n",
    "# Print DataFrame\n",
    "# print(df.head())\n",
    "\n",
    "print(df.to_string(max_rows=None))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n",
      "[('tumor.stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_D)_QC_mutation_calls.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                   From          To  Similarity\n",
      "3  tumor_sample_barcode  tumor_code       0.474\n",
      "\n",
      "\n",
      "Checking similarity values for column: tumor_sample_barcode tumor_code\n",
      "Domain similarity for tumor_sample_barcode and tumor_code is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('hugo_symbol', 'tumor_focality'), ('variant_type', 'morphology'), ('variant_classification', 'tumor_grade'), ('tumor_sample_barcode', 'ajcc_pathologic_t'), ('passed_filters', 'primary_diagnosis'), ('matched_norm_sample_barcode', 'tumor_code')]\n",
      "Domain similarity for hugo_symbol and tumor_focality is 0.1455\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_A)_Metadata.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.738\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('top2a.proteogenomic.status', 'tumor_focality'), ('cd3.tils.status', 'morphology'), ('tmt.channel', 'tumor_grade'), ('akt1.mutation.type', 'ajcc_pathologic_t'), ('sample.id', 'primary_diagnosis'), ('tnbc.updated.clinical.status', 'tumor_code'), ('apobec.signature', 'ajcc_pathologic_n'), ('arid1a.mutation.type', 'ajcc_pathologic_stage'), ('cd3.tils.counts', 'vital_status'), ('cbfb.mutation.type', 'race'), ('pten.mutation.type', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for top2a.proteogenomic.status and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_B)_QC_RNA-seq.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('sample.id', 'tumor_focality'), ('acgt tag id', 'morphology'), ('qc', 'tumor_grade'), ('notes', 'ajcc_pathologic_t')]\n",
      "Domain similarity for sample.id and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_C)_QC_WXS.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('participant', 'tumor_focality'), ('sample_id', 'morphology'), ('type', 'tumor_grade')]\n",
      "Domain similarity for participant and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Krug, we have:\n",
      "Groundtruth: [('tumor.stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "Found pairs [('ethnicity', 'ethnicity'), ('gender', 'gender')]\n",
      "Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "9\n",
      "[('gender', 'gender'), ('race', 'race'), ('ethnicity_self_identify', 'ethnicity'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('grade', 'tumor_grade'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('path_stage_primary_tumor_pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes_pn', 'ajcc_pathologic_n')]\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_data_sample_availability.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 1', 'tumor_focality'), ('unnamed: 2', 'morphology'), ('unnamed: 4', 'tumor_grade'), ('table: sample size of different data sets after excluding samples with low quality.', 'ajcc_pathologic_t'), ('unnamed: 3', 'primary_diagnosis')]\n",
      "Domain similarity for unnamed: 1 and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_RNA-seq_sequencing_depth.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case', 'tumor_focality'), ('tissue type', 'morphology'), ('sample aliquot id', 'tumor_grade')]\n",
      "Domain similarity for case and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_WGS-WES_sequencing_depth.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('bam file', 'tumor_focality'), ('case', 'morphology'), ('sample aliquot', 'tumor_grade')]\n",
      "Domain similarity for bam file and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_ccrcc_clinical_characteristics.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['tumor_focality', 'race', 'gender']\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                        From                     To  Similarity\n",
      "5    ethnicity_self_identify              ethnicity       0.553\n",
      "10                     grade            tumor_grade       0.684\n",
      "11  tumor_stage_pathological  ajcc_pathologic_stage       0.499\n",
      "\n",
      "\n",
      "Checking similarity values for column: ethnicity_self_identify ethnicity\n",
      "Domain similarity for ethnicity_self_identify and ethnicity is 0.0705\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: grade tumor_grade\n",
      "Domain similarity for grade and tumor_grade is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: tumor_stage_pathological ajcc_pathologic_stage\n",
      "Domain similarity for tumor_stage_pathological and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('tumor_site', 'morphology'), ('tumor_site_other', 'tumor_grade'), ('grade', 'ajcc_pathologic_t'), ('ethnicity_self_identify', 'primary_diagnosis'), ('path_stage_primary_tumor_pt', 'tumor_code'), ('path_stage_reg_lymph_nodes_pn', 'ethnicity'), ('case_id', 'ajcc_pathologic_n'), ('tumor/normal', 'vital_status'), ('histologic_type', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for tumor_site and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Clark, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('race', 'race'), ('ethnicity_self_identify', 'ethnicity'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('grade', 'tumor_grade'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('path_stage_primary_tumor_pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes_pn', 'ajcc_pathologic_n')]\n",
      "Found pairs [('tumor_focality', 'tumor_focality'), ('race', 'race'), ('gender', 'gender'), ('tumor_stage_pathological', 'ajcc_pathologic_stage')]\n",
      "Precision: 1.0, Recall: 0.4444444444444444, F1 Score: 0.6153846153846153\n",
      "5\n",
      "[('gender', 'gender'), ('pt', 'ajcc_pathologic_t'), ('pn', 'ajcc_pathologic_n'), ('stage', 'ajcc_pathologic_stage'), ('vital.status', 'vital_status')]\n",
      "\n",
      "Processing ../data/extracted-tables/Vasaikar_B-ClinicalData.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['gender']\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "     From                     To  Similarity\n",
      "17  stage  ajcc_pathologic_stage       0.408\n",
      "\n",
      "\n",
      "Checking similarity values for column: stage ajcc_pathologic_stage\n",
      "Domain similarity for stage and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('pt', 'tumor_focality'), ('cea', 'morphology'), ('mirna-seq', 'tumor_grade'), ('msi_pcr_result', 'ajcc_pathologic_t'), ('polyps_history', 'primary_diagnosis'), ('pole_mutation', 'tumor_code'), ('msh2_mutation', 'ajcc_pathologic_n'), ('mrna-seq', 'ethnicity'), ('pros', 'vital_status'), ('age', 'race'), ('immunescore', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for pt and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Vasaikar_A-Annotation.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('data type', 'tumor_focality'), ('column definition', 'morphology'), ('column name', 'tumor_grade'), ('definition', 'ajcc_pathologic_t')]\n",
      "Domain similarity for data type and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Vasaikar, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('pt', 'ajcc_pathologic_t'), ('pn', 'ajcc_pathologic_n'), ('stage', 'ajcc_pathologic_stage'), ('vital.status', 'vital_status')]\n",
      "Found pairs [('gender', 'gender'), ('stage', 'ajcc_pathologic_stage')]\n",
      "Precision: 1.0, Recall: 0.4, F1 Score: 0.5714285714285715\n",
      "3\n",
      "[('gender', 'gender'), ('ethnicity', 'ethnicity'), ('dominant.histological.subtype', 'primary_diagnosis')]\n",
      "\n",
      "Processing ../data/extracted-tables/Gillette_Annotions_S1A.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.729\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "    From                     To  Similarity\n",
      "6  stage  ajcc_pathologic_stage       0.426\n",
      "\n",
      "\n",
      "Checking similarity values for column: stage ajcc_pathologic_stage\n",
      "Domain similarity for stage and ajcc_pathologic_stage is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('keap1.mutation', 'tumor_focality'), ('methylation.qc', 'morphology'), ('qc.status', 'tumor_grade'), ('smoking.history', 'ajcc_pathologic_t'), ('dominant.signature.wgs.notsmoking.50perc', 'primary_diagnosis'), ('cimp.status', 'tumor_code'), ('kras.mutation', 'ajcc_pathologic_n'), ('aliquot', 'ajcc_pathologic_stage'), ('sample.id', 'vital_status'), ('dominant.histological.subtype', 'race'), ('erbb2.mutation', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for keap1.mutation and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Gillette, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('ethnicity', 'ethnicity'), ('dominant.histological.subtype', 'primary_diagnosis')]\n",
      "Found pairs [('ethnicity', 'ethnicity'), ('gender', 'gender')]\n",
      "Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "4\n",
      "[('gender', 'gender'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('vital_status', 'vital_status')]\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_additional_annotations.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('tp53_alter_status', 'tumor_focality'), ('pdgfra_alter_status', 'morphology'), ('nf1_alter_status', 'tumor_grade'), ('atrx_alter_status', 'ajcc_pathologic_t'), ('telomere', 'primary_diagnosis'), ('mirna', 'tumor_code'), ('ancestry_prediction', 'gender'), ('nmf_consensus', 'ajcc_pathologic_n'), ('multiomic', 'ethnicity'), ('tert_alter_status', 'ajcc_pathologic_stage'), ('braf_alter_status', 'vital_status'), ('dna_methyl_mnp', 'race'), ('case', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for tp53_alter_status and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_survival_analysis.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('stratification group', 'tumor_focality'), ('group value', 'morphology')]\n",
      "Domain similarity for stratification group and tumor_focality is 0.338\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_cbttc_sample_info.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('aliquot_id', 'tumor_focality'), ('kids_first_id', 'morphology'), ('research_id', 'tumor_grade'), ('clinical_status_at_collection_event', 'ajcc_pathologic_t'), ('sample_id', 'primary_diagnosis'), ('original_aliquot_id', 'tumor_code'), ('tmt_channel', 'gender')]\n",
      "Domain similarity for aliquot_id and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_clinical_data.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'vital_status', 'race', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: vital_status\n",
      "Domain similarity for vital_status is 0.202\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                      From         To  Similarity\n",
      "5  ethnicity_self_identify  ethnicity       0.523\n",
      "\n",
      "\n",
      "Checking similarity values for column: ethnicity_self_identify ethnicity\n",
      "Domain similarity for ethnicity_self_identify and ethnicity is 0.058\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('exclusion_reason', 'tumor_focality'), ('cause_of_death', 'morphology'), ('tumor_laterality', 'tumor_grade'), ('lost_to_follow_up', 'ajcc_pathologic_t'), ('expert_path_review_details', 'primary_diagnosis'), ('smoking_history', 'tumor_code'), ('vital_status', 'ajcc_pathologic_stage'), ('tumor_site_curation_notes', 'ajcc_pathologic_n'), ('country_of_origin', 'vital_status'), ('ethnicity_self_identify', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for exclusion_reason and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Wang, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('vital_status', 'vital_status')]\n",
      "Found pairs [('ethnicity', 'ethnicity'), ('race', 'race'), ('gender', 'gender')]\n",
      "Precision: 1.0, Recall: 0.75, F1 Score: 0.8571428571428571\n",
      "3\n",
      "[('stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1A.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.124\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "             From                     To  Similarity\n",
      "6           stage  ajcc_pathologic_stage       0.412\n",
      "11  ethnicity_mod              ethnicity       0.805\n",
      "16      qc.status           vital_status       0.403\n",
      "\n",
      "\n",
      "Checking similarity values for column: stage ajcc_pathologic_stage\n",
      "Domain similarity for stage and ajcc_pathologic_stage is 0.828\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: ethnicity_mod ethnicity\n",
      "Domain similarity for ethnicity_mod and ethnicity is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: qc.status vital_status\n",
      "Domain similarity for qc.status and vital_status is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('tcga pan-can immune clustering', 'tumor_focality'), ('pten.mutation', 'morphology'), ('sample.id', 'tumor_grade'), ('pathology.based.histology.assessment', 'ajcc_pathologic_t'), ('cdkn2a.mutation', 'primary_diagnosis'), ('secondhand.smoke', 'tumor_code'), ('participant', 'ajcc_pathologic_n'), ('tp53.mutation', 'ethnicity'), ('nmf.subtype.core', 'vital_status'), ('type', 'race'), ('immunecluster.rna', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for tcga pan-can immune clustering and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1C.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 5', 'tumor_focality'), ('unnamed: 7', 'morphology'), ('unnamed: 1', 'tumor_grade'), ('unnamed: 11', 'ajcc_pathologic_t'), ('unnamed: 8', 'primary_diagnosis'), ('unnamed: 4', 'tumor_code'), ('unnamed: 3', 'ajcc_pathologic_n'), ('nmf cluster (p<0.05)', 'ethnicity'), ('unnamed: 10', 'vital_status'), ('unnamed: 9', 'race'), ('unnamed: 6', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for unnamed: 5 and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1E.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 5', 'tumor_focality'), ('unnamed: 7', 'morphology'), ('unnamed: 1', 'tumor_grade'), ('unnamed: 8', 'ajcc_pathologic_t'), ('counts', 'primary_diagnosis'), ('unnamed: 0', 'tumor_code'), ('unnamed: 4', 'ajcc_pathologic_n'), ('unnamed: 3', 'ethnicity'), ('p-values (bonferroni corrected)', 'vital_status'), ('unnamed: 9', 'race')]\n",
      "Domain similarity for unnamed: 5 and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1D.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 5', 'tumor_focality'), ('unnamed: 1', 'morphology'), ('unnamed: 11', 'tumor_grade'), ('unnamed: 8', 'ajcc_pathologic_t'), ('counts', 'primary_diagnosis'), ('unnamed: 0', 'tumor_code'), ('unnamed: 4', 'ajcc_pathologic_n'), ('unnamed: 3', 'ethnicity'), ('p-values (bonferroni corrected)', 'vital_status'), ('unnamed: 10', 'race'), ('unnamed: 6', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for unnamed: 5 and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Satpathy, we have:\n",
      "Groundtruth: [('stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "Found pairs [('gender', 'gender'), ('stage', 'ajcc_pathologic_stage')]\n",
      "Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "7\n",
      "[('sex', 'gender'), ('race', 'race'), ('tumor_focality', 'tumor_focality'), ('pathologic_staging_regional_lymph_nodes_pn', 'ajcc_pathologic_n'), ('pathologic_staging_primary_tumor_pt', 'ajcc_pathologic_t'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('vital_status', 'vital_status')]\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_circRNAForNAT.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 0', 'tumor_focality')]\n",
      "Domain similarity for unnamed: 0 and tumor_focality is 0.0765\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_Clinical_data.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['tumor_focality', 'vital_status', 'race']\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: vital_status\n",
      "Domain similarity for vital_status is 0.202\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                                        From                     To  \\\n",
      "3                        histology_diagnosis      primary_diagnosis   \n",
      "14       pathologic_staging_primary_tumor_pt  ajcc_pathologic_stage   \n",
      "15  pathologic_staging_distant_metastasis_pm  ajcc_pathologic_stage   \n",
      "18                  tumor_stage_pathological  ajcc_pathologic_stage   \n",
      "\n",
      "    Similarity  \n",
      "3        0.535  \n",
      "14       0.414  \n",
      "15       0.406  \n",
      "18       0.488  \n",
      "\n",
      "\n",
      "Checking similarity values for column: histology_diagnosis primary_diagnosis\n",
      "Domain similarity for histology_diagnosis and primary_diagnosis is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: pathologic_staging_primary_tumor_pt ajcc_pathologic_stage\n",
      "Domain similarity for pathologic_staging_primary_tumor_pt and ajcc_pathologic_stage is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: pathologic_staging_distant_metastasis_pm ajcc_pathologic_stage\n",
      "Domain similarity for pathologic_staging_distant_metastasis_pm and ajcc_pathologic_stage is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: tumor_stage_pathological ajcc_pathologic_stage\n",
      "Domain similarity for tumor_stage_pathological and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('clinical_staging_distant_metastasis_cm', 'morphology'), ('cause_of_death', 'tumor_grade'), ('tumor_included_for_the_study', 'ajcc_pathologic_t'), ('stromal_fraction', 'tumor_code'), ('tumor_necrosis', 'gender'), ('is_this_patient_lost_to_follow_up', 'ethnicity'), ('pathologic_staging_primary_tumor_pt', 'ajcc_pathologic_n'), ('muscle_fraction', 'vital_status'), ('residual_tumor', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for clinical_staging_distant_metastasis_cm and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_TMT_labeling_scheme.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('channel_name', 'morphology'), ('sample_type', 'tumor_grade'), ('case_id', 'ajcc_pathologic_t')]\n",
      "Domain similarity for channel_name and morphology is 0.383\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_Molecular_phenotype_data.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "         From            To  Similarity\n",
      "4  msi_status  vital_status       0.435\n",
      "\n",
      "\n",
      "Checking similarity values for column: msi_status vital_status\n",
      "Domain similarity for msi_status and vital_status is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('bailey', 'morphology'), ('collisson', 'tumor_grade'), ('moffitt', 'ajcc_pathologic_t'), ('case_id', 'tumor_code'), ('msi_status', 'gender')]\n",
      "Domain similarity for bailey and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_circRNAforTumor.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 0', 'morphology')]\n",
      "Domain similarity for unnamed: 0 and morphology is 0.261\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_Proteins_phospho_glyco_IDs.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_id', 'morphology'), ('type', 'tumor_grade'), ('tmt_channel', 'ajcc_pathologic_t')]\n",
      "Domain similarity for case_id and morphology is 0.321\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Cao, we have:\n",
      "Groundtruth: [('sex', 'gender'), ('race', 'race'), ('tumor_focality', 'tumor_focality'), ('pathologic_staging_regional_lymph_nodes_pn', 'ajcc_pathologic_n'), ('pathologic_staging_primary_tumor_pt', 'ajcc_pathologic_t'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('vital_status', 'vital_status')]\n",
      "Found pairs [('tumor_focality', 'tumor_focality'), ('race', 'race'), ('histology_diagnosis', 'primary_diagnosis'), ('tumor_stage_pathological', 'ajcc_pathologic_stage')]\n",
      "Precision: 0.75, Recall: 0.42857142857142855, F1 Score: 0.5454545454545454\n",
      "7\n",
      "[('gender', 'gender'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('histologic_grade', 'tumor_grade'), ('patho_staging_pt', 'ajcc_pathologic_t'), ('patho_staging_pn', 'ajcc_pathologic_n'), ('tumor_stage', 'ajcc_pathologic_stage')]\n",
      "\n",
      "Processing ../data/extracted-tables/Huang_Meta_table.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['tumor_focality', 'gender']\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                      From          To  Similarity\n",
      "24  tumor_pathology_review  morphology        0.41\n",
      "\n",
      "\n",
      "Checking similarity values for column: tumor_pathology_review morphology\n",
      "Domain similarity for tumor_pathology_review and morphology is 0.251\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('smoke_age_stop', 'morphology'), ('patho_staging_pn', 'tumor_grade'), ('smoking_inferred', 'ajcc_pathologic_t'), ('tumor_site_original', 'primary_diagnosis'), ('patho_staging_pt', 'tumor_code'), ('smoking_history', 'ajcc_pathologic_n'), ('tumor_necrosis', 'ethnicity'), ('integrated_subtype', 'ajcc_pathologic_stage'), ('histologic_type', 'vital_status'), ('tumor_stage', 'race'), ('country', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for smoke_age_stop and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Huang_Data_availabilty.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('1: available; 0: unavailable', 'morphology'), ('case_id', 'tumor_grade')]\n",
      "Domain similarity for 1: available; 0: unavailable and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Huang_TMT-set.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('channel #', 'morphology'), ('case_id', 'tumor_grade'), ('combined_id', 'ajcc_pathologic_t'), ('type', 'primary_diagnosis')]\n",
      "Domain similarity for channel # and morphology is 0.383\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Huang, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('histologic_grade', 'tumor_grade'), ('patho_staging_pt', 'ajcc_pathologic_t'), ('patho_staging_pn', 'ajcc_pathologic_n'), ('tumor_stage', 'ajcc_pathologic_stage')]\n",
      "Found pairs [('tumor_focality', 'tumor_focality'), ('gender', 'gender')]\n",
      "Precision: 1.0, Recall: 0.2857142857142857, F1 Score: 0.4444444444444445\n",
      "10\n",
      "[('histologic_grade_figo', 'tumor_grade'), ('histologic_type', 'primary_diagnosis'), ('path_stage_primary_tumor-pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes-pn', 'ajcc_pathologic_n'), ('tumor_stage-pathological', 'ajcc_pathologic_stage'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('gender', 'gender'), ('tumor_site', 'tissue_or_organ_of_origin'), ('tumor_focality', 'tumor_focality')]\n",
      "\n",
      "Processing ../data/extracted-tables/Dou_UCEC_CPTAC3_meta_table_V2.1.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['tumor_focality', 'race', 'ethnicity', 'gender']\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.8955\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                        From                     To  Similarity\n",
      "18  tumor_stage-pathological  ajcc_pathologic_stage       0.531\n",
      "38                msi_status           vital_status       0.456\n",
      "\n",
      "\n",
      "Checking similarity values for column: tumor_stage-pathological ajcc_pathologic_stage\n",
      "Domain similarity for tumor_stage-pathological and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: msi_status vital_status\n",
      "Domain similarity for msi_status and vital_status is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_excluded', 'morphology'), ('myometrial_invasion_specify', 'tumor_grade'), ('wxs_tumor_sample_type', 'ajcc_pathologic_t'), ('wgs_normal_uuid', 'primary_diagnosis'), ('wxs_normal_uuid', 'tumor_code'), ('rnaseq_r2_sample_type', 'ajcc_pathologic_n'), ('p53', 'vital_status'), ('proteomics_oct', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for case_excluded and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Dou, we have:\n",
      "Groundtruth: [('histologic_grade_figo', 'tumor_grade'), ('histologic_type', 'primary_diagnosis'), ('path_stage_primary_tumor-pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes-pn', 'ajcc_pathologic_n'), ('tumor_stage-pathological', 'ajcc_pathologic_stage'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('gender', 'gender'), ('tumor_site', 'tissue_or_organ_of_origin'), ('tumor_focality', 'tumor_focality')]\n",
      "Found pairs [('tumor_focality', 'tumor_focality'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('gender', 'gender'), ('tumor_stage-pathological', 'ajcc_pathologic_stage')]\n",
      "Precision: 1.0, Recall: 0.5, F1 Score: 0.6666666666666666\n",
      "\n",
      "\n",
      "Krug: Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "Clark: Precision: 1.0, Recall: 0.4444444444444444, F1 Score: 0.6153846153846153\n",
      "Vasaikar: Precision: 1.0, Recall: 0.4, F1 Score: 0.5714285714285715\n",
      "Gillette: Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "Wang: Precision: 1.0, Recall: 0.75, F1 Score: 0.8571428571428571\n",
      "Satpathy: Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "Cao: Precision: 0.75, Recall: 0.42857142857142855, F1 Score: 0.5454545454545454\n",
      "Huang: Precision: 1.0, Recall: 0.2857142857142857, F1 Score: 0.4444444444444445\n",
      "Dou: Precision: 1.0, Recall: 0.5, F1 Score: 0.6666666666666666\n"
     ]
    }
   ],
   "source": [
    "scores = {}\n",
    "for file in os.listdir(ground_truth_path):\n",
    "\n",
    "    # if 'Dou' not in file:\n",
    "    #     continue\n",
    "\n",
    "    study = file.split('.csv')[0]\n",
    "\n",
    "    groundtruth_path = os.path.join(ground_truth_path, file)\n",
    "    groundtruth = ut.load_table_matching_groundtruth(groundtruth_path)\n",
    "    groundtruth = [\n",
    "        entry for entry in groundtruth if entry[1] in set(gdc_df.columns)]\n",
    "    groundtruth = [(entry[0].lower(), entry[1].lower()) for entry in groundtruth]\n",
    "    print(len(groundtruth))\n",
    "    print(groundtruth)\n",
    "\n",
    "    files = glob.glob(os.path.join(csv_data_path, f\"{study}*.csv\"))\n",
    "\n",
    "    matching_pairs = []\n",
    "\n",
    "    for i, file in enumerate(files):\n",
    "        \n",
    "        \n",
    "\n",
    "        print(f\"\\nProcessing {file}\")\n",
    "        candidate_df = pd.read_csv(file)\n",
    "        candidate_df.columns = candidate_df.columns.str.lower()\n",
    "        candidate_df = candidate_df.select_dtypes(exclude='number')\n",
    "\n",
    "        # Exact match based on column names\n",
    "        common_cols = list(\n",
    "            set(candidate_df.columns).intersection(set(gdc_df.columns)))\n",
    "        \n",
    "        print('\\nExact column name matches between candidate table and gdc', common_cols)\n",
    "        for col in common_cols:\n",
    "            print('Checking similarity values for column:', col)\n",
    "            candidate_col_values =  candidate_df[col].dropna().astype(str).unique()\n",
    "            candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]# remove null values and make it lower case\n",
    "            target_col_values = gdc_df[col].dropna().astype(str).unique()\n",
    "            target_col_values = [c.lower() for c in target_col_values] # mae it lower case\n",
    "            domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "            if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "                matching_pairs.append((col, col))\n",
    "\n",
    "            print(f\"Domain similarity for {col} is {domain_similarity_avg}\")\n",
    "            print('-'*60)\n",
    "        print('\\n')\n",
    "\n",
    "        ## Checking similarity based on column names\n",
    "\n",
    "        # TF-IDF      \n",
    "        model = PolyFuzz(\"TF-IDF\").match(list(candidate_df.columns), list(gdc_df.columns))\n",
    "        name_matches = model.get_matches()\n",
    "        # Edit Distance model\n",
    "        # jellyfish_matcher = EditDistance(n_jobs=1, scorer=jaro_winkler_similarity)\n",
    "        # model = PolyFuzz(jellyfish_matcher).match(list(candidate_df.columns), list(gdc_df.columns))\n",
    "        # name_matches = model.get_matches()\n",
    "                                          \n",
    "        name_matches = name_matches[(name_matches['Similarity'] >= 0.4) & (name_matches['Similarity'] < 1.0)] ## get similar matches\n",
    "        print('Similar column name matches between candidate table and gdc')\n",
    "        print(name_matches) \n",
    "        print('\\n')\n",
    "\n",
    "        for match in name_matches.itertuples():\n",
    "            print('Checking similarity values for column:', match[1], match[2])\n",
    "            candidate_col_values =  candidate_df[match[1]].dropna().astype(str).unique()\n",
    "            candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]\n",
    "            \n",
    "            \n",
    "            target_col_values = gdc_df[match[2]].dropna().astype(str).unique()\n",
    "            target_col_values = [c.lower() for c in target_col_values ]\n",
    "            \n",
    "            domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "            if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "                matching_pairs.append((match[1], match[2]))\n",
    "            print(f\"Domain similarity for {match[1]} and {match[2]} is {domain_similarity_avg}\")\n",
    "            print('-'*60) \n",
    "\n",
    "\n",
    "        # checking the remaining of the candidates\n",
    "        print('\\nChecking similarity values for remaining columns')\n",
    "            \n",
    "        candidate_cols = set(candidate_df.columns)\n",
    "        remaining_cols = list(candidate_cols - set([pair[0] for pair in matching_pairs]))\n",
    "\n",
    "        target_cols = set(gdc_df.columns)\n",
    "        remaining_target_cols = list(target_cols - set([pair[1] for pair in matching_pairs]))\n",
    "        \n",
    "        pairwise_elements = list(zip(remaining_cols, remaining_target_cols))\n",
    "        print(pairwise_elements)\n",
    "\n",
    "        for pair in pairwise_elements:\n",
    "            # print('\\nChecking similarity values for column:', pair[0], pair[1])\n",
    "            candidate_col_values =  candidate_df[pair[0]].dropna().astype(str).unique()\n",
    "            candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]\n",
    "            \n",
    "            target_col_values = gdc_df[pair[1]].dropna().astype(str).unique()\n",
    "            target_col_values = [c.lower() for c in target_col_values ]\n",
    "            domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "            # domain_similarity_avg = check_domains_emb(candidate_col_values, target_col_values)\n",
    "            if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "                matching_pairs.append(pair)\n",
    "            print(f\"Domain similarity for {pair[0]} and {pair[1]} is {domain_similarity_avg}\")\n",
    "            print('-'*60)\n",
    "            break\n",
    "        \n",
    "\n",
    "    print(f'\\nFor {study}, we have:')\n",
    "    print('Groundtruth:', groundtruth)\n",
    "    print('Found pairs', matching_pairs)\n",
    "    precision, recall, f1_score = calculate_metrics(groundtruth, matching_pairs)\n",
    "    print(f\"Precision: {precision}, Recall: {recall}, F1 Score: {f1_score}\")\n",
    "    scores[study] = (precision, recall, f1_score)\n",
    "\n",
    "        # ## check remaining of ground truth\n",
    "        # remaining_pairs = list(set(groundtruth) - set(matching_pairs))\n",
    "        # print('Remaining pairs:', remaining_pairs)\n",
    "\n",
    "        # for pair in remaining_pairs:\n",
    "        #     print('\\nChecking similarity values for column:', pair[0], pair[1])\n",
    "        #     candidate_col_values =  candidate_df[pair[0]].dropna().astype(str).unique()\n",
    "        #     candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]\n",
    "            \n",
    "        #     target_col_values = gdc_df[pair[1]].dropna().astype(str).unique()\n",
    "        #     target_col_values = [c.lower() for c in target_col_values ]\n",
    "        #     domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "        #     if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "        #         matching_pairs.append(pair)\n",
    "        #     print(f\"Domain similarity for {pair[0]} and {pair[1]} is {domain_similarity_avg}\")\n",
    "        #     print('-'*60)\n",
    "\n",
    "        # print('\\nSo far, we have:')\n",
    "        # print('Groundtruth:', groundtruth)\n",
    "        # print('Found pairs', matching_pairs)\n",
    "        # precision, recall, f1_score = calculate_metrics(groundtruth, matching_pairs)\n",
    "        # print(f\"Precision: {precision}, Recall: {recall}, F1 Score: {f1_score}\")\n",
    "print('\\n')\n",
    "for k,v in scores.items():\n",
    "    precision, recall, f1_score = v\n",
    "    print(f'{k}: Precision: {precision}, Recall: {recall}, F1 Score: {f1_score}')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "arpah",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
