{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import pandas as pd\n",
    "from valentine.algorithms import Coma\n",
    "from valentine.algorithms import JaccardDistanceMatcher\n",
    "from valentine.algorithms import SimilarityFlooding\n",
    "from valentine.algorithms import DistributionBased\n",
    "from valentine.algorithms import Cupid\n",
    "from valentine.metrics import F1Score, PrecisionTopNPercent\n",
    "from valentine import valentine_match\n",
    "import pprint\n",
    "import glob\n",
    "from column_representation import ColumnRepresentation, gdc_target_columns, gdc_df\n",
    "\n",
    "from polyfuzz import PolyFuzz\n",
    "from polyfuzz.models import EditDistance\n",
    "from jellyfish import jaro_winkler_similarity\n",
    "from polyfuzz.models import Embeddings\n",
    "from flair.embeddings import TransformerWordEmbeddings, WordEmbeddings\n",
    "\n",
    "from gdc.gdc_api import GDCSchema\n",
    "import utils.file_utils as ut\n",
    "import sys\n",
    "import os\n",
    "sys.path.append(os.path.join(os.path.abspath(''), '..'))\n",
    "\n",
    "\n",
    "sns.set_theme(style=\"white\")\n",
    "\n",
    "\n",
    "pp = pprint.PrettyPrinter(indent=4, sort_dicts=True)\n",
    "\n",
    "ground_truth_path = os.path.join(\n",
    "    '..', 'data', 'table-matching-ground-truth', 'ground-truth')\n",
    "csv_data_path = os.path.join('..', 'data', 'extracted-tables')\n",
    "\n",
    "cat_null_values = set(['n/a', 'na', 'nan', 'null', 'lost to follow-up', 'not available', 'unable to obtain',\n",
    "                       'unknown', 'not reported', 'not allowed to collect',\n",
    "                       'unspecified', 'not specified', 'other, specify','none'])\n",
    "\n",
    "DOMAIN_SIMILARITY_THRESHOLD = 0.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1068, 16)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case_submitter_id</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>race</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>gender</th>\n",
       "      <th>vital_status</th>\n",
       "      <th>ajcc_pathologic_t</th>\n",
       "      <th>ajcc_pathologic_n</th>\n",
       "      <th>ajcc_pathologic_stage</th>\n",
       "      <th>tumor_grade</th>\n",
       "      <th>tumor_focality</th>\n",
       "      <th>tumor_largest_dimension_diameter</th>\n",
       "      <th>primary_diagnosis</th>\n",
       "      <th>morphology</th>\n",
       "      <th>tissue_or_organ_of_origin</th>\n",
       "      <th>tumor_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>01BR001</td>\n",
       "      <td>20089.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Alive</td>\n",
       "      <td>T2</td>\n",
       "      <td>N1c</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Invasive carcinoma of no special type</td>\n",
       "      <td>8500/3</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>01BR008</td>\n",
       "      <td>17532.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>01BR009</td>\n",
       "      <td>23376.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>01BR010</td>\n",
       "      <td>23741.0</td>\n",
       "      <td>black or african american</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>01BR015</td>\n",
       "      <td>12784.0</td>\n",
       "      <td>white</td>\n",
       "      <td>not hispanic or latino</td>\n",
       "      <td>female</td>\n",
       "      <td>Alive</td>\n",
       "      <td>T2</td>\n",
       "      <td>N1</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>GX</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Invasive carcinoma of no special type</td>\n",
       "      <td>8500/3</td>\n",
       "      <td>Breast, NOS</td>\n",
       "      <td>BRCA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  case_submitter_id  age_at_diagnosis                       race  \\\n",
       "0           01BR001           20089.0  black or african american   \n",
       "1           01BR008           17532.0  black or african american   \n",
       "2           01BR009           23376.0  black or african american   \n",
       "3           01BR010           23741.0  black or african american   \n",
       "4           01BR015           12784.0                      white   \n",
       "\n",
       "                ethnicity  gender  vital_status ajcc_pathologic_t  \\\n",
       "0  not hispanic or latino  female         Alive                T2   \n",
       "1  not hispanic or latino  female  Not Reported      Not Reported   \n",
       "2  not hispanic or latino  female  Not Reported      Not Reported   \n",
       "3  not hispanic or latino  female  Not Reported      Not Reported   \n",
       "4  not hispanic or latino  female         Alive                T2   \n",
       "\n",
       "  ajcc_pathologic_n ajcc_pathologic_stage tumor_grade tumor_focality  \\\n",
       "0               N1c              Stage II          GX   Not Reported   \n",
       "1      Not Reported          Not Reported          GX   Not Reported   \n",
       "2      Not Reported          Not Reported          GX   Not Reported   \n",
       "3      Not Reported          Not Reported          GX   Not Reported   \n",
       "4                N1              Stage II          GX   Not Reported   \n",
       "\n",
       "  tumor_largest_dimension_diameter                      primary_diagnosis  \\\n",
       "0                     Not Reported  Invasive carcinoma of no special type   \n",
       "1                     Not Reported                           Not Reported   \n",
       "2                     Not Reported                           Not Reported   \n",
       "3                     Not Reported                           Not Reported   \n",
       "4                     Not Reported  Invasive carcinoma of no special type   \n",
       "\n",
       "     morphology tissue_or_organ_of_origin tumor_code  \n",
       "0        8500/3               Breast, NOS       BRCA  \n",
       "1  Not Reported               Breast, NOS       BRCA  \n",
       "2  Not Reported               Breast, NOS       BRCA  \n",
       "3  Not Reported               Breast, NOS       BRCA  \n",
       "4        8500/3               Breast, NOS       BRCA  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_target = pd.read_csv('../data/target.csv')\n",
    "df_target.drop('study', axis=1, inplace=True)\n",
    "print(df_target.shape)\n",
    "df_target.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'age_at_diagnosis',\n",
       " 'ajcc_pathologic_n',\n",
       " 'ajcc_pathologic_stage',\n",
       " 'ajcc_pathologic_t',\n",
       " 'ethnicity',\n",
       " 'gender',\n",
       " 'morphology',\n",
       " 'primary_diagnosis',\n",
       " 'race',\n",
       " 'tissue_or_organ_of_origin',\n",
       " 'tumor_code',\n",
       " 'tumor_focality',\n",
       " 'tumor_grade',\n",
       " 'tumor_largest_dimension_diameter',\n",
       " 'vital_status'}"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "interested_cols = set(df_target.columns)\n",
    "interested_cols.remove('case_submitter_id')\n",
    "interested_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['stage iva', 'stage ib', 'stage x', 'stage ivb', 'stage ia3',\n",
       "       'stage ic', 'stage i', 'stage ib2', 'stage iic', 'stage 0a',\n",
       "       'stage 0', 'stage iiic2', 'stage iii', 'stage ii', 'stage ia1',\n",
       "       'stage is', 'stage tis', 'stage iib', 'stage iiia', 'stage ia2',\n",
       "       'stage ivc', 'stage iiic', 'stage ia', 'stage ib1', 'stage 0is',\n",
       "       'stage iia1', 'stage iiid', 'stage iiia1', 'stage iiib',\n",
       "       'stage iiia2', 'stage iiic1', 'stage iv', 'stage iia2',\n",
       "       'stage iia', <NA>], dtype=object)"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gdc_df\n",
    "gdc_df.ajcc_pathologic_stage.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "def calculate_metrics(groundtruth, matching_pairs):\n",
    "    \n",
    "    true_positives = len(set(groundtruth) & set(matching_pairs))\n",
    "    false_positives = len(set(matching_pairs) - set(groundtruth))\n",
    "    false_negatives = len(set(groundtruth) - set(matching_pairs))\n",
    "    \n",
    "    precision = true_positives / (true_positives + false_positives)\n",
    "    recall = true_positives / (true_positives + false_negatives) \n",
    "    f1_score = 2 * (precision * recall) / (precision + recall)\n",
    "    \n",
    "    return precision, recall, f1_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check_domains_tf(colvalues1, colvalues2):\n",
    "    model = PolyFuzz(\"TF-IDF\").match(colvalues1, colvalues2)\n",
    "    value_matches = model.get_matches()\n",
    "    # model.visualize_precision_recall()\n",
    "    # print(value_matches)\n",
    "    value_matches = value_matches.dropna()\n",
    "    # print(value_matches)\n",
    "    n = len(colvalues1)\n",
    "    # get the first elements according to the domain size of the candidate\n",
    "    value_matches = value_matches.head(n)\n",
    "\n",
    "    return value_matches['Similarity'].median()\n",
    "\n",
    "\n",
    "def check_domains_edit(colvalues1, colvalues2):\n",
    "    jellyfish_matcher = EditDistance(n_jobs=1, scorer=jaro_winkler_similarity)\n",
    "    model = PolyFuzz(jellyfish_matcher).match(colvalues1, colvalues2)\n",
    "    value_matches = model.get_matches()\n",
    "    # print(value_matches)\n",
    "    value_matches = value_matches.dropna()\n",
    "\n",
    "    n = len(colvalues1)\n",
    "    # get the first elements according to the domain size of the candidate\n",
    "    value_matches = value_matches.head(n)\n",
    "\n",
    "    return value_matches['Similarity'].median()\n",
    "\n",
    "\n",
    "def check_domains_emb(colvalues1, colvalues2):\n",
    "\n",
    "    bert = TransformerWordEmbeddings('bert-base-multilingual-cased')\n",
    "    bert_matcher = Embeddings(bert, min_similarity=0)\n",
    "\n",
    "    # fasttext = WordEmbeddings('en-crawl')\n",
    "    # fasttext_matcher = Embeddings(fasttext, min_similarity=0)\n",
    "\n",
    "    # matchers = [bert_matcher, fasttext_matcher]\n",
    "\n",
    "    model = PolyFuzz(bert_matcher).match(colvalues1, colvalues2)\n",
    "\n",
    "    value_matches = model.get_matches()\n",
    "\n",
    "    print(value_matches)\n",
    "\n",
    "    # value_matches = value_matches.dropna()\n",
    "\n",
    "    # n = len(colvalues1)\n",
    "    # # get the first elements according to the domain size of the candidate\n",
    "    # value_matches = value_matches.head(n)\n",
    "\n",
    "    # return value_matches['Similarity'].median()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3\n",
      "[('tumor.stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_D)_QC_mutation_calls.csv\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                   From          To  Similarity\n",
      "3  tumor_sample_barcode  tumor_code       0.474\n",
      "\n",
      "\n",
      "Checking similarity values for column: tumor_sample_barcode tumor_code\n",
      "Domain similarity for tumor_sample_barcode and tumor_code is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('variant_type', 'primary_diagnosis'), ('hugo_symbol', 'tumor_focality'), ('variant_classification', 'morphology'), ('passed_filters', 'tumor_grade'), ('tumor_sample_barcode', 'ethnicity'), ('matched_norm_sample_barcode', 'ajcc_pathologic_stage')]\n",
      "Domain similarity for variant_type and primary_diagnosis is 0.3665\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_A)_Metadata.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.738\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('cd3.tils.counts', 'primary_diagnosis'), ('pr.clinical.status', 'tumor_focality'), ('gata3.mutation.type', 'morphology'), ('pten.mutation.type', 'tumor_grade'), ('erbb2.proteogenomic.status', 'ajcc_pathologic_stage'), ('sample.ids', 'race'), ('cbfb.mutation.type', 'vital_status'), ('apobec.signature', 'ajcc_pathologic_n'), ('pam50', 'tissue_or_organ_of_origin'), ('er.updated.clinical.status', 'tumor_code'), ('arid1a.mutation.type', 'ajcc_pathologic_t')]\n",
      "Domain similarity for cd3.tils.counts and primary_diagnosis is 0.255\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_B)_QC_RNA-seq.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('sample.id', 'primary_diagnosis'), ('acgt tag id', 'tumor_focality'), ('qc', 'morphology'), ('notes', 'tumor_grade')]\n",
      "Domain similarity for sample.id and primary_diagnosis is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Krug_C)_QC_WXS.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('type', 'primary_diagnosis'), ('sample_id', 'tumor_focality'), ('participant', 'morphology')]\n",
      "Domain similarity for type and primary_diagnosis is 0.626\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Krug, we have:\n",
      "Groundtruth: [('tumor.stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "Found pairs [('ethnicity', 'ethnicity'), ('gender', 'gender'), ('type', 'primary_diagnosis')]\n",
      "Precision: 0.6666666666666666, Recall: 0.6666666666666666, F1 Score: 0.6666666666666666\n",
      "9\n",
      "[('gender', 'gender'), ('race', 'race'), ('ethnicity_self_identify', 'ethnicity'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('grade', 'tumor_grade'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('path_stage_primary_tumor_pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes_pn', 'ajcc_pathologic_n')]\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_data_sample_availability.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 1', 'primary_diagnosis'), ('unnamed: 4', 'tumor_focality'), ('unnamed: 3', 'morphology'), ('unnamed: 2', 'tumor_grade'), ('table: sample size of different data sets after excluding samples with low quality.', 'ethnicity')]\n",
      "Domain similarity for unnamed: 1 and primary_diagnosis is 0.27\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_RNA-seq_sequencing_depth.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('tissue type', 'primary_diagnosis'), ('sample aliquot id', 'tumor_focality'), ('case', 'morphology')]\n",
      "Domain similarity for tissue type and primary_diagnosis is 0.5305\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_WGS-WES_sequencing_depth.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case', 'tumor_focality'), ('bam file', 'morphology'), ('sample aliquot', 'tumor_grade')]\n",
      "Domain similarity for case and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Clark_ccrcc_clinical_characteristics.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['tumor_focality', 'race', 'gender']\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                        From                     To  Similarity\n",
      "5    ethnicity_self_identify              ethnicity       0.553\n",
      "10                     grade            tumor_grade       0.684\n",
      "11  tumor_stage_pathological  ajcc_pathologic_stage       0.499\n",
      "\n",
      "\n",
      "Checking similarity values for column: ethnicity_self_identify ethnicity\n",
      "Domain similarity for ethnicity_self_identify and ethnicity is 0.0705\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: grade tumor_grade\n",
      "Domain similarity for grade and tumor_grade is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: tumor_stage_pathological ajcc_pathologic_stage\n",
      "Domain similarity for tumor_stage_pathological and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_id', 'morphology'), ('tumor_site_other', 'tumor_grade'), ('tumor/normal', 'ethnicity'), ('tumor_site', 'vital_status'), ('ethnicity_self_identify', 'ajcc_pathologic_n'), ('grade', 'tissue_or_organ_of_origin'), ('path_stage_reg_lymph_nodes_pn', 'tumor_code'), ('age', 'ajcc_pathologic_t')]\n",
      "Domain similarity for case_id and morphology is 0.3205\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Clark, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('race', 'race'), ('ethnicity_self_identify', 'ethnicity'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('grade', 'tumor_grade'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('path_stage_primary_tumor_pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes_pn', 'ajcc_pathologic_n')]\n",
      "Found pairs [('tissue type', 'primary_diagnosis'), ('tumor_focality', 'tumor_focality'), ('race', 'race'), ('gender', 'gender'), ('tumor_stage_pathological', 'ajcc_pathologic_stage')]\n",
      "Precision: 0.8, Recall: 0.4444444444444444, F1 Score: 0.5714285714285714\n",
      "5\n",
      "[('gender', 'gender'), ('pt', 'ajcc_pathologic_t'), ('pn', 'ajcc_pathologic_n'), ('stage', 'ajcc_pathologic_stage'), ('vital.status', 'vital_status')]\n",
      "\n",
      "Processing ../data/extracted-tables/Vasaikar_B-ClinicalData.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['gender']\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "     From                     To  Similarity\n",
      "17  stage  ajcc_pathologic_stage       0.408\n",
      "\n",
      "\n",
      "Checking similarity values for column: stage ajcc_pathologic_stage\n",
      "Domain similarity for stage and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('vital.status', 'primary_diagnosis'), ('cms', 'tumor_focality'), ('mrna-seq', 'morphology'), ('polyps_history', 'tumor_grade'), ('pole_mutation', 'ethnicity'), ('sampleid', 'race'), ('pn', 'vital_status'), ('mutation_rate', 'ajcc_pathologic_n'), ('kras_mutation', 'tissue_or_organ_of_origin'), ('mirna-seq', 'tumor_code'), ('clinical', 'ajcc_pathologic_t')]\n",
      "Domain similarity for vital.status and primary_diagnosis is 0.369\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Vasaikar_A-Annotation.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('column name', 'primary_diagnosis'), ('data type', 'tumor_focality'), ('column definition', 'morphology'), ('definition', 'tumor_grade')]\n",
      "Domain similarity for column name and primary_diagnosis is 0.27\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Vasaikar, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('pt', 'ajcc_pathologic_t'), ('pn', 'ajcc_pathologic_n'), ('stage', 'ajcc_pathologic_stage'), ('vital.status', 'vital_status')]\n",
      "Found pairs [('gender', 'gender'), ('stage', 'ajcc_pathologic_stage')]\n",
      "Precision: 1.0, Recall: 0.4, F1 Score: 0.5714285714285715\n",
      "3\n",
      "[('gender', 'gender'), ('ethnicity', 'ethnicity'), ('dominant.histological.subtype', 'primary_diagnosis')]\n",
      "\n",
      "Processing ../data/extracted-tables/Gillette_Annotions_S1A.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.729\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "    From                     To  Similarity\n",
      "6  stage  ajcc_pathologic_stage       0.426\n",
      "\n",
      "\n",
      "Checking similarity values for column: stage ajcc_pathologic_stage\n",
      "Domain similarity for stage and ajcc_pathologic_stage is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('secondhand.smoke', 'primary_diagnosis'), ('braf.mutation', 'tumor_focality'), ('aliquot', 'morphology'), ('type', 'tumor_grade'), ('qc.status', 'ajcc_pathologic_stage'), ('araf.mutation', 'race'), ('dominant.signature.wgs.notsmoking.50perc', 'vital_status'), ('dominant.histological.subtype', 'ajcc_pathologic_n'), ('smoking.status', 'tissue_or_organ_of_origin'), ('stage', 'tumor_code'), ('mrna.expression.subtype.tcga', 'ajcc_pathologic_t')]\n",
      "Domain similarity for secondhand.smoke and primary_diagnosis is 0.264\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Gillette, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('ethnicity', 'ethnicity'), ('dominant.histological.subtype', 'primary_diagnosis')]\n",
      "Found pairs [('ethnicity', 'ethnicity'), ('gender', 'gender')]\n",
      "Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "4\n",
      "[('gender', 'gender'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('vital_status', 'vital_status')]\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_additional_annotations.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('idh1_alter_status', 'primary_diagnosis'), ('telomere', 'tumor_focality'), ('pik3ca_alter_status', 'morphology'), ('tert_alter_status', 'tumor_grade'), ('atrx_alter_status', 'ethnicity'), ('mgmt_methyl_clinical_test_method', 'ajcc_pathologic_stage'), ('pik3r1_alter_status', 'race'), ('immune', 'vital_status'), ('multiomic', 'ajcc_pathologic_n'), ('dna_methyl', 'tissue_or_organ_of_origin'), ('pten_alter_status', 'tumor_code'), ('braf_alter_status', 'ajcc_pathologic_t'), ('nf1_alter_status', 'gender')]\n",
      "Domain similarity for idh1_alter_status and primary_diagnosis is 0.189\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_survival_analysis.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('group value', 'primary_diagnosis'), ('stratification group', 'tumor_focality')]\n",
      "Domain similarity for group value and primary_diagnosis is 0.552\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_cbttc_sample_info.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('clinical_status_at_collection_event', 'tumor_focality'), ('aliquot_id', 'morphology'), ('tmt_channel', 'tumor_grade'), ('research_id', 'ethnicity'), ('kids_first_id', 'ajcc_pathologic_stage'), ('sample_id', 'race'), ('original_aliquot_id', 'vital_status')]\n",
      "Domain similarity for clinical_status_at_collection_event and tumor_focality is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Wang_clinical_data.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'vital_status', 'race', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: vital_status\n",
      "Domain similarity for vital_status is 0.202\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                      From         To  Similarity\n",
      "5  ethnicity_self_identify  ethnicity       0.523\n",
      "\n",
      "\n",
      "Checking similarity values for column: ethnicity_self_identify ethnicity\n",
      "Domain similarity for ethnicity_self_identify and ethnicity is 0.058\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_id', 'morphology'), ('expert_path_review', 'tumor_focality'), ('smoking_history', 'tumor_grade'), ('tumor_laterality', 'ajcc_pathologic_stage'), ('cause_of_death', 'vital_status'), ('ethnicity_self_identify', 'ajcc_pathologic_n'), ('secondhand_smoke_exposure', 'tissue_or_organ_of_origin'), ('tumor_site_curated', 'tumor_code'), ('excluded_in_downstream_analysis', 'ajcc_pathologic_t')]\n",
      "Domain similarity for case_id and morphology is 0.307\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Wang, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('vital_status', 'vital_status')]\n",
      "Found pairs [('group value', 'primary_diagnosis'), ('ethnicity', 'ethnicity'), ('race', 'race'), ('gender', 'gender')]\n",
      "Precision: 0.75, Recall: 0.75, F1 Score: 0.75\n",
      "3\n",
      "[('stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1A.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.124\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "             From                     To  Similarity\n",
      "6           stage  ajcc_pathologic_stage       0.412\n",
      "11  ethnicity_mod              ethnicity       0.805\n",
      "16      qc.status           vital_status       0.403\n",
      "\n",
      "\n",
      "Checking similarity values for column: stage ajcc_pathologic_stage\n",
      "Domain similarity for stage and ajcc_pathologic_stage is 0.828\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: ethnicity_mod ethnicity\n",
      "Domain similarity for ethnicity_mod and ethnicity is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: qc.status vital_status\n",
      "Domain similarity for qc.status and vital_status is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('nmf.subtype.core', 'primary_diagnosis'), ('secondhand.smoke', 'tumor_focality'), ('ethnicity', 'morphology'), ('smoking.score.category.wes', 'tumor_grade'), ('arid1a.mutation', 'ethnicity'), ('suz12.mutation', 'race'), ('tcga pan-can immune clustering', 'vital_status'), ('cul3.mutation', 'ajcc_pathologic_n'), ('pten.mutation', 'tissue_or_organ_of_origin'), ('cimp.status.meth', 'tumor_code'), ('pik3ca.mutation', 'ajcc_pathologic_t')]\n",
      "Domain similarity for nmf.subtype.core and primary_diagnosis is 0.552\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1C.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 5', 'morphology'), ('unnamed: 6', 'tumor_focality'), ('unnamed: 4', 'tumor_grade'), ('unnamed: 9', 'ethnicity'), ('nmf cluster (p<0.05)', 'race'), ('unnamed: 11', 'vital_status'), ('unnamed: 2', 'ajcc_pathologic_n'), ('unnamed: 8', 'tissue_or_organ_of_origin'), ('unnamed: 1', 'tumor_code'), ('unnamed: 10', 'ajcc_pathologic_t')]\n",
      "Domain similarity for unnamed: 5 and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1E.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 5', 'morphology'), ('unnamed: 4', 'tumor_focality'), ('unnamed: 9', 'tumor_grade'), ('unnamed: 0', 'ethnicity'), ('unnamed: 8', 'race'), ('p-values (bonferroni corrected)', 'vital_status'), ('counts', 'ajcc_pathologic_n'), ('unnamed: 1', 'tissue_or_organ_of_origin'), ('unnamed: 3', 'tumor_code'), ('unnamed: 7', 'ajcc_pathologic_t')]\n",
      "Domain similarity for unnamed: 5 and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Satpathy_Table_S1D.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 5', 'morphology'), ('unnamed: 6', 'tumor_focality'), ('unnamed: 4', 'tumor_grade'), ('unnamed: 9', 'ethnicity'), ('unnamed: 11', 'race'), ('unnamed: 0', 'vital_status'), ('unnamed: 8', 'ajcc_pathologic_n'), ('p-values (bonferroni corrected)', 'tissue_or_organ_of_origin'), ('counts', 'tumor_code'), ('unnamed: 1', 'ajcc_pathologic_t')]\n",
      "Domain similarity for unnamed: 5 and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Satpathy, we have:\n",
      "Groundtruth: [('stage', 'ajcc_pathologic_stage'), ('gender', 'gender'), ('ethnicity', 'ethnicity')]\n",
      "Found pairs [('gender', 'gender'), ('stage', 'ajcc_pathologic_stage'), ('nmf.subtype.core', 'primary_diagnosis')]\n",
      "Precision: 0.6666666666666666, Recall: 0.6666666666666666, F1 Score: 0.6666666666666666\n",
      "7\n",
      "[('sex', 'gender'), ('race', 'race'), ('tumor_focality', 'tumor_focality'), ('pathologic_staging_regional_lymph_nodes_pn', 'ajcc_pathologic_n'), ('pathologic_staging_primary_tumor_pt', 'ajcc_pathologic_t'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('vital_status', 'vital_status')]\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_circRNAForNAT.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 0', 'primary_diagnosis')]\n",
      "Domain similarity for unnamed: 0 and primary_diagnosis is 0.04\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_Clinical_data.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['vital_status', 'tumor_focality', 'race']\n",
      "Checking similarity values for column: vital_status\n",
      "Domain similarity for vital_status is 0.202\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                                        From                     To  \\\n",
      "3                        histology_diagnosis      primary_diagnosis   \n",
      "14       pathologic_staging_primary_tumor_pt  ajcc_pathologic_stage   \n",
      "15  pathologic_staging_distant_metastasis_pm  ajcc_pathologic_stage   \n",
      "18                  tumor_stage_pathological  ajcc_pathologic_stage   \n",
      "\n",
      "    Similarity  \n",
      "3        0.535  \n",
      "14       0.414  \n",
      "15       0.406  \n",
      "18       0.488  \n",
      "\n",
      "\n",
      "Checking similarity values for column: histology_diagnosis primary_diagnosis\n",
      "Domain similarity for histology_diagnosis and primary_diagnosis is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: pathologic_staging_primary_tumor_pt ajcc_pathologic_stage\n",
      "Domain similarity for pathologic_staging_primary_tumor_pt and ajcc_pathologic_stage is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: pathologic_staging_distant_metastasis_pm ajcc_pathologic_stage\n",
      "Domain similarity for pathologic_staging_distant_metastasis_pm and ajcc_pathologic_stage is nan\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: tumor_stage_pathological ajcc_pathologic_stage\n",
      "Domain similarity for tumor_stage_pathological and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('fat_fraction', 'morphology'), ('cause_of_death', 'tumor_grade'), ('tumor_included_for_the_study', 'ethnicity'), ('inflammation_fraction', 'vital_status'), ('tumor_necrosis', 'ajcc_pathologic_n'), ('muscle_fraction', 'tissue_or_organ_of_origin'), ('is_this_patient_lost_to_follow_up', 'tumor_code'), ('pathologic_staging_regional_lymph_nodes_pn', 'ajcc_pathologic_t'), ('neoplastic_cellularity', 'gender')]\n",
      "Domain similarity for fat_fraction and morphology is 0.761\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_TMT_labeling_scheme.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('channel_name', 'tumor_grade'), ('sample_type', 'ethnicity'), ('case_id', 'vital_status')]\n",
      "Domain similarity for channel_name and tumor_grade is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_Molecular_phenotype_data.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "         From            To  Similarity\n",
      "4  msi_status  vital_status       0.435\n",
      "\n",
      "\n",
      "Checking similarity values for column: msi_status vital_status\n",
      "Domain similarity for msi_status and vital_status is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_id', 'tumor_grade'), ('collisson', 'ethnicity'), ('msi_status', 'vital_status'), ('bailey', 'ajcc_pathologic_n'), ('moffitt', 'tissue_or_organ_of_origin')]\n",
      "Domain similarity for case_id and tumor_grade is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_circRNAforTumor.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('unnamed: 0', 'tumor_grade')]\n",
      "Domain similarity for unnamed: 0 and tumor_grade is 0.06\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Cao_Proteins_phospho_glyco_IDs.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_id', 'tumor_grade'), ('type', 'ethnicity'), ('tmt_channel', 'vital_status')]\n",
      "Domain similarity for case_id and tumor_grade is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Cao, we have:\n",
      "Groundtruth: [('sex', 'gender'), ('race', 'race'), ('tumor_focality', 'tumor_focality'), ('pathologic_staging_regional_lymph_nodes_pn', 'ajcc_pathologic_n'), ('pathologic_staging_primary_tumor_pt', 'ajcc_pathologic_t'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('vital_status', 'vital_status')]\n",
      "Found pairs [('tumor_focality', 'tumor_focality'), ('race', 'race'), ('histology_diagnosis', 'primary_diagnosis'), ('tumor_stage_pathological', 'ajcc_pathologic_stage'), ('fat_fraction', 'morphology')]\n",
      "Precision: 0.6, Recall: 0.42857142857142855, F1 Score: 0.5\n",
      "7\n",
      "[('gender', 'gender'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('histologic_grade', 'tumor_grade'), ('patho_staging_pt', 'ajcc_pathologic_t'), ('patho_staging_pn', 'ajcc_pathologic_n'), ('tumor_stage', 'ajcc_pathologic_stage')]\n",
      "\n",
      "Processing ../data/extracted-tables/Huang_Meta_table.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['tumor_focality', 'gender']\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                      From          To  Similarity\n",
      "24  tumor_pathology_review  morphology        0.41\n",
      "\n",
      "\n",
      "Checking similarity values for column: tumor_pathology_review morphology\n",
      "Domain similarity for tumor_pathology_review and morphology is 0.251\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('clinic_staging_dist_metas', 'primary_diagnosis'), ('num_yrs_alc_con', 'morphology'), ('num_smoke_per_day', 'tumor_grade'), ('smoking_history', 'ethnicity'), ('alcohol_consum', 'ajcc_pathologic_stage'), ('tumor_stage', 'race'), ('tumor_pathology_review', 'vital_status'), ('patho_staging_pt', 'ajcc_pathologic_n'), ('tumor_site_original', 'tissue_or_organ_of_origin'), ('hpv_inference', 'tumor_code'), ('tumor_necrosis', 'ajcc_pathologic_t')]\n",
      "Domain similarity for clinic_staging_dist_metas and primary_diagnosis is 0.394\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Huang_Data_availabilty.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_id', 'primary_diagnosis'), ('1: available; 0: unavailable', 'morphology')]\n",
      "Domain similarity for case_id and primary_diagnosis is 0.0765\n",
      "------------------------------------------------------------\n",
      "\n",
      "Processing ../data/extracted-tables/Huang_TMT-set.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc []\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "Empty DataFrame\n",
      "Columns: [From, To, Similarity]\n",
      "Index: []\n",
      "\n",
      "\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('case_id', 'primary_diagnosis'), ('type', 'morphology'), ('combined_id', 'tumor_grade'), ('channel #', 'ethnicity')]\n",
      "Domain similarity for case_id and primary_diagnosis is 0.10999999999999999\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Huang, we have:\n",
      "Groundtruth: [('gender', 'gender'), ('tumor_focality', 'tumor_focality'), ('histologic_type', 'primary_diagnosis'), ('histologic_grade', 'tumor_grade'), ('patho_staging_pt', 'ajcc_pathologic_t'), ('patho_staging_pn', 'ajcc_pathologic_n'), ('tumor_stage', 'ajcc_pathologic_stage')]\n",
      "Found pairs [('tumor_focality', 'tumor_focality'), ('gender', 'gender')]\n",
      "Precision: 1.0, Recall: 0.2857142857142857, F1 Score: 0.4444444444444445\n",
      "10\n",
      "[('histologic_grade_figo', 'tumor_grade'), ('histologic_type', 'primary_diagnosis'), ('path_stage_primary_tumor-pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes-pn', 'ajcc_pathologic_n'), ('tumor_stage-pathological', 'ajcc_pathologic_stage'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('gender', 'gender'), ('tumor_site', 'tissue_or_organ_of_origin'), ('tumor_focality', 'tumor_focality')]\n",
      "\n",
      "Processing ../data/extracted-tables/Dou_UCEC_CPTAC3_meta_table_V2.1.csv\n",
      "\n",
      "Exact column name matches between candidate table and gdc ['ethnicity', 'tumor_focality', 'race', 'gender']\n",
      "Checking similarity values for column: ethnicity\n",
      "Domain similarity for ethnicity is 0.8955\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: tumor_focality\n",
      "Domain similarity for tumor_focality is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: race\n",
      "Domain similarity for race is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: gender\n",
      "Domain similarity for gender is 1.0\n",
      "------------------------------------------------------------\n",
      "\n",
      "\n",
      "Similar column name matches between candidate table and gdc\n",
      "                        From                     To  Similarity\n",
      "18  tumor_stage-pathological  ajcc_pathologic_stage       0.531\n",
      "38                msi_status           vital_status       0.456\n",
      "\n",
      "\n",
      "Checking similarity values for column: tumor_stage-pathological ajcc_pathologic_stage\n",
      "Domain similarity for tumor_stage-pathological and ajcc_pathologic_stage is 1.0\n",
      "------------------------------------------------------------\n",
      "Checking similarity values for column: msi_status vital_status\n",
      "Domain similarity for msi_status and vital_status is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "Checking similarity values for remaining columns\n",
      "[('proteomics_oct', 'morphology'), ('tumor_site_other', 'primary_diagnosis'), ('p53', 'tumor_grade'), ('wxs_tumor_filename', 'vital_status'), ('tumor_purity', 'ajcc_pathologic_n'), ('wgs_normal_uuid', 'tissue_or_organ_of_origin'), ('wgs_normal_sample_type', 'tumor_code'), ('path_stage_dist_mets-pm', 'ajcc_pathologic_t')]\n",
      "Domain similarity for proteomics_oct and morphology is nan\n",
      "------------------------------------------------------------\n",
      "\n",
      "For Dou, we have:\n",
      "Groundtruth: [('histologic_grade_figo', 'tumor_grade'), ('histologic_type', 'primary_diagnosis'), ('path_stage_primary_tumor-pt', 'ajcc_pathologic_t'), ('path_stage_reg_lymph_nodes-pn', 'ajcc_pathologic_n'), ('tumor_stage-pathological', 'ajcc_pathologic_stage'), ('race', 'race'), ('ethnicity', 'ethnicity'), ('gender', 'gender'), ('tumor_site', 'tissue_or_organ_of_origin'), ('tumor_focality', 'tumor_focality')]\n",
      "Found pairs [('ethnicity', 'ethnicity'), ('tumor_focality', 'tumor_focality'), ('race', 'race'), ('gender', 'gender'), ('tumor_stage-pathological', 'ajcc_pathologic_stage')]\n",
      "Precision: 1.0, Recall: 0.5, F1 Score: 0.6666666666666666\n",
      "Krug: Precision: 0.6666666666666666, Recall: 0.6666666666666666, F1 Score: 0.6666666666666666\n",
      "Clark: Precision: 0.8, Recall: 0.4444444444444444, F1 Score: 0.5714285714285714\n",
      "Vasaikar: Precision: 1.0, Recall: 0.4, F1 Score: 0.5714285714285715\n",
      "Gillette: Precision: 1.0, Recall: 0.6666666666666666, F1 Score: 0.8\n",
      "Wang: Precision: 0.75, Recall: 0.75, F1 Score: 0.75\n",
      "Satpathy: Precision: 0.6666666666666666, Recall: 0.6666666666666666, F1 Score: 0.6666666666666666\n",
      "Cao: Precision: 0.6, Recall: 0.42857142857142855, F1 Score: 0.5\n",
      "Huang: Precision: 1.0, Recall: 0.2857142857142857, F1 Score: 0.4444444444444445\n",
      "Dou: Precision: 1.0, Recall: 0.5, F1 Score: 0.6666666666666666\n"
     ]
    }
   ],
   "source": [
    "scores = {}\n",
    "for file in os.listdir(ground_truth_path):\n",
    "\n",
    "    # if 'Dou' not in file:\n",
    "    #     continue\n",
    "\n",
    "    study = file.split('.csv')[0]\n",
    "\n",
    "    groundtruth_path = os.path.join(ground_truth_path, file)\n",
    "    groundtruth = ut.load_table_matching_groundtruth(groundtruth_path)\n",
    "    groundtruth = [\n",
    "        entry for entry in groundtruth if entry[1] in set(gdc_df.columns)]\n",
    "    groundtruth = [(entry[0].lower(), entry[1].lower()) for entry in groundtruth]\n",
    "    print(len(groundtruth))\n",
    "    print(groundtruth)\n",
    "\n",
    "    files = glob.glob(os.path.join(csv_data_path, f\"{study}*.csv\"))\n",
    "\n",
    "    matching_pairs = []\n",
    "\n",
    "    for i, file in enumerate(files):\n",
    "        \n",
    "        \n",
    "\n",
    "        print(f\"\\nProcessing {file}\")\n",
    "        candidate_df = pd.read_csv(file)\n",
    "        candidate_df.columns = candidate_df.columns.str.lower()\n",
    "        candidate_df = candidate_df.select_dtypes(exclude='number')\n",
    "\n",
    "        # Exact match based on column names\n",
    "        common_cols = list(\n",
    "            set(candidate_df.columns).intersection(set(gdc_df.columns)))\n",
    "        \n",
    "        print('\\nExact column name matches between candidate table and gdc', common_cols)\n",
    "        for col in common_cols:\n",
    "            print('Checking similarity values for column:', col)\n",
    "            candidate_col_values =  candidate_df[col].dropna().astype(str).unique()\n",
    "            candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]# remove null values and make it lower case\n",
    "            target_col_values = gdc_df[col].dropna().astype(str).unique()\n",
    "            target_col_values = [c.lower() for c in target_col_values] # mae it lower case\n",
    "            domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "            if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "                matching_pairs.append((col, col))\n",
    "\n",
    "            print(f\"Domain similarity for {col} is {domain_similarity_avg}\")\n",
    "            print('-'*60)\n",
    "        print('\\n')\n",
    "\n",
    "        ## Checking similarity based on column names\n",
    "\n",
    "        # TF-IDF      \n",
    "        model = PolyFuzz(\"TF-IDF\").match(list(candidate_df.columns), list(gdc_df.columns))\n",
    "        name_matches = model.get_matches()\n",
    "        # Edit Distance model\n",
    "        # jellyfish_matcher = EditDistance(n_jobs=1, scorer=jaro_winkler_similarity)\n",
    "        # model = PolyFuzz(jellyfish_matcher).match(list(candidate_df.columns), list(gdc_df.columns))\n",
    "        # name_matches = model.get_matches()\n",
    "                                          \n",
    "        name_matches = name_matches[(name_matches['Similarity'] >= 0.4) & (name_matches['Similarity'] < 1.0)] ## get similar matches\n",
    "        print('Similar column name matches between candidate table and gdc')\n",
    "        print(name_matches) \n",
    "        print('\\n')\n",
    "\n",
    "        for match in name_matches.itertuples():\n",
    "            print('Checking similarity values for column:', match[1], match[2])\n",
    "            candidate_col_values =  candidate_df[match[1]].dropna().astype(str).unique()\n",
    "            candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]\n",
    "            \n",
    "            \n",
    "            target_col_values = gdc_df[match[2]].dropna().astype(str).unique()\n",
    "            target_col_values = [c.lower() for c in target_col_values ]\n",
    "            \n",
    "            domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "            if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "                matching_pairs.append((match[1], match[2]))\n",
    "            print(f\"Domain similarity for {match[1]} and {match[2]} is {domain_similarity_avg}\")\n",
    "            print('-'*60) \n",
    "\n",
    "\n",
    "        # checking the remaining of the candidates\n",
    "        print('\\nChecking similarity values for remaining columns')\n",
    "            \n",
    "        candidate_cols = set(candidate_df.columns)\n",
    "        remaining_cols = list(candidate_cols - set([pair[0] for pair in matching_pairs]))\n",
    "\n",
    "        target_cols = set(gdc_df.columns)\n",
    "        remaining_target_cols = list(target_cols - set([pair[1] for pair in matching_pairs]))\n",
    "        \n",
    "        pairwise_elements = list(zip(remaining_cols, remaining_target_cols))\n",
    "        print(pairwise_elements)\n",
    "\n",
    "        for pair in pairwise_elements:\n",
    "            # print('\\nChecking similarity values for column:', pair[0], pair[1])\n",
    "            candidate_col_values =  candidate_df[pair[0]].dropna().astype(str).unique()\n",
    "            candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]\n",
    "            \n",
    "            target_col_values = gdc_df[pair[1]].dropna().astype(str).unique()\n",
    "            target_col_values = [c.lower() for c in target_col_values ]\n",
    "            domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "            # domain_similarity_avg = check_domains_emb(candidate_col_values, target_col_values)\n",
    "            if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "                matching_pairs.append(pair)\n",
    "            print(f\"Domain similarity for {pair[0]} and {pair[1]} is {domain_similarity_avg}\")\n",
    "            print('-'*60)\n",
    "            break\n",
    "        \n",
    "\n",
    "    print(f'\\nFor {study}, we have:')\n",
    "    print('Groundtruth:', groundtruth)\n",
    "    print('Found pairs', matching_pairs)\n",
    "    precision, recall, f1_score = calculate_metrics(groundtruth, matching_pairs)\n",
    "    print(f\"Precision: {precision}, Recall: {recall}, F1 Score: {f1_score}\")\n",
    "    scores[study] = (precision, recall, f1_score)\n",
    "\n",
    "        # ## check remaining of ground truth\n",
    "        # remaining_pairs = list(set(groundtruth) - set(matching_pairs))\n",
    "        # print('Remaining pairs:', remaining_pairs)\n",
    "\n",
    "        # for pair in remaining_pairs:\n",
    "        #     print('\\nChecking similarity values for column:', pair[0], pair[1])\n",
    "        #     candidate_col_values =  candidate_df[pair[0]].dropna().astype(str).unique()\n",
    "        #     candidate_col_values = [c.lower() for c in candidate_col_values if c.lower() not in cat_null_values]\n",
    "            \n",
    "        #     target_col_values = gdc_df[pair[1]].dropna().astype(str).unique()\n",
    "        #     target_col_values = [c.lower() for c in target_col_values ]\n",
    "        #     domain_similarity_avg = check_domains_tf(candidate_col_values, target_col_values)\n",
    "        #     if domain_similarity_avg > DOMAIN_SIMILARITY_THRESHOLD:\n",
    "        #         matching_pairs.append(pair)\n",
    "        #     print(f\"Domain similarity for {pair[0]} and {pair[1]} is {domain_similarity_avg}\")\n",
    "        #     print('-'*60)\n",
    "\n",
    "        # print('\\nSo far, we have:')\n",
    "        # print('Groundtruth:', groundtruth)\n",
    "        # print('Found pairs', matching_pairs)\n",
    "        # precision, recall, f1_score = calculate_metrics(groundtruth, matching_pairs)\n",
    "        # print(f\"Precision: {precision}, Recall: {recall}, F1 Score: {f1_score}\")\n",
    "print('\\n')\n",
    "for k,v in scores.items():\n",
    "    precision, recall, f1_score = v\n",
    "    print(f'{k}: Precision: {precision}, Recall: {recall}, F1 Score: {f1_score}')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "arpah",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
