{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import os\n",
    "sys.path.append(\"..\")\n",
    "\n",
    "from database.database import Database\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading a single file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "file_name = \"Cao_Clinical_data.csv\"\n",
    "file_path = \"../data/extracted-tables/Cao_Clinical_data.csv\"  \n",
    "static_path = os.path.abspath(file_path)\n",
    "\n",
    "\n",
    "db = Database()\n",
    "db.load_data(file_name, static_path)\n",
    "# db.describe_database()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading a folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Database contains the following dataframes:\n",
      "\t- Cao_circRNAForNAT.csv\n",
      "\t\t- Shape: (3979, 22)\n",
      "\t\t- Columns: Index(['Unnamed: 0', 'C3L-02610', 'C3N-01998', 'C3L-01637', 'C3L-02606',\n",
      "       'C3N-00436', 'C3L-00819', 'C3N-00518', 'C3L-00102', 'C3N-00517',\n",
      "       'C3L-00640', 'C3N-01011', 'C3L-03123', 'C3N-02768', 'C3L-02809',\n",
      "       'C3N-02589', 'C3N-03780', 'C3N-03440', 'C3N-02010', 'C3L-04072',\n",
      "       'C3N-03069', 'C3L-02899'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "                               Unnamed: 0  C3L-02610  C3N-01998  C3L-01637  \\\n",
      "0  circ_chr10_100260218_100262063_CWF19L1   0.000000   4.129283   0.000000   \n",
      "1     circ_chr10_100923975_100926019_SLF2   7.808321   8.042261   8.052351   \n",
      "2     circ_chr10_100937402_100950753_SLF2   0.000000   0.000000   0.000000   \n",
      "3    circ_chr10_101667886_101676436_FBXW4   7.152893   7.419876   7.051263   \n",
      "4    circ_chr10_101672915_101676436_FBXW4   0.000000   6.008317   0.000000   \n",
      "\n",
      "   C3L-02606  C3N-00436  C3L-00819  C3N-00518  C3L-00102  C3N-00517  ...  \\\n",
      "0   0.000000   0.000000   0.000000   0.000000   0.000000   0.000000  ...   \n",
      "1   7.922971   8.423494   8.410324   8.632341   8.003715   8.211548  ...   \n",
      "2   0.000000   0.000000   0.000000   6.275938   0.000000   0.000000  ...   \n",
      "3   7.402159   7.672496   7.465403   8.220330   7.761019   0.000000  ...   \n",
      "4   0.000000   0.000000   0.000000   5.236875   0.000000   0.000000  ...   \n",
      "\n",
      "   C3L-03123  C3N-02768  C3L-02809  C3N-02589  C3N-03780  C3N-03440  \\\n",
      "0   0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
      "1   8.154312   8.515660   8.007868   8.681484   8.243698   7.131548   \n",
      "2   0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
      "3   6.080231   8.138988   7.839645   7.523562   6.931447   5.485749   \n",
      "4   0.000000   0.000000   0.000000   6.186461   0.000000   0.000000   \n",
      "\n",
      "   C3N-02010  C3L-04072  C3N-03069  C3L-02899  \n",
      "0   0.000000   0.000000   0.000000   0.000000  \n",
      "1   8.152640   8.067811   7.928548   8.374909  \n",
      "2   0.000000   0.000000   0.000000   0.000000  \n",
      "3   7.190812   7.677015   6.211791   7.479538  \n",
      "4   0.000000   0.000000   0.000000   0.000000  \n",
      "\n",
      "[5 rows x 22 columns]\n",
      "\t- Wang_additional_annotations.csv\n",
      "\t\t- Shape: (109, 43)\n",
      "\t\t- Columns: Index(['case', 'sample_type', 'multiomic', 'nmf_consensus',\n",
      "       'nmf_cluster_membership', 'rna_wang_cancer_cell_2017',\n",
      "       'mRNA_stemness_index', 'dna_methyl', 'dna_methyl_consensus',\n",
      "       'dna_methyl_mnp', 'is_gcimp', 'mgmt_methyl_stp27_prediction',\n",
      "       'mgmt_methyl_stp27_probability', 'mgmt_methyl_clinical_test',\n",
      "       'mgmt_methyl_clinical_test_method', 'immune', 'xcell_immune_score',\n",
      "       'xcell_stroma_score', 'xcell_microenvironment_score', 'telomere',\n",
      "       'mirna', 'ancestry_prediction', 'ancestry_prediction_afr_prob',\n",
      "       'ancestry_prediction_amr_prob', 'ancestry_prediction_eas_prob',\n",
      "       'ancestry_prediction_eur_prob', 'ancestry_prediction_sas_prob',\n",
      "       'wxs_total_mutation', 'wgs_total_mutation', 'EGFR_alter_status',\n",
      "       'PTEN_alter_status', 'TP53_alter_status', 'NF1_alter_status',\n",
      "       'PDGFRA_alter_status', 'PIK3CA_alter_status', 'BRAF_alter_status',\n",
      "       'RB1_alter_status', 'ATRX_alter_status', 'IDH1_alter_status',\n",
      "       'PIK3R1_alter_status', 'TERT_alter_status', 'chr7_amp', 'chr10_del'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "        case sample_type multiomic nmf_consensus  nmf_cluster_membership  \\\n",
      "0  C3L-00104       tumor      nmf1          nmf1                   0.743   \n",
      "1  C3L-00365       tumor      nmf3          nmf3                   0.614   \n",
      "2  C3L-00674       tumor      nmf1          nmf1                   0.507   \n",
      "3  C3L-00677       tumor      nmf1          nmf1                   0.536   \n",
      "4  C3L-01040       tumor      nmf1          nmf1                   0.589   \n",
      "\n",
      "  rna_wang_cancer_cell_2017  mRNA_stemness_index dna_methyl  \\\n",
      "0                 Proneural             0.678244        dm2   \n",
      "1                 Classical             0.681122        dm4   \n",
      "2               Mesenchymal             0.744635        dm5   \n",
      "3                 Proneural             0.900896        dm5   \n",
      "4                 Classical             0.647288        dm5   \n",
      "\n",
      "   dna_methyl_consensus dna_methyl_mnp  ...  PDGFRA_alter_status  \\\n",
      "0              0.314081    GBM, RTK II  ...                  NaN   \n",
      "1              0.891149    GBM, RTK II  ...                  NaN   \n",
      "2              0.625634       GBM, MES  ...                  NaN   \n",
      "3              0.406474       GBM, MES  ...                  NaN   \n",
      "4              0.623458       GBM, MES  ...                  NaN   \n",
      "\n",
      "   PIK3CA_alter_status  BRAF_alter_status  RB1_alter_status ATRX_alter_status  \\\n",
      "0                  NaN                NaN       Splice_Site               NaN   \n",
      "1                  NaN                NaN       Frame_Shift               NaN   \n",
      "2                  NaN                NaN               NaN               NaN   \n",
      "3                  NaN                NaN       Splice_Site          Nonsense   \n",
      "4                  NaN                NaN               NaN               NaN   \n",
      "\n",
      "  IDH1_alter_status  PIK3R1_alter_status  TERT_alter_status  chr7_amp  \\\n",
      "0               NaN                  NaN           Promoter       1.0   \n",
      "1               NaN                  NaN           Promoter       1.0   \n",
      "2               NaN                  NaN           Promoter       1.0   \n",
      "3               NaN                  NaN                NaN       1.0   \n",
      "4               NaN                  NaN           Promoter       1.0   \n",
      "\n",
      "  chr10_del  \n",
      "0       1.0  \n",
      "1       1.0  \n",
      "2       1.0  \n",
      "3       1.0  \n",
      "4       1.0  \n",
      "\n",
      "[5 rows x 43 columns]\n",
      "\t- Clark_data_sample_availability.csv\n",
      "\t\t- Shape: (11, 5)\n",
      "\t\t- Columns: Index(['Table: Sample size of different data sets after excluding samples with low quality.',\n",
      "       'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  Table: Sample size of different data sets after excluding samples with low quality.  \\\n",
      "0                                                NaN                                    \n",
      "1                                    Protein/phospho                                    \n",
      "2                                       CNV/Mutation                                    \n",
      "3                                                RNA                                    \n",
      "4                                        Methylation                                    \n",
      "\n",
      "    Unnamed: 1  Unnamed: 2       Unnamed: 3     Unnamed: 4  \n",
      "0  CCRCC-tumor  CCRCC-NAT*  Non-CCRCC tumor  Non-CCRCC NAT  \n",
      "1          103          80                7              3  \n",
      "2          103          80                7              3  \n",
      "3          103          72                7              2  \n",
      "4        100**         NaN                7            NaN  \n",
      "\t- Cao_Clinical_data.csv\n",
      "\t\t- Shape: (140, 39)\n",
      "\t\t- Columns: Index(['case_id', 'tumor_included_for_the_study',\n",
      "       'normal_included_for_the_study', 'histology_diagnosis', 'age', 'sex',\n",
      "       'race', 'participant_country', 'tumor_site', 'tumor_focality',\n",
      "       'tumor_size_cm', 'tumor_necrosis', 'lymph_vascular_invasion',\n",
      "       'perineural_invasion', 'number_of_lymph_nodes_examined',\n",
      "       'number_of_lymph_nodes_positive_for_tumor',\n",
      "       'pathologic_staging_regional_lymph_nodes_pn',\n",
      "       'pathologic_staging_primary_tumor_pt',\n",
      "       'pathologic_staging_distant_metastasis_pm',\n",
      "       'clinical_staging_distant_metastasis_cm', 'residual_tumor',\n",
      "       'tumor_stage_pathological', 'additional_pathologic_findings', 'bmi',\n",
      "       'alcohol_consumption', 'tobacco_smoking_history', 'medical_condition',\n",
      "       'Neoplastic_cellularity', 'Acinar_fraction', 'Islet_fraction',\n",
      "       'Stromal_fraction', 'Non_neoplastic_duct', 'Fat_fraction',\n",
      "       'Inflammation_fraction', 'Muscle_fraction', 'follow_up_days',\n",
      "       'vital_status', 'is_this_patient_lost_to_follow_up', 'cause_of_death'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     case_id tumor_included_for_the_study normal_included_for_the_study  \\\n",
      "0  C3L-00102                          yes                           yes   \n",
      "1  C3L-00189                          yes                           yes   \n",
      "2  C3L-00277                          yes                           yes   \n",
      "3  C3L-00401                          yes                           yes   \n",
      "4  C3L-00640                          yes                           yes   \n",
      "\n",
      "  histology_diagnosis  age     sex   race participant_country tumor_site  \\\n",
      "0                PDAC   42    Male  White       United States       head   \n",
      "1                PDAC   68  Female    NaN              Canada       head   \n",
      "2                PDAC   69    Male  White               Other       tail   \n",
      "3                PDAC   62  Female    NaN              Canada       body   \n",
      "4                PDAC   59  Female  White       United States       body   \n",
      "\n",
      "  tumor_focality  ...  Islet_fraction Stromal_fraction Non_neoplastic_duct  \\\n",
      "0       Unifocal  ...               2               60                   1   \n",
      "1       Unifocal  ...           1;3;2         75;55;53               2;1;0   \n",
      "2       Unifocal  ...               0               55                   0   \n",
      "3       Unifocal  ...           2;2;3         35;30;32             3;10;10   \n",
      "4       Unifocal  ...             0;0            55;44                 0;0   \n",
      "\n",
      "  Fat_fraction Inflammation_fraction  Muscle_fraction follow_up_days  \\\n",
      "0            4                     7                0          249.0   \n",
      "1        0;6;0               7;30;25            0;0;0         1035.0   \n",
      "2            0                     5                0          610.0   \n",
      "3        0;0;0               10;13;5            0;0;0         1228.0   \n",
      "4          0;0                 10;15              0;1          594.0   \n",
      "\n",
      "  vital_status is_this_patient_lost_to_follow_up        cause_of_death  \n",
      "0     Deceased                                No  pancreatic carcinoma  \n",
      "1     Deceased                                No  pancreatic carcinoma  \n",
      "2     Deceased                               Yes  pancreatic carcinoma  \n",
      "3       Living                                No                    na  \n",
      "4       Living                                No                    na  \n",
      "\n",
      "[5 rows x 39 columns]\n",
      "\t- Cao_TMT_labeling_scheme.csv\n",
      "\t\t- Shape: (247, 4)\n",
      "\t\t- Columns: Index(['TMT_set', 'channel_name', 'case_id', 'sample_type'], dtype='object')\n",
      "\t\t- Head: \n",
      "   TMT_set channel_name    case_id sample_type\n",
      "0        1         128C  C3L-03123         NAT\n",
      "1        1         131N  C3N-03884         NAT\n",
      "2        2         128C  C3L-03630         NAT\n",
      "3        2         130C  C3L-03356         NAT\n",
      "4        2         131N  C3L-01054         NAT\n",
      "\t- Vasaikar_B-ClinicalData.csv\n",
      "\t\t- Shape: (112, 45)\n",
      "\t\t- Columns: Index(['SampleID', 'Clinical', 'Mutation', 'CNV', 'miRNA-seq', 'mRNA-seq',\n",
      "       'Label-free Proteome', 'TMT Proteome', 'TMT Proteome (Normal)',\n",
      "       'TMT Phosphoproteome', 'TMT Phosphoproteome (Normal)', 'Age', 'Gender',\n",
      "       'Mucinous', 'Subsite', 'PT', 'PN', 'Stage', 'CEA', 'Vascular_Invasion',\n",
      "       'Lymphatic_Invasion', 'Perineural_Invasion', 'Synchronous_Tumors',\n",
      "       'Polyps_History', 'Polyps_Present', 'Vital.Status', 'Tumor.Status',\n",
      "       'POLE_mutation', 'MSH2_mutation', 'MSH6_mutation', 'PMS2_mutation',\n",
      "       'KRAS_mutation', 'NRAS_mutation', 'BRAF_mutation', 'mutation_rate',\n",
      "       'Hypermutation_status', 'MSI_PCR_Result', 'MSmutectFishercall', 'CMS',\n",
      "       'ProS', 'UMS', 'StromalScore', 'ImmuneScore', 'ESTIMATEScore',\n",
      "       'TumorPurity'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  SampleID Clinical Mutation  CNV miRNA-seq mRNA-seq Label-free Proteome  \\\n",
      "0     Type      BIN      BIN  BIN       BIN      BIN                 BIN   \n",
      "1  01CO001        1        1    1         1        1                   1   \n",
      "2  01CO005        1        1    1         1        1                   1   \n",
      "3  01CO006        1        1    1         1        1                   1   \n",
      "4  01CO008        1        1    1         1        1                   1   \n",
      "\n",
      "  TMT Proteome TMT Proteome (Normal) TMT Phosphoproteome  ...  \\\n",
      "0          BIN                   BIN                 BIN  ...   \n",
      "1          NaN                     1                 NaN  ...   \n",
      "2            1                     1                   1  ...   \n",
      "3            1                     1                   1  ...   \n",
      "4            1                     1                   1  ...   \n",
      "\n",
      "  Hypermutation_status MSI_PCR_Result MSmutectFishercall   CMS ProS  UMS  \\\n",
      "0                  BIN            BIN                BIN   CAT  CAT  CAT   \n",
      "1             nonhyper            NaN                MSS  CMS3    C  NaN   \n",
      "2             nonhyper            NaN                MSS  CMS2    E  CIN   \n",
      "3             nonhyper            NaN                MSS  CMS3    C  NaN   \n",
      "4             nonhyper            NaN                MSS  CMS3    E  CIN   \n",
      "\n",
      "   StromalScore   ImmuneScore ESTIMATEScore  TumorPurity  \n",
      "0           CON           CON           CON          CON  \n",
      "1   -368.880599   573.3158586   204.4352597  0.805074113  \n",
      "2  -1102.297876   -427.026762  -1529.324638  0.928479028  \n",
      "3    466.174616   1542.023347   2008.197963  0.621792179  \n",
      "4   -1122.84127  -54.71768692  -1177.558957  0.908073573  \n",
      "\n",
      "[5 rows x 45 columns]\n",
      "\t- Wang_survival_analysis.csv\n",
      "\t\t- Shape: (47, 8)\n",
      "\t\t- Columns: Index(['Stratification group', 'Group value', 'n', 'events', 'median',\n",
      "       '0.95LCL', '0.95UCL', 'log-rank p-value'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "                    Stratification group Group value   n  events  median  \\\n",
      "0  DNA methylation subtype\\n(dna_methyl)         dm1  15      12    1.04   \n",
      "1                                    NaN         dm2  18      10    1.10   \n",
      "2                                    NaN         dm3  21      14    1.07   \n",
      "3                                    NaN         dm4  13       8    1.78   \n",
      "4                                    NaN         dm5  15       8    2.20   \n",
      "\n",
      "   0.95LCL  0.95UCL  log-rank p-value  \n",
      "0    0.805      NaN              0.55  \n",
      "1    0.797      NaN               NaN  \n",
      "2    0.931      NaN               NaN  \n",
      "3    0.882      NaN               NaN  \n",
      "4    1.309      NaN               NaN  \n",
      "\t- McDermott_OneYearFollowUp.csv\n",
      "\t\t- Shape: (256, 71)\n",
      "\t\t- Columns: Index(['CPTAC Case ID', 'Tumor analyzed', 'Normal FT analyzed', 'CRF Name',\n",
      "       'Date of Last Contact (Do not answer if patient is deceased)',\n",
      "       'Vital Status (at time of last contact)', 'Date of Death',\n",
      "       'Tumor Status at Time of Last Contact or Death',\n",
      "       'Was a Review of the Initial Pathological Findings Done?',\n",
      "       'Was the Pathology Review consistent with the Diagnosis?',\n",
      "       'Adjuvant (Post-Operative) Radiation Therapy',\n",
      "       'Adjuvant (Post-Operative) Pharmaceutical Therapy',\n",
      "       'Adjuvant (Post-Operative) Immunotherapy',\n",
      "       'Adjuvant (Post-Operative) Hormone Therapy',\n",
      "       'Adjuvant (Post-Operative) Targeted Molecular Therapy',\n",
      "       'Measure of Success of Outcome at the Completion of Initial First Course Treatment (surgery and adjuvant therapies)',\n",
      "       ' New Tumor Event After Initial Treatment?', 'Type of New Tumor Event',\n",
      "       'Anatomic Site of New Tumor Event',\n",
      "       'Other Site of New Tumor Event or Lymph Node Location',\n",
      "       'Date of New Tumor Event', 'Method Of Diagnosis of New Tumor Event',\n",
      "       'Other Method Of Diagnosis of New Tumor Event',\n",
      "       'Additional Surgery for New Tumor Event',\n",
      "       'Date of Additional Surgery for New Tumor Event',\n",
      "       'Additional Chemotherapy Treatment of New Tumor Event',\n",
      "       'Additional Immunotherapy Treatment of New Tumor Event',\n",
      "       'Additional Hormone Therapy Treatment of New Tumor Event',\n",
      "       'Additional Targeted Molecular Therapy Treatment of New Tumor Event',\n",
      "       'Radiation Type', 'Location of Radiation Treatment',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis to the Date Radiation Therapy Started',\n",
      "       'Total Dose', 'Units', 'Total Number of Fractions',\n",
      "       'Radiation Treatment Ongoing',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis to the Date Radiation Therapy Ended',\n",
      "       'Measure of Best Response of Radiation Treatment',\n",
      "       'Was Patient Treated on a Clinical Trial?',\n",
      "       'Drug Name (Brand or Generic)', 'Clinical Trial Drug Classification',\n",
      "       'Pharmaceutical Type',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis to Date of Therapy Start',\n",
      "       'Therapy Ongoing',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis to Date of Therapy End',\n",
      "       'Measure of Best Response of Pharmaceutical Treatment',\n",
      "       'What Type of Malignancy was This?', 'Primary Site of Disease',\n",
      "       'Laterality of the Disease', 'Histological Type',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis of the Tumor Submitted for CPTAC to the Date of Initial Diagnosis of Other Malignancy',\n",
      "       'Did the patient have surgery for this malignancy?', 'Type of Surgery',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis of the Tumor Submitted for CPTAC to the Date of Surgical Resection for this Other Malignancy',\n",
      "       'Did the patient receive pharmaceutical therapy for this malignancy?',\n",
      "       'Extent of Pharmaceutical Therapy', 'Drug Name(s) (Brand or Generic)',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis of the Tumor Submitted for CPTAC to Date Pharmaceutical Therapy Started for this Other Malignancy',\n",
      "       'Did the patient receive radiation therapy for this malignancy?',\n",
      "       'Extent of Radiation Therapy',\n",
      "       'If the patient received locoregional radiation, was the radiation therapy received in the same field as the tumor submitted for CPTAC?',\n",
      "       'Number of Days from Date of Initial Pathologic Diagnosis of the Tumor Submitted to CPTAC to Date Radiation Therapy Started for this Other Malignancy',\n",
      "       'Was the patient staged using FIGO?',\n",
      "       'FIGO Staging System (Gynecologic Tumors Only)', 'FIGO Stage',\n",
      "       ' Was the patient staged using AJCC?', 'AJCC Cancer Staging Edition',\n",
      "       'Pathologic Spread: Primary Tumor (pT)',\n",
      "       'Pathologic Spread: Lymph Nodes (pN)', 'Distant Metastases (M)',\n",
      "       'AJCC Tumor Stage'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  CPTAC Case ID Tumor analyzed Normal FT analyzed  \\\n",
      "0       01OV007       C01OV007           N01OV007   \n",
      "1       01OV007            NaN                NaN   \n",
      "2       01OV007            NaN                NaN   \n",
      "3       01OV008            NaN           N01OV008   \n",
      "4       01OV008            NaN                NaN   \n",
      "\n",
      "                           CRF Name  \\\n",
      "0                          One Year   \n",
      "1  Pharmaceutical Supplemental form   \n",
      "2  Pharmaceutical Supplemental form   \n",
      "3                          One Year   \n",
      "4  Pharmaceutical Supplemental form   \n",
      "\n",
      "  Date of Last Contact (Do not answer if patient is deceased)  \\\n",
      "0                                         2015-03-10            \n",
      "1                                                NaN            \n",
      "2                                                NaN            \n",
      "3                                         2015-03-10            \n",
      "4                                                NaN            \n",
      "\n",
      "  Vital Status (at time of last contact) Date of Death  \\\n",
      "0                                 Living           NaN   \n",
      "1                                    NaN           NaN   \n",
      "2                                    NaN           NaN   \n",
      "3                                 Living           NaN   \n",
      "4                                    NaN           NaN   \n",
      "\n",
      "  Tumor Status at Time of Last Contact or Death  \\\n",
      "0                                    Tumor free   \n",
      "1                                           NaN   \n",
      "2                                           NaN   \n",
      "3                                    Tumor free   \n",
      "4                                           NaN   \n",
      "\n",
      "  Was a Review of the Initial Pathological Findings Done?  \\\n",
      "0                                                Yes        \n",
      "1                                                NaN        \n",
      "2                                                NaN        \n",
      "3                                                Yes        \n",
      "4                                                NaN        \n",
      "\n",
      "  Was the Pathology Review consistent with the Diagnosis?  ...  \\\n",
      "0                                                Yes       ...   \n",
      "1                                                NaN       ...   \n",
      "2                                                NaN       ...   \n",
      "3                                                Yes       ...   \n",
      "4                                                NaN       ...   \n",
      "\n",
      "  Number of Days from Date of Initial Pathologic Diagnosis of the Tumor Submitted to CPTAC to Date Radiation Therapy Started for this Other Malignancy  \\\n",
      "0                                                NaN                                                                                                     \n",
      "1                                                NaN                                                                                                     \n",
      "2                                                NaN                                                                                                     \n",
      "3                                                NaN                                                                                                     \n",
      "4                                                NaN                                                                                                     \n",
      "\n",
      "  Was the patient staged using FIGO?  \\\n",
      "0                                NaN   \n",
      "1                                NaN   \n",
      "2                                NaN   \n",
      "3                                NaN   \n",
      "4                                NaN   \n",
      "\n",
      "  FIGO Staging System (Gynecologic Tumors Only) FIGO Stage  \\\n",
      "0                                           NaN        NaN   \n",
      "1                                           NaN        NaN   \n",
      "2                                           NaN        NaN   \n",
      "3                                           NaN        NaN   \n",
      "4                                           NaN        NaN   \n",
      "\n",
      "   Was the patient staged using AJCC? AJCC Cancer Staging Edition  \\\n",
      "0                                 NaN                         NaN   \n",
      "1                                 NaN                         NaN   \n",
      "2                                 NaN                         NaN   \n",
      "3                                 NaN                         NaN   \n",
      "4                                 NaN                         NaN   \n",
      "\n",
      "  Pathologic Spread: Primary Tumor (pT) Pathologic Spread: Lymph Nodes (pN)  \\\n",
      "0                                   NaN                                 NaN   \n",
      "1                                   NaN                                 NaN   \n",
      "2                                   NaN                                 NaN   \n",
      "3                                   NaN                                 NaN   \n",
      "4                                   NaN                                 NaN   \n",
      "\n",
      "  Distant Metastases (M) AJCC Tumor Stage  \n",
      "0                    NaN              NaN  \n",
      "1                    NaN              NaN  \n",
      "2                    NaN              NaN  \n",
      "3                    NaN              NaN  \n",
      "4                    NaN              NaN  \n",
      "\n",
      "[5 rows x 71 columns]\n",
      "\t- Wang_cbttc_sample_info.csv\n",
      "\t\t- Shape: (41, 9)\n",
      "\t\t- Columns: Index(['aliquot_id', 'sample_id', 'original_aliquot_id', 'batch', 'plex',\n",
      "       'tmt_channel', 'research_id', 'kids_first_id',\n",
      "       'clinical_status_at_collection_event'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "                    aliquot_id   sample_id       original_aliquot_id  batch  \\\n",
      "0       7316-288-TISS [894482]    7316-288    7316-288-TISS [894482]      1   \n",
      "1        7316-89-TISS [894435]     7316-89     7316-89-TISS [894435]      1   \n",
      "2      7316-2176-TISS [894450]   7316-2176   7316-2176-TISS [894450]      1   \n",
      "3     7316UP-904-TISS [900638]  7316UP-904  7316UP-904-TISS [900638]      1   \n",
      "4  7316-3521-TISS [894433]...7   7316-3521   7316-3521-TISS [894433]      1   \n",
      "\n",
      "   plex tmt_channel research_id kids_first_id  \\\n",
      "0     1        127N      C54735   PT_84Q11Z5X   \n",
      "1     1        128N      C25461   PT_N32BYY8A   \n",
      "2     1        129N     C624225   PT_VAJN5QP8   \n",
      "3     1        129C     C326688           NaN   \n",
      "4     1        130N     C944025   PT_63F63QY0   \n",
      "\n",
      "  clinical_status_at_collection_event  \n",
      "0                               Alive  \n",
      "1                               Alive  \n",
      "2                               Alive  \n",
      "3                               Alive  \n",
      "4                               Alive  \n",
      "\t- Krug_D)_QC_mutation_calls.csv\n",
      "\t\t- Shape: (29673, 15)\n",
      "\t\t- Columns: Index(['Hugo_Symbol', 'Chromosome', 'Start_Position', 'End_position',\n",
      "       'Variant_Classification', 'Variant_Type', 'Tumor_Sample_Barcode',\n",
      "       'Matched_Norm_Sample_Barcode', 'n_ref_count', 't_ref_count',\n",
      "       't_alt_count', 'n_alt_count', 't_depth', 'n_depth', 'Passed_Filters'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  Hugo_Symbol  Chromosome  Start_Position  End_position  \\\n",
      "0        A1CF          10        52580407      52580407   \n",
      "1        A1CF          10        52573790      52573790   \n",
      "2        A1CF          10        52595869      52595869   \n",
      "3        A1CF          10        52610433      52610433   \n",
      "4         A2M          12         9256906       9256906   \n",
      "\n",
      "  Variant_Classification Variant_Type Tumor_Sample_Barcode  \\\n",
      "0      Missense_Mutation          SNP               106_DM   \n",
      "1      Missense_Mutation          SNP               329_DM   \n",
      "2      Missense_Mutation          SNP               329_DM   \n",
      "3                 Intron          SNP               329_DM   \n",
      "4      Missense_Mutation          SNP               132_DM   \n",
      "\n",
      "  Matched_Norm_Sample_Barcode  n_ref_count  t_ref_count  t_alt_count  \\\n",
      "0                      106_DN         73.0           58            4   \n",
      "1                      329_DN         80.0           96           39   \n",
      "2                      329_DN         89.0          105           18   \n",
      "3                      329_DN         75.0          104           15   \n",
      "4                      132_DN        197.0          105           38   \n",
      "\n",
      "   n_alt_count  t_depth  n_depth                           Passed_Filters  \n",
      "0          0.0     62.0     73.0  oxogOBF,ffpeOBF,ICE_PON,blat,TCGA_8334,  \n",
      "1          1.0    135.0     81.0  oxogOBF,ffpeOBF,ICE_PON,blat,TCGA_8334,  \n",
      "2          0.0    123.0     89.0  oxogOBF,ffpeOBF,ICE_PON,blat,TCGA_8334,  \n",
      "3          0.0    119.0     75.0  oxogOBF,ffpeOBF,ICE_PON,blat,TCGA_8334,  \n",
      "4          0.0    143.0    197.0  oxogOBF,ffpeOBF,ICE_PON,blat,TCGA_8334,  \n",
      "\t- Krug_A)_Metadata.csv\n",
      "\t\t- Shape: (122, 54)\n",
      "\t\t- Columns: Index(['Sample.ID', 'Sample.IDs', 'TMT.Plex', 'TMT.Channel', 'Tumor.Stage',\n",
      "       'Ischemia.Time.in.Minutes', 'PAM50', 'NMF.Cluster',\n",
      "       'NMF.Cluster.Membership.Score', 'Age.in.Month', 'Gender', 'Ethnicity',\n",
      "       'ER.Updated.Clinical.Status', 'PR.Clinical.Status',\n",
      "       'ERBB2.Updated.Clinical.Status', 'TNBC.Updated.Clinical.Status',\n",
      "       'ERBB2.Proteogenomic.Status', 'TOP2A.Proteogenomic.Status',\n",
      "       'ERBB2.Gene.Amplified', 'TOP2A.Gene.Amplified', 'ESTIMATE.TumorPurity',\n",
      "       'CIBERSORT.Absolute.Score', 'ESTIMATE.Immune.Score',\n",
      "       'xCell.Immune.Score', 'ESTIMATE.Stromal.Score', 'xCell.Stromal.Score',\n",
      "       'CD3.TILS.Status', 'CD3.TILS.Counts',\n",
      "       'Number.of.non.synonymous.Mutations', 'APOBEC.Signature',\n",
      "       'Chromosome.INstability.index.CIN.', 'Stemness.Score',\n",
      "       'TP53.Mutation.Type', 'PIK3CA.Mutation.Type', 'PTEN.Mutation.Type',\n",
      "       'MAP3K1.Mutation.Type', 'AKT1.Mutation.Type', 'GATA3.Mutation.Type',\n",
      "       'CBFB.Mutation.Type', 'KMT2C.Mutation.Type', 'SF3B1.Mutation.Type',\n",
      "       'ARID1A.Mutation.Type', 'MLLT4.Mutation.Type', 'TP53.Mutation.Status',\n",
      "       'PIK3CA.Mutation.Status', 'PTEN.Mutation.Status',\n",
      "       'MAP3K1.Mutation.Status', 'AKT1.Mutation.Status',\n",
      "       'GATA3.Mutation.Status', 'CBFB.Mutation.Status',\n",
      "       'KMT2C.Mutation.Status', 'SF3B1.Mutation.Status',\n",
      "       'ARID1A.Mutation.Status', 'MLLT4.Mutation.Status'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "   Sample.ID Sample.IDs TMT.Plex TMT.Channel Tumor.Stage  \\\n",
      "0  CPT000814  CPT000814       13        127C   Stage IIA   \n",
      "1  CPT001846  CPT001846       12        128C   Stage III   \n",
      "2   X01BR001   X01BR001        2        129N   Stage IIB   \n",
      "3   X01BR008   X01BR008       16        127C     unknown   \n",
      "4   X01BR009   X01BR009       16        127N     unknown   \n",
      "\n",
      "   Ischemia.Time.in.Minutes  PAM50 NMF.Cluster  NMF.Cluster.Membership.Score  \\\n",
      "0                       NaN  Basal     Basal-I                         1.000   \n",
      "1                       NaN  Basal     Basal-I                         0.672   \n",
      "2                       0.0  Basal     Basal-I                         0.782   \n",
      "3                       NaN  Basal     Basal-I                         0.958   \n",
      "4                       NaN  Basal     Basal-I                         0.825   \n",
      "\n",
      "  Age.in.Month  ... PIK3CA.Mutation.Status PTEN.Mutation.Status  \\\n",
      "0      unknown  ...                      0                    0   \n",
      "1      unknown  ...                      1                    0   \n",
      "2          660  ...                      0                    0   \n",
      "3      unknown  ...                      0                    0   \n",
      "4      unknown  ...                      0                    0   \n",
      "\n",
      "  MAP3K1.Mutation.Status AKT1.Mutation.Status GATA3.Mutation.Status  \\\n",
      "0                      0                    0                     0   \n",
      "1                      0                    0                     0   \n",
      "2                      0                    0                     0   \n",
      "3                      0                    0                     0   \n",
      "4                      0                    0                     0   \n",
      "\n",
      "  CBFB.Mutation.Status KMT2C.Mutation.Status SF3B1.Mutation.Status  \\\n",
      "0                    0                     0                     0   \n",
      "1                    0                     0                     0   \n",
      "2                    0                     0                     0   \n",
      "3                    0                     0                     0   \n",
      "4                    0                     0                     0   \n",
      "\n",
      "   ARID1A.Mutation.Status  MLLT4.Mutation.Status  \n",
      "0                       0                      0  \n",
      "1                       0                      0  \n",
      "2                       0                      0  \n",
      "3                       0                      0  \n",
      "4                       0                      0  \n",
      "\n",
      "[5 rows x 54 columns]\n",
      "\t- Cao_Molecular_phenotype_data.csv\n",
      "\t\t- Shape: (140, 50)\n",
      "\t\t- Columns: Index(['case_id', 'immune_deconv', 'epithelial_cancer_deconv',\n",
      "       'stromal_deconv', 'mature_exocrine_endocrine_deconv',\n",
      "       'neoplastic_cellularity_histology_estimate',\n",
      "       'acinar_histology_estimate', 'islet_histology_estimate',\n",
      "       'stromal_histology_estimate', 'Non_neoplastic_duct_histology_estimate',\n",
      "       'Fat_histology_estimate', 'inflammation_histology_estimate',\n",
      "       'Muscle_histology_estimate',\n",
      "       'necrosis_(%OF_TUMOR_WITH_NECROSIS)_histology_estimate', 'Bailey',\n",
      "       'Collisson', 'Moffitt', 'StromalScore_ESTIMATE', 'ImmuneScore_ESTIMATE',\n",
      "       'mutation_count', 'KRAS_VAF', 'chr9p_del', 'chr18q_del', 'chr17p_del',\n",
      "       'chrIdx', 'MSI_status', 'Androgen_pathway', 'EGFR_pathway',\n",
      "       'Estrogen_pathway', 'Hypoxia_pathway', 'JAK-STAT_pathway',\n",
      "       'MAPK_pathway', 'NFkB_pathway', 'p53_pathway', 'PI3K_pathway',\n",
      "       'TGFb_pathway', 'TNFa_pathway', 'Trail_pathway', 'VEGF_pathway',\n",
      "       'WNT_pathway', 'T cells_MCPCounter', 'CD8 T cells_MCPCounter',\n",
      "       'Cytotoxic lymphocytes_MCPCounter', 'B lineage_MCPCounter',\n",
      "       'NK cells_MCPCounter', 'Monocytic lineage_MCPCounter',\n",
      "       'Myeloid dendritic cells_MCPCounter', 'Neutrophils_MCPCounter',\n",
      "       'Endothelial cells_MCPCounter', 'Fibroblasts_MCPCounter'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     case_id  immune_deconv  epithelial_cancer_deconv  stromal_deconv  \\\n",
      "0  C3L-03394            NaN                       NaN             NaN   \n",
      "1  C3N-03428            NaN                       NaN             NaN   \n",
      "2  C3L-02112       0.417512                  0.073393        0.236795   \n",
      "3  C3N-01719            NaN                       NaN             NaN   \n",
      "4  C3N-03670       0.412693                  0.087876        0.294358   \n",
      "\n",
      "   mature_exocrine_endocrine_deconv  \\\n",
      "0                               NaN   \n",
      "1                               NaN   \n",
      "2                          0.272300   \n",
      "3                               NaN   \n",
      "4                          0.205074   \n",
      "\n",
      "   neoplastic_cellularity_histology_estimate  acinar_histology_estimate  \\\n",
      "0                                   0.095238                   0.095238   \n",
      "1                                   0.125000                   0.000000   \n",
      "2                                   0.100000                   0.250000   \n",
      "3                                   0.200000                   0.000000   \n",
      "4                                   0.117647                   0.098039   \n",
      "\n",
      "   islet_histology_estimate  stromal_histology_estimate  \\\n",
      "0                  0.009524                    0.647619   \n",
      "1                  0.000000                    0.685000   \n",
      "2                  0.020000                    0.515000   \n",
      "3                  0.000000                    0.400000   \n",
      "4                  0.009804                    0.637255   \n",
      "\n",
      "   Non_neoplastic_duct_histology_estimate  ...  T cells_MCPCounter  \\\n",
      "0                                0.009524  ...            8.178878   \n",
      "1                                0.015000  ...            9.107655   \n",
      "2                                0.015000  ...            7.362932   \n",
      "3                                0.000000  ...            7.836231   \n",
      "4                                0.009804  ...            7.816005   \n",
      "\n",
      "   CD8 T cells_MCPCounter  Cytotoxic lymphocytes_MCPCounter  \\\n",
      "0                7.705238                          7.445063   \n",
      "1                8.859476                          7.852227   \n",
      "2                7.741258                          6.702642   \n",
      "3                7.780979                          7.351599   \n",
      "4                7.675142                          7.076496   \n",
      "\n",
      "   B lineage_MCPCounter NK cells_MCPCounter Monocytic lineage_MCPCounter  \\\n",
      "0             10.104551            2.422822                    10.251721   \n",
      "1             11.744701            3.097248                    10.812730   \n",
      "2              9.097383            2.167536                    10.212629   \n",
      "3              8.174878            3.120686                    10.987916   \n",
      "4              8.691956            2.564830                    10.492650   \n",
      "\n",
      "  Myeloid dendritic cells_MCPCounter  Neutrophils_MCPCounter  \\\n",
      "0                           7.845322                7.902072   \n",
      "1                           8.775130                6.851306   \n",
      "2                           8.140795                8.059085   \n",
      "3                           6.296944                8.376835   \n",
      "4                           6.970909                8.283781   \n",
      "\n",
      "   Endothelial cells_MCPCounter  Fibroblasts_MCPCounter  \n",
      "0                      9.258399               14.181165  \n",
      "1                      8.931949               13.529618  \n",
      "2                      9.114213               13.967631  \n",
      "3                      9.589114               15.431378  \n",
      "4                      9.278416               14.252854  \n",
      "\n",
      "[5 rows x 50 columns]\n",
      "\t- Gillette_Annotions_S1A.csv\n",
      "\t\t- Shape: (225, 67)\n",
      "\t\t- Columns: Index(['Sample.ID', 'Participant', 'Experiment', 'Channel', 'Type', 'Aliquot',\n",
      "       'Smoking.Status', 'Stage', 'Region.of.Origin', 'Country.of.Origin',\n",
      "       'Age', 'Gender', 'Ethnicity', 'Height.cm', 'Weight.kg', 'BMI',\n",
      "       'Cigarettes.per.Day', 'Pack.Years.Smoked', 'Smoking.History',\n",
      "       'Secondhand.Smoke', 'Smoking.Score.WGS',\n",
      "       'Smoking.Signature.Fraction.WGS',\n",
      "       'Dominant.Signature.WGS.notSmoking.50perc',\n",
      "       'Dominant.Signature.Fraction.WGS.notSmoking',\n",
      "       'DNP.GG.to.TT.or.CC.to.AA.Count.WGS', 'NMF.consensus',\n",
      "       'NMF.cluster.membership', 'mRNA.Expression.Subtype.TCGA',\n",
      "       'mRNA.stemness.index', 'CIMP.status', 'Tumor.Purity.byESTIMATE.RNAseq',\n",
      "       'TSNet Purity', 'ESTIMATEScore', 'ESTIMATE ImmuneScore',\n",
      "       'ESTIMATE StromalScore', 'QC.status', 'TP53.mutation', 'KRAS.mutation',\n",
      "       'STK11.mutation', 'EGFR.mutation', 'KEAP1.mutation', 'RB1.mutation',\n",
      "       'BRAF.mutation', 'ARAF.mutation', 'ERBB2.mutation',\n",
      "       'TP53.mutation.status', 'KRAS.mutation.status', 'STK11.mutation.status',\n",
      "       'EGFR.mutation.status', 'KEAP1.mutation.status', 'RB1.mutation.status',\n",
      "       'BRAF.mutation.status', 'ARAF.mutation.status', 'ERBB2.mutation.status',\n",
      "       'Total.Mutation.Count.WGS', 'Mutation.Count.ExcludingINDELs.WGS',\n",
      "       'Total.DNP.Count.WGS', 'Number.somatic.mutations',\n",
      "       'Mutation.Signature.Activity.W1.COSMIC5',\n",
      "       'Mutation.Signature.Activity.W2.COSMIC4',\n",
      "       'Mutation.Signature.Activity.W3.COSMIC2', 'ALK.fusion', 'ROS1.fusion',\n",
      "       'RET.fusion', 'Putative.driver.mutation', 'Methylation.QC',\n",
      "       'Dominant.Histological.Subtype'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     Sample.ID Participant  Experiment Channel   Type        Aliquot  \\\n",
      "0    C3N.01799   C3N-01799           1     126  Tumor  CPT0089340003   \n",
      "1  C3N.01799.N   C3N-01799           1    127N    NAT  CPT0089360003   \n",
      "2    C3L.01890   C3L-01890           1    127C  Tumor  CPT0092930003   \n",
      "3  C3L.01890.N   C3L-01890           1    128N    NAT  CPT0092940003   \n",
      "4    C3N.00572   C3N-00572           1    128C  Tumor  CPT0015410003   \n",
      "\n",
      "  Smoking.Status Stage Region.of.Origin Country.of.Origin  ...  \\\n",
      "0         smoker    2A          western               usa  ...   \n",
      "1         smoker   NaN          western               usa  ...   \n",
      "2         smoker    1A          western               usa  ...   \n",
      "3         smoker   NaN          western               usa  ...   \n",
      "4     non-smoker    1A            asian           vietnam  ...   \n",
      "\n",
      "   Number.somatic.mutations Mutation.Signature.Activity.W1.COSMIC5  \\\n",
      "0                      40.0                              18.752534   \n",
      "1                       1.0                                    NaN   \n",
      "2                       NaN                                    NaN   \n",
      "3                       NaN                                    NaN   \n",
      "4                      19.0                              10.000456   \n",
      "\n",
      "  Mutation.Signature.Activity.W2.COSMIC4  \\\n",
      "0                               4.651348   \n",
      "1                                    NaN   \n",
      "2                                    NaN   \n",
      "3                                    NaN   \n",
      "4                               5.567000   \n",
      "\n",
      "   Mutation.Signature.Activity.W3.COSMIC2  ALK.fusion  ROS1.fusion  \\\n",
      "0                               26.924018           0            0   \n",
      "1                                     NaN           0            0   \n",
      "2                                     NaN           0            0   \n",
      "3                                     NaN           0            0   \n",
      "4                                1.959127           1            0   \n",
      "\n",
      "   RET.fusion  Putative.driver.mutation Methylation.QC  \\\n",
      "0           0                       NaN           pass   \n",
      "1           0                       NaN           pass   \n",
      "2           0                       NaN           pass   \n",
      "3           0                       NaN           pass   \n",
      "4           0                ALK-Fusion           pass   \n",
      "\n",
      "  Dominant.Histological.Subtype  \n",
      "0                        acinar  \n",
      "1                           NaN  \n",
      "2                        acinar  \n",
      "3                           NaN  \n",
      "4                micropapillary  \n",
      "\n",
      "[5 rows x 67 columns]\n",
      "\t- Cao_circRNAforTumor.csv\n",
      "\t\t- Shape: (3979, 141)\n",
      "\t\t- Columns: Index(['Unnamed: 0', 'C3L-03394', 'C3N-03428', 'C3L-02112', 'C3N-01719',\n",
      "       'C3N-03670', 'C3N-02998', 'C3N-03211', 'C3L-01052', 'C3N-00512',\n",
      "       ...\n",
      "       'C3L-01031', 'C3N-01167', 'C3L-01971', 'C3L-02604', 'C3L-00277',\n",
      "       'C3L-01328', 'C3L-03639', 'C3N-04282', 'C3L-04072', 'C3L-02613'],\n",
      "      dtype='object', length=141)\n",
      "\t\t- Head: \n",
      "                               Unnamed: 0  C3L-03394  C3N-03428  C3L-02112  \\\n",
      "0  circ_chr10_100260218_100262063_CWF19L1   0.000000   4.332708   0.000000   \n",
      "1     circ_chr10_100923975_100926019_SLF2   0.000000   8.219895   8.214708   \n",
      "2     circ_chr10_100937402_100950753_SLF2   0.000000   0.000000   0.000000   \n",
      "3    circ_chr10_101667886_101676436_FBXW4   7.111344   7.941752   7.379032   \n",
      "4    circ_chr10_101672915_101676436_FBXW4   0.000000   0.000000   5.903761   \n",
      "\n",
      "   C3N-01719  C3N-03670  C3N-02998  C3N-03211  C3L-01052  C3N-00512  ...  \\\n",
      "0   0.000000   0.000000   0.000000   0.000000   0.000000   3.068671  ...   \n",
      "1   7.453024   7.927245   7.848435   0.000000   8.334497   7.767721  ...   \n",
      "2   0.000000   0.000000   0.000000   0.000000   0.000000   0.000000  ...   \n",
      "3   6.715070   7.124121   7.485185   7.007532   8.111083   7.697732  ...   \n",
      "4   0.000000   0.000000   0.000000   0.000000   0.000000   5.521679  ...   \n",
      "\n",
      "   C3L-01031  C3N-01167  C3L-01971  C3L-02604  C3L-00277  C3L-01328  \\\n",
      "0   0.000000   0.000000   0.000000     0.0000   0.000000   0.000000   \n",
      "1   7.769309   7.702381   7.836366     0.0000   8.471635   7.441533   \n",
      "2   0.000000   0.000000   0.000000     0.0000   0.000000   0.000000   \n",
      "3   7.181301   7.709291   7.370862     6.4266   7.142720   6.873075   \n",
      "4   5.275380   0.000000   0.000000     0.0000   0.000000   0.000000   \n",
      "\n",
      "   C3L-03639  C3N-04282  C3L-04072  C3L-02613  \n",
      "0    0.00000   0.000000        0.0   0.000000  \n",
      "1    7.10507   7.846493        0.0   7.443358  \n",
      "2    0.00000   0.000000        0.0   0.000000  \n",
      "3    0.00000   6.851874        0.0   6.520737  \n",
      "4    0.00000   0.000000        0.0   0.000000  \n",
      "\n",
      "[5 rows x 141 columns]\n",
      "\t- McDermott_BaselineClinicalData.csv\n",
      "\t\t- Shape: (93, 22)\n",
      "\t\t- Columns: Index(['CPTAC Case ID', 'Tumor analyzed', 'Normal FT analyzed',\n",
      "       'Age in Months at Time of Tissue Procurement \\t', 'Gender', 'Race',\n",
      "       'Ethnicity', 'Jewish Religion/Heritage', 'Site of Disease',\n",
      "       'Anatomic site of Tumor Specimen',\n",
      "       'Anatomic Lateral Position of Tumor Specimen', 'Histological Subtype',\n",
      "       'Method of Initial Pathologic Diagnosis',\n",
      "       'Tumor Stage (Pathological) Ovary FIGO Staging System', 'Tumor Grade',\n",
      "       'Tumor Residual Disease (for the max diameter of the largest remaining tumor nodule)',\n",
      "       'History of Other Malignancy?', 'History of Systemic Chemotherapy?',\n",
      "       'History or Radiation Therapy for Any Prior Malignancy That Involves Treatment to the Abdomen or Pelvis?',\n",
      "       'History of Hormonal Therapy for Cancer within the Last Five Years?',\n",
      "       'History of Neo-Adjuvant Treatment to Tumor Specimen Submitted for CPTAC',\n",
      "       'Ischemia Time for Tumor Tissue'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  CPTAC Case ID Tumor analyzed Normal FT analyzed  \\\n",
      "0       01OV007       C01OV007           N01OV007   \n",
      "1       01OV008            NaN           N01OV008   \n",
      "2       01OV010            NaN           N01OV010   \n",
      "3       01OV013            NaN           N01OV013   \n",
      "4       01OV017       C01OV017           N01OV017   \n",
      "\n",
      "   Age in Months at Time of Tissue Procurement \\t  Gender  \\\n",
      "0                                           820.0  Female   \n",
      "1                                           665.0  Female   \n",
      "2                                           625.0  Female   \n",
      "3                                           828.0  Female   \n",
      "4                                           672.0  Female   \n",
      "\n",
      "                        Race               Ethnicity Jewish Religion/Heritage  \\\n",
      "0                      White  Not Hispanic or Latino                  Unknown   \n",
      "1                      White  Not Hispanic or Latino               Not Jewish   \n",
      "2  Black or African American                 Unknown                  Unknown   \n",
      "3                      White  Not Hispanic or Latino               Not Jewish   \n",
      "4                      White  Not Hispanic or Latino               Not Jewish   \n",
      "\n",
      "  Site of Disease Anatomic site of Tumor Specimen  ...  \\\n",
      "0      Peritoneum                         Omentum  ...   \n",
      "1  Fallopian tube                           Ovary  ...   \n",
      "2  Fallopian tube                           Ovary  ...   \n",
      "3           Ovary                           Ovary  ...   \n",
      "4           Ovary                           Ovary  ...   \n",
      "\n",
      "  Method of Initial Pathologic Diagnosis  \\\n",
      "0                        Tumor resection   \n",
      "1                        Tumor resection   \n",
      "2                        Tumor resection   \n",
      "3                        Tumor resection   \n",
      "4                        Tumor resection   \n",
      "\n",
      "  Tumor Stage (Pathological) Ovary FIGO Staging System Tumor Grade  \\\n",
      "0                                                 IV            G3   \n",
      "1                                               IIIC            G3   \n",
      "2                                               IIIC            G3   \n",
      "3                                               IIIC            G3   \n",
      "4                                               IIIC            G3   \n",
      "\n",
      "  Tumor Residual Disease (for the max diameter of the largest remaining tumor nodule)  \\\n",
      "0                                                NaN                                    \n",
      "1                                                NaN                                    \n",
      "2                                            1-10 mm                                    \n",
      "3                                                NaN                                    \n",
      "4                                                NaN                                    \n",
      "\n",
      "  History of Other Malignancy? History of Systemic Chemotherapy?  \\\n",
      "0                           No                                No   \n",
      "1                           No                                No   \n",
      "2                           No                                No   \n",
      "3                           No                                No   \n",
      "4                           No                                No   \n",
      "\n",
      "  History or Radiation Therapy for Any Prior Malignancy That Involves Treatment to the Abdomen or Pelvis?  \\\n",
      "0                                                 No                                                        \n",
      "1                                                 No                                                        \n",
      "2                                                 No                                                        \n",
      "3                                                 No                                                        \n",
      "4                                                 No                                                        \n",
      "\n",
      "  History of Hormonal Therapy for Cancer within the Last Five Years?  \\\n",
      "0                                                 No                   \n",
      "1                                                 No                   \n",
      "2                                                 No                   \n",
      "3                                                 No                   \n",
      "4                                                 No                   \n",
      "\n",
      "  History of Neo-Adjuvant Treatment to Tumor Specimen Submitted for CPTAC  \\\n",
      "0                                                 No                        \n",
      "1                                                 No                        \n",
      "2                                                 No                        \n",
      "3                                                 No                        \n",
      "4                                                 No                        \n",
      "\n",
      "  Ischemia Time for Tumor Tissue  \n",
      "0                            3.0  \n",
      "1                            7.0  \n",
      "2                            7.0  \n",
      "3                            5.0  \n",
      "4                            5.0  \n",
      "\n",
      "[5 rows x 22 columns]\n",
      "\t- Satpathy_Table_S1A.csv\n",
      "\t\t- Shape: (207, 82)\n",
      "\t\t- Columns: Index(['Sample.ID', 'Participant', 'Experiment', 'Channel', 'Type',\n",
      "       'Smoking.History', 'Smoking.History_modified', 'Stage', 'Stage_mod',\n",
      "       'Country.of.Origin', 'Age', 'Gender', 'Ethnicity', 'Ethnicity_mod',\n",
      "       'Pathology.based.histology.assessment', 'Cigarettes.per.Day',\n",
      "       'Pack.Years.Smoked', 'Secondhand.Smoke', 'CIMP.status.meth',\n",
      "       'QC.status', 'TP53.mutation', 'PTEN.mutation', 'CDKN2A.mutation',\n",
      "       'KMT2D.mutation', 'NFE2L2.mutation', 'ARID1A.mutation', 'CUL3.mutation',\n",
      "       'BRCA2.mutation', 'KEAP1.mutation', 'SUZ12.mutation', 'NF1.mutation',\n",
      "       'PIK3CA.mutation', 'NOTCH1.mutation', 'TP53.mutation.status',\n",
      "       'PTEN.mutation.status', 'CDKN2A.mutation.status',\n",
      "       'KMT2D.mutation.status', 'NFE2L2.mutation.status',\n",
      "       'ARID1A.mutation.status', 'CUL3.mutation.status',\n",
      "       'BRCA2.mutation.status', 'KEAP1.mutation.status',\n",
      "       'SUZ12.mutation.status', 'NF1.mutation.status',\n",
      "       'PIK3CA.mutation.status', 'NOTCH1.mutation.status',\n",
      "       'CUL3.NFE2L2.KEAP1.mutation.status', 'KAT6A.cna.wes', 'SOX2.cna.wes',\n",
      "       'TP63.cna.wes', 'FGFR1.cna.wes', 'CDKN2A.cna.wes',\n",
      "       'CDKN2A.pathway.alteration', 'CDKN2A.pathway.alteration.status',\n",
      "       'PIK3CA.pathway.alteration', 'PIK3CA.pathway.alteration.status',\n",
      "       'FGFR3.TACC3.fusion.rna', 'TCGA.RNA.cluster', 'NMF.subtype',\n",
      "       'NMF.subtype.core', 'NMF.subtype.membership.score', 'CIN.wes',\n",
      "       'Smoking.Signature.Fraction.wes', 'Smoking.Signature.Count.wes',\n",
      "       'Total.Mutation.Count.wes', 'Mutation.Count.ExcludingINDELs.wes',\n",
      "       'DNP.Count.wes', 'DNP.Count.GG.to.TT.or.CC.to.AA.wes',\n",
      "       'Smoking.score.wes', 'Smoking.Score.Category.wes',\n",
      "       'Mutation.Count.Excluding.Silent.wes',\n",
      "       'Total.Mutation.Count.per.Mb.wes', 'TCGA pan-can immune clustering',\n",
      "       'ESTIMATE.StromalScore.rna', 'ESTIMATE.ImmuneScore.rna',\n",
      "       'ESTIMATE.TumorPurity.rna', 'TSNet.Purity.rna', 'ImmuneCluster.rna',\n",
      "       'xCell.ImmuneScore.rna', 'xCell.StromaScore.rna',\n",
      "       'xCell.MicroenvironmentScore.rna', 'CIBERSORT.AbsoluteScore.rna'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     Sample.ID Participant  Experiment Channel   Type  \\\n",
      "0  C3L.00081.N   C3L-00081           3    129N    NAT   \n",
      "1    C3L.00081   C3L-00081           3    128C  Tumor   \n",
      "2  C3L.00415.N   C3L-00415          16    131N    NAT   \n",
      "3    C3L.00415   C3L-00415          16    130C  Tumor   \n",
      "4  C3L.00445.N   C3L-00445          15    130N    NAT   \n",
      "\n",
      "                                     Smoking.History Smoking.History_modified  \\\n",
      "0  Lifelong non-smoker: Less than 100 cigarettes ...               Non-Smoker   \n",
      "1  Lifelong non-smoker: Less than 100 cigarettes ...               Non-Smoker   \n",
      "2       Current reformed smoker within past 15 years          Reformed Smoker   \n",
      "3       Current reformed smoker within past 15 years          Reformed Smoker   \n",
      "4       Current reformed smoker within past 15 years          Reformed Smoker   \n",
      "\n",
      "  Stage Stage_mod Country.of.Origin  ...  TCGA pan-can immune clustering  \\\n",
      "0   NaN       NaN               usa  ...                    Inflammatory   \n",
      "1   IIA        II               usa  ...             Lymphocyte depleted   \n",
      "2   NaN       NaN               usa  ...                    Inflammatory   \n",
      "3   IIA        II               usa  ...              IFN-gamma dominant   \n",
      "4   NaN       NaN          bulgaria  ...             Lymphocyte depleted   \n",
      "\n",
      "  ESTIMATE.StromalScore.rna ESTIMATE.ImmuneScore.rna ESTIMATE.TumorPurity.rna  \\\n",
      "0               2639.267606              4990.694721                      NaN   \n",
      "1               1977.164185              1850.506227                 0.393055   \n",
      "2               3142.667264              5471.937274                      NaN   \n",
      "3                316.433455              3634.439750                 0.376362   \n",
      "4               2426.103463              5157.233664                      NaN   \n",
      "\n",
      "  TSNet.Purity.rna ImmuneCluster.rna  xCell.ImmuneScore.rna  \\\n",
      "0         0.113597      NAT-enriched                 0.2745   \n",
      "1         0.781920        Warm Tumor                 0.0834   \n",
      "2         0.124839      NAT-enriched                 0.2831   \n",
      "3         0.564154        Warm Tumor                 0.1537   \n",
      "4         0.038895      NAT-enriched                 0.3606   \n",
      "\n",
      "  xCell.StromaScore.rna xCell.MicroenvironmentScore.rna  \\\n",
      "0                0.3206                          0.5951   \n",
      "1                0.0061                          0.0895   \n",
      "2                0.2626                          0.5457   \n",
      "3                0.0028                          0.1565   \n",
      "4                0.2508                          0.6114   \n",
      "\n",
      "  CIBERSORT.AbsoluteScore.rna  \n",
      "0                    0.792101  \n",
      "1                    0.371968  \n",
      "2                    0.997112  \n",
      "3                    0.813605  \n",
      "4                    0.726055  \n",
      "\n",
      "[5 rows x 82 columns]\n",
      "\t- Dou_UCEC_CPTAC3_meta_table_V2.1.csv\n",
      "\t\t- Shape: (153, 179)\n",
      "\t\t- Columns: Index(['idx', 'Proteomics_Participant_ID', 'Case_excluded',\n",
      "       'Proteomics_TMT_batch', 'Proteomics_TMT_plex', 'Proteomics_TMT_channel',\n",
      "       'Proteomics_Parent_Sample_IDs', 'Proteomics_Aliquot_ID',\n",
      "       'Proteomics_Tumor_Normal', 'Proteomics_OCT',\n",
      "       ...\n",
      "       'RNAseq_R1_sample_type', 'RNAseq_R1_filename', 'RNAseq_R1_UUID',\n",
      "       'RNAseq_R2_sample_type', 'RNAseq_R2_filename', 'RNAseq_R2_UUID',\n",
      "       'miRNAseq_sample_type', 'miRNAseq_UUID', 'Methylation_available',\n",
      "       'Methylation_quality'],\n",
      "      dtype='object', length=179)\n",
      "\t\t- Head: \n",
      "    idx Proteomics_Participant_ID Case_excluded  Proteomics_TMT_batch  \\\n",
      "0  S001                 C3L-00006            No                     2   \n",
      "1  S002                 C3L-00008            No                     4   \n",
      "2  S003                 C3L-00032            No                     1   \n",
      "3  S004                 C3L-00084           Yes                     3   \n",
      "4  S005                 C3L-00090            No                     3   \n",
      "\n",
      "   Proteomics_TMT_plex Proteomics_TMT_channel Proteomics_Parent_Sample_IDs  \\\n",
      "0                    5                   128N                 C3L-00006-01   \n",
      "1                   16                   130N                 C3L-00008-01   \n",
      "2                    2                    131                 C3L-00032-01   \n",
      "3                   11                   129N                 C3L-00084-01   \n",
      "4                   12                   129C                 C3L-00090-01   \n",
      "\n",
      "  Proteomics_Aliquot_ID Proteomics_Tumor_Normal Proteomics_OCT  ...  \\\n",
      "0         CPT0001460012                   Tumor             No  ...   \n",
      "1         CPT0001300009                   Tumor             No  ...   \n",
      "2         CPT0001420009                   Tumor             No  ...   \n",
      "3         CPT0000820012                   Tumor             No  ...   \n",
      "4         CPT0001140003                   Tumor             No  ...   \n",
      "\n",
      "  RNAseq_R1_sample_type                                 RNAseq_R1_filename  \\\n",
      "0                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
      "1                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
      "2                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
      "3                 Tumor  170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
      "4                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
      "\n",
      "                         RNAseq_R1_UUID RNAseq_R2_sample_type  \\\n",
      "0  8a1efc47-1c29-417f-a425-cdbd09565dcb                 Tumor   \n",
      "1  555725e8-cba5-4676-9b0a-80100cbf9f47                 Tumor   \n",
      "2  9ae968f3-691d-4db3-9977-1ab3e5af9085                 Tumor   \n",
      "3  b0a7cdf2-2ad8-4442-91b0-548ea4975554                 Tumor   \n",
      "4  8ce5618d-9ff6-40f9-aeea-8d8e1633ae38                 Tumor   \n",
      "\n",
      "                                  RNAseq_R2_filename  \\\n",
      "0  170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
      "1  170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
      "2  170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
      "3  170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
      "4  170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
      "\n",
      "                         RNAseq_R2_UUID miRNAseq_sample_type  \\\n",
      "0  8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda                Tumor   \n",
      "1  15235b12-b67a-4678-acc4-ed03d642bd5e                Tumor   \n",
      "2  423b6b09-02aa-4f47-9241-f75c1dad1161                Tumor   \n",
      "3  c83987a5-1c13-4af4-b46c-218fe5f60c34                  NaN   \n",
      "4  06d3fd4a-a623-4146-8500-4f1f17235253                Tumor   \n",
      "\n",
      "                          miRNAseq_UUID Methylation_available  \\\n",
      "0  37bcba98-1094-459e-83ae-c23a602416fb                   YES   \n",
      "1  492b50d8-ec35-46e7-a65d-06512aaee394                   YES   \n",
      "2  1794ff56-db2d-4d1a-8758-cab7fe3d98c1                   YES   \n",
      "3                                   NaN                   YES   \n",
      "4  a6524c2d-d7dd-4629-980e-b45dbdc92c49                   YES   \n",
      "\n",
      "  Methylation_quality  \n",
      "0                PASS  \n",
      "1                PASS  \n",
      "2                PASS  \n",
      "3                PASS  \n",
      "4                PASS  \n",
      "\n",
      "[5 rows x 179 columns]\n",
      "\t- Satpathy_Table_S1C.csv\n",
      "\t\t- Shape: (19, 12)\n",
      "\t\t- Columns: Index(['NMF cluster (p<0.05)', 'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3',\n",
      "       'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6', 'Unnamed: 7', 'Unnamed: 8',\n",
      "       'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  NMF cluster (p<0.05)         Unnamed: 1              Unnamed: 2  \\\n",
      "0                  NaN                NaN                  counts   \n",
      "1             variable           category  Basal.inclusive.counts   \n",
      "2     CIMP.status.meth          CIMP-high                       5   \n",
      "3     CIMP.status.meth  CIMP-intermediate                       7   \n",
      "4     CIMP.status.meth           CIMP-low                       3   \n",
      "\n",
      "         Unnamed: 3           Unnamed: 4                 Unnamed: 5  \\\n",
      "0               NaN                  NaN                        NaN   \n",
      "1  Classical.counts  EMT.enriched.counts  Inflamed.secretory.counts   \n",
      "2                16                    1                          0   \n",
      "3                 2                   11                         17   \n",
      "4                 3                    1                          0   \n",
      "\n",
      "                       Unnamed: 6                       Unnamed: 7  \\\n",
      "0                             NaN  p-values (Bonferroni corrected)   \n",
      "1  Proliferative.primitive.counts             Basal.inclusive.pval   \n",
      "2                               2                                1   \n",
      "3                               1                                1   \n",
      "4                               8                                1   \n",
      "\n",
      "            Unnamed: 8         Unnamed: 9              Unnamed: 10  \\\n",
      "0                  NaN                NaN                      NaN   \n",
      "1       Classical.pval  EMT.enriched.pval  Inflamed.secretory.pval   \n",
      "2  0.00289022574432088                  1                        1   \n",
      "3                    1                  1     0.000110121446619861   \n",
      "4                    1                  1                        1   \n",
      "\n",
      "                    Unnamed: 11  \n",
      "0                           NaN  \n",
      "1  Proliferative.primitive.pval  \n",
      "2                             1  \n",
      "3                             1  \n",
      "4            0.0141483486690702  \n",
      "\t- McDermott_Summary.csv\n",
      "\t\t- Shape: (17, 4)\n",
      "\t\t- Columns: Index(['Patient Information', 'Unnamed: 1', 'Cancer', 'Normal'], dtype='object')\n",
      "\t\t- Head: \n",
      "                         Patient Information  \\\n",
      "0  Age at Time of Tissue Procurement (year)    \n",
      "1                                        NaN   \n",
      "2         Weight of Tumor Tissue Sample (mg)   \n",
      "3                                        NaN   \n",
      "4                                       Race   \n",
      "\n",
      "                         Unnamed: 1       Cancer      Normal  \n",
      "0                             Range    36.6-85.9   44.0-71.8  \n",
      "1                      Average + SD  59.4  10.7  57.5  7.4  \n",
      "2                             Range      60-2300         NaN  \n",
      "3                      Average + SD    481  401         NaN  \n",
      "4  American Indian or Alaska Native            1           1  \n",
      "\t- Huang_Meta_table.csv\n",
      "\t\t- Shape: (110, 37)\n",
      "\t\t- Columns: Index(['case_id', 'age', 'gender', 'country', 'smoking_history',\n",
      "       'smoke_age_start', 'smoke_age_stop', 'num_smoke_per_day',\n",
      "       'num_pack_years_sm', 'smoking_second_hand', 'smoking_inferred',\n",
      "       'alcohol_consum', 'num_yrs_alc_con', 'tumor_site_original',\n",
      "       'tumor_site_curated', 'tumor_focality', 'tumor_size_cm',\n",
      "       'histologic_type', 'histologic_grade', 'tumor_necrosis',\n",
      "       'patho_staging_pt', 'patho_staging_pn', 'clinic_staging_dist_metas',\n",
      "       'tumor_stage', 'P16', 'HPV_inference', 'NAT_pathology_review',\n",
      "       'tumor_pathology_review', 'ESTIMATE_stromal_score',\n",
      "       'ESTIMATE_immune_score', 'CD3_IHC_count', 'stemness_score',\n",
      "       'mutation_count', 'neoAntigen_count', 'chr_instability_idx',\n",
      "       'integrated_subtype', 'transcriptomic_subtype'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     case_id  age gender  country  \\\n",
      "0  C3L-00977   56   Male   Russia   \n",
      "1  C3L-00987   61   Male  Ukraine   \n",
      "2  C3L-00994   50   Male  Ukraine   \n",
      "3  C3L-00995   56   Male  Ukraine   \n",
      "4  C3L-00997   47   Male  Ukraine   \n",
      "\n",
      "                                     smoking_history smoke_age_start  \\\n",
      "0             Current reformed smoker, years unknown             NaN   \n",
      "1  Current smoker: Includes daily and non-daily s...              18   \n",
      "2       Current reformed smoker within past 15 years              16   \n",
      "3        Current reformed smoker, more than 15 years              25   \n",
      "4        Current reformed smoker, more than 15 years              12   \n",
      "\n",
      "  smoke_age_stop num_smoke_per_day  num_pack_years_sm  \\\n",
      "0            NaN                20                NaN   \n",
      "1            NaN                20               43.0   \n",
      "2             50                 6               10.2   \n",
      "3             56                20               31.0   \n",
      "4             27                20               15.0   \n",
      "\n",
      "                                 smoking_second_hand  ...  \\\n",
      "0                                                Yes  ...   \n",
      "1                                                Yes  ...   \n",
      "2  Exposure to secondhand smoke history not avail...  ...   \n",
      "3  Exposure to secondhand smoke history not avail...  ...   \n",
      "4                                                Yes  ...   \n",
      "\n",
      "            tumor_pathology_review ESTIMATE_stromal_score  \\\n",
      "0                          SCC 80%            6825.995755   \n",
      "1  SCC 90%;SCC 90%;SCC 80%;SCC 70%            5999.793467   \n",
      "2  SCC 40%;SCC 70%;SCC 40%;SCC 75%            8924.036564   \n",
      "3  SCC 90%;SCC 80%;SCC 40%;SCC 70%            8723.429667   \n",
      "4                  SCC 70%;SCC 70%            7025.911695   \n",
      "\n",
      "  ESTIMATE_immune_score CD3_IHC_count stemness_score mutation_count  \\\n",
      "0           7989.115925           1.0       0.953243            106   \n",
      "1           4772.409716           0.0       0.825330             83   \n",
      "2           8176.233903           5.0       0.664581             67   \n",
      "3           8342.246345          70.0       0.539918             64   \n",
      "4           7445.251991          60.0       0.843765            129   \n",
      "\n",
      "   neoAntigen_count chr_instability_idx integrated_subtype  \\\n",
      "0                 0            2.003654              Basal   \n",
      "1                 0            5.205612                CIN   \n",
      "2                 0            1.684475             Immune   \n",
      "3                 0            1.340483             Immune   \n",
      "4                 0            3.906370                CIN   \n",
      "\n",
      "  transcriptomic_subtype  \n",
      "0            Mesenchymal  \n",
      "1              Classical  \n",
      "2            Mesenchymal  \n",
      "3            Mesenchymal  \n",
      "4              Classical  \n",
      "\n",
      "[5 rows x 37 columns]\n",
      "\t- Satpathy_Table_S1E.csv\n",
      "\t\t- Shape: (139, 10)\n",
      "\t\t- Columns: Index(['Unnamed: 0', 'Unnamed: 1', 'counts', 'Unnamed: 3', 'Unnamed: 4',\n",
      "       'Unnamed: 5', 'p-values (Bonferroni corrected)', 'Unnamed: 7',\n",
      "       'Unnamed: 8', 'Unnamed: 9'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "   Unnamed: 0 Unnamed: 1             counts        Unnamed: 3  \\\n",
      "0    variable   category  Cold.Tumor.counts  Hot.Tumor.counts   \n",
      "1  Experiment          1                  1                 0   \n",
      "2  Experiment         10                  0                 2   \n",
      "3  Experiment         11                  4                 2   \n",
      "4  Experiment         12                  1                 1   \n",
      "\n",
      "            Unnamed: 4         Unnamed: 5 p-values (Bonferroni corrected)  \\\n",
      "0  NAT.enriched.counts  Warm.Tumor.counts                 Cold.Tumor.pval   \n",
      "1                    4                  4                               1   \n",
      "2                    5                  3                               1   \n",
      "3                    3                  0                               1   \n",
      "4                    4                  3                               1   \n",
      "\n",
      "       Unnamed: 7         Unnamed: 8       Unnamed: 9  \n",
      "0  Hot.Tumor.pval  NAT.enriched.pval  Warm.Tumor.pval  \n",
      "1               1                  1                1  \n",
      "2               1                  1                1  \n",
      "3               1                  1                1  \n",
      "4               1                  1                1  \n",
      "\t- Clark_RNA-seq_sequencing_depth.csv\n",
      "\t\t- Shape: (175, 15)\n",
      "\t\t- Columns: Index(['Case', 'Tissue Type', 'Sample Aliquot ID',\n",
      "       'Uniquely mapped reads number', 'Uniquely mapped reads %',\n",
      "       '% of reads mapped to multiple loci',\n",
      "       '% of reads mapped to too many loci', '% of reads unmapped: other',\n",
      "       '% of reads unmapped: too short', 'Insertion rate per base',\n",
      "       'Mismatch rate per base, %', 'Number of input reads',\n",
      "       'Number of reads mapped to multiple loci',\n",
      "       'Number of reads mapped to too many loci', 'Number of splices: Total'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "        Case          Tissue Type                     Sample Aliquot ID  \\\n",
      "0        NaN                  NaN                                   NaN   \n",
      "1  C3L-01287  Solid Tissue Normal  ee722bb1-db91-46ff-bbf7-989ee5954824   \n",
      "2  C3L-00561  Solid Tissue Normal  703aaad7-3056-4938-8099-108b6679ded4   \n",
      "3  C3L-00561        Primary Tumor  da873fe0-1299-471c-9020-c61bd9b507dd   \n",
      "4  C3L-01287        Primary Tumor  3197d373-5a6f-4e45-bc8e-2f5a6c7186e8   \n",
      "\n",
      "   Uniquely mapped reads number  Uniquely mapped reads %  \\\n",
      "0                           NaN                      NaN   \n",
      "1                    94846820.0                    92.72   \n",
      "2                    73020918.0                    93.11   \n",
      "3                    87486665.0                    89.90   \n",
      "4                    90613424.0                    91.56   \n",
      "\n",
      "   % of reads mapped to multiple loci  % of reads mapped to too many loci  \\\n",
      "0                                 NaN                                 NaN   \n",
      "1                                5.70                                0.28   \n",
      "2                                5.32                                0.29   \n",
      "3                                6.77                                0.31   \n",
      "4                                5.79                                0.39   \n",
      "\n",
      "   % of reads unmapped: other  % of reads unmapped: too short  \\\n",
      "0                         NaN                             NaN   \n",
      "1                        0.11                            1.20   \n",
      "2                        0.11                            1.17   \n",
      "3                        0.11                            2.91   \n",
      "4                        0.11                            2.15   \n",
      "\n",
      "   Insertion rate per base  Mismatch rate per base, %  Number of input reads  \\\n",
      "0                      NaN                        NaN                    NaN   \n",
      "1                     0.01                       0.20            102294273.0   \n",
      "2                     0.01                       0.19             78427086.0   \n",
      "3                     0.00                       0.22             97314711.0   \n",
      "4                     0.01                       0.17             98964138.0   \n",
      "\n",
      "   Number of reads mapped to multiple loci  \\\n",
      "0                                      NaN   \n",
      "1                                5828078.0   \n",
      "2                                4175421.0   \n",
      "3                                6585325.0   \n",
      "4                                5726980.0   \n",
      "\n",
      "   Number of reads mapped to too many loci  Number of splices: Total  \n",
      "0                                      NaN                       NaN  \n",
      "1                                 282490.0                22475023.0  \n",
      "2                                 224692.0                15823428.0  \n",
      "3                                 306171.0                22855561.0  \n",
      "4                                 387341.0                23645783.0  \n",
      "\t- Krug_B)_QC_RNA-seq.csv\n",
      "\t\t- Shape: (122, 7)\n",
      "\t\t- Columns: Index(['Sample.ID', 'ACGT tag ID', 'Total reads', 'QC', 'mapping rates(%)',\n",
      "       'No. of transcripts', 'Notes'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  Sample.ID ACGT tag ID  Total reads    QC  mapping rates(%)  \\\n",
      "0  X01BR001        25-R     33444989  pass            0.9611   \n",
      "1  X01BR008       350-R     38418218  pass            0.9761   \n",
      "2  X01BR009       345-R     35745238  pass            0.9680   \n",
      "3  X01BR010       349-R     36915705  pass            0.9731   \n",
      "4  X01BR015        26-R     33513916  pass            0.9400   \n",
      "\n",
      "   No. of transcripts Notes  \n",
      "0               77112   NaN  \n",
      "1               78337   NaN  \n",
      "2               84231   NaN  \n",
      "3               81885   NaN  \n",
      "4               82366   NaN  \n",
      "\t- Satpathy_Table_S1D.csv\n",
      "\t\t- Shape: (124, 12)\n",
      "\t\t- Columns: Index(['Unnamed: 0', 'Unnamed: 1', 'counts', 'Unnamed: 3', 'Unnamed: 4',\n",
      "       'Unnamed: 5', 'Unnamed: 6', 'p-values (Bonferroni corrected)',\n",
      "       'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "   Unnamed: 0 Unnamed: 1                  counts        Unnamed: 3  \\\n",
      "0    variable   category  Basal.inclusive.counts  Classical.counts   \n",
      "1  Experiment          1                       0                 1   \n",
      "2  Experiment         10                       1                 0   \n",
      "3  Experiment         11                       0                 4   \n",
      "4  Experiment         12                       2                 1   \n",
      "\n",
      "            Unnamed: 4                 Unnamed: 5  \\\n",
      "0  EMT.enriched.counts  Inflamed.secretory.counts   \n",
      "1                    2                          0   \n",
      "2                    1                          2   \n",
      "3                    0                          0   \n",
      "4                    2                          0   \n",
      "\n",
      "                       Unnamed: 6 p-values (Bonferroni corrected)  \\\n",
      "0  Proliferative.primitive.counts            Basal.inclusive.pval   \n",
      "1                               2                               1   \n",
      "2                               1                               1   \n",
      "3                               1                               1   \n",
      "4                               0                               1   \n",
      "\n",
      "       Unnamed: 8         Unnamed: 9              Unnamed: 10  \\\n",
      "0  Classical.pval  EMT.enriched.pval  Inflamed.secretory.pval   \n",
      "1               1                  1                        1   \n",
      "2               1                  1                        1   \n",
      "3               1                  1                        1   \n",
      "4               1                  1                        1   \n",
      "\n",
      "                    Unnamed: 11  \n",
      "0  Proliferative.primitive.pval  \n",
      "1                             1  \n",
      "2                             1  \n",
      "3                             1  \n",
      "4                             1  \n",
      "\t- Vasaikar_A-Annotation.csv\n",
      "\t\t- Shape: (45, 6)\n",
      "\t\t- Columns: Index(['Column name', 'Column definition', 'Unnamed: 2', 'Unnamed: 3',\n",
      "       'Data type', 'Definition'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "  Column name                              Column definition  Unnamed: 2  \\\n",
      "0    SampleID              ID of the prospective participant         NaN   \n",
      "1    Clinical                     Clinical data availability         NaN   \n",
      "2    Mutation                     Mutation data availability         NaN   \n",
      "3         CNV  Copy number variation (CNV) data availability         NaN   \n",
      "4   miRNA-seq                 microRNA-Seq data availability         NaN   \n",
      "\n",
      "   Unnamed: 3 Data type   Definition  \n",
      "0         NaN       Bin       Binary  \n",
      "1         NaN       Cat  Categorical  \n",
      "2         NaN       Con   Continuous  \n",
      "3         NaN       Ord      Ordinal  \n",
      "4         NaN       NaN          NaN  \n",
      "\t- Cao_Proteins_phospho_glyco_IDs.csv\n",
      "\t\t- Shape: (215, 14)\n",
      "\t\t- Columns: Index(['Case_ID', 'Type', 'TMT_Set', 'TMT_Channel', 'Proteins', 'Peptides',\n",
      "       'Phosphoproteins', 'Phosphopeptides', 'Phosphosites', 'N-Glycoproteins',\n",
      "       'N-Glycopeptides', 'N-Glycosites', 'O-Glycoproteins',\n",
      "       'O-Glycopeptides'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     Case_ID   Type  TMT_Set TMT_Channel  Proteins  Peptides  Phosphoproteins  \\\n",
      "0  C3L-00017  Tumor       20        129C      9043     76356             5370   \n",
      "1  C3L-00102  Tumor       16        127N      9127     79937             5456   \n",
      "2  C3L-00102    NAT       16        131N      9126     79706             5459   \n",
      "3  C3L-00189  Tumor       14        128N      8903     77642             5552   \n",
      "4  C3L-00189    NAT       14        131C      8904     77639             5552   \n",
      "\n",
      "   Phosphopeptides  Phosphosites  N-Glycoproteins  N-Glycopeptides  \\\n",
      "0            18995         24900              767            10468   \n",
      "1            19592         26034              761             9549   \n",
      "2            19591         26026              751             9018   \n",
      "3            20010         27117              665             8251   \n",
      "4            20006         27103              660             7833   \n",
      "\n",
      "   N-Glycosites  O-Glycoproteins  O-Glycopeptides  \n",
      "0          1618              307             1003  \n",
      "1          1580              294              958  \n",
      "2          1549              285              911  \n",
      "3          1394              254              903  \n",
      "4          1369              246              865  \n",
      "\t- Huang_Data_availabilty.csv\n",
      "\t\t- Shape: (110, 17)\n",
      "\t\t- Columns: Index(['case_id', 'WES', 'WES_BN', 'WGS', 'WGS_BN', 'RNA-Seq', 'RNA-Seq_AN',\n",
      "       'MiRNA', 'MiRNA_AN', 'Methy', 'Methy_AN', 'Global_TMT', 'Global_TMT_AN',\n",
      "       'PhosphoTMT', 'PhosphoTMT_AN', 'Unnamed: 15',\n",
      "       '1: available; 0: unavailable'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     case_id  WES  WES_BN  WGS  WGS_BN  RNA-Seq  RNA-Seq_AN  MiRNA  MiRNA_AN  \\\n",
      "0  C3L-00977    1       1    1       1        1           1      1         1   \n",
      "1  C3L-00987    1       1    1       1        1           0      1         0   \n",
      "2  C3L-00994    1       1    1       1        1           1      1         1   \n",
      "3  C3L-00995    1       1    1       1        1           1      1         1   \n",
      "4  C3L-00997    1       1    1       1        1           1      1         1   \n",
      "\n",
      "   Methy  Methy_AN  Global_TMT  Global_TMT_AN  PhosphoTMT  PhosphoTMT_AN  \\\n",
      "0      1         1           1              0           1              0   \n",
      "1      0         0           1              0           1              0   \n",
      "2      1         1           1              1           1              1   \n",
      "3      1         1           1              1           1              1   \n",
      "4      1         1           1              1           1              1   \n",
      "\n",
      "   Unnamed: 15              1: available; 0: unavailable  \n",
      "0          NaN  _BN: blood normal; _AN: adjacent  normal  \n",
      "1          NaN                                       NaN  \n",
      "2          NaN                                       NaN  \n",
      "3          NaN                                       NaN  \n",
      "4          NaN                                       NaN  \n",
      "\t- Clark_WGS-WES_sequencing_depth.csv\n",
      "\t\t- Shape: (441, 9)\n",
      "\t\t- Columns: Index(['Case', 'Sample Aliquot', 'Bam File', 'Total Reads',\n",
      "       'Proper Mapped Reads', 'Coverage', 'Cov_100x', 'Template Mean',\n",
      "       'Template SD'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "        Case Sample Aliquot           Bam File  Total Reads  \\\n",
      "0        NaN            NaN                NaN          NaN   \n",
      "1  C3L-00096  CPT0000050163  CPT0000050163-wgs  965308021.0   \n",
      "2  C3L-00096  CPT0000050163  CPT0000050163-wxs  375791912.0   \n",
      "3  C3L-00010  CPT0000060163  CPT0000060163-wgs  926236236.0   \n",
      "4  C3L-00010  CPT0000060163  CPT0000060163-wxs  400609052.0   \n",
      "\n",
      "   Proper Mapped Reads  Coverage  Cov_100x  Template Mean  Template SD  \n",
      "0                  NaN       NaN       NaN            NaN          NaN  \n",
      "1                 0.88     41.19       NaN         376.77       115.93  \n",
      "2                 0.90    170.48      0.73            NaN          NaN  \n",
      "3                 0.88     39.65       NaN         362.63       103.11  \n",
      "4                 0.91    180.88      0.86            NaN          NaN  \n",
      "\t- Huang_TMT-set.csv\n",
      "\t\t- Shape: (190, 6)\n",
      "\t\t- Columns: Index(['TMT set#', 'channel #', 'peptide amount for labeling (ug)', 'case_id',\n",
      "       'type', 'combined_id'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "   TMT set# channel #  peptide amount for labeling (ug)    case_id   type  \\\n",
      "0         1      127N                             300.0  C3L-00997  Tumor   \n",
      "1         1      127C                             300.0  C3N-03849  Tumor   \n",
      "2         1      128N                             300.0  C3N-01858    NAT   \n",
      "3         1      128C                             300.0        QC1    NaN   \n",
      "4         1      129N                             300.0  C3N-03487  Tumor   \n",
      "\n",
      "   combined_id  \n",
      "0  C3L-00997-T  \n",
      "1  C3N-03849-T  \n",
      "2  C3N-01858-N  \n",
      "3          QC1  \n",
      "4  C3N-03487-T  \n",
      "\t- Clark_ccrcc_clinical_characteristics.csv\n",
      "\t\t- Shape: (110, 16)\n",
      "\t\t- Columns: Index(['Case_ID', 'Tumor/Normal', 'Gender', 'Age', 'BMI', 'Race',\n",
      "       'Ethnicity_Self_Identify', 'Tumor_Site', 'Tumor_Site_Other',\n",
      "       'Tumor_Size_cm', 'Tumor_Focality', 'Histologic_Type', 'Grade',\n",
      "       'Tumor_Stage_Pathological', 'Path_Stage_Primary_Tumor_pT',\n",
      "       'Path_Stage_Reg_Lymph_Nodes_pN'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     Case_ID Tumor/Normal  Gender Age    BMI   Race Ethnicity_Self_Identify  \\\n",
      "0  C3L-00004           TN    Male  72  22.80  White                   White   \n",
      "1  C3L-00010           TN    Male  30  34.15  White                   White   \n",
      "2  C3L-00011           TN  Female  63  27.47  White                   White   \n",
      "3  C3L-00026           TN  Female  65  28.23  White                   White   \n",
      "4  C3L-00079           TN    Male  49  37.88  White                   white   \n",
      "\n",
      "   Tumor_Site Tumor_Site_Other  Tumor_Size_cm Tumor_Focality  \\\n",
      "0  Lower pole              NaN           12.0       Unifocal   \n",
      "1  Upper pole              NaN            6.5       Unifocal   \n",
      "2  Lower pole              NaN           12.0       Unifocal   \n",
      "3  Lower pole              NaN            2.0       Unifocal   \n",
      "4       Other     mid-inferior           14.0       Unifocal   \n",
      "\n",
      "                   Histologic_Type Grade Tumor_Stage_Pathological  \\\n",
      "0  Clear cell renal cell carcinoma    G3                Stage III   \n",
      "1  Clear cell renal cell carcinoma    G3                  Stage I   \n",
      "2  Clear cell renal cell carcinoma    G4                 Stage IV   \n",
      "3  Clear cell renal cell carcinoma    G3                  Stage I   \n",
      "4  Clear cell renal cell carcinoma    G3                Stage III   \n",
      "\n",
      "  Path_Stage_Primary_Tumor_pT Path_Stage_Reg_Lymph_Nodes_pN  \n",
      "0                         pT3                           pNX  \n",
      "1                        pT1b                           pN0  \n",
      "2                        pT3a                           pNX  \n",
      "3                        pT1a                           pNX  \n",
      "4                        pT3a                           pN1  \n",
      "\t- Krug_C)_QC_WXS.csv\n",
      "\t\t- Shape: (244, 14)\n",
      "\t\t- Columns: Index(['Type', 'Sample_id', 'participant', 'TOTAL_READS', 'PF_READS',\n",
      "       'READ_LENGTH', 'TOTAL_BASES', 'PF_BASES', 'Q20_BASES', 'PF_Q20_BASES',\n",
      "       'Q30_BASES', 'PF_Q30_BASES', 'Q20_EQUIVALENT_YIELD',\n",
      "       'PF_Q20_EQUIVALENT_YIELD'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     Type Sample_id participant  TOTAL_READS   PF_READS  READ_LENGTH  \\\n",
      "0   Tumor    100_DM     11BR020    214780004  214780004          137   \n",
      "1  Normal    100_DN     11BR020    201972266  201972266          137   \n",
      "2   Tumor    101_DM     11BR022    184087888  184087888          136   \n",
      "3  Normal    101_DN     11BR022    221378874  221378874          138   \n",
      "4   Tumor    102_DM     11BR023    195699454  195699454          137   \n",
      "\n",
      "   TOTAL_BASES     PF_BASES    Q20_BASES  PF_Q20_BASES    Q30_BASES  \\\n",
      "0  29525511437  29525511437  29276291481   29276291481  28729049806   \n",
      "1  27709470651  27709470651  27470610354   27470610354  26948379346   \n",
      "2  25087738426  25087738426  24839964699   24839964699  24277332849   \n",
      "3  30568366545  30568366545  30313820727   30313820727  29707980567   \n",
      "4  26941326289  26941326289  26681350066   26681350066  26096412597   \n",
      "\n",
      "   PF_Q30_BASES  Q20_EQUIVALENT_YIELD  PF_Q20_EQUIVALENT_YIELD  \n",
      "0   28729049806           43755154395              43755154395  \n",
      "1   26948379346           41054557180              41054557180  \n",
      "2   24277332849           37093349447              37093349447  \n",
      "3   29707980567           45283444190              45283444190  \n",
      "4   26096412597           39848786099              39848786099  \n",
      "\t- Wang_clinical_data.csv\n",
      "\t\t- Shape: (110, 28)\n",
      "\t\t- Columns: Index(['case_id', 'age', 'gender', 'height', 'weight', 'bmi',\n",
      "       'country_of_origin', 'race', 'ethnicity', 'ethnicity_self_identify',\n",
      "       'smoking_history', 'secondhand_smoke_exposure', 'tumor_laterality',\n",
      "       'tumor_site_curated', 'tumor_site_curation_notes', 'follow_up_period',\n",
      "       'vital_status', 'path_diag_to_last_contact_days',\n",
      "       'path_diag_to_death_days', 'cause_of_death', 'lost_to_follow_up',\n",
      "       'tumor_reoccur_after_treatment', 'path_diag_to_tumor_reoccur_days',\n",
      "       'expert_path_review', 'expert_path_review_num_segments',\n",
      "       'expert_path_review_details', 'excluded_in_downstream_analysis',\n",
      "       'exclusion_reason'],\n",
      "      dtype='object')\n",
      "\t\t- Head: \n",
      "     case_id  age  gender  height  weight    bmi country_of_origin   race  \\\n",
      "0  C3L-00104   58    Male   188.0   115.0  32.54     United States  White   \n",
      "1  C3L-00365   59  Female   162.0    54.0  20.61     United States  White   \n",
      "2  C3L-00674   45    Male   193.0   102.0  27.44               NaN  White   \n",
      "3  C3L-00677   69  Female   164.0    52.0  19.32               NaN  White   \n",
      "4  C3L-01040   77    Male   170.0    70.0  24.22            Russia    NaN   \n",
      "\n",
      "                ethnicity ethnicity_self_identify  ...  \\\n",
      "0  Not-Hispanic or Latino               Caucasian  ...   \n",
      "1  Not-Hispanic or Latino               Caucasian  ...   \n",
      "2  Not-Hispanic or Latino               Caucasian  ...   \n",
      "3  Not-Hispanic or Latino               Caucasian  ...   \n",
      "4                     NaN               Caucasian  ...   \n",
      "\n",
      "  path_diag_to_death_days cause_of_death lost_to_follow_up  \\\n",
      "0                   129.0            NaN                No   \n",
      "1                   322.0            NaN                No   \n",
      "2                   478.0            GBM                No   \n",
      "3                   155.0            GBM                No   \n",
      "4                     NaN            NaN               Yes   \n",
      "\n",
      "  tumor_reoccur_after_treatment path_diag_to_tumor_reoccur_days  \\\n",
      "0                           0.0                             NaN   \n",
      "1                           NaN                             NaN   \n",
      "2                           1.0                           245.0   \n",
      "3                           NaN                             NaN   \n",
      "4                           0.0                             NaN   \n",
      "\n",
      "  expert_path_review expert_path_review_num_segments  \\\n",
      "0               PASS                             1.0   \n",
      "1               FAIL                             1.0   \n",
      "2               PASS                             1.0   \n",
      "3               PASS                             1.0   \n",
      "4               PASS                             2.0   \n",
      "\n",
      "       expert_path_review_details  excluded_in_downstream_analysis  \\\n",
      "0                             NaN                            False   \n",
      "1  C3L-00365-01(low tumor nuclei)                            False   \n",
      "2                             NaN                            False   \n",
      "3                             NaN                            False   \n",
      "4                             NaN                            False   \n",
      "\n",
      "  exclusion_reason  \n",
      "0              NaN  \n",
      "1              NaN  \n",
      "2              NaN  \n",
      "3              NaN  \n",
      "4              NaN  \n",
      "\n",
      "[5 rows x 28 columns]\n"
     ]
    }
   ],
   "source": [
    "folder_path = \"../data/extracted-tables/\"\n",
    "static_path = os.path.abspath(folder_path)\n",
    "db = Database()\n",
    "\n",
    "db.load_data_from_folder(folder_path)\n",
    "db.describe_database()\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "arpah",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
