{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b6a5938c-d32e-4816-ba31-a0b54e4a9826",
   "metadata": {},
   "source": [
    "# Data Integration with BDI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f01f77a6-f349-4f71-a847-c26b69a60c41",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/rlopez/opt/anaconda3/envs/askem/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "from bdi import APIManager"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "99afdf9a-b68a-4a30-9735-f8ef36027757",
   "metadata": {},
   "outputs": [],
   "source": [
    "manager = APIManager()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb0b6170-c8e5-4ff3-9948-a55acb913563",
   "metadata": {},
   "source": [
    "## Dataset Loading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "34e5d030-a50d-4076-9e96-7e4e73fe7cda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>idx</th>\n",
       "      <th>Proteomics_Participant_ID</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>Proteomics_TMT_batch</th>\n",
       "      <th>Proteomics_TMT_plex</th>\n",
       "      <th>Proteomics_TMT_channel</th>\n",
       "      <th>Proteomics_Parent_Sample_IDs</th>\n",
       "      <th>Proteomics_Aliquot_ID</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>Proteomics_OCT</th>\n",
       "      <th>...</th>\n",
       "      <th>RNAseq_R1_sample_type</th>\n",
       "      <th>RNAseq_R1_filename</th>\n",
       "      <th>RNAseq_R1_UUID</th>\n",
       "      <th>RNAseq_R2_sample_type</th>\n",
       "      <th>RNAseq_R2_filename</th>\n",
       "      <th>RNAseq_R2_UUID</th>\n",
       "      <th>miRNAseq_sample_type</th>\n",
       "      <th>miRNAseq_UUID</th>\n",
       "      <th>Methylation_available</th>\n",
       "      <th>Methylation_quality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>No</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>128N</td>\n",
       "      <td>C3L-00006-01</td>\n",
       "      <td>CPT0001460012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8a1efc47-1c29-417f-a425-cdbd09565dcb</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>37bcba98-1094-459e-83ae-c23a602416fb</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S002</td>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>No</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>130N</td>\n",
       "      <td>C3L-00008-01</td>\n",
       "      <td>CPT0001300009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>555725e8-cba5-4676-9b0a-80100cbf9f47</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>15235b12-b67a-4678-acc4-ed03d642bd5e</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>492b50d8-ec35-46e7-a65d-06512aaee394</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S003</td>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>No</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>131</td>\n",
       "      <td>C3L-00032-01</td>\n",
       "      <td>CPT0001420009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>9ae968f3-691d-4db3-9977-1ab3e5af9085</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>423b6b09-02aa-4f47-9241-f75c1dad1161</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>1794ff56-db2d-4d1a-8758-cab7fe3d98c1</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S004</td>\n",
       "      <td>C3L-00084</td>\n",
       "      <td>Yes</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>129N</td>\n",
       "      <td>C3L-00084-01</td>\n",
       "      <td>CPT0000820012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>b0a7cdf2-2ad8-4442-91b0-548ea4975554</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>c83987a5-1c13-4af4-b46c-218fe5f60c34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S005</td>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>No</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>129C</td>\n",
       "      <td>C3L-00090-01</td>\n",
       "      <td>CPT0001140003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>8ce5618d-9ff6-40f9-aeea-8d8e1633ae38</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>06d3fd4a-a623-4146-8500-4f1f17235253</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>a6524c2d-d7dd-4629-980e-b45dbdc92c49</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>S149</td>\n",
       "      <td>NX14</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>128C</td>\n",
       "      <td>C3N-03699-01,C3N-03699-02</td>\n",
       "      <td>CPT0230130002,CPT0230130003,CPT0230130004,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>S150</td>\n",
       "      <td>NX15</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>131</td>\n",
       "      <td>C3N-03700-01</td>\n",
       "      <td>CPT0230190002,CPT0230190003,CPT0230190004</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>S151</td>\n",
       "      <td>NX16</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>129C</td>\n",
       "      <td>C3N-03701-01,C3N-03701-02,C3N-03701-03</td>\n",
       "      <td>CPT0230250002,CPT0230260002,CPT0230270002,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>S152</td>\n",
       "      <td>NX17</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>127C</td>\n",
       "      <td>C3N-03706-01,C3N-03706-02,C3N-03706-03</td>\n",
       "      <td>CPT0230400002,CPT0230400003,CPT0230400004,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>S153</td>\n",
       "      <td>NX18</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>130N</td>\n",
       "      <td>C3N-03708-01,C3N-03708-02,C3N-03708-03</td>\n",
       "      <td>CPT0230460002,CPT0230460003,CPT0230460004,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      idx Proteomics_Participant_ID Case_excluded  Proteomics_TMT_batch  \\\n",
       "0    S001                 C3L-00006            No                     2   \n",
       "1    S002                 C3L-00008            No                     4   \n",
       "2    S003                 C3L-00032            No                     1   \n",
       "3    S004                 C3L-00084           Yes                     3   \n",
       "4    S005                 C3L-00090            No                     3   \n",
       "..    ...                       ...           ...                   ...   \n",
       "148  S149                      NX14            No                     5   \n",
       "149  S150                      NX15            No                     5   \n",
       "150  S151                      NX16            No                     5   \n",
       "151  S152                      NX17            No                     5   \n",
       "152  S153                      NX18            No                     5   \n",
       "\n",
       "     Proteomics_TMT_plex Proteomics_TMT_channel  \\\n",
       "0                      5                   128N   \n",
       "1                     16                   130N   \n",
       "2                      2                    131   \n",
       "3                     11                   129N   \n",
       "4                     12                   129C   \n",
       "..                   ...                    ...   \n",
       "148                   17                   128C   \n",
       "149                   17                    131   \n",
       "150                   17                   129C   \n",
       "151                   17                   127C   \n",
       "152                   17                   130N   \n",
       "\n",
       "               Proteomics_Parent_Sample_IDs  \\\n",
       "0                              C3L-00006-01   \n",
       "1                              C3L-00008-01   \n",
       "2                              C3L-00032-01   \n",
       "3                              C3L-00084-01   \n",
       "4                              C3L-00090-01   \n",
       "..                                      ...   \n",
       "148               C3N-03699-01,C3N-03699-02   \n",
       "149                            C3N-03700-01   \n",
       "150  C3N-03701-01,C3N-03701-02,C3N-03701-03   \n",
       "151  C3N-03706-01,C3N-03706-02,C3N-03706-03   \n",
       "152  C3N-03708-01,C3N-03708-02,C3N-03708-03   \n",
       "\n",
       "                                 Proteomics_Aliquot_ID  \\\n",
       "0                                        CPT0001460012   \n",
       "1                                        CPT0001300009   \n",
       "2                                        CPT0001420009   \n",
       "3                                        CPT0000820012   \n",
       "4                                        CPT0001140003   \n",
       "..                                                 ...   \n",
       "148  CPT0230130002,CPT0230130003,CPT0230130004,CPT0...   \n",
       "149          CPT0230190002,CPT0230190003,CPT0230190004   \n",
       "150  CPT0230250002,CPT0230260002,CPT0230270002,CPT0...   \n",
       "151  CPT0230400002,CPT0230400003,CPT0230400004,CPT0...   \n",
       "152  CPT0230460002,CPT0230460003,CPT0230460004,CPT0...   \n",
       "\n",
       "    Proteomics_Tumor_Normal Proteomics_OCT  ... RNAseq_R1_sample_type  \\\n",
       "0                     Tumor             No  ...                 Tumor   \n",
       "1                     Tumor             No  ...                 Tumor   \n",
       "2                     Tumor             No  ...                 Tumor   \n",
       "3                     Tumor             No  ...                 Tumor   \n",
       "4                     Tumor             No  ...                 Tumor   \n",
       "..                      ...            ...  ...                   ...   \n",
       "148         Enriched_normal             No  ...                   NaN   \n",
       "149         Enriched_normal             No  ...                   NaN   \n",
       "150         Enriched_normal             No  ...                   NaN   \n",
       "151         Enriched_normal             No  ...                   NaN   \n",
       "152         Enriched_normal             No  ...                   NaN   \n",
       "\n",
       "                                    RNAseq_R1_filename  \\\n",
       "0    170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "1    170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "2    170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "3    170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "4    170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "..                                                 ...   \n",
       "148                                                NaN   \n",
       "149                                                NaN   \n",
       "150                                                NaN   \n",
       "151                                                NaN   \n",
       "152                                                NaN   \n",
       "\n",
       "                           RNAseq_R1_UUID RNAseq_R2_sample_type  \\\n",
       "0    8a1efc47-1c29-417f-a425-cdbd09565dcb                 Tumor   \n",
       "1    555725e8-cba5-4676-9b0a-80100cbf9f47                 Tumor   \n",
       "2    9ae968f3-691d-4db3-9977-1ab3e5af9085                 Tumor   \n",
       "3    b0a7cdf2-2ad8-4442-91b0-548ea4975554                 Tumor   \n",
       "4    8ce5618d-9ff6-40f9-aeea-8d8e1633ae38                 Tumor   \n",
       "..                                    ...                   ...   \n",
       "148                                   NaN                   NaN   \n",
       "149                                   NaN                   NaN   \n",
       "150                                   NaN                   NaN   \n",
       "151                                   NaN                   NaN   \n",
       "152                                   NaN                   NaN   \n",
       "\n",
       "                                    RNAseq_R2_filename  \\\n",
       "0    170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "1    170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "2    170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "3    170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "4    170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "..                                                 ...   \n",
       "148                                                NaN   \n",
       "149                                                NaN   \n",
       "150                                                NaN   \n",
       "151                                                NaN   \n",
       "152                                                NaN   \n",
       "\n",
       "                           RNAseq_R2_UUID miRNAseq_sample_type  \\\n",
       "0    8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda                Tumor   \n",
       "1    15235b12-b67a-4678-acc4-ed03d642bd5e                Tumor   \n",
       "2    423b6b09-02aa-4f47-9241-f75c1dad1161                Tumor   \n",
       "3    c83987a5-1c13-4af4-b46c-218fe5f60c34                  NaN   \n",
       "4    06d3fd4a-a623-4146-8500-4f1f17235253                Tumor   \n",
       "..                                    ...                  ...   \n",
       "148                                   NaN                  NaN   \n",
       "149                                   NaN                  NaN   \n",
       "150                                   NaN                  NaN   \n",
       "151                                   NaN                  NaN   \n",
       "152                                   NaN                  NaN   \n",
       "\n",
       "                            miRNAseq_UUID Methylation_available  \\\n",
       "0    37bcba98-1094-459e-83ae-c23a602416fb                   YES   \n",
       "1    492b50d8-ec35-46e7-a65d-06512aaee394                   YES   \n",
       "2    1794ff56-db2d-4d1a-8758-cab7fe3d98c1                   YES   \n",
       "3                                     NaN                   YES   \n",
       "4    a6524c2d-d7dd-4629-980e-b45dbdc92c49                   YES   \n",
       "..                                    ...                   ...   \n",
       "148                                   NaN                   NaN   \n",
       "149                                   NaN                   NaN   \n",
       "150                                   NaN                   NaN   \n",
       "151                                   NaN                   NaN   \n",
       "152                                   NaN                   NaN   \n",
       "\n",
       "    Methylation_quality  \n",
       "0                  PASS  \n",
       "1                  PASS  \n",
       "2                  PASS  \n",
       "3                  PASS  \n",
       "4                  PASS  \n",
       "..                  ...  \n",
       "148                 NaN  \n",
       "149                 NaN  \n",
       "150                 NaN  \n",
       "151                 NaN  \n",
       "152                 NaN  \n",
       "\n",
       "[153 rows x 179 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset_path =  './datasets/dou.csv'\n",
    "dataset = manager.load_dataset(dataset_path)\n",
    "dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6661dabe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>a260_a280_ratio</th>\n",
       "      <th>aa_change</th>\n",
       "      <th>adapter_content</th>\n",
       "      <th>adapter_name</th>\n",
       "      <th>adapter_sequence</th>\n",
       "      <th>additional_pathology_findings</th>\n",
       "      <th>adrenal_hormone</th>\n",
       "      <th>adverse_event</th>\n",
       "      <th>adverse_event_grade</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>...</th>\n",
       "      <th>workflow_start_datetime</th>\n",
       "      <th>workflow_type</th>\n",
       "      <th>workflow_version</th>\n",
       "      <th>year_of_birth</th>\n",
       "      <th>year_of_death</th>\n",
       "      <th>year_of_diagnosis</th>\n",
       "      <th>year_of_follow_up</th>\n",
       "      <th>years_smoked</th>\n",
       "      <th>zone_of_origin_prostate</th>\n",
       "      <th>zygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.85</td>\n",
       "      <td>V600E</td>\n",
       "      <td>FAIL</td>\n",
       "      <td>Illumina_Universal_Adapter</td>\n",
       "      <td>AGATCGGAAGAGCACACGTCTGAACTCCAGTCA</td>\n",
       "      <td>Asbestos bodies</td>\n",
       "      <td>No</td>\n",
       "      <td>Diarrhea</td>\n",
       "      <td>Grade 1</td>\n",
       "      <td>7305</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-01-15T08:30:00</td>\n",
       "      <td>ABSOLUTE LiftOver</td>\n",
       "      <td>1.0.0</td>\n",
       "      <td>1902</td>\n",
       "      <td>2015</td>\n",
       "      <td>2005</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>Central zone</td>\n",
       "      <td>Hemizygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.90</td>\n",
       "      <td>G12D</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Nextera_Transposase_Sequence</td>\n",
       "      <td>CTGTCTCTTATACACATCTCCGAGCCCACGAGAC</td>\n",
       "      <td>Bilateral ovaries with endometriotic cyst and ...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Dry Skin</td>\n",
       "      <td>Grade 2</td>\n",
       "      <td>14610</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-02-20T09:45:00</td>\n",
       "      <td>AscatNGS</td>\n",
       "      <td>1.2.1</td>\n",
       "      <td>1906</td>\n",
       "      <td>2016</td>\n",
       "      <td>2006</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Overlapping/multiple zones</td>\n",
       "      <td>Heterozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.95</td>\n",
       "      <td>T790M</td>\n",
       "      <td>PASS</td>\n",
       "      <td>TruSeq_Universal_Adapter</td>\n",
       "      <td>GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT</td>\n",
       "      <td>Bone marrow concordant histology</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Duodenal Ulcer</td>\n",
       "      <td>Grade 3</td>\n",
       "      <td>8035</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-03-10T14:00:00</td>\n",
       "      <td>BCGSC miRNA Profiling</td>\n",
       "      <td>2.0.0</td>\n",
       "      <td>1910</td>\n",
       "      <td>2017</td>\n",
       "      <td>2007</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>Peripheral zone</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.00</td>\n",
       "      <td>L858R</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOLiD_Adapter</td>\n",
       "      <td>TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG</td>\n",
       "      <td>Clostridioides difficile (c. diff)</td>\n",
       "      <td>No</td>\n",
       "      <td>Generalized Muscle Weakness</td>\n",
       "      <td>Grade 4</td>\n",
       "      <td>21915</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-04-05T16:30:00</td>\n",
       "      <td>DNAcopy</td>\n",
       "      <td>2.5.3</td>\n",
       "      <td>1914</td>\n",
       "      <td>2018</td>\n",
       "      <td>2008</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Not Reported</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2.05</td>\n",
       "      <td>E545K</td>\n",
       "      <td>WARN</td>\n",
       "      <td>Ion_Torrent_Adapter</td>\n",
       "      <td>GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG</td>\n",
       "      <td>Diffuse and early nodular diabetic glomerulosc...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Hearing Impaired</td>\n",
       "      <td>Grade 5</td>\n",
       "      <td>25550</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-05-25T11:15:00</td>\n",
       "      <td>GISTIC - Focal Deletion</td>\n",
       "      <td>3.0.0</td>\n",
       "      <td>1918</td>\n",
       "      <td>2019</td>\n",
       "      <td>2009</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>Unknown zone</td>\n",
       "      <td>Nullizygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.10</td>\n",
       "      <td>R175H</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>P5_Dual_Index_Adapter</td>\n",
       "      <td>AATGATACGGCGACCACCGAGATCTACAC</td>\n",
       "      <td>Dysplasia; high grade</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Hematosalpinx</td>\n",
       "      <td>Grade 4</td>\n",
       "      <td>18250</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-06-30T13:45:00</td>\n",
       "      <td>HTSeq - Counts</td>\n",
       "      <td>3.1.4</td>\n",
       "      <td>1922</td>\n",
       "      <td>2020</td>\n",
       "      <td>2010</td>\n",
       "      <td>6</td>\n",
       "      <td>15</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.80</td>\n",
       "      <td>Y220C</td>\n",
       "      <td>WARN</td>\n",
       "      <td>P7_Multiplexing_Sequence</td>\n",
       "      <td>CAAGCAGAAGACGGCATACGAGAT</td>\n",
       "      <td>Epithelial hyperplasia</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Optic Nerve Disorder</td>\n",
       "      <td>Grade 2</td>\n",
       "      <td>29220</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-07-22T10:00:00</td>\n",
       "      <td>Pindel Variant Aggregation and Masking</td>\n",
       "      <td>4.0.0</td>\n",
       "      <td>1926</td>\n",
       "      <td>2021</td>\n",
       "      <td>2011</td>\n",
       "      <td>7</td>\n",
       "      <td>20</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Hemizygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1.75</td>\n",
       "      <td>D816V</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>MiSeq_Adapter</td>\n",
       "      <td>GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT</td>\n",
       "      <td>Keratinizing dysplasia; severe (carcinoma in s...</td>\n",
       "      <td>No</td>\n",
       "      <td>Scalp Pain</td>\n",
       "      <td>Grade 2</td>\n",
       "      <td>6570</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-08-18T15:30:00</td>\n",
       "      <td>Read Group Quality Control</td>\n",
       "      <td>4.2.0</td>\n",
       "      <td>1930</td>\n",
       "      <td>2022</td>\n",
       "      <td>2012</td>\n",
       "      <td>8</td>\n",
       "      <td>25</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.15</td>\n",
       "      <td>KIT_D816V</td>\n",
       "      <td>FAIL</td>\n",
       "      <td>HiSeq_X_Ten_Adapter</td>\n",
       "      <td>AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG</td>\n",
       "      <td>Other Complex Extravascular Matrix Patterns</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Skin and Subcutaneous Tissue Disorders - Other</td>\n",
       "      <td>Grade 4</td>\n",
       "      <td>9490</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-09-05T09:00:00</td>\n",
       "      <td>STAR - Counts</td>\n",
       "      <td>5.0.0</td>\n",
       "      <td>1934</td>\n",
       "      <td>2023</td>\n",
       "      <td>2013</td>\n",
       "      <td>9</td>\n",
       "      <td>30</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Nullizygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2.20</td>\n",
       "      <td>C797S</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>NovaSeq_6000_Adapter</td>\n",
       "      <td>CGACAGGTTCAGAGTTCTACAGTCCGACGATC</td>\n",
       "      <td>Platinum-resistant</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Spermatic Cord Hemorrhage</td>\n",
       "      <td>Grade 4</td>\n",
       "      <td>12775</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-10-13T14:15:00</td>\n",
       "      <td>VarScan2</td>\n",
       "      <td>5.3.2</td>\n",
       "      <td>1938</td>\n",
       "      <td>2024</td>\n",
       "      <td>2014</td>\n",
       "      <td>10</td>\n",
       "      <td>35</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Nullizygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1.70</td>\n",
       "      <td>E17K</td>\n",
       "      <td>FAIL</td>\n",
       "      <td>PacBio_SMRTbell_Adapter</td>\n",
       "      <td>TCTTCCGATCTTAGGCATCTCCTATGTAGGC</td>\n",
       "      <td>Dysplasia; high grade</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Diarrhea</td>\n",
       "      <td>Grade 3</td>\n",
       "      <td>16425</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-11-02T08:45:00</td>\n",
       "      <td>AscatNGS</td>\n",
       "      <td>6.0.0</td>\n",
       "      <td>1942</td>\n",
       "      <td>2025</td>\n",
       "      <td>2015</td>\n",
       "      <td>11</td>\n",
       "      <td>40</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1.65</td>\n",
       "      <td>R249S</td>\n",
       "      <td>PASS</td>\n",
       "      <td>Oxford_Nanopore_Adapter</td>\n",
       "      <td>GATCTCGTATGCCGTCTTCTGCTTGAAA</td>\n",
       "      <td>Clostridioides difficile (c. diff)</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Skin and Subcutaneous Tissue Disorders - Other</td>\n",
       "      <td>Grade 2</td>\n",
       "      <td>23725</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-12-07T16:00:00</td>\n",
       "      <td>VarScan2</td>\n",
       "      <td>6.1.1</td>\n",
       "      <td>1946</td>\n",
       "      <td>2026</td>\n",
       "      <td>2016</td>\n",
       "      <td>12</td>\n",
       "      <td>45</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2.25</td>\n",
       "      <td>N345K</td>\n",
       "      <td>FAIL</td>\n",
       "      <td>Roche_454_Adapter</td>\n",
       "      <td>CAGACGTGTGCTCTTCCGATCTGATC</td>\n",
       "      <td>Diffuse and early nodular diabetic glomerulosc...</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Dry Skin</td>\n",
       "      <td>Grade 1</td>\n",
       "      <td>29200</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-01-28T10:30:00</td>\n",
       "      <td>BCGSC miRNA Profiling</td>\n",
       "      <td>7.0.0</td>\n",
       "      <td>1950</td>\n",
       "      <td>2027</td>\n",
       "      <td>2017</td>\n",
       "      <td>13</td>\n",
       "      <td>50</td>\n",
       "      <td>Overlapping/multiple zones</td>\n",
       "      <td>Homozygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1.60</td>\n",
       "      <td>G13C</td>\n",
       "      <td>FAIL</td>\n",
       "      <td>Mega_RAP_Adapter</td>\n",
       "      <td>TCGTATGCCGTCTTCTGCTTGATCG</td>\n",
       "      <td>Dysplasia; high grade</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Skin and Subcutaneous Tissue Disorders - Other</td>\n",
       "      <td>Grade 3</td>\n",
       "      <td>32850</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-02-15T15:45:00</td>\n",
       "      <td>BCGSC miRNA Profiling</td>\n",
       "      <td>7.4.5</td>\n",
       "      <td>1954</td>\n",
       "      <td>2028</td>\n",
       "      <td>2018</td>\n",
       "      <td>14</td>\n",
       "      <td>55</td>\n",
       "      <td>Overlapping/multiple zones</td>\n",
       "      <td>Hemizygous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2.30</td>\n",
       "      <td>P13L</td>\n",
       "      <td>FAIL</td>\n",
       "      <td>SureSelectXT_Adapter</td>\n",
       "      <td>AGACGTGTGCTCTTCCGATCATGCA</td>\n",
       "      <td>Dysplasia; high grade</td>\n",
       "      <td>No</td>\n",
       "      <td>Generalized Muscle Weakness</td>\n",
       "      <td>Grade 4</td>\n",
       "      <td>36500</td>\n",
       "      <td>...</td>\n",
       "      <td>2023-03-23T12:30:00</td>\n",
       "      <td>HTSeq - Counts</td>\n",
       "      <td>8.0.0</td>\n",
       "      <td>1958</td>\n",
       "      <td>2029</td>\n",
       "      <td>2019</td>\n",
       "      <td>15</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Transition zone</td>\n",
       "      <td>Nullizygous</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15 rows × 734 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    a260_a280_ratio  aa_change adapter_content                  adapter_name  \\\n",
       "0              1.85      V600E            FAIL    Illumina_Universal_Adapter   \n",
       "1              1.90       G12D    Not Reported  Nextera_Transposase_Sequence   \n",
       "2              1.95      T790M            PASS      TruSeq_Universal_Adapter   \n",
       "3              2.00      L858R         Unknown                 SOLiD_Adapter   \n",
       "4              2.05      E545K            WARN           Ion_Torrent_Adapter   \n",
       "5              2.10      R175H    Not Reported         P5_Dual_Index_Adapter   \n",
       "6              1.80      Y220C            WARN      P7_Multiplexing_Sequence   \n",
       "7              1.75      D816V    Not Reported                 MiSeq_Adapter   \n",
       "8              2.15  KIT_D816V            FAIL           HiSeq_X_Ten_Adapter   \n",
       "9              2.20      C797S    Not Reported          NovaSeq_6000_Adapter   \n",
       "10             1.70       E17K            FAIL       PacBio_SMRTbell_Adapter   \n",
       "11             1.65      R249S            PASS       Oxford_Nanopore_Adapter   \n",
       "12             2.25      N345K            FAIL             Roche_454_Adapter   \n",
       "13             1.60       G13C            FAIL              Mega_RAP_Adapter   \n",
       "14             2.30       P13L            FAIL          SureSelectXT_Adapter   \n",
       "\n",
       "                      adapter_sequence  \\\n",
       "0    AGATCGGAAGAGCACACGTCTGAACTCCAGTCA   \n",
       "1   CTGTCTCTTATACACATCTCCGAGCCCACGAGAC   \n",
       "2     GATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT   \n",
       "3    TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG   \n",
       "4   GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG   \n",
       "5        AATGATACGGCGACCACCGAGATCTACAC   \n",
       "6             CAAGCAGAAGACGGCATACGAGAT   \n",
       "7   GTGACTGGAGTTCAGACGTGTGCTCTTCCGATCT   \n",
       "8    AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAG   \n",
       "9     CGACAGGTTCAGAGTTCTACAGTCCGACGATC   \n",
       "10     TCTTCCGATCTTAGGCATCTCCTATGTAGGC   \n",
       "11        GATCTCGTATGCCGTCTTCTGCTTGAAA   \n",
       "12          CAGACGTGTGCTCTTCCGATCTGATC   \n",
       "13           TCGTATGCCGTCTTCTGCTTGATCG   \n",
       "14           AGACGTGTGCTCTTCCGATCATGCA   \n",
       "\n",
       "                        additional_pathology_findings adrenal_hormone  \\\n",
       "0                                     Asbestos bodies              No   \n",
       "1   Bilateral ovaries with endometriotic cyst and ...    Not Reported   \n",
       "2                    Bone marrow concordant histology             Yes   \n",
       "3                  Clostridioides difficile (c. diff)              No   \n",
       "4   Diffuse and early nodular diabetic glomerulosc...    Not Reported   \n",
       "5                               Dysplasia; high grade    Not Reported   \n",
       "6                              Epithelial hyperplasia             Yes   \n",
       "7   Keratinizing dysplasia; severe (carcinoma in s...              No   \n",
       "8         Other Complex Extravascular Matrix Patterns             Yes   \n",
       "9                                  Platinum-resistant    Not Reported   \n",
       "10                              Dysplasia; high grade    Not Reported   \n",
       "11                 Clostridioides difficile (c. diff)    Not Reported   \n",
       "12  Diffuse and early nodular diabetic glomerulosc...    Not Reported   \n",
       "13                              Dysplasia; high grade             Yes   \n",
       "14                              Dysplasia; high grade              No   \n",
       "\n",
       "                                     adverse_event adverse_event_grade  \\\n",
       "0                                         Diarrhea             Grade 1   \n",
       "1                                         Dry Skin             Grade 2   \n",
       "2                                   Duodenal Ulcer             Grade 3   \n",
       "3                      Generalized Muscle Weakness             Grade 4   \n",
       "4                                 Hearing Impaired             Grade 5   \n",
       "5                                    Hematosalpinx             Grade 4   \n",
       "6                             Optic Nerve Disorder             Grade 2   \n",
       "7                                       Scalp Pain             Grade 2   \n",
       "8   Skin and Subcutaneous Tissue Disorders - Other             Grade 4   \n",
       "9                        Spermatic Cord Hemorrhage             Grade 4   \n",
       "10                                        Diarrhea             Grade 3   \n",
       "11  Skin and Subcutaneous Tissue Disorders - Other             Grade 2   \n",
       "12                                        Dry Skin             Grade 1   \n",
       "13  Skin and Subcutaneous Tissue Disorders - Other             Grade 3   \n",
       "14                     Generalized Muscle Weakness             Grade 4   \n",
       "\n",
       "    age_at_diagnosis  ...  workflow_start_datetime  \\\n",
       "0               7305  ...      2023-01-15T08:30:00   \n",
       "1              14610  ...      2023-02-20T09:45:00   \n",
       "2               8035  ...      2023-03-10T14:00:00   \n",
       "3              21915  ...      2023-04-05T16:30:00   \n",
       "4              25550  ...      2023-05-25T11:15:00   \n",
       "5              18250  ...      2023-06-30T13:45:00   \n",
       "6              29220  ...      2023-07-22T10:00:00   \n",
       "7               6570  ...      2023-08-18T15:30:00   \n",
       "8               9490  ...      2023-09-05T09:00:00   \n",
       "9              12775  ...      2023-10-13T14:15:00   \n",
       "10             16425  ...      2023-11-02T08:45:00   \n",
       "11             23725  ...      2023-12-07T16:00:00   \n",
       "12             29200  ...      2023-01-28T10:30:00   \n",
       "13             32850  ...      2023-02-15T15:45:00   \n",
       "14             36500  ...      2023-03-23T12:30:00   \n",
       "\n",
       "                             workflow_type  workflow_version  year_of_birth  \\\n",
       "0                        ABSOLUTE LiftOver             1.0.0           1902   \n",
       "1                                 AscatNGS             1.2.1           1906   \n",
       "2                    BCGSC miRNA Profiling             2.0.0           1910   \n",
       "3                                  DNAcopy             2.5.3           1914   \n",
       "4                  GISTIC - Focal Deletion             3.0.0           1918   \n",
       "5                           HTSeq - Counts             3.1.4           1922   \n",
       "6   Pindel Variant Aggregation and Masking             4.0.0           1926   \n",
       "7               Read Group Quality Control             4.2.0           1930   \n",
       "8                            STAR - Counts             5.0.0           1934   \n",
       "9                                 VarScan2             5.3.2           1938   \n",
       "10                                AscatNGS             6.0.0           1942   \n",
       "11                                VarScan2             6.1.1           1946   \n",
       "12                   BCGSC miRNA Profiling             7.0.0           1950   \n",
       "13                   BCGSC miRNA Profiling             7.4.5           1954   \n",
       "14                          HTSeq - Counts             8.0.0           1958   \n",
       "\n",
       "   year_of_death year_of_diagnosis year_of_follow_up years_smoked  \\\n",
       "0           2015              2005                 1            0   \n",
       "1           2016              2006                 2            1   \n",
       "2           2017              2007                 3            2   \n",
       "3           2018              2008                 4            5   \n",
       "4           2019              2009                 5           10   \n",
       "5           2020              2010                 6           15   \n",
       "6           2021              2011                 7           20   \n",
       "7           2022              2012                 8           25   \n",
       "8           2023              2013                 9           30   \n",
       "9           2024              2014                10           35   \n",
       "10          2025              2015                11           40   \n",
       "11          2026              2016                12           45   \n",
       "12          2027              2017                13           50   \n",
       "13          2028              2018                14           55   \n",
       "14          2029              2019                15      unknown   \n",
       "\n",
       "       zone_of_origin_prostate      zygosity  \n",
       "0                 Central zone    Hemizygous  \n",
       "1   Overlapping/multiple zones  Heterozygous  \n",
       "2              Peripheral zone    Homozygous  \n",
       "3              Transition zone  Not Reported  \n",
       "4                 Unknown zone   Nullizygous  \n",
       "5              Transition zone       Unknown  \n",
       "6              Transition zone    Hemizygous  \n",
       "7              Transition zone       Unknown  \n",
       "8              Transition zone   Nullizygous  \n",
       "9              Transition zone   Nullizygous  \n",
       "10             Transition zone       Unknown  \n",
       "11             Transition zone       Unknown  \n",
       "12  Overlapping/multiple zones    Homozygous  \n",
       "13  Overlapping/multiple zones    Hemizygous  \n",
       "14             Transition zone   Nullizygous  \n",
       "\n",
       "[15 rows x 734 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "manager.global_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d4f458b-610e-4911-b386-53a8261f5188",
   "metadata": {},
   "source": [
    "## Column Mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ecbdaa25-5731-45fe-bc3a-cd3b3a7b6080",
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/Users/rlopez/ASKEM/askem-arpa-h-project/bdi/mapping_algorithms/scope_reducing/_algorithms/contrastive_learning/models/model_20_1.pt'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mmanager\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mreduce_scope\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      2\u001b[0m column_mappings \u001b[38;5;241m=\u001b[39m manager\u001b[38;5;241m.\u001b[39mmap_columns()\n\u001b[1;32m      3\u001b[0m column_mappings\n",
      "File \u001b[0;32m~/ASKEM/askem-arpa-h-project/bdi/api.py:45\u001b[0m, in \u001b[0;36mAPIManager.reduce_scope\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m     44\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mreduce_scope\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m---> 45\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mscope_manager \u001b[38;5;241m=\u001b[39m \u001b[43mScopeReducingManager\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     46\u001b[0m \u001b[43m        \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdataset\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mglobal_table\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     47\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mreduced_scope \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mscope_manager\u001b[38;5;241m.\u001b[39mreduce()\n\u001b[1;32m     48\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mreduced_scope\n",
      "File \u001b[0;32m~/ASKEM/askem-arpa-h-project/bdi/mapping_recommendation/scope_reducing_manager.py:8\u001b[0m, in \u001b[0;36mScopeReducingManager.__init__\u001b[0;34m(self, dataset, target_domain)\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdataset \u001b[38;5;241m=\u001b[39m dataset\n\u001b[1;32m      7\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtarget_domain \u001b[38;5;241m=\u001b[39m target_domain\n\u001b[0;32m----> 8\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbest_method \u001b[38;5;241m=\u001b[39m \u001b[43mYurongReducer\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/ASKEM/askem-arpa-h-project/bdi/mapping_algorithms/scope_reducing/algorithms.py:17\u001b[0m, in \u001b[0;36mYurongReducer.__init__\u001b[0;34m(self, model_path)\u001b[0m\n\u001b[1;32m     15\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28mself\u001b[39m, model_path\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmodels/model_20_1.pt\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[1;32m     16\u001b[0m     \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__init__\u001b[39m()\n\u001b[0;32m---> 17\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mapi \u001b[38;5;241m=\u001b[39m \u001b[43mContrastiveLearningAPI\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/ASKEM/askem-arpa-h-project/bdi/mapping_algorithms/scope_reducing/_algorithms/contrastive_learning/cl_api.py:25\u001b[0m, in \u001b[0;36mContrastiveLearningAPI.__init__\u001b[0;34m(self, model_path, top_k, batch_size)\u001b[0m\n\u001b[1;32m     23\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mbatch_size \u001b[38;5;241m=\u001b[39m batch_size\n\u001b[1;32m     24\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdevice \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcuda\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m torch\u001b[38;5;241m.\u001b[39mcuda\u001b[38;5;241m.\u001b[39mis_available() \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcpu\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m---> 25\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmodel \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mload_checkpoint\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     26\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtop_k \u001b[38;5;241m=\u001b[39m top_k\n",
      "File \u001b[0;32m~/ASKEM/askem-arpa-h-project/bdi/mapping_algorithms/scope_reducing/_algorithms/contrastive_learning/cl_api.py:29\u001b[0m, in \u001b[0;36mContrastiveLearningAPI.load_checkpoint\u001b[0;34m(self, lm)\u001b[0m\n\u001b[1;32m     28\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mload_checkpoint\u001b[39m(\u001b[38;5;28mself\u001b[39m, lm\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mroberta\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[0;32m---> 29\u001b[0m     ckpt \u001b[38;5;241m=\u001b[39m \u001b[43mtorch\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mload\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmodel_path\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmap_location\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtorch\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdevice\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcpu\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     30\u001b[0m     scale_loss \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0.1\u001b[39m\n\u001b[1;32m     31\u001b[0m     lambd \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m3.9\u001b[39m\n",
      "File \u001b[0;32m~/opt/anaconda3/envs/askem/lib/python3.10/site-packages/torch/serialization.py:998\u001b[0m, in \u001b[0;36mload\u001b[0;34m(f, map_location, pickle_module, weights_only, mmap, **pickle_load_args)\u001b[0m\n\u001b[1;32m    995\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mencoding\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m pickle_load_args\u001b[38;5;241m.\u001b[39mkeys():\n\u001b[1;32m    996\u001b[0m     pickle_load_args[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mencoding\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mutf-8\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[0;32m--> 998\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[43m_open_file_like\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mrb\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m \u001b[38;5;28;01mas\u001b[39;00m opened_file:\n\u001b[1;32m    999\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m _is_zipfile(opened_file):\n\u001b[1;32m   1000\u001b[0m         \u001b[38;5;66;03m# The zipfile reader is going to advance the current file position.\u001b[39;00m\n\u001b[1;32m   1001\u001b[0m         \u001b[38;5;66;03m# If we want to actually tail call to torch.jit.load, we need to\u001b[39;00m\n\u001b[1;32m   1002\u001b[0m         \u001b[38;5;66;03m# reset back to the original position.\u001b[39;00m\n\u001b[1;32m   1003\u001b[0m         orig_position \u001b[38;5;241m=\u001b[39m opened_file\u001b[38;5;241m.\u001b[39mtell()\n",
      "File \u001b[0;32m~/opt/anaconda3/envs/askem/lib/python3.10/site-packages/torch/serialization.py:445\u001b[0m, in \u001b[0;36m_open_file_like\u001b[0;34m(name_or_buffer, mode)\u001b[0m\n\u001b[1;32m    443\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_open_file_like\u001b[39m(name_or_buffer, mode):\n\u001b[1;32m    444\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m _is_path(name_or_buffer):\n\u001b[0;32m--> 445\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_open_file\u001b[49m\u001b[43m(\u001b[49m\u001b[43mname_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    446\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    447\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;129;01min\u001b[39;00m mode:\n",
      "File \u001b[0;32m~/opt/anaconda3/envs/askem/lib/python3.10/site-packages/torch/serialization.py:426\u001b[0m, in \u001b[0;36m_open_file.__init__\u001b[0;34m(self, name, mode)\u001b[0m\n\u001b[1;32m    425\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28mself\u001b[39m, name, mode):\n\u001b[0;32m--> 426\u001b[0m     \u001b[38;5;28msuper\u001b[39m()\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mname\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m)\u001b[49m)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/Users/rlopez/ASKEM/askem-arpa-h-project/bdi/mapping_algorithms/scope_reducing/_algorithms/contrastive_learning/models/model_20_1.pt'"
     ]
    }
   ],
   "source": [
    "\n",
    "manager.reduce_scope()\n",
    "column_mappings = manager.map_columns()\n",
    "column_mappings"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81d7a0fd-3077-43e3-9a91-69e6a280075a",
   "metadata": {},
   "source": [
    "## Value Mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ed0ff201-3984-4b3b-8a0f-eacd91c56bac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# manager.map_values()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "445405ab-782e-4168-a5b7-8044f0aeb45f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
