{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b6a5938c-d32e-4816-ba31-a0b54e4a9826",
   "metadata": {},
   "source": [
    "# Data Integration with BDI"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1af374aa-f7b2-4329-b605-faf71c17e6b0",
   "metadata": {},
   "source": [
    "First, import the class `APIManager`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f01f77a6-f349-4f71-a847-c26b69a60c41",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/rlopez/opt/anaconda3/envs/askem/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "from bdi import APIManager"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f1d11fa-9ff2-4e9e-8548-48d58109d9a3",
   "metadata": {},
   "source": [
    "Add the path to the pre-trained model for mapping recommendations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8214c039-1189-4a2f-b1e4-899b221bcff8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ['BDI_MODEL_PATH'] = '/Users/rlopez/Downloads/model_20_1.pt' #YOUR PATH HERE"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb0b6170-c8e5-4ff3-9948-a55acb913563",
   "metadata": {},
   "source": [
    "## Dataset Loading"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee01a713-6cea-42a7-9a64-7aeaba7b0857",
   "metadata": {},
   "source": [
    "In this example, we are mapping the Dou dataset to the GDC format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "99afdf9a-b68a-4a30-9735-f8ef36027757",
   "metadata": {},
   "outputs": [],
   "source": [
    "manager = APIManager()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "34e5d030-a50d-4076-9e96-7e4e73fe7cda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>idx</th>\n",
       "      <th>Proteomics_Participant_ID</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>Proteomics_TMT_batch</th>\n",
       "      <th>Proteomics_TMT_plex</th>\n",
       "      <th>Proteomics_TMT_channel</th>\n",
       "      <th>Proteomics_Parent_Sample_IDs</th>\n",
       "      <th>Proteomics_Aliquot_ID</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>Proteomics_OCT</th>\n",
       "      <th>...</th>\n",
       "      <th>RNAseq_R1_sample_type</th>\n",
       "      <th>RNAseq_R1_filename</th>\n",
       "      <th>RNAseq_R1_UUID</th>\n",
       "      <th>RNAseq_R2_sample_type</th>\n",
       "      <th>RNAseq_R2_filename</th>\n",
       "      <th>RNAseq_R2_UUID</th>\n",
       "      <th>miRNAseq_sample_type</th>\n",
       "      <th>miRNAseq_UUID</th>\n",
       "      <th>Methylation_available</th>\n",
       "      <th>Methylation_quality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>No</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>128N</td>\n",
       "      <td>C3L-00006-01</td>\n",
       "      <td>CPT0001460012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8a1efc47-1c29-417f-a425-cdbd09565dcb</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>37bcba98-1094-459e-83ae-c23a602416fb</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S002</td>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>No</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>130N</td>\n",
       "      <td>C3L-00008-01</td>\n",
       "      <td>CPT0001300009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>555725e8-cba5-4676-9b0a-80100cbf9f47</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>15235b12-b67a-4678-acc4-ed03d642bd5e</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>492b50d8-ec35-46e7-a65d-06512aaee394</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S003</td>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>No</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>131</td>\n",
       "      <td>C3L-00032-01</td>\n",
       "      <td>CPT0001420009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>9ae968f3-691d-4db3-9977-1ab3e5af9085</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>423b6b09-02aa-4f47-9241-f75c1dad1161</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>1794ff56-db2d-4d1a-8758-cab7fe3d98c1</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S004</td>\n",
       "      <td>C3L-00084</td>\n",
       "      <td>Yes</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>129N</td>\n",
       "      <td>C3L-00084-01</td>\n",
       "      <td>CPT0000820012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>b0a7cdf2-2ad8-4442-91b0-548ea4975554</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>c83987a5-1c13-4af4-b46c-218fe5f60c34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S005</td>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>No</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>129C</td>\n",
       "      <td>C3L-00090-01</td>\n",
       "      <td>CPT0001140003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>8ce5618d-9ff6-40f9-aeea-8d8e1633ae38</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>06d3fd4a-a623-4146-8500-4f1f17235253</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>a6524c2d-d7dd-4629-980e-b45dbdc92c49</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>S149</td>\n",
       "      <td>NX14</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>128C</td>\n",
       "      <td>C3N-03699-01,C3N-03699-02</td>\n",
       "      <td>CPT0230130002,CPT0230130003,CPT0230130004,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>S150</td>\n",
       "      <td>NX15</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>131</td>\n",
       "      <td>C3N-03700-01</td>\n",
       "      <td>CPT0230190002,CPT0230190003,CPT0230190004</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>S151</td>\n",
       "      <td>NX16</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>129C</td>\n",
       "      <td>C3N-03701-01,C3N-03701-02,C3N-03701-03</td>\n",
       "      <td>CPT0230250002,CPT0230260002,CPT0230270002,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>S152</td>\n",
       "      <td>NX17</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>127C</td>\n",
       "      <td>C3N-03706-01,C3N-03706-02,C3N-03706-03</td>\n",
       "      <td>CPT0230400002,CPT0230400003,CPT0230400004,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>S153</td>\n",
       "      <td>NX18</td>\n",
       "      <td>No</td>\n",
       "      <td>5</td>\n",
       "      <td>17</td>\n",
       "      <td>130N</td>\n",
       "      <td>C3N-03708-01,C3N-03708-02,C3N-03708-03</td>\n",
       "      <td>CPT0230460002,CPT0230460003,CPT0230460004,CPT0...</td>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      idx Proteomics_Participant_ID Case_excluded  Proteomics_TMT_batch  \\\n",
       "0    S001                 C3L-00006            No                     2   \n",
       "1    S002                 C3L-00008            No                     4   \n",
       "2    S003                 C3L-00032            No                     1   \n",
       "3    S004                 C3L-00084           Yes                     3   \n",
       "4    S005                 C3L-00090            No                     3   \n",
       "..    ...                       ...           ...                   ...   \n",
       "148  S149                      NX14            No                     5   \n",
       "149  S150                      NX15            No                     5   \n",
       "150  S151                      NX16            No                     5   \n",
       "151  S152                      NX17            No                     5   \n",
       "152  S153                      NX18            No                     5   \n",
       "\n",
       "     Proteomics_TMT_plex Proteomics_TMT_channel  \\\n",
       "0                      5                   128N   \n",
       "1                     16                   130N   \n",
       "2                      2                    131   \n",
       "3                     11                   129N   \n",
       "4                     12                   129C   \n",
       "..                   ...                    ...   \n",
       "148                   17                   128C   \n",
       "149                   17                    131   \n",
       "150                   17                   129C   \n",
       "151                   17                   127C   \n",
       "152                   17                   130N   \n",
       "\n",
       "               Proteomics_Parent_Sample_IDs  \\\n",
       "0                              C3L-00006-01   \n",
       "1                              C3L-00008-01   \n",
       "2                              C3L-00032-01   \n",
       "3                              C3L-00084-01   \n",
       "4                              C3L-00090-01   \n",
       "..                                      ...   \n",
       "148               C3N-03699-01,C3N-03699-02   \n",
       "149                            C3N-03700-01   \n",
       "150  C3N-03701-01,C3N-03701-02,C3N-03701-03   \n",
       "151  C3N-03706-01,C3N-03706-02,C3N-03706-03   \n",
       "152  C3N-03708-01,C3N-03708-02,C3N-03708-03   \n",
       "\n",
       "                                 Proteomics_Aliquot_ID  \\\n",
       "0                                        CPT0001460012   \n",
       "1                                        CPT0001300009   \n",
       "2                                        CPT0001420009   \n",
       "3                                        CPT0000820012   \n",
       "4                                        CPT0001140003   \n",
       "..                                                 ...   \n",
       "148  CPT0230130002,CPT0230130003,CPT0230130004,CPT0...   \n",
       "149          CPT0230190002,CPT0230190003,CPT0230190004   \n",
       "150  CPT0230250002,CPT0230260002,CPT0230270002,CPT0...   \n",
       "151  CPT0230400002,CPT0230400003,CPT0230400004,CPT0...   \n",
       "152  CPT0230460002,CPT0230460003,CPT0230460004,CPT0...   \n",
       "\n",
       "    Proteomics_Tumor_Normal Proteomics_OCT  ... RNAseq_R1_sample_type  \\\n",
       "0                     Tumor             No  ...                 Tumor   \n",
       "1                     Tumor             No  ...                 Tumor   \n",
       "2                     Tumor             No  ...                 Tumor   \n",
       "3                     Tumor             No  ...                 Tumor   \n",
       "4                     Tumor             No  ...                 Tumor   \n",
       "..                      ...            ...  ...                   ...   \n",
       "148         Enriched_normal             No  ...                   NaN   \n",
       "149         Enriched_normal             No  ...                   NaN   \n",
       "150         Enriched_normal             No  ...                   NaN   \n",
       "151         Enriched_normal             No  ...                   NaN   \n",
       "152         Enriched_normal             No  ...                   NaN   \n",
       "\n",
       "                                    RNAseq_R1_filename  \\\n",
       "0    170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "1    170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "2    170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "3    170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "4    170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "..                                                 ...   \n",
       "148                                                NaN   \n",
       "149                                                NaN   \n",
       "150                                                NaN   \n",
       "151                                                NaN   \n",
       "152                                                NaN   \n",
       "\n",
       "                           RNAseq_R1_UUID RNAseq_R2_sample_type  \\\n",
       "0    8a1efc47-1c29-417f-a425-cdbd09565dcb                 Tumor   \n",
       "1    555725e8-cba5-4676-9b0a-80100cbf9f47                 Tumor   \n",
       "2    9ae968f3-691d-4db3-9977-1ab3e5af9085                 Tumor   \n",
       "3    b0a7cdf2-2ad8-4442-91b0-548ea4975554                 Tumor   \n",
       "4    8ce5618d-9ff6-40f9-aeea-8d8e1633ae38                 Tumor   \n",
       "..                                    ...                   ...   \n",
       "148                                   NaN                   NaN   \n",
       "149                                   NaN                   NaN   \n",
       "150                                   NaN                   NaN   \n",
       "151                                   NaN                   NaN   \n",
       "152                                   NaN                   NaN   \n",
       "\n",
       "                                    RNAseq_R2_filename  \\\n",
       "0    170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "1    170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "2    170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "3    170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "4    170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "..                                                 ...   \n",
       "148                                                NaN   \n",
       "149                                                NaN   \n",
       "150                                                NaN   \n",
       "151                                                NaN   \n",
       "152                                                NaN   \n",
       "\n",
       "                           RNAseq_R2_UUID miRNAseq_sample_type  \\\n",
       "0    8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda                Tumor   \n",
       "1    15235b12-b67a-4678-acc4-ed03d642bd5e                Tumor   \n",
       "2    423b6b09-02aa-4f47-9241-f75c1dad1161                Tumor   \n",
       "3    c83987a5-1c13-4af4-b46c-218fe5f60c34                  NaN   \n",
       "4    06d3fd4a-a623-4146-8500-4f1f17235253                Tumor   \n",
       "..                                    ...                  ...   \n",
       "148                                   NaN                  NaN   \n",
       "149                                   NaN                  NaN   \n",
       "150                                   NaN                  NaN   \n",
       "151                                   NaN                  NaN   \n",
       "152                                   NaN                  NaN   \n",
       "\n",
       "                            miRNAseq_UUID Methylation_available  \\\n",
       "0    37bcba98-1094-459e-83ae-c23a602416fb                   YES   \n",
       "1    492b50d8-ec35-46e7-a65d-06512aaee394                   YES   \n",
       "2    1794ff56-db2d-4d1a-8758-cab7fe3d98c1                   YES   \n",
       "3                                     NaN                   YES   \n",
       "4    a6524c2d-d7dd-4629-980e-b45dbdc92c49                   YES   \n",
       "..                                    ...                   ...   \n",
       "148                                   NaN                   NaN   \n",
       "149                                   NaN                   NaN   \n",
       "150                                   NaN                   NaN   \n",
       "151                                   NaN                   NaN   \n",
       "152                                   NaN                   NaN   \n",
       "\n",
       "    Methylation_quality  \n",
       "0                  PASS  \n",
       "1                  PASS  \n",
       "2                  PASS  \n",
       "3                  PASS  \n",
       "4                  PASS  \n",
       "..                  ...  \n",
       "148                 NaN  \n",
       "149                 NaN  \n",
       "150                 NaN  \n",
       "151                 NaN  \n",
       "152                 NaN  \n",
       "\n",
       "[153 rows x 179 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dataset_path =  './datasets/dou.csv'\n",
    "dataset = manager.load_dataset(dataset_path)\n",
    "dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d4f458b-610e-4911-b386-53a8261f5188",
   "metadata": {},
   "source": [
    "## Column Mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ca7e99b8-d571-4d66-9737-1ab0fd4d6d2c",
   "metadata": {},
   "source": [
    "Reduce the scope of GDC selecting top k candidates for each column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "18eedf76-fbd9-4b86-8f1e-3f8c7a222e43",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some weights of RobertaModel were not initialized from the model checkpoint at roberta-base and are newly initialized: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n",
      "  0%|                                                                                                                                                                                                                                                  | 0/179 [00:00<?, ?it/s]We strongly recommend passing in an `attention_mask` since your input_ids may be padded. See https://huggingface.co/docs/transformers/troubleshooting#incorrect-output-when-padding-tokens-arent-masked.\n",
      "100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 179/179 [00:03<00:00, 50.28it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Table features extracted from 179 columns\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 734/734 [00:16<00:00, 45.44it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Table features extracted from 734 columns\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[{'Candidate column': 'idx',\n",
       "  'Top k columns': [('id', '0.9042'),\n",
       "   ('doi', '0.6512'),\n",
       "   ('name', '0.6272'),\n",
       "   ('dbgap_accession_number', '0.6259'),\n",
       "   ('proc_internal', '0.616'),\n",
       "   ('project_id', '0.6102'),\n",
       "   ('status', '0.59'),\n",
       "   ('code', '0.5863'),\n",
       "   ('undescended_testis_corrected_method', '0.58'),\n",
       "   ('copy_number', '0.5741')]},\n",
       " {'Candidate column': 'Proteomics_Participant_ID',\n",
       "  'Top k columns': [('test_analyte_type', '0.4357'),\n",
       "   ('analyte_type', '0.3729'),\n",
       "   ('histologic_progression_type', '0.3727'),\n",
       "   ('analyte_type_id', '0.3701'),\n",
       "   ('clinical_trial_indicator', '0.3636'),\n",
       "   ('secondary_expression_analyses', '0.3574'),\n",
       "   ('experiment_name', '0.3553'),\n",
       "   ('analytes', '0.3552'),\n",
       "   ('hormonal_contraceptive_type', '0.3545'),\n",
       "   ('perineural_invasion_present', '0.3471')]},\n",
       " {'Candidate column': 'Case_excluded',\n",
       "  'Top k columns': [('cases', '0.6179'),\n",
       "   ('described_cases', '0.541'),\n",
       "   ('undescended_testis_history', '0.4446'),\n",
       "   ('undescended_testis_history_laterality', '0.4329'),\n",
       "   ('zone_of_origin_prostate', '0.428'),\n",
       "   ('undescended_testis_corrected_laterality', '0.4068'),\n",
       "   ('undescended_testis_corrected', '0.4036'),\n",
       "   ('undescended_testis_corrected_method', '0.3872'),\n",
       "   ('method_of_sample_procurement', '0.3789'),\n",
       "   ('procedures_performed', '0.3769')]},\n",
       " {'Candidate column': 'Proteomics_TMT_batch',\n",
       "  'Top k columns': [('analyte_type_id', '0.4612'),\n",
       "   ('tumor_infiltrating_macrophages', '0.4482'),\n",
       "   ('analyte_type', '0.442'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.4092'),\n",
       "   ('analytes', '0.4077'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.4065'),\n",
       "   ('residual_tumor_measurement', '0.3915'),\n",
       "   ('nononcologic_therapeutic_agents', '0.3888'),\n",
       "   ('ensat_pathologic_t', '0.3858'),\n",
       "   ('percent_tumor_cells', '0.3754')]},\n",
       " {'Candidate column': 'Proteomics_TMT_plex',\n",
       "  'Top k columns': [('residual_tumor_measurement', '0.5041'),\n",
       "   ('percent_tumor_cells', '0.4259'),\n",
       "   ('double_expressor_lymphoma', '0.423'),\n",
       "   ('extrathyroid_extension', '0.4181'),\n",
       "   ('medulloblastoma_molecular_classification', '0.4143'),\n",
       "   ('tumor_infiltrating_macrophages', '0.4067'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.4057'),\n",
       "   ('tissue_microarray_coordinates', '0.3925'),\n",
       "   ('micrometastasis_present', '0.3897'),\n",
       "   ('tumor_descriptor', '0.3872')]},\n",
       " {'Candidate column': 'Proteomics_TMT_channel',\n",
       "  'Top k columns': [('micrometastasis_present', '0.5361'),\n",
       "   ('tumor_infiltrating_macrophages', '0.5264'),\n",
       "   ('residual_tumor_measurement', '0.4746'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.4536'),\n",
       "   ('tissue_microarray_coordinates', '0.4424'),\n",
       "   ('analyte_type_id', '0.4294'),\n",
       "   ('raw_methylation_arrays', '0.4201'),\n",
       "   ('analyte_type', '0.4194'),\n",
       "   ('methylation_liftover_workflows', '0.411'),\n",
       "   ('percent_tumor_cells', '0.4078')]},\n",
       " {'Candidate column': 'Proteomics_Parent_Sample_IDs',\n",
       "  'Top k columns': [('analyte_type_id', '0.5622'),\n",
       "   ('analyte_type', '0.5615'),\n",
       "   ('analytes', '0.5379'),\n",
       "   ('test_analyte_type', '0.503'),\n",
       "   ('gene_expressions', '0.4887'),\n",
       "   ('method_of_sample_procurement', '0.4645'),\n",
       "   ('scan_tracer_used', '0.4618'),\n",
       "   ('gene_symbol', '0.4615'),\n",
       "   ('protein_expressions', '0.4588'),\n",
       "   ('sample_ordinal', '0.4457')]},\n",
       " {'Candidate column': 'Proteomics_Aliquot_ID',\n",
       "  'Top k columns': [('analytes', '0.4196'),\n",
       "   ('aliquots', '0.4104'),\n",
       "   ('chemo_concurrent_to_radiation', '0.4066'),\n",
       "   ('analyte_type_id', '0.388'),\n",
       "   ('prostatic_chips_positive_count', '0.3794'),\n",
       "   ('analyte_type', '0.3688'),\n",
       "   ('type_of_tobacco_used', '0.3647'),\n",
       "   ('nononcologic_therapeutic_agents', '0.3552'),\n",
       "   ('spectrophotometer_method', '0.3371'),\n",
       "   ('scan_tracer_used', '0.326')]},\n",
       " {'Candidate column': 'Proteomics_Tumor_Normal',\n",
       "  'Top k columns': [('residual_tumor_measurement', '0.5188'),\n",
       "   ('residual_tumor', '0.5073'),\n",
       "   ('percent_tumor_cells', '0.5009'),\n",
       "   ('percent_tumor_invasion', '0.4944'),\n",
       "   ('percent_tumor_nuclei', '0.4716'),\n",
       "   ('other_clinical_attributes', '0.4524'),\n",
       "   ('classification_of_tumor', '0.446'),\n",
       "   ('tumor_purity', '0.4448'),\n",
       "   ('tumor_level_prostate', '0.4307'),\n",
       "   ('percent_normal_cells', '0.4223')]},\n",
       " {'Candidate column': 'Proteomics_OCT',\n",
       "  'Top k columns': [('analytes', '0.5444'),\n",
       "   ('analyte_type', '0.5213'),\n",
       "   ('analyte_type_id', '0.4955'),\n",
       "   ('chemo_concurrent_to_radiation', '0.4701'),\n",
       "   ('nononcologic_therapeutic_agents', '0.4541'),\n",
       "   ('spectrophotometer_method', '0.4475'),\n",
       "   ('test_analyte_type', '0.4383'),\n",
       "   ('secondary_expression_analyses', '0.4371'),\n",
       "   ('molecular_analysis_method', '0.4363'),\n",
       "   ('site_of_resection_or_biopsy', '0.4294')]},\n",
       " {'Candidate column': 'Country',\n",
       "  'Top k columns': [('country_of_birth', '0.8066'),\n",
       "   ('country_of_residence_at_enrollment', '0.6309'),\n",
       "   ('zone_of_origin_prostate', '0.5379'),\n",
       "   ('ethnicity', '0.5185'),\n",
       "   ('variant_origin', '0.4986'),\n",
       "   ('tumor_confined_to_organ_of_origin', '0.4203'),\n",
       "   ('tissue_or_organ_of_origin', '0.4156'),\n",
       "   ('international_prognostic_index', '0.4004'),\n",
       "   ('demographics', '0.3991'),\n",
       "   ('locus', '0.3925')]},\n",
       " {'Candidate column': 'Histologic_Grade_FIGO',\n",
       "  'Top k columns': [('histologic_progression_type', '0.6528'),\n",
       "   ('tumor_grade_category', '0.553'),\n",
       "   ('inpc_histologic_group', '0.5197'),\n",
       "   ('tumor_grade', '0.5011'),\n",
       "   ('histone_variant', '0.5006'),\n",
       "   ('who_nte_grade', '0.4882'),\n",
       "   ('ensat_pathologic_stage', '0.4856'),\n",
       "   ('histone_family', '0.4648'),\n",
       "   ('uicc_pathologic_stage', '0.4646'),\n",
       "   ('tumor_regression_grade', '0.4602')]},\n",
       " {'Candidate column': 'Myometrial_invasion_Specify',\n",
       "  'Top k columns': [('perineural_invasion_present', '0.6745'),\n",
       "   ('percent_tumor_invasion', '0.5577'),\n",
       "   ('mirna_expressions', '0.4977'),\n",
       "   ('ovarian_surface_involvement', '0.4913'),\n",
       "   ('tumor_depth_measurement', '0.4774'),\n",
       "   ('imaging_findings', '0.4719'),\n",
       "   ('tumor_depth', '0.4584'),\n",
       "   ('imaging_result', '0.4286'),\n",
       "   ('mirna_expression_workflows', '0.4205'),\n",
       "   ('tumor_depth_descriptor', '0.4179')]},\n",
       " {'Candidate column': 'Histologic_type',\n",
       "  'Top k columns': [('inpc_histologic_group', '0.7529'),\n",
       "   ('family_histories', '0.6628'),\n",
       "   ('histone_family', '0.5595'),\n",
       "   ('morphologic_architectural_pattern', '0.5401'),\n",
       "   ('ensat_pathologic_stage', '0.5202'),\n",
       "   ('histologic_progression_type', '0.5131'),\n",
       "   ('history_of_tumor_type', '0.5081'),\n",
       "   ('histone_variant', '0.5071'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.5041'),\n",
       "   ('history_of_tumor', '0.4901')]},\n",
       " {'Candidate column': 'Treatment_naive',\n",
       "  'Top k columns': [('prior_treatment', '0.6518'),\n",
       "   ('treatments', '0.6296'),\n",
       "   ('reason_treatment_not_given', '0.6179'),\n",
       "   ('reason_treatment_ended', '0.6094'),\n",
       "   ('treatment_effect', '0.5983'),\n",
       "   ('treatment_or_therapy', '0.5978'),\n",
       "   ('treatment_type', '0.5946'),\n",
       "   ('treatment_intent_type', '0.5797'),\n",
       "   ('treatment_outcome', '0.5791'),\n",
       "   ('therapeutic_target_level', '0.5788')]},\n",
       " {'Candidate column': 'Tumor_purity',\n",
       "  'Top k columns': [('residual_tumor', '0.7139'),\n",
       "   ('residual_tumor_measurement', '0.663'),\n",
       "   ('classification_of_tumor', '0.6294'),\n",
       "   ('percent_tumor_invasion', '0.6156'),\n",
       "   ('history_of_tumor', '0.5965'),\n",
       "   ('history_of_tumor_type', '0.594'),\n",
       "   ('tumor_purity', '0.5919'),\n",
       "   ('greatest_tumor_dimension', '0.5551'),\n",
       "   ('percent_tumor_nuclei', '0.5416'),\n",
       "   ('percent_tumor_cells', '0.5047')]},\n",
       " {'Candidate column': 'Path_Stage_Primary_Tumor-pT',\n",
       "  'Top k columns': [('residual_tumor_measurement', '0.5583'),\n",
       "   ('classification_of_tumor', '0.529'),\n",
       "   ('residual_tumor', '0.5231'),\n",
       "   ('history_of_tumor_type', '0.501'),\n",
       "   ('history_of_tumor', '0.4963'),\n",
       "   ('treatment_effect_indicator', '0.4757'),\n",
       "   ('other_clinical_attributes', '0.4669'),\n",
       "   ('treatment_type', '0.4588'),\n",
       "   ('percent_tumor_invasion', '0.4564'),\n",
       "   ('tumor_descriptor', '0.4513')]},\n",
       " {'Candidate column': 'Path_Stage_Reg_Lymph_Nodes-pN',\n",
       "  'Top k columns': [('non_nodal_regional_disease', '0.4671'),\n",
       "   ('cog_renal_stage', '0.4519'),\n",
       "   ('inrg_stage', '0.4226'),\n",
       "   ('inss_stage', '0.4154'),\n",
       "   ('ensat_pathologic_stage', '0.4071'),\n",
       "   ('lymph_nodes_removed', '0.3925'),\n",
       "   ('lymph_nodes_tested', '0.3924'),\n",
       "   ('ensat_pathologic_n', '0.3902'),\n",
       "   ('enneking_msts_stage', '0.3831'),\n",
       "   ('enneking_msts_grade', '0.3789')]},\n",
       " {'Candidate column': 'Clin_Stage_Dist_Mets-cM',\n",
       "  'Top k columns': [('somatic_mutation_calling_workflows', '0.6143'),\n",
       "   ('ensat_clinical_m', '0.5676'),\n",
       "   ('enneking_msts_stage', '0.5529'),\n",
       "   ('specialized_molecular_test', '0.5487'),\n",
       "   ('enneking_msts_metastasis', '0.5352'),\n",
       "   ('uicc_clinical_m', '0.5315'),\n",
       "   ('uicc_pathologic_m', '0.5092'),\n",
       "   ('enneking_msts_grade', '0.4988'),\n",
       "   ('somatic_aggregation_workflows', '0.4866'),\n",
       "   ('msi_status', '0.4847')]},\n",
       " {'Candidate column': 'Path_Stage_Dist_Mets-pM',\n",
       "  'Top k columns': [('somatic_mutation_calling_workflows', '0.6366'),\n",
       "   ('specialized_molecular_test', '0.6093'),\n",
       "   ('enneking_msts_stage', '0.5867'),\n",
       "   ('enneking_msts_metastasis', '0.5692'),\n",
       "   ('ensat_clinical_m', '0.5537'),\n",
       "   ('msi_status', '0.5508'),\n",
       "   ('somatic_aggregation_workflows', '0.5263'),\n",
       "   ('mirna_expression_workflows', '0.5202'),\n",
       "   ('somatic_mutation_indexes', '0.5097'),\n",
       "   ('enneking_msts_grade', '0.5058')]},\n",
       " {'Candidate column': 'tumor_Stage-Pathological',\n",
       "  'Top k columns': [('metastasis_at_diagnosis_site', '0.6505'),\n",
       "   ('metastasis_at_diagnosis', '0.6176'),\n",
       "   ('ensat_pathologic_stage', '0.583'),\n",
       "   ('tumor_descriptor', '0.5825'),\n",
       "   ('tumor_grade', '0.5701'),\n",
       "   ('tumor_grade_category', '0.5624'),\n",
       "   ('histologic_progression_type', '0.53'),\n",
       "   ('uicc_pathologic_stage', '0.53'),\n",
       "   ('tumor_level_prostate', '0.5183'),\n",
       "   ('uicc_clinical_stage', '0.5138')]},\n",
       " {'Candidate column': 'FIGO_stage',\n",
       "  'Top k columns': [('figo_stage', '0.7615'),\n",
       "   ('inrg_stage', '0.6144'),\n",
       "   ('iss_stage', '0.5767'),\n",
       "   ('ensat_pathologic_stage', '0.5592'),\n",
       "   ('inss_stage', '0.5226'),\n",
       "   ('uicc_pathologic_stage', '0.5049'),\n",
       "   ('histologic_progression_type', '0.4687'),\n",
       "   ('uicc_clinical_t', '0.4378'),\n",
       "   ('irs_stage', '0.4159'),\n",
       "   ('uicc_clinical_stage', '0.4055')]},\n",
       " {'Candidate column': 'LVSI',\n",
       "  'Top k columns': [('includes_spike_ins', '0.42'),\n",
       "   ('vascular_invasion_type', '0.3905'),\n",
       "   ('vascular_invasion_present', '0.3893'),\n",
       "   ('magnification', '0.3464'),\n",
       "   ('sites_of_involvement', '0.346'),\n",
       "   ('distributor_reference', '0.3392'),\n",
       "   ('lane_number', '0.3372'),\n",
       "   ('adapter_name', '0.3359'),\n",
       "   ('ulceration_indicator', '0.3311'),\n",
       "   ('margin_status', '0.32')]},\n",
       " {'Candidate column': 'BMI',\n",
       "  'Top k columns': [('weight', '0.7613'),\n",
       "   ('initial_weight', '0.6398'),\n",
       "   ('current_weight', '0.5384'),\n",
       "   ('height', '0.5144'),\n",
       "   ('gross_tumor_weight', '0.4286'),\n",
       "   ('menopause_status', '0.4029'),\n",
       "   ('gender', '0.3856'),\n",
       "   ('demographics', '0.3789'),\n",
       "   ('age_at_diagnosis', '0.3699'),\n",
       "   ('age_at_index', '0.3627')]},\n",
       " {'Candidate column': 'Age',\n",
       "  'Top k columns': [('age_at_onset', '0.9307'),\n",
       "   ('age_at_index', '0.8667'),\n",
       "   ('age_at_diagnosis', '0.8658'),\n",
       "   ('age_at_last_exposure', '0.8531'),\n",
       "   ('age_is_obfuscated', '0.8395'),\n",
       "   ('relationship_age_at_diagnosis', '0.7915'),\n",
       "   ('year_of_birth', '0.7152'),\n",
       "   ('menopause_status', '0.6797'),\n",
       "   ('year_of_death', '0.6449'),\n",
       "   ('tobacco_smoking_onset_year', '0.5639')]},\n",
       " {'Candidate column': 'Diabetes',\n",
       "  'Top k columns': [('diabetes_treatment_type', '0.7134'),\n",
       "   ('disease_type', '0.4525'),\n",
       "   ('disease_response', '0.4431'),\n",
       "   ('pregnant_at_diagnosis', '0.4169'),\n",
       "   ('initial_disease_status', '0.4147'),\n",
       "   ('relationship_age_at_diagnosis', '0.4024'),\n",
       "   ('last_known_disease_status', '0.3827'),\n",
       "   ('days_to_treatment_end', '0.371'),\n",
       "   ('alcohol_drinks_per_day', '0.369'),\n",
       "   ('days_to_diagnosis', '0.3622')]},\n",
       " {'Candidate column': 'Race',\n",
       "  'Top k columns': [('race', '0.918'),\n",
       "   ('rin', '0.3374'),\n",
       "   ('zone_of_origin_prostate', '0.3352'),\n",
       "   ('chromosome_arm', '0.3146'),\n",
       "   ('course_number', '0.3091'),\n",
       "   ('ethnicity', '0.2982'),\n",
       "   ('lane_number', '0.2975'),\n",
       "   ('histologic_progression_type', '0.2944'),\n",
       "   ('chip_position', '0.2933'),\n",
       "   ('evidence_of_progression_type', '0.2921')]},\n",
       " {'Candidate column': 'Ethnicity',\n",
       "  'Top k columns': [('ethnicity', '0.8949'),\n",
       "   ('demographics', '0.5743'),\n",
       "   ('country_of_residence_at_enrollment', '0.5062'),\n",
       "   ('country_of_birth', '0.4913'),\n",
       "   ('occupation_type', '0.4128'),\n",
       "   ('variant_origin', '0.4116'),\n",
       "   ('locus', '0.3913'),\n",
       "   ('gender', '0.384'),\n",
       "   ('roots', '0.3777'),\n",
       "   ('stain_type', '0.3553')]},\n",
       " {'Candidate column': 'Gender',\n",
       "  'Top k columns': [('gender', '0.9501'),\n",
       "   ('relationship_gender', '0.7476'),\n",
       "   ('pregnancy_outcome', '0.5293'),\n",
       "   ('demographics', '0.5272'),\n",
       "   ('category', '0.4912'),\n",
       "   ('pregnant_at_diagnosis', '0.4713'),\n",
       "   ('comorbidity', '0.4514'),\n",
       "   ('morphologic_architectural_pattern', '0.4348'),\n",
       "   ('relationship_type', '0.427'),\n",
       "   ('comorbidities', '0.4223')]},\n",
       " {'Candidate column': 'Tumor_Site',\n",
       "  'Top k columns': [('residual_tumor_measurement', '0.7299'),\n",
       "   ('residual_tumor', '0.724'),\n",
       "   ('history_of_tumor_type', '0.6364'),\n",
       "   ('history_of_tumor', '0.6339'),\n",
       "   ('classification_of_tumor', '0.6323'),\n",
       "   ('percent_tumor_invasion', '0.6204'),\n",
       "   ('greatest_tumor_dimension', '0.58'),\n",
       "   ('tumor_descriptor', '0.5446'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.5338'),\n",
       "   ('non_nodal_tumor_deposits', '0.5216')]},\n",
       " {'Candidate column': 'Tumor_Site_Other',\n",
       "  'Top k columns': [('residual_tumor', '0.6941'),\n",
       "   ('residual_tumor_measurement', '0.6846'),\n",
       "   ('history_of_tumor', '0.6312'),\n",
       "   ('history_of_tumor_type', '0.6306'),\n",
       "   ('classification_of_tumor', '0.5991'),\n",
       "   ('percent_tumor_invasion', '0.595'),\n",
       "   ('non_nodal_tumor_deposits', '0.542'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.514'),\n",
       "   ('greatest_tumor_dimension', '0.5111'),\n",
       "   ('tumor_descriptor', '0.5103')]},\n",
       " {'Candidate column': 'Tumor_Focality',\n",
       "  'Top k columns': [('tumor_focality', '0.6894'),\n",
       "   ('residual_tumor', '0.6855'),\n",
       "   ('residual_tumor_measurement', '0.6483'),\n",
       "   ('classification_of_tumor', '0.6482'),\n",
       "   ('percent_tumor_invasion', '0.5713'),\n",
       "   ('greatest_tumor_dimension', '0.5316'),\n",
       "   ('tumor_descriptor', '0.4944'),\n",
       "   ('history_of_tumor_type', '0.4808'),\n",
       "   ('tumor_shape', '0.4689'),\n",
       "   ('percent_tumor_cells', '0.4671')]},\n",
       " {'Candidate column': 'Tumor_Size_cm',\n",
       "  'Top k columns': [('tumor_largest_dimension_diameter', '0.6984'),\n",
       "   ('greatest_tumor_dimension', '0.6599'),\n",
       "   ('residual_tumor_measurement', '0.6178'),\n",
       "   ('tumor_shape', '0.6099'),\n",
       "   ('distance_normal_to_tumor', '0.5903'),\n",
       "   ('body_surface_area', '0.5583'),\n",
       "   ('gross_tumor_weight', '0.5573'),\n",
       "   ('tumor_descriptor', '0.5538'),\n",
       "   ('max_tumor_bulk_site', '0.5456'),\n",
       "   ('tumor_focality', '0.5276')]},\n",
       " {'Candidate column': 'Estrogen_Receptor',\n",
       "  'Top k columns': [('hormone_replacement_therapy_type', '0.5277'),\n",
       "   ('hormonal_contraceptive_type', '0.5153'),\n",
       "   ('hormonal_contraceptive_use', '0.4837'),\n",
       "   ('chemo_concurrent_to_radiation', '0.4763'),\n",
       "   ('chemical_exposure_type', '0.469'),\n",
       "   ('adrenal_hormone', '0.4385'),\n",
       "   ('radon_exposure', '0.4187'),\n",
       "   ('nononcologic_therapeutic_agents', '0.396'),\n",
       "   ('protein_expressions', '0.3706'),\n",
       "   ('gene_symbol', '0.3692')]},\n",
       " {'Candidate column': 'Estrogen_Receptor_%',\n",
       "  'Top k columns': [('hormone_replacement_therapy_type', '0.5255'),\n",
       "   ('chemo_concurrent_to_radiation', '0.5044'),\n",
       "   ('hormonal_contraceptive_type', '0.4983'),\n",
       "   ('chemical_exposure_type', '0.4917'),\n",
       "   ('hormonal_contraceptive_use', '0.4805'),\n",
       "   ('adrenal_hormone', '0.4556'),\n",
       "   ('radon_exposure', '0.4335'),\n",
       "   ('nononcologic_therapeutic_agents', '0.3844'),\n",
       "   ('raw_methylation_arrays', '0.3713'),\n",
       "   ('therapeutic_agents', '0.3554')]},\n",
       " {'Candidate column': 'Progesterone_Receptor',\n",
       "  'Top k columns': [('hormone_replacement_therapy_type', '0.564'),\n",
       "   ('adrenal_hormone', '0.5513'),\n",
       "   ('therapeutic_agents', '0.4814'),\n",
       "   ('nononcologic_therapeutic_agents', '0.4813'),\n",
       "   ('hormonal_contraceptive_type', '0.4781'),\n",
       "   ('hormonal_contraceptive_use', '0.4556'),\n",
       "   ('protein_expressions', '0.4496'),\n",
       "   ('chemo_concurrent_to_radiation', '0.406'),\n",
       "   ('therapeutic_target_level', '0.3986'),\n",
       "   ('drug_category', '0.36')]},\n",
       " {'Candidate column': 'Progesterone_Receptor_%',\n",
       "  'Top k columns': [('hormone_replacement_therapy_type', '0.5567'),\n",
       "   ('adrenal_hormone', '0.5444'),\n",
       "   ('nononcologic_therapeutic_agents', '0.4752'),\n",
       "   ('therapeutic_agents', '0.4736'),\n",
       "   ('hormonal_contraceptive_type', '0.467'),\n",
       "   ('hormonal_contraceptive_use', '0.4556'),\n",
       "   ('protein_expressions', '0.435'),\n",
       "   ('chemo_concurrent_to_radiation', '0.4083'),\n",
       "   ('therapeutic_target_level', '0.3947'),\n",
       "   ('drug_category', '0.3621')]},\n",
       " {'Candidate column': 'MLH1',\n",
       "  'Top k columns': [('enneking_msts_metastasis', '0.4529'),\n",
       "   ('somatic_mutation_indexes', '0.3949'),\n",
       "   ('enneking_msts_stage', '0.3869'),\n",
       "   ('msi_status', '0.3807'),\n",
       "   ('enneking_msts_grade', '0.369'),\n",
       "   ('bmi', '0.3679'),\n",
       "   ('proportion_base_mismatch', '0.3647'),\n",
       "   ('metaplasia_present', '0.3502'),\n",
       "   ('histone_family', '0.3485'),\n",
       "   ('uicc_pathologic_m', '0.3457')]},\n",
       " {'Candidate column': 'MLH2',\n",
       "  'Top k columns': [('enneking_msts_metastasis', '0.4942'),\n",
       "   ('enneking_msts_stage', '0.4439'),\n",
       "   ('enneking_msts_grade', '0.4239'),\n",
       "   ('somatic_mutation_indexes', '0.4173'),\n",
       "   ('proportion_base_mismatch', '0.4138'),\n",
       "   ('msi_status', '0.4073'),\n",
       "   ('bmi', '0.3877'),\n",
       "   ('specialized_molecular_test', '0.3854'),\n",
       "   ('metaplasia_present', '0.3648'),\n",
       "   ('uicc_pathologic_m', '0.3556')]},\n",
       " {'Candidate column': 'MSH6',\n",
       "  'Top k columns': [('enneking_msts_grade', '0.4663'),\n",
       "   ('enneking_msts_stage', '0.4425'),\n",
       "   ('msi_status', '0.4389'),\n",
       "   ('enneking_msts_metastasis', '0.4337'),\n",
       "   ('specialized_molecular_test', '0.4276'),\n",
       "   ('bmi', '0.3866'),\n",
       "   ('msi_score', '0.3675'),\n",
       "   ('proportion_base_mismatch', '0.3632'),\n",
       "   ('ensat_clinical_m', '0.3542'),\n",
       "   ('medulloblastoma_molecular_classification', '0.3426')]},\n",
       " {'Candidate column': 'PMS2',\n",
       "  'Top k columns': [('pmid', '0.6376'),\n",
       "   ('simple_somatic_mutations', '0.5259'),\n",
       "   ('msi_status', '0.5241'),\n",
       "   ('procedures_performed', '0.5217'),\n",
       "   ('parent_samples', '0.5064'),\n",
       "   ('inss_stage', '0.4909'),\n",
       "   ('follow_ups', '0.4574'),\n",
       "   ('type', '0.4558'),\n",
       "   ('projects', '0.4341'),\n",
       "   ('mirna_expression_workflows', '0.4313')]},\n",
       " {'Candidate column': 'p53',\n",
       "  'Top k columns': [('prostatic_chips_positive_count', '0.4936'),\n",
       "   ('number_proliferating_cells', '0.4305'),\n",
       "   ('prostatic_chips_total_count', '0.4089'),\n",
       "   ('fev1_ref_post_bronch_percent', '0.3974'),\n",
       "   ('gleason_score', '0.3913'),\n",
       "   ('fev1_ref_pre_bronch_percent', '0.3897'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3792'),\n",
       "   ('includes_spike_ins', '0.3778'),\n",
       "   ('gleason_patterns_percent', '0.3756'),\n",
       "   ('protein_expressions', '0.3537')]},\n",
       " {'Candidate column': 'Other_IHC_specify',\n",
       "  'Top k columns': [('uicc_pathologic_m', '0.4508'),\n",
       "   ('hiv_viral_load', '0.4425'),\n",
       "   ('uicc_pathologic_t', '0.4095'),\n",
       "   ('uicc_pathologic_n', '0.3852'),\n",
       "   ('cdc_hiv_risk_factors', '0.3747'),\n",
       "   ('uicc_pathologic_stage', '0.3525'),\n",
       "   ('tumor_confined_to_organ_of_origin', '0.3518'),\n",
       "   ('max_tumor_bulk_site', '0.3486'),\n",
       "   ('batch_id', '0.321'),\n",
       "   ('described_cases', '0.3164')]},\n",
       " {'Candidate column': 'MLH1_Promoter_Hypermethylation',\n",
       "  'Top k columns': [('methylation_array_harmonization_workflows', '0.4179'),\n",
       "   ('methylation_liftover_workflows', '0.3856'),\n",
       "   ('masked_methylation_arrays', '0.322'),\n",
       "   ('histone_family', '0.3174'),\n",
       "   ('enneking_msts_metastasis', '0.3102'),\n",
       "   ('raw_methylation_arrays', '0.2968'),\n",
       "   ('somatic_mutation_indexes', '0.2903'),\n",
       "   ('double_hit_lymphoma', '0.2835'),\n",
       "   ('metaplasia_present', '0.2785'),\n",
       "   ('methylation_beta_values', '0.263')]},\n",
       " {'Candidate column': 'Num_full_term_pregnancies',\n",
       "  'Top k columns': [('pregnancy_outcome', '0.6625'),\n",
       "   ('pregnancy_count', '0.6385'),\n",
       "   ('pregnant_at_diagnosis', '0.6206'),\n",
       "   ('marital_status', '0.5975'),\n",
       "   ('weeks_gestation_at_birth', '0.5749'),\n",
       "   ('days_to_birth', '0.5465'),\n",
       "   ('premature_at_birth', '0.5208'),\n",
       "   ('menopause_status', '0.4979'),\n",
       "   ('relationship_gender', '0.4859'),\n",
       "   ('gender', '0.473')]},\n",
       " {'Candidate column': 'EPIC_Bcells',\n",
       "  'Top k columns': [('number_proliferating_cells', '0.6183'),\n",
       "   ('flow_cell_barcode', '0.5541'),\n",
       "   ('number_expect_cells', '0.4695'),\n",
       "   ('percent_normal_cells', '0.4489'),\n",
       "   ('percent_stromal_cells', '0.4375'),\n",
       "   ('single_cell_library', '0.4222'),\n",
       "   ('protein_expressions', '0.4089'),\n",
       "   ('epithelioid_cell_percent', '0.402'),\n",
       "   ('cell_count', '0.3984'),\n",
       "   ('prostatic_chips_positive_count', '0.3917')]},\n",
       " {'Candidate column': 'EPIC_CAFs',\n",
       "  'Top k columns': [('adapter_sequence', '0.4417'),\n",
       "   ('library_strand', '0.4164'),\n",
       "   ('type', '0.4154'),\n",
       "   ('creator', '0.413'),\n",
       "   ('derived_files', '0.4044'),\n",
       "   ('adapter_name', '0.3973'),\n",
       "   ('library_strategy', '0.3966'),\n",
       "   ('is_ffpe', '0.3947'),\n",
       "   ('read_group_qcs', '0.3846'),\n",
       "   ('to_trim_adapter_sequence', '0.3837')]},\n",
       " {'Candidate column': 'EPIC_CD4_Tcells',\n",
       "  'Top k columns': [('single_cell_library', '0.4446'),\n",
       "   ('cd4_count', '0.423'),\n",
       "   ('flow_cell_barcode', '0.4175'),\n",
       "   ('cytoband', '0.417'),\n",
       "   ('cancer_detection_method', '0.3928'),\n",
       "   ('number_proliferating_cells', '0.3881'),\n",
       "   ('uicc_pathologic_t', '0.3746'),\n",
       "   ('cell_count', '0.3725'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.3682'),\n",
       "   ('uicc_pathologic_n', '0.3634')]},\n",
       " {'Candidate column': 'EPIC_CD8_Tcells',\n",
       "  'Top k columns': [('flow_cell_barcode', '0.4208'),\n",
       "   ('single_cell_library', '0.4173'),\n",
       "   ('icd_10', '0.3991'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.3795'),\n",
       "   ('chipseq_target', '0.3719'),\n",
       "   ('icd_10_code', '0.3647'),\n",
       "   ('cell_count', '0.3606'),\n",
       "   ('number_proliferating_cells', '0.3593'),\n",
       "   ('chipseq_antibody', '0.3533'),\n",
       "   ('tumor_infiltrating_macrophages', '0.3517')]},\n",
       " {'Candidate column': 'EPIC_Endothelial',\n",
       "  'Top k columns': [('epithelioid_cell_percent', '0.3595'),\n",
       "   ('primary_disease', '0.3433'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.3424'),\n",
       "   ('diagnosis_is_primary_disease', '0.3421'),\n",
       "   ('extrascleral_extension', '0.3396'),\n",
       "   ('prostatic_chips_positive_count', '0.3355'),\n",
       "   ('extrathyroid_extension', '0.3274'),\n",
       "   ('primary_site', '0.3266'),\n",
       "   ('prostatic_chips_total_count', '0.3201'),\n",
       "   ('tumor_depth', '0.3199')]},\n",
       " {'Candidate column': 'EPIC_Macrophages',\n",
       "  'Top k columns': [('tumor_infiltrating_macrophages', '0.4912'),\n",
       "   ('mitotic_total_area', '0.4774'),\n",
       "   ('tissue_microarray_coordinates', '0.4748'),\n",
       "   ('micropapillary_features', '0.4516'),\n",
       "   ('size_extraocular_nodule', '0.417'),\n",
       "   ('medulloblastoma_molecular_classification', '0.412'),\n",
       "   ('size_selection_range', '0.4053'),\n",
       "   ('average_insert_size', '0.4031'),\n",
       "   ('specialized_molecular_test', '0.4002'),\n",
       "   ('hysterectomy_margins_involved', '0.3927')]},\n",
       " {'Candidate column': 'EPIC_NKcells',\n",
       "  'Top k columns': [('number_proliferating_cells', '0.6398'),\n",
       "   ('lymph_nodes_positive', '0.5691'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.5294'),\n",
       "   ('protein_expressions', '0.5053'),\n",
       "   ('number_expect_cells', '0.5052'),\n",
       "   ('percent_normal_cells', '0.4882'),\n",
       "   ('prostatic_chips_positive_count', '0.4881'),\n",
       "   ('percent_stromal_cells', '0.4878'),\n",
       "   ('epithelioid_cell_percent', '0.4584'),\n",
       "   ('peripancreatic_lymph_nodes_positive', '0.4549')]},\n",
       " {'Candidate column': 'EPIC_otherCells',\n",
       "  'Top k columns': [('flow_cell_barcode', '0.4086'),\n",
       "   ('chemo_concurrent_to_radiation', '0.3805'),\n",
       "   ('cytoband', '0.3672'),\n",
       "   ('single_cell_library', '0.3656'),\n",
       "   ('lymph_node_dissection_site', '0.357'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.3547'),\n",
       "   ('lymph_node_dissection_method', '0.3492'),\n",
       "   ('number_proliferating_cells', '0.3466'),\n",
       "   ('papillary_renal_cell_type', '0.3453'),\n",
       "   ('lymph_nodes_tested', '0.3271')]},\n",
       " {'Candidate column': 'CIBERSORT_B _cells _naive',\n",
       "  'Top k columns': [('single_cell_library', '0.5987'),\n",
       "   ('flow_cell_barcode', '0.5801'),\n",
       "   ('base_caller_name', '0.5418'),\n",
       "   ('bcr_id', '0.5379'),\n",
       "   ('chipseq_antibody', '0.5211'),\n",
       "   ('cell_count', '0.5192'),\n",
       "   ('batch_id', '0.5'),\n",
       "   ('bone_marrow_malignant_cells', '0.4976'),\n",
       "   ('number_expect_cells', '0.4927'),\n",
       "   ('multiplex_barcode', '0.4702')]},\n",
       " {'Candidate column': 'CIBERSORT_B _cells _memory',\n",
       "  'Top k columns': [('single_cell_library', '0.7394'),\n",
       "   ('multiplex_barcode', '0.6744'),\n",
       "   ('chipseq_antibody', '0.6287'),\n",
       "   ('flow_cell_barcode', '0.613'),\n",
       "   ('read_group_qcs', '0.5867'),\n",
       "   ('base_caller_name', '0.5815'),\n",
       "   ('library_strand', '0.5603'),\n",
       "   ('chipseq_target', '0.5468'),\n",
       "   ('batch_id', '0.5437'),\n",
       "   ('copy_number_auxiliary_files', '0.5404')]},\n",
       " {'Candidate column': 'CIBERSORT_Plasma _cells',\n",
       "  'Top k columns': [('number_proliferating_cells', '0.5313'),\n",
       "   ('number_expect_cells', '0.5136'),\n",
       "   ('cell_count', '0.4745'),\n",
       "   ('flow_cell_barcode', '0.4239'),\n",
       "   ('percent_stromal_cells', '0.4059'),\n",
       "   ('bone_marrow_malignant_cells', '0.4024'),\n",
       "   ('single_cell_library', '0.3907'),\n",
       "   ('percent_normal_cells', '0.3837'),\n",
       "   ('goblet_cells_columnar_mucosa_present', '0.3731'),\n",
       "   ('medulloblastoma_molecular_classification', '0.3654')]},\n",
       " {'Candidate column': 'CIBERSORT_T _cells _CD8',\n",
       "  'Top k columns': [('icd_10_code', '0.6154'),\n",
       "   ('multiplex_barcode', '0.5762'),\n",
       "   ('single_cell_library', '0.5657'),\n",
       "   ('icd_10', '0.5589'),\n",
       "   ('base_caller_version', '0.5418'),\n",
       "   ('chipseq_target', '0.5375'),\n",
       "   ('read_group_qcs', '0.529'),\n",
       "   ('base_caller_name', '0.5103'),\n",
       "   ('copy_number_auxiliary_files', '0.5077'),\n",
       "   ('chipseq_antibody', '0.5069')]},\n",
       " {'Candidate column': 'CIBERSORT_T _cells _CD4 _naive',\n",
       "  'Top k columns': [('single_cell_library', '0.6527'),\n",
       "   ('multiplex_barcode', '0.6506'),\n",
       "   ('read_group_qcs', '0.6064'),\n",
       "   ('icd_10_code', '0.5941'),\n",
       "   ('target_capture_kit_catalog_number', '0.5844'),\n",
       "   ('per_sequence_gc_content', '0.5806'),\n",
       "   ('uicc_pathologic_t', '0.5747'),\n",
       "   ('chipseq_target', '0.5718'),\n",
       "   ('tumor_code', '0.5624'),\n",
       "   ('uicc_pathologic_n', '0.5568')]},\n",
       " {'Candidate column': 'CIBERSORT_T _cells _CD4 _memory _resting',\n",
       "  'Top k columns': [('multiplex_barcode', '0.6769'),\n",
       "   ('single_cell_library', '0.6604'),\n",
       "   ('extranodal_extension', '0.6101'),\n",
       "   ('read_group_qcs', '0.5888'),\n",
       "   ('progression_or_recurrence_anatomic_site', '0.5877'),\n",
       "   ('library_strand', '0.5753'),\n",
       "   ('copy_number_auxiliary_files', '0.5585'),\n",
       "   ('chipseq_target', '0.5526'),\n",
       "   ('target_capture_kit_catalog_number', '0.551'),\n",
       "   ('base_caller_version', '0.5414')]},\n",
       " {'Candidate column': 'CIBERSORT_T _cells _CD4 _memory _activated',\n",
       "  'Top k columns': [('multiplex_barcode', '0.6694'),\n",
       "   ('single_cell_library', '0.6554'),\n",
       "   ('target_capture_kit_catalog_number', '0.5897'),\n",
       "   ('extranodal_extension', '0.5803'),\n",
       "   ('chipseq_target', '0.5788'),\n",
       "   ('read_group_qcs', '0.5694'),\n",
       "   ('copy_number_auxiliary_files', '0.557'),\n",
       "   ('target_capture_kit_version', '0.5509'),\n",
       "   ('library_strand', '0.5429'),\n",
       "   ('progression_or_recurrence_anatomic_site', '0.5403')]},\n",
       " {'Candidate column': 'CIBERSORT_T _cells _follicular _helper',\n",
       "  'Top k columns': [('single_cell_library', '0.5251'),\n",
       "   ('extranodal_extension', '0.4998'),\n",
       "   ('chipseq_target', '0.4985'),\n",
       "   ('read_group_qcs', '0.4983'),\n",
       "   ('base_caller_name', '0.4953'),\n",
       "   ('library_strand', '0.4852'),\n",
       "   ('intron', '0.4838'),\n",
       "   ('library_strategy', '0.4769'),\n",
       "   ('target_capture_kit_catalog_number', '0.4765'),\n",
       "   ('percent_follicular_component', '0.454')]},\n",
       " {'Candidate column': 'CIBERSORT_T _cells _regulatory _(Tregs)',\n",
       "  'Top k columns': [('single_cell_library', '0.5228'),\n",
       "   ('recist_targeted_regions_sum', '0.5033'),\n",
       "   ('extranodal_extension', '0.4653'),\n",
       "   ('multiplex_barcode', '0.4637'),\n",
       "   ('undescended_testis_corrected_method', '0.4559'),\n",
       "   ('library_strand', '0.4549'),\n",
       "   ('recist_targeted_regions_number', '0.4545'),\n",
       "   ('tumor_code', '0.4541'),\n",
       "   ('per_sequence_gc_content', '0.4431'),\n",
       "   ('library_strategy', '0.4417')]},\n",
       " {'Candidate column': 'CIBERSORT_T _cells _gamma _delta',\n",
       "  'Top k columns': [('fragment_maximum_length', '0.5147'),\n",
       "   ('fragment_minimum_length', '0.4914'),\n",
       "   ('fragment_standard_deviation_length', '0.4803'),\n",
       "   ('transglottic_extension', '0.476'),\n",
       "   ('fragment_mean_length', '0.4744'),\n",
       "   ('tumor_width_measurement', '0.4603'),\n",
       "   ('no_matched_normal_wgs', '0.459'),\n",
       "   ('tumor_length_measurement', '0.4563'),\n",
       "   ('no_matched_normal_low_pass_wgs', '0.4538'),\n",
       "   ('pairs_on_diff_chr', '0.4521')]},\n",
       " {'Candidate column': 'CIBERSORT_NK _cells _resting',\n",
       "  'Top k columns': [('cell_count', '0.5298'),\n",
       "   ('number_expect_cells', '0.513'),\n",
       "   ('flow_cell_barcode', '0.4843'),\n",
       "   ('base_caller_name', '0.4827'),\n",
       "   ('chipseq_antibody', '0.4759'),\n",
       "   ('fev1_ref_pre_bronch_percent', '0.464'),\n",
       "   ('single_cell_library', '0.4584'),\n",
       "   ('proc_internal', '0.4488'),\n",
       "   ('chipseq_target', '0.4435'),\n",
       "   ('copy_number_liftover_workflows', '0.4408')]},\n",
       " {'Candidate column': 'CIBERSORT_NK _cells _activated',\n",
       "  'Top k columns': [('cell_count', '0.5725'),\n",
       "   ('number_expect_cells', '0.5105'),\n",
       "   ('number_proliferating_cells', '0.5045'),\n",
       "   ('percent_stromal_cells', '0.4715'),\n",
       "   ('prostatic_chips_positive_count', '0.4442'),\n",
       "   ('percent_normal_cells', '0.443'),\n",
       "   ('flow_cell_barcode', '0.4418'),\n",
       "   ('chipseq_target', '0.4415'),\n",
       "   ('synchronous_malignancy', '0.4343'),\n",
       "   ('bone_marrow_malignant_cells', '0.4281')]},\n",
       " {'Candidate column': 'CIBERSORT_Monocytes',\n",
       "  'Top k columns': [('bone_marrow_malignant_cells', '0.5555'),\n",
       "   ('cell_count', '0.5074'),\n",
       "   ('tumor_infiltrating_lymphocytes', '0.4628'),\n",
       "   ('percent_tumor_cells', '0.4425'),\n",
       "   ('percent_monocyte_infiltration', '0.4411'),\n",
       "   ('medulloblastoma_molecular_classification', '0.4296'),\n",
       "   ('percent_tumor_nuclei', '0.415'),\n",
       "   ('cytoband', '0.3904'),\n",
       "   ('percent_stromal_cells', '0.3765'),\n",
       "   ('papillary_renal_cell_type', '0.3627')]},\n",
       " {'Candidate column': 'CIBERSORT_Macrophages _M0',\n",
       "  'Top k columns': [('specialized_molecular_test', '0.6746'),\n",
       "   ('enneking_msts_metastasis', '0.5373'),\n",
       "   ('tissue_microarray_coordinates', '0.5236'),\n",
       "   ('enneking_msts_grade', '0.5197'),\n",
       "   ('proportion_base_mismatch', '0.5091'),\n",
       "   ('size_selection_range', '0.5001'),\n",
       "   ('mitotic_total_area', '0.4826'),\n",
       "   ('micropapillary_features', '0.4791'),\n",
       "   ('tumor_infiltrating_macrophages', '0.4625'),\n",
       "   ('uicc_pathologic_m', '0.4607')]},\n",
       " {'Candidate column': 'CIBERSORT_Macrophages _M1',\n",
       "  'Top k columns': [('specialized_molecular_test', '0.6'),\n",
       "   ('tissue_microarray_coordinates', '0.5233'),\n",
       "   ('enneking_msts_metastasis', '0.5177'),\n",
       "   ('enneking_msts_grade', '0.4836'),\n",
       "   ('proportion_base_mismatch', '0.4802'),\n",
       "   ('micropapillary_features', '0.4696'),\n",
       "   ('uicc_pathologic_m', '0.4342'),\n",
       "   ('mitotic_total_area', '0.4341'),\n",
       "   ('micrometastasis_present', '0.4333'),\n",
       "   ('tumor_infiltrating_macrophages', '0.4322')]},\n",
       " {'Candidate column': 'CIBERSORT_Macrophages _M2',\n",
       "  'Top k columns': [('specialized_molecular_test', '0.6164'),\n",
       "   ('enneking_msts_metastasis', '0.529'),\n",
       "   ('tissue_microarray_coordinates', '0.5162'),\n",
       "   ('enneking_msts_grade', '0.4975'),\n",
       "   ('proportion_base_mismatch', '0.4911'),\n",
       "   ('micropapillary_features', '0.4482'),\n",
       "   ('size_selection_range', '0.4469'),\n",
       "   ('uicc_pathologic_m', '0.4399'),\n",
       "   ('tumor_infiltrating_macrophages', '0.4392'),\n",
       "   ('germline_mutation_calling_workflows', '0.4382')]},\n",
       " {'Candidate column': 'CIBERSORT_Dendritic _cells _resting',\n",
       "  'Top k columns': [('sequence_length_distribution', '0.4004'),\n",
       "   ('recist_targeted_regions_sum', '0.3789'),\n",
       "   ('dlco_ref_predictive_percent', '0.3709'),\n",
       "   ('tumor_length_measurement', '0.3699'),\n",
       "   ('best_overall_response', '0.3681'),\n",
       "   ('sequence_duplication_levels', '0.3562'),\n",
       "   ('residual_disease', '0.3559'),\n",
       "   ('per_sequence_quality_score', '0.3531'),\n",
       "   ('course_number', '0.3492'),\n",
       "   ('test_value_range', '0.3477')]},\n",
       " {'Candidate column': 'CIBERSORT_Dendritic _cells _activated',\n",
       "  'Top k columns': [('percent_tumor_cells', '0.3896'),\n",
       "   ('residual_disease', '0.3747'),\n",
       "   ('diagnosis_is_primary_disease', '0.3717'),\n",
       "   ('percent_tumor_nuclei', '0.3517'),\n",
       "   ('total_sequences', '0.3437'),\n",
       "   ('other_clinical_attributes', '0.3416'),\n",
       "   ('goblet_cells_columnar_mucosa_present', '0.3413'),\n",
       "   ('cytoband', '0.336'),\n",
       "   ('residual_tumor', '0.3316'),\n",
       "   ('selected_normal_targeted_sequencing', '0.3276')]},\n",
       " {'Candidate column': 'CIBERSORT_Mast _cells _resting',\n",
       "  'Top k columns': [('single_cell_library', '0.4594'),\n",
       "   ('somatic_mutation_calling_workflows', '0.4292'),\n",
       "   ('micrometastasis_present', '0.4153'),\n",
       "   ('tissue_microarray_coordinates', '0.4107'),\n",
       "   ('enneking_msts_metastasis', '0.4091'),\n",
       "   ('recist_targeted_regions_sum', '0.4073'),\n",
       "   ('proportion_base_mismatch', '0.3963'),\n",
       "   ('medulloblastoma_molecular_classification', '0.3767'),\n",
       "   ('recist_targeted_regions_number', '0.3766'),\n",
       "   ('enneking_msts_stage', '0.3701')]},\n",
       " {'Candidate column': 'CIBERSORT_Mast _cells _activated',\n",
       "  'Top k columns': [('single_cell_library', '0.4796'),\n",
       "   ('micrometastasis_present', '0.4494'),\n",
       "   ('cell_count', '0.428'),\n",
       "   ('chipseq_target', '0.4264'),\n",
       "   ('germline_mutation_calling_workflows', '0.4138'),\n",
       "   ('goblet_cells_columnar_mucosa_present', '0.4059'),\n",
       "   ('flow_cell_barcode', '0.4053'),\n",
       "   ('medulloblastoma_molecular_classification', '0.4053'),\n",
       "   ('bone_marrow_malignant_cells', '0.3948'),\n",
       "   ('proportion_base_mismatch', '0.3903')]},\n",
       " {'Candidate column': 'CIBERSORT_Eosinophils',\n",
       "  'Top k columns': [('gastric_esophageal_junction_involvement', '0.5389'),\n",
       "   ('papillary_renal_cell_type', '0.4526'),\n",
       "   ('peritoneal_fluid_cytological_status', '0.3933'),\n",
       "   ('extracapsular_extension', '0.3826'),\n",
       "   ('extracapsular_extension_present', '0.3777'),\n",
       "   ('extrascleral_extension', '0.3638'),\n",
       "   ('transglottic_extension', '0.3589'),\n",
       "   ('to_trim_adapter_sequence', '0.338'),\n",
       "   ('site_of_resection_or_biopsy', '0.3321'),\n",
       "   ('tumor_infiltrating_macrophages', '0.3268')]},\n",
       " {'Candidate column': 'CIBERSORT_Neutrophils',\n",
       "  'Top k columns': [('tumor_infiltrating_macrophages', '0.4836'),\n",
       "   ('papillary_renal_cell_type', '0.4745'),\n",
       "   ('biospecimen_anatomic_site', '0.3943'),\n",
       "   ('tissue_type', '0.3891'),\n",
       "   ('gastric_esophageal_junction_involvement', '0.3863'),\n",
       "   ('cytoband', '0.3833'),\n",
       "   ('tissue_microarray_coordinates', '0.381'),\n",
       "   ('chipseq_antibody', '0.373'),\n",
       "   ('ensat_pathologic_n', '0.3713'),\n",
       "   ('site_of_resection_or_biopsy', '0.3705')]},\n",
       " {'Candidate column': 'CIBERSORT_Absolute _score',\n",
       "  'Top k columns': [('weiss_assessment_score', '0.628'),\n",
       "   ('per_sequence_quality_score', '0.5752'),\n",
       "   ('msi_score', '0.5669'),\n",
       "   ('margin_status', '0.5151'),\n",
       "   ('ishak_fibrosis_score', '0.484'),\n",
       "   ('pairs_on_diff_chr', '0.448'),\n",
       "   ('per_base_sequence_quality', '0.4475'),\n",
       "   ('average_base_quality', '0.4358'),\n",
       "   ('best_overall_response', '0.4315'),\n",
       "   ('course_number', '0.4277')]},\n",
       " {'Candidate column': 'ESTIMATE_StromalScore',\n",
       "  'Top k columns': [('sample_ordinal', '0.3874'),\n",
       "   ('who_nte_grade', '0.3799'),\n",
       "   ('test_analyte_type', '0.3786'),\n",
       "   ('ensat_pathologic_n', '0.3667'),\n",
       "   ('classification_of_tumor', '0.3662'),\n",
       "   ('tumor_regression_grade', '0.3658'),\n",
       "   ('percent_stromal_cells', '0.3607'),\n",
       "   ('ensat_pathologic_t', '0.3579'),\n",
       "   ('analyte_type', '0.349'),\n",
       "   ('ensat_pathologic_stage', '0.3481')]},\n",
       " {'Candidate column': 'ESTIMATE_ImmuneScore',\n",
       "  'Top k columns': [('measurement_unit', '0.4294'),\n",
       "   ('msi_score', '0.4208'),\n",
       "   ('test_units', '0.408'),\n",
       "   ('residual_tumor_measurement', '0.4014'),\n",
       "   ('per_tile_sequence_quality', '0.3986'),\n",
       "   ('specialized_molecular_test', '0.3769'),\n",
       "   ('enneking_msts_tumor_site', '0.3614'),\n",
       "   ('radiosensitizing_agent', '0.3593'),\n",
       "   ('contamination', '0.3537'),\n",
       "   ('weiss_assessment_score', '0.3495')]},\n",
       " {'Candidate column': 'ESTIMATE_ESTIMATEScore',\n",
       "  'Top k columns': [('supratentorial_localization', '0.4371'),\n",
       "   ('state', '0.435'),\n",
       "   ('described_cases', '0.4232'),\n",
       "   ('experiment_name', '0.4139'),\n",
       "   ('extranodal_extension', '0.3973'),\n",
       "   ('undescended_testis_history', '0.3821'),\n",
       "   ('undescended_testis_history_laterality', '0.3819'),\n",
       "   ('zone_of_origin_prostate', '0.3748'),\n",
       "   ('data_formats', '0.3721'),\n",
       "   ('namespace', '0.372')]},\n",
       " {'Candidate column': 'Stemness_score',\n",
       "  'Top k columns': [('staining_intensity_value', '0.5731'),\n",
       "   ('gleason_score', '0.5525'),\n",
       "   ('staining_intensity_scale', '0.5474'),\n",
       "   ('msi_score', '0.5068'),\n",
       "   ('ulceration_indicator', '0.4198'),\n",
       "   ('secondary_gleason_grade', '0.4054'),\n",
       "   ('primary_gleason_grade', '0.3907'),\n",
       "   ('weiss_assessment_score', '0.3892'),\n",
       "   ('per_sequence_quality_score', '0.3787'),\n",
       "   ('per_tile_sequence_quality', '0.344')]},\n",
       " {'Candidate column': 'ER_ESR1',\n",
       "  'Top k columns': [('rin', '0.5183'),\n",
       "   ('adapter_sequence', '0.4293'),\n",
       "   ('adapter_name', '0.3969'),\n",
       "   ('protocol_identifier', '0.383'),\n",
       "   ('intron', '0.3563'),\n",
       "   ('rna_expression_workflows', '0.3532'),\n",
       "   ('ensat_pathologic_t', '0.3394'),\n",
       "   ('gene_symbol', '0.3344'),\n",
       "   ('releasable', '0.3326'),\n",
       "   ('scan_tracer_used', '0.3298')]},\n",
       " {'Candidate column': 'PR_PGR',\n",
       "  'Top k columns': [('platform', '0.5278'),\n",
       "   ('platforms', '0.5162'),\n",
       "   ('experimental_protocol_type', '0.5156'),\n",
       "   ('test_result', '0.4984'),\n",
       "   ('consistent_pathology_review', '0.4968'),\n",
       "   ('international_prognostic_index', '0.489'),\n",
       "   ('projects', '0.4859'),\n",
       "   ('programs', '0.4841'),\n",
       "   ('project', '0.4736'),\n",
       "   ('in_review', '0.4711')]},\n",
       " {'Candidate column': 'Pathway_activity_EGFR',\n",
       "  'Top k columns': [('additional_pathology_findings', '0.4948'),\n",
       "   ('ensat_pathologic_t', '0.4548'),\n",
       "   ('route_of_administration', '0.4373'),\n",
       "   ('rna_expression_workflows', '0.4287'),\n",
       "   ('workflow_type', '0.4101'),\n",
       "   ('library_strategy', '0.4037'),\n",
       "   ('to_trim_adapter_sequence', '0.3999'),\n",
       "   ('extranodal_extension', '0.3993'),\n",
       "   ('ann_arbor_extranodal_involvement', '0.3967'),\n",
       "   ('protocol_identifier', '0.3886')]},\n",
       " {'Candidate column': 'Pathway_activity_Hypoxia',\n",
       "  'Top k columns': [('therapeutic_level_achieved', '0.5316'),\n",
       "   ('therapeutic_levels_achieved', '0.5183'),\n",
       "   ('treatment_effect_indicator', '0.4912'),\n",
       "   ('treatment_effect', '0.4907'),\n",
       "   ('therapeutic_target_level', '0.4881'),\n",
       "   ('treatment_outcome', '0.4342'),\n",
       "   ('prostatic_involvement_percent', '0.4333'),\n",
       "   ('therapeutic_agents', '0.3981'),\n",
       "   ('treatment_intent_type', '0.387'),\n",
       "   ('treatment_outcome_duration', '0.3813')]},\n",
       " {'Candidate column': 'Pathway_activity_JAK.STAT',\n",
       "  'Top k columns': [('ajcc_pathologic_t', '0.523'),\n",
       "   ('ajcc_pathologic_n', '0.5077'),\n",
       "   ('ajcc_pathologic_m', '0.4844'),\n",
       "   ('lymph_node_involvement', '0.4621'),\n",
       "   ('lymph_node_dissection_method', '0.4567'),\n",
       "   ('ann_arbor_extranodal_involvement', '0.4543'),\n",
       "   ('workflow_link', '0.423'),\n",
       "   ('lymph_node_dissection_site', '0.4175'),\n",
       "   ('lymph_node_involved_site', '0.407'),\n",
       "   ('rna_expression_workflows', '0.3997')]},\n",
       " {'Candidate column': 'Pathway_activity_MAPK',\n",
       "  'Top k columns': [('prostatic_involvement_percent', '0.426'),\n",
       "   ('somatic_mutation_calling_workflows', '0.4074'),\n",
       "   ('metastasis_at_diagnosis', '0.4006'),\n",
       "   ('ensat_pathologic_t', '0.3972'),\n",
       "   ('lymph_node_involved_site', '0.3964'),\n",
       "   ('additional_pathology_findings', '0.3951'),\n",
       "   ('sites_of_involvement', '0.3925'),\n",
       "   ('metastasis_at_diagnosis_site', '0.3921'),\n",
       "   ('somatic_mutation_indexes', '0.3808'),\n",
       "   ('lymph_node_involvement', '0.379')]},\n",
       " {'Candidate column': 'Pathway_activity_NFkB',\n",
       "  'Top k columns': [('rna_expression_workflows', '0.4857'),\n",
       "   ('dbgap_accession_number', '0.453'),\n",
       "   ('biospecimen_anatomic_site', '0.453'),\n",
       "   ('ensat_pathologic_n', '0.45'),\n",
       "   ('progression_or_recurrence_anatomic_site', '0.4482'),\n",
       "   ('target_capture_kit_target_region', '0.4345'),\n",
       "   ('ajcc_pathologic_n', '0.4319'),\n",
       "   ('scan_tracer_used', '0.4232'),\n",
       "   ('flow_cell_barcode', '0.4223'),\n",
       "   ('lymph_node_involvement', '0.419')]},\n",
       " {'Candidate column': 'Pathway_activity_PI3K',\n",
       "  'Top k columns': [('primary_site', '0.5519'),\n",
       "   ('additional_pathology_findings', '0.4751'),\n",
       "   ('sites_of_involvement', '0.4621'),\n",
       "   ('primary_disease', '0.454'),\n",
       "   ('perineural_invasion_present', '0.4424'),\n",
       "   ('workflow_link', '0.4281'),\n",
       "   ('sites_of_involvement_count', '0.422'),\n",
       "   ('ensat_pathologic_t', '0.4215'),\n",
       "   ('procedures_performed', '0.4182'),\n",
       "   ('includes_spike_ins', '0.418')]},\n",
       " {'Candidate column': 'Pathway_activity_TGFb',\n",
       "  'Top k columns': [('ensat_pathologic_t', '0.4666'),\n",
       "   ('additional_pathology_findings', '0.4624'),\n",
       "   ('ajcc_pathologic_t', '0.3966'),\n",
       "   ('to_trim_adapter_sequence', '0.3815'),\n",
       "   ('barretts_esophagus_goblet_cells_present', '0.3786'),\n",
       "   ('ensat_pathologic_n', '0.372'),\n",
       "   ('rna_expression_workflows', '0.3683'),\n",
       "   ('library_strategy', '0.366'),\n",
       "   ('route_of_administration', '0.3562'),\n",
       "   ('oct_embedded', '0.3535')]},\n",
       " {'Candidate column': 'Pathway_activity_TNFa',\n",
       "  'Top k columns': [('ensat_pathologic_t', '0.4468'),\n",
       "   ('ensat_pathologic_n', '0.4302'),\n",
       "   ('lymph_node_involved_site', '0.4069'),\n",
       "   ('non_nodal_tumor_deposits', '0.4014'),\n",
       "   ('lymph_node_involvement', '0.3789'),\n",
       "   ('route_of_administration', '0.3689'),\n",
       "   ('rna_expression_workflows', '0.3621'),\n",
       "   ('somatic_annotation_workflows', '0.3571'),\n",
       "   ('recist_targeted_regions_number', '0.3569'),\n",
       "   ('additional_pathology_findings', '0.3508')]},\n",
       " {'Candidate column': 'Pathway_activity_Trail',\n",
       "  'Top k columns': [('additional_pathology_findings', '0.4976'),\n",
       "   ('ensat_pathologic_t', '0.4736'),\n",
       "   ('route_of_administration', '0.4561'),\n",
       "   ('ensat_pathologic_stage', '0.4364'),\n",
       "   ('ensat_pathologic_n', '0.4164'),\n",
       "   ('lane_number', '0.4063'),\n",
       "   ('zone_of_origin_prostate', '0.3863'),\n",
       "   ('sites_of_involvement', '0.3854'),\n",
       "   ('to_trim_adapter_sequence', '0.3624'),\n",
       "   ('undescended_testis_history_laterality', '0.3513')]},\n",
       " {'Candidate column': 'Pathway_activity_VEGF',\n",
       "  'Top k columns': [('additional_pathology_findings', '0.4321'),\n",
       "   ('international_prognostic_index', '0.396'),\n",
       "   ('sites_of_involvement', '0.3958'),\n",
       "   ('zone_of_origin_prostate', '0.3942'),\n",
       "   ('extranodal_extension', '0.3792'),\n",
       "   ('library_strategy', '0.3683'),\n",
       "   ('vascular_invasion_present', '0.352'),\n",
       "   ('experimental_strategy', '0.3492'),\n",
       "   ('vascular_invasion_type', '0.3478'),\n",
       "   ('oct_embedded', '0.3469')]},\n",
       " {'Candidate column': 'Pathway_activity_p53',\n",
       "  'Top k columns': [('target_capture_kit_catalog_number', '0.4498'),\n",
       "   ('ensat_pathologic_t', '0.4427'),\n",
       "   ('ensat_pathologic_n', '0.4405'),\n",
       "   ('proc_internal', '0.4257'),\n",
       "   ('chipseq_target', '0.4157'),\n",
       "   ('lane_number', '0.4157'),\n",
       "   ('prostatic_chips_positive_count', '0.4136'),\n",
       "   ('workflow_link', '0.4095'),\n",
       "   ('target_capture_kit_name', '0.3969'),\n",
       "   ('read_group_qcs', '0.3939')]},\n",
       " {'Candidate column': 'TP53_ATM',\n",
       "  'Top k columns': [('enneking_msts_metastasis', '0.3781'),\n",
       "   ('tumor_infiltrating_macrophages', '0.3752'),\n",
       "   ('double_expressor_lymphoma', '0.346'),\n",
       "   ('metastasis_at_diagnosis', '0.3336'),\n",
       "   ('treatment_anatomic_sites', '0.3306'),\n",
       "   ('chipseq_target', '0.3269'),\n",
       "   ('metastasis_at_diagnosis_site', '0.3097'),\n",
       "   ('prostatic_chips_total_count', '0.308'),\n",
       "   ('double_hit_lymphoma', '0.3018'),\n",
       "   ('burkitt_lymphoma_clinical_variant', '0.2979')]},\n",
       " {'Candidate column': 'TP53_CHEK2',\n",
       "  'Top k columns': [('prostatic_chips_positive_count', '0.4056'),\n",
       "   ('chip_id', '0.3998'),\n",
       "   ('prostatic_chips_total_count', '0.3766'),\n",
       "   ('number_proliferating_cells', '0.373'),\n",
       "   ('chip_position', '0.3672'),\n",
       "   ('primary_gleason_grade', '0.3589'),\n",
       "   ('chipseq_target', '0.3577'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3577'),\n",
       "   ('clark_level', '0.3438'),\n",
       "   ('lane_number', '0.3276')]},\n",
       " {'Candidate column': 'TP53_MDM4',\n",
       "  'Top k columns': [('metastasis_at_diagnosis', '0.3788'),\n",
       "   ('ensat_clinical_m', '0.3731'),\n",
       "   ('medulloblastoma_molecular_classification', '0.37'),\n",
       "   ('metaplasia_present', '0.3527'),\n",
       "   ('enneking_msts_metastasis', '0.3471'),\n",
       "   ('metastasis_at_diagnosis_site', '0.3455'),\n",
       "   ('double_expressor_lymphoma', '0.3365'),\n",
       "   ('melanoma_known_primary', '0.3289'),\n",
       "   ('enneking_msts_stage', '0.3254'),\n",
       "   ('uicc_pathologic_m', '0.3065')]},\n",
       " {'Candidate column': 'TP53_RPS6KA3',\n",
       "  'Top k columns': [('number_proliferating_cells', '0.3952'),\n",
       "   ('fastq_name', '0.3906'),\n",
       "   ('chipseq_target', '0.3585'),\n",
       "   ('chipseq_antibody', '0.3518'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3436'),\n",
       "   ('protein_expressions', '0.3427'),\n",
       "   ('chip_id', '0.3419'),\n",
       "   ('prostatic_chips_positive_count', '0.3418'),\n",
       "   ('includes_spike_ins', '0.3408'),\n",
       "   ('read_group_qcs', '0.3379')]},\n",
       " {'Candidate column': 'TP53_TP53',\n",
       "  'Top k columns': [('prostatic_chips_positive_count', '0.379'),\n",
       "   ('prostatic_chips_total_count', '0.3673'),\n",
       "   ('adapter_name', '0.3661'),\n",
       "   ('chip_id', '0.3654'),\n",
       "   ('number_proliferating_cells', '0.3479'),\n",
       "   ('is_ffpe', '0.329'),\n",
       "   ('burkitt_lymphoma_clinical_variant', '0.3269'),\n",
       "   ('chipseq_target', '0.3255'),\n",
       "   ('includes_spike_ins', '0.3221'),\n",
       "   ('pediatric_kidney_staging', '0.3144')]},\n",
       " {'Candidate column': 'TP53_pathway',\n",
       "  'Top k columns': [('ensat_pathologic_t', '0.4499'),\n",
       "   ('lane_number', '0.4477'),\n",
       "   ('ensat_pathologic_n', '0.4133'),\n",
       "   ('additional_pathology_findings', '0.3861'),\n",
       "   ('to_trim_adapter_sequence', '0.3794'),\n",
       "   ('extracapsular_extension_present', '0.378'),\n",
       "   ('section_location', '0.3595'),\n",
       "   ('extracapsular_extension', '0.3541'),\n",
       "   ('primary_site', '0.3334'),\n",
       "   ('lymph_node_dissection_site', '0.3293')]},\n",
       " {'Candidate column': 'PI3K_AKT1',\n",
       "  'Top k columns': [('includes_spike_ins', '0.4639'),\n",
       "   ('is_ffpe', '0.45'),\n",
       "   ('prostatic_chips_positive_count', '0.4358'),\n",
       "   ('mitosis_karyorrhexis_index', '0.4256'),\n",
       "   ('protein_expressions', '0.4113'),\n",
       "   ('chip_id', '0.4053'),\n",
       "   ('primary_gleason_grade', '0.4008'),\n",
       "   ('chipseq_target', '0.3975'),\n",
       "   ('chip_position', '0.3921'),\n",
       "   ('prostatic_chips_total_count', '0.3786')]},\n",
       " {'Candidate column': 'PI3K_AKT2',\n",
       "  'Top k columns': [('includes_spike_ins', '0.4726'),\n",
       "   ('is_ffpe', '0.4538'),\n",
       "   ('protein_expressions', '0.4122'),\n",
       "   ('prostatic_chips_positive_count', '0.4111'),\n",
       "   ('chipseq_target', '0.4005'),\n",
       "   ('primary_gleason_grade', '0.3984'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3971'),\n",
       "   ('chip_id', '0.3903'),\n",
       "   ('target_capture_kit_name', '0.3828'),\n",
       "   ('chip_position', '0.3767')]},\n",
       " {'Candidate column': 'PI3K_AKT3',\n",
       "  'Top k columns': [('protein_expressions', '0.4263'),\n",
       "   ('includes_spike_ins', '0.4146'),\n",
       "   ('primary_gleason_grade', '0.4095'),\n",
       "   ('is_ffpe', '0.405'),\n",
       "   ('chipseq_target', '0.4015'),\n",
       "   ('chip_id', '0.3999'),\n",
       "   ('kmer_content', '0.3906'),\n",
       "   ('target_capture_kit_name', '0.3862'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3838'),\n",
       "   ('target_capture_kit', '0.3741')]},\n",
       " {'Candidate column': 'PI3K_DEPDC5',\n",
       "  'Top k columns': [('primary_disease', '0.4146'),\n",
       "   ('diagnosis_is_primary_disease', '0.3789'),\n",
       "   ('primary_gleason_grade', '0.3771'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3674'),\n",
       "   ('pmid', '0.3671'),\n",
       "   ('cancer_detection_method', '0.3471'),\n",
       "   ('is_ffpe', '0.3412'),\n",
       "   ('md5sum', '0.3256'),\n",
       "   ('protein_expressions', '0.3224'),\n",
       "   ('evidence_of_progression_type', '0.3219')]},\n",
       " {'Candidate column': 'PI3K_DEPTOR',\n",
       "  'Top k columns': [('primary_disease', '0.5455'),\n",
       "   ('perineural_invasion_present', '0.5106'),\n",
       "   ('primary_site', '0.5096'),\n",
       "   ('pmid', '0.4966'),\n",
       "   ('primary_diagnosis', '0.4348'),\n",
       "   ('procedures_performed', '0.4319'),\n",
       "   ('diagnosis_is_primary_disease', '0.4256'),\n",
       "   ('is_ffpe', '0.4247'),\n",
       "   ('projects', '0.4178'),\n",
       "   ('project', '0.4145')]},\n",
       " {'Candidate column': 'PI3K_INPP4B',\n",
       "  'Top k columns': [('chip_position', '0.4462'),\n",
       "   ('chip_id', '0.4441'),\n",
       "   ('includes_spike_ins', '0.4191'),\n",
       "   ('prostatic_chips_positive_count', '0.419'),\n",
       "   ('perineural_invasion_present', '0.4161'),\n",
       "   ('primary_gleason_grade', '0.4061'),\n",
       "   ('protein_expressions', '0.4023'),\n",
       "   ('chipseq_target', '0.3819'),\n",
       "   ('chipseq_antibody', '0.3787'),\n",
       "   ('fab_morphology_code', '0.3704')]},\n",
       " {'Candidate column': 'PI3K_MAPKAP1',\n",
       "  'Top k columns': [('enneking_msts_metastasis', '0.4025'),\n",
       "   ('enneking_msts_stage', '0.3966'),\n",
       "   ('enneking_msts_grade', '0.3915'),\n",
       "   ('primary_gleason_grade', '0.3871'),\n",
       "   ('specialized_molecular_test', '0.3816'),\n",
       "   ('slide_images', '0.3779'),\n",
       "   ('pmid', '0.3766'),\n",
       "   ('msi_status', '0.3735'),\n",
       "   ('protein_expressions', '0.3697'),\n",
       "   ('primary_site', '0.3688')]},\n",
       " {'Candidate column': 'PI3K_MLST8',\n",
       "  'Top k columns': [('simple_germline_variations', '0.4952'),\n",
       "   ('specialized_molecular_test', '0.4911'),\n",
       "   ('enneking_msts_grade', '0.4892'),\n",
       "   ('pairs_on_diff_chr', '0.4453'),\n",
       "   ('icd_10_code', '0.4424'),\n",
       "   ('primary_gleason_grade', '0.4361'),\n",
       "   ('fab_morphology_code', '0.4275'),\n",
       "   ('enneking_msts_stage', '0.4234'),\n",
       "   ('fragment_minimum_length', '0.4058'),\n",
       "   ('fragment_maximum_length', '0.4015')]},\n",
       " {'Candidate column': 'PI3K_MTOR',\n",
       "  'Top k columns': [('pmid', '0.4599'),\n",
       "   ('enneking_msts_metastasis', '0.4401'),\n",
       "   ('masked_somatic_mutations', '0.4327'),\n",
       "   ('msi_status', '0.4252'),\n",
       "   ('perineural_invasion_present', '0.4191'),\n",
       "   ('residual_tumor_measurement', '0.4181'),\n",
       "   ('instrument_model', '0.4154'),\n",
       "   ('specialized_molecular_test', '0.415'),\n",
       "   ('enneking_msts_tumor_site', '0.4067'),\n",
       "   ('supratentorial_localization', '0.3986')]},\n",
       " {'Candidate column': 'PI3K_NPRL2',\n",
       "  'Top k columns': [('includes_spike_ins', '0.4284'),\n",
       "   ('primary_gleason_grade', '0.4113'),\n",
       "   ('mitosis_karyorrhexis_index', '0.393'),\n",
       "   ('perineural_invasion_present', '0.381'),\n",
       "   ('prostatic_chips_positive_count', '0.3763'),\n",
       "   ('pmid', '0.3714'),\n",
       "   ('is_ffpe', '0.3669'),\n",
       "   ('protein_expressions', '0.3621'),\n",
       "   ('primary_site', '0.3614'),\n",
       "   ('primary_disease', '0.3611')]},\n",
       " {'Candidate column': 'PI3K_NPRL3',\n",
       "  'Top k columns': [('primary_gleason_grade', '0.4261'),\n",
       "   ('perineural_invasion_present', '0.389'),\n",
       "   ('includes_spike_ins', '0.3799'),\n",
       "   ('pmid', '0.3791'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3785'),\n",
       "   ('protein_expressions', '0.3679'),\n",
       "   ('primary_disease', '0.3677'),\n",
       "   ('evidence_of_progression_type', '0.3404'),\n",
       "   ('additional_pathology_findings', '0.3312'),\n",
       "   ('primary_site', '0.3247')]},\n",
       " {'Candidate column': 'PI3K_PDK1',\n",
       "  'Top k columns': [('prostatic_chips_positive_count', '0.4597'),\n",
       "   ('primary_disease', '0.4431'),\n",
       "   ('primary_gleason_grade', '0.4409'),\n",
       "   ('includes_spike_ins', '0.4363'),\n",
       "   ('gleason_patterns_percent', '0.4149'),\n",
       "   ('diagnosis_is_primary_disease', '0.4075'),\n",
       "   ('mitosis_karyorrhexis_index', '0.4023'),\n",
       "   ('hpv_positive_type', '0.4'),\n",
       "   ('melanoma_known_primary', '0.3932'),\n",
       "   ('protein_expressions', '0.3928')]},\n",
       " {'Candidate column': 'PI3K_PIK3CA',\n",
       "  'Top k columns': [('chip_position', '0.4268'),\n",
       "   ('chip_id', '0.4211'),\n",
       "   ('enneking_msts_grade', '0.4081'),\n",
       "   ('simple_germline_variations', '0.407'),\n",
       "   ('chipseq_target', '0.4066'),\n",
       "   ('pairs_on_diff_chr', '0.4059'),\n",
       "   ('kmer_content', '0.4048'),\n",
       "   ('protein_expressions', '0.4042'),\n",
       "   ('international_prognostic_index', '0.3999'),\n",
       "   ('additional_pathology_findings', '0.3968')]},\n",
       " {'Candidate column': 'PI3K_PIK3CB',\n",
       "  'Top k columns': [('ajcc_pathologic_m', '0.4783'),\n",
       "   ('igcccg_stage', '0.4739'),\n",
       "   ('chip_id', '0.467'),\n",
       "   ('chipseq_target', '0.4517'),\n",
       "   ('read_group_qcs', '0.4489'),\n",
       "   ('uicc_pathologic_m', '0.4329'),\n",
       "   ('ajcc_pathologic_t', '0.4298'),\n",
       "   ('cd4_count', '0.4297'),\n",
       "   ('multiplex_barcode', '0.4225'),\n",
       "   ('uicc_pathologic_t', '0.4212')]},\n",
       " {'Candidate column': 'PI3K_PIK3R1',\n",
       "  'Top k columns': [('perineural_invasion_present', '0.4313'),\n",
       "   ('primary_gleason_grade', '0.4225'),\n",
       "   ('protein_expressions', '0.4072'),\n",
       "   ('includes_spike_ins', '0.4047'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3791'),\n",
       "   ('rin', '0.3725'),\n",
       "   ('primary_disease', '0.369'),\n",
       "   ('chip_position', '0.3676'),\n",
       "   ('primary_site', '0.3672'),\n",
       "   ('evidence_of_progression_type', '0.3597')]},\n",
       " {'Candidate column': 'PI3K_PIK3R2',\n",
       "  'Top k columns': [('primary_gleason_grade', '0.4194'),\n",
       "   ('includes_spike_ins', '0.4018'),\n",
       "   ('protein_expressions', '0.3927'),\n",
       "   ('perineural_invasion_present', '0.3899'),\n",
       "   ('evidence_of_progression_type', '0.3765'),\n",
       "   ('primary_site', '0.3532'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3498'),\n",
       "   ('primary_disease', '0.3456'),\n",
       "   ('additional_pathology_findings', '0.3401'),\n",
       "   ('scan_tracer_used', '0.3398')]},\n",
       " {'Candidate column': 'PI3K_PPP2R1A',\n",
       "  'Top k columns': [('protein_expressions', '0.4507'),\n",
       "   ('perineural_invasion_present', '0.3828'),\n",
       "   ('primary_gleason_grade', '0.3714'),\n",
       "   ('includes_spike_ins', '0.3668'),\n",
       "   ('evidence_of_progression_type', '0.3639'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3553'),\n",
       "   ('is_ffpe', '0.3517'),\n",
       "   ('prostatic_chips_positive_count', '0.3515'),\n",
       "   ('primary_site', '0.3334'),\n",
       "   ('rin', '0.3286')]},\n",
       " {'Candidate column': 'PI3K_PTEN',\n",
       "  'Top k columns': [('primary_site', '0.5371'),\n",
       "   ('is_ffpe', '0.5059'),\n",
       "   ('perineural_invasion_present', '0.4658'),\n",
       "   ('procedures_performed', '0.4612'),\n",
       "   ('pmid', '0.4536'),\n",
       "   ('international_prognostic_index', '0.4521'),\n",
       "   ('id', '0.4444'),\n",
       "   ('protein_expressions', '0.4384'),\n",
       "   ('includes_spike_ins', '0.4376'),\n",
       "   ('prostatic_chips_positive_count', '0.4313')]},\n",
       " {'Candidate column': 'PI3K_RHEB',\n",
       "  'Top k columns': [('protein_expressions', '0.4776'),\n",
       "   ('rin', '0.4766'),\n",
       "   ('mitosis_karyorrhexis_index', '0.4413'),\n",
       "   ('additional_pathology_findings', '0.4084'),\n",
       "   ('perineural_invasion_present', '0.3734'),\n",
       "   ('histone_family', '0.3659'),\n",
       "   ('analytes', '0.3543'),\n",
       "   ('primary_site', '0.3414'),\n",
       "   ('fab_morphology_code', '0.3382'),\n",
       "   ('chip_position', '0.3331')]},\n",
       " {'Candidate column': 'PI3K_RICTOR',\n",
       "  'Top k columns': [('perineural_invasion_present', '0.5335'),\n",
       "   ('rin', '0.5077'),\n",
       "   ('includes_spike_ins', '0.426'),\n",
       "   ('primary_disease', '0.4121'),\n",
       "   ('primary_site', '0.4052'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3903'),\n",
       "   ('margins_involved_site', '0.3739'),\n",
       "   ('is_ffpe', '0.3732'),\n",
       "   ('largest_extrapelvic_peritoneal_focus', '0.3653'),\n",
       "   ('percent_tumor_invasion', '0.359')]},\n",
       " {'Candidate column': 'PI3K_RPS6',\n",
       "  'Top k columns': [('rin', '0.4198'),\n",
       "   ('protein_expressions', '0.4153'),\n",
       "   ('programs', '0.4143'),\n",
       "   ('procedures_performed', '0.3994'),\n",
       "   ('library_preparation_kit_vendor', '0.3964'),\n",
       "   ('scan_tracer_used', '0.3924'),\n",
       "   ('protocol_identifier', '0.3841'),\n",
       "   ('code', '0.3711'),\n",
       "   ('includes_spike_ins', '0.3703'),\n",
       "   ('library_preparation_kit_name', '0.3689')]},\n",
       " {'Candidate column': 'PI3K_RPS6KB1',\n",
       "  'Top k columns': [('multiplex_barcode', '0.4766'),\n",
       "   ('kmer_content', '0.4332'),\n",
       "   ('read_group_qcs', '0.429'),\n",
       "   ('library_preparation_kit_vendor', '0.4264'),\n",
       "   ('library_preparation_kit_name', '0.4177'),\n",
       "   ('target_capture_kit_name', '0.417'),\n",
       "   ('chipseq_target', '0.417'),\n",
       "   ('target_capture_kit_catalog_number', '0.4155'),\n",
       "   ('library_strand', '0.4154'),\n",
       "   ('chip_id', '0.4074')]},\n",
       " {'Candidate column': 'PI3K_RPTOR',\n",
       "  'Top k columns': [('rin', '0.5505'),\n",
       "   ('perineural_invasion_present', '0.4989'),\n",
       "   ('procedures_performed', '0.4945'),\n",
       "   ('primary_site', '0.4858'),\n",
       "   ('additional_pathology_findings', '0.4751'),\n",
       "   ('scan_tracer_used', '0.4619'),\n",
       "   ('type', '0.4583'),\n",
       "   ('analytes', '0.4483'),\n",
       "   ('protocol_identifier', '0.4451'),\n",
       "   ('analyte_type_id', '0.4424')]},\n",
       " {'Candidate column': 'PI3K_STK11',\n",
       "  'Top k columns': [('chipseq_target', '0.4824'),\n",
       "   ('chip_id', '0.4715'),\n",
       "   ('icd_10_code', '0.469'),\n",
       "   ('target_capture_kit_name', '0.4516'),\n",
       "   ('icd_10', '0.4509'),\n",
       "   ('target_capture_kit_version', '0.4465'),\n",
       "   ('target_capture_kit', '0.4377'),\n",
       "   ('target_capture_kit_vendor', '0.424'),\n",
       "   ('programs', '0.4215'),\n",
       "   ('chip_position', '0.4167')]},\n",
       " {'Candidate column': 'PI3K_TSC1',\n",
       "  'Top k columns': [('is_ffpe', '0.4283'),\n",
       "   ('includes_spike_ins', '0.4254'),\n",
       "   ('primary_site', '0.4211'),\n",
       "   ('prostatic_chips_positive_count', '0.4035'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3984'),\n",
       "   ('primary_gleason_grade', '0.3756'),\n",
       "   ('protein_expressions', '0.3734'),\n",
       "   ('chipseq_target', '0.3657'),\n",
       "   ('chip_position', '0.3654'),\n",
       "   ('analytes', '0.3607')]},\n",
       " {'Candidate column': 'PI3K_TSC2',\n",
       "  'Top k columns': [('includes_spike_ins', '0.4423'),\n",
       "   ('is_ffpe', '0.4363'),\n",
       "   ('primary_site', '0.4329'),\n",
       "   ('prostatic_chips_positive_count', '0.3811'),\n",
       "   ('primary_gleason_grade', '0.3798'),\n",
       "   ('mitosis_karyorrhexis_index', '0.3605'),\n",
       "   ('evidence_of_progression_type', '0.3585'),\n",
       "   ('protein_expressions', '0.358'),\n",
       "   ('international_prognostic_index', '0.3503'),\n",
       "   ('procedures_performed', '0.3495')]},\n",
       " {'Candidate column': 'PI3K_pathway',\n",
       "  'Top k columns': [('primary_site', '0.5069'),\n",
       "   ('additional_pathology_findings', '0.5023'),\n",
       "   ('sites_of_involvement', '0.4578'),\n",
       "   ('perineural_invasion_present', '0.4447'),\n",
       "   ('to_trim_adapter_sequence', '0.4273'),\n",
       "   ('ensat_pathologic_t', '0.415'),\n",
       "   ('includes_spike_ins', '0.4108'),\n",
       "   ('procedures_performed', '0.408'),\n",
       "   ('protein_expressions', '0.4061'),\n",
       "   ('locus', '0.4006')]},\n",
       " {'Candidate column': 'HRD_BRCA1',\n",
       "  'Top k columns': [('percent_rhabdoid_features', '0.4272'),\n",
       "   ('rhabdoid_percent', '0.3777'),\n",
       "   ('adrenal_hormone', '0.3682'),\n",
       "   ('bcr_id', '0.3487'),\n",
       "   ('histone_family', '0.3478'),\n",
       "   ('ajcc_clinical_m', '0.3413'),\n",
       "   ('ajcc_pathologic_stage', '0.3368'),\n",
       "   ('rhabdoid_present', '0.3277'),\n",
       "   ('ribosomal_rna_28s_16s_ratio', '0.3176'),\n",
       "   ('ajcc_pathologic_m', '0.3155')]},\n",
       " {'Candidate column': 'HRD_BRCA2',\n",
       "  'Top k columns': [('percent_rhabdoid_features', '0.4212'),\n",
       "   ('rhabdoid_percent', '0.3706'),\n",
       "   ('adrenal_hormone', '0.3656'),\n",
       "   ('bcr_id', '0.3607'),\n",
       "   ('extrathyroid_extension', '0.3453'),\n",
       "   ('ajcc_clinical_m', '0.3351'),\n",
       "   ('ajcc_pathologic_stage', '0.3346'),\n",
       "   ('ribosomal_rna_28s_16s_ratio', '0.3271'),\n",
       "   ('histone_family', '0.3263'),\n",
       "   ('rhabdoid_present', '0.3185')]},\n",
       " {'Candidate column': 'HRD_BRCA1_or_BRCA2',\n",
       "  'Top k columns': [('library_preparation_kit_name', '0.4796'),\n",
       "   ('ajcc_pathologic_stage', '0.4769'),\n",
       "   ('ajcc_pathologic_m', '0.4705'),\n",
       "   ('library_preparation_kit_vendor', '0.4679'),\n",
       "   ('library_preparation_kit_catalog_number', '0.4622'),\n",
       "   ('library_preparation_kit_version', '0.456'),\n",
       "   ('library_strategy', '0.4466'),\n",
       "   ('ajcc_pathologic_t', '0.4451'),\n",
       "   ('icd_10_code', '0.4352'),\n",
       "   ('igcccg_stage', '0.4244')]},\n",
       " {'Candidate column': 'CNV_DEL',\n",
       "  'Top k columns': [('extranodal_extension', '0.4501'),\n",
       "   ('library_name', '0.4014'),\n",
       "   ('dlco_ref_predictive_percent', '0.3981'),\n",
       "   ('library_strand', '0.3969'),\n",
       "   ('molecular_consequence', '0.3949'),\n",
       "   ('library_strategy', '0.3748'),\n",
       "   ('adapter_sequence', '0.3622'),\n",
       "   ('variant_type', '0.3596'),\n",
       "   ('namespace', '0.3594'),\n",
       "   ('simple_germline_variations', '0.3562')]},\n",
       " {'Candidate column': 'CNV_AMP',\n",
       "  'Top k columns': [('base_caller_version', '0.5442'),\n",
       "   ('base_caller_name', '0.4892'),\n",
       "   ('library_strand', '0.4674'),\n",
       "   ('library_preparation_kit_version', '0.4637'),\n",
       "   ('library_name', '0.4513'),\n",
       "   ('single_cell_library', '0.4483'),\n",
       "   ('multiplex_barcode', '0.439'),\n",
       "   ('library_strategy', '0.4384'),\n",
       "   ('chipseq_antibody', '0.438'),\n",
       "   ('copy_number_variation_workflows', '0.4342')]},\n",
       " {'Candidate column': 'CNV_class',\n",
       "  'Top k columns': [('library_name', '0.5486'),\n",
       "   ('library_strand', '0.5423'),\n",
       "   ('base_caller_version', '0.5376'),\n",
       "   ('base_caller_name', '0.5317'),\n",
       "   ('library_strategy', '0.5228'),\n",
       "   ('fab_morphology_code', '0.5042'),\n",
       "   ('structural_variant_calling_workflows', '0.4996'),\n",
       "   ('copy_number_variation_workflows', '0.497'),\n",
       "   ('variant_type', '0.4952'),\n",
       "   ('chipseq_antibody', '0.4858')]},\n",
       " {'Candidate column': 'CNV_idx',\n",
       "  'Top k columns': [('library_name', '0.5912'),\n",
       "   ('undescended_testis_corrected', '0.5713'),\n",
       "   ('base_caller_version', '0.5676'),\n",
       "   ('library_strand', '0.5601'),\n",
       "   ('copy_number_variation_workflows', '0.5526'),\n",
       "   ('library_strategy', '0.5473'),\n",
       "   ('tumor_code_id', '0.534'),\n",
       "   ('tumor_code', '0.5317'),\n",
       "   ('undescended_testis_corrected_method', '0.5316'),\n",
       "   ('chipseq_antibody', '0.5305')]},\n",
       " {'Candidate column': 'CNV_1q_DEL',\n",
       "  'Top k columns': [('library_strategy', '0.4995'),\n",
       "   ('library_strand', '0.4955'),\n",
       "   ('library_name', '0.4922'),\n",
       "   ('extranodal_extension', '0.4882'),\n",
       "   ('dlco_ref_predictive_percent', '0.4819'),\n",
       "   ('structural_variant_calling_workflows', '0.4769'),\n",
       "   ('enneking_msts_stage', '0.4613'),\n",
       "   ('intron', '0.457'),\n",
       "   ('copy_number_variation_workflows', '0.4544'),\n",
       "   ('enneking_msts_metastasis', '0.4481')]},\n",
       " {'Candidate column': 'CNV_3q_DEL',\n",
       "  'Top k columns': [('library_strategy', '0.564'),\n",
       "   ('library_strand', '0.5543'),\n",
       "   ('extranodal_extension', '0.5424'),\n",
       "   ('library_name', '0.5322'),\n",
       "   ('sequence_duplication_levels', '0.5141'),\n",
       "   ('structural_variant_calling_workflows', '0.5027'),\n",
       "   ('variant_type', '0.4962'),\n",
       "   ('simple_germline_variations', '0.4928'),\n",
       "   ('project_id', '0.4746'),\n",
       "   ('fab_morphology_code', '0.4713')]},\n",
       " {'Candidate column': 'CNV_4q_DEL',\n",
       "  'Top k columns': [('extranodal_extension', '0.5623'),\n",
       "   ('library_strategy', '0.55'),\n",
       "   ('library_strand', '0.5453'),\n",
       "   ('library_name', '0.5342'),\n",
       "   ('structural_variant_calling_workflows', '0.4971'),\n",
       "   ('dlco_ref_predictive_percent', '0.4776'),\n",
       "   ('simple_germline_variations', '0.4774'),\n",
       "   ('intron', '0.4718'),\n",
       "   ('read_group_qcs', '0.4668'),\n",
       "   ('enneking_msts_stage', '0.4631')]},\n",
       " {'Candidate column': 'CNV_1q_AMP',\n",
       "  'Top k columns': [('base_caller_version', '0.6163'),\n",
       "   ('base_caller_name', '0.5612'),\n",
       "   ('chipseq_antibody', '0.5363'),\n",
       "   ('read_group_qcs', '0.5183'),\n",
       "   ('chipseq_target', '0.5049'),\n",
       "   ('library_preparation_kit_version', '0.5'),\n",
       "   ('target_capture_kit_version', '0.4948'),\n",
       "   ('single_cell_library', '0.4902'),\n",
       "   ('multiplex_barcode', '0.4883'),\n",
       "   ('library_strand', '0.4845')]},\n",
       " {'Candidate column': 'CNV_3q_AMP',\n",
       "  'Top k columns': [('base_caller_version', '0.5911'),\n",
       "   ('library_strand', '0.5401'),\n",
       "   ('base_caller_name', '0.5396'),\n",
       "   ('read_group_qcs', '0.5317'),\n",
       "   ('library_strategy', '0.5306'),\n",
       "   ('library_preparation_kit_version', '0.529'),\n",
       "   ('multiplex_barcode', '0.5238'),\n",
       "   ('chipseq_antibody', '0.521'),\n",
       "   ('library_name', '0.5105'),\n",
       "   ('single_cell_library', '0.5028')]},\n",
       " {'Candidate column': 'CNV_4q_AMP',\n",
       "  'Top k columns': [('base_caller_version', '0.6241'),\n",
       "   ('base_caller_name', '0.5805'),\n",
       "   ('read_group_qcs', '0.5718'),\n",
       "   ('chipseq_antibody', '0.5502'),\n",
       "   ('multiplex_barcode', '0.5287'),\n",
       "   ('library_strand', '0.5284'),\n",
       "   ('single_cell_library', '0.5219'),\n",
       "   ('library_preparation_kit_version', '0.5153'),\n",
       "   ('library_strategy', '0.5138'),\n",
       "   ('chipseq_target', '0.5092')]},\n",
       " {'Candidate column': 'Purity_Immune',\n",
       "  'Top k columns': [('biospecimen_supplements', '0.3847'),\n",
       "   ('biospecimen_type', '0.3762'),\n",
       "   ('tumor_purity', '0.3383'),\n",
       "   ('mirna_expressions', '0.3338'),\n",
       "   ('molecular_tests', '0.3311'),\n",
       "   ('contamination', '0.3277'),\n",
       "   ('state', '0.3261'),\n",
       "   ('is_ffpe', '0.3184'),\n",
       "   ('risk_factors', '0.3166'),\n",
       "   ('biospecimen_volume', '0.315')]},\n",
       " {'Candidate column': 'Purity_Cancer',\n",
       "  'Top k columns': [('tumor_purity', '0.4593'),\n",
       "   ('bone_marrow_malignant_cells', '0.4237'),\n",
       "   ('synchronous_malignancy', '0.4118'),\n",
       "   ('percent_tumor_nuclei', '0.3946'),\n",
       "   ('relative_with_cancer_history', '0.3625'),\n",
       "   ('cancer_detection_method', '0.3539'),\n",
       "   ('is_ffpe', '0.3494'),\n",
       "   ('relatives_with_cancer_history_count', '0.3466'),\n",
       "   ('melanoma_known_primary', '0.3416'),\n",
       "   ('number_expect_cells', '0.336')]},\n",
       " {'Candidate column': 'Purity_Stroma',\n",
       "  'Top k columns': [('stain_type', '0.5347'),\n",
       "   ('melanoma_known_primary', '0.4947'),\n",
       "   ('eye_color', '0.4586'),\n",
       "   ('staining_intensity_value', '0.4415'),\n",
       "   ('staining_intensity_scale', '0.4367'),\n",
       "   ('respirable_crystalline_silica_exposure', '0.4264'),\n",
       "   ('diagnosis_is_primary_disease', '0.4126'),\n",
       "   ('parent_with_radiation_exposure', '0.4042'),\n",
       "   ('lesions_treated_number', '0.4006'),\n",
       "   ('radon_exposure', '0.3945')]},\n",
       " {'Candidate column': 'MSI_status',\n",
       "  'Top k columns': [('msi_status', '0.5876'),\n",
       "   ('programs', '0.4932'),\n",
       "   ('pmid', '0.4577'),\n",
       "   ('proportion_base_mismatch', '0.456'),\n",
       "   ('msi_score', '0.4548'),\n",
       "   ('mirna_expression_workflows', '0.4481'),\n",
       "   ('mirna_expressions', '0.4417'),\n",
       "   ('specialized_molecular_test', '0.4349'),\n",
       "   ('run_metadata_files', '0.4321'),\n",
       "   ('masked_somatic_mutations', '0.4299')]},\n",
       " {'Candidate column': 'POLE_subtype',\n",
       "  'Top k columns': [('transglottic_extension', '0.4514'),\n",
       "   ('discontiguous_lesion_count', '0.4283'),\n",
       "   ('satellite_nodule_present', '0.4229'),\n",
       "   ('supratentorial_localization', '0.3956'),\n",
       "   ('clark_level', '0.3744'),\n",
       "   ('primary_gleason_grade', '0.3556'),\n",
       "   ('is_ffpe', '0.3543'),\n",
       "   ('zygosity', '0.3436'),\n",
       "   ('extrascleral_extension', '0.342'),\n",
       "   ('includes_spike_ins', '0.3412')]},\n",
       " {'Candidate column': 'JAK1_MS_INDEL',\n",
       "  'Top k columns': [('ajcc_pathologic_m', '0.5961'),\n",
       "   ('enneking_msts_stage', '0.5332'),\n",
       "   ('simple_germline_variations', '0.5322'),\n",
       "   ('ajcc_pathologic_n', '0.5083'),\n",
       "   ('ajcc_pathologic_t', '0.5008'),\n",
       "   ('base_caller_version', '0.4995'),\n",
       "   ('germline_mutation_calling_workflows', '0.4993'),\n",
       "   ('fab_morphology_code', '0.4888'),\n",
       "   ('enneking_msts_grade', '0.4855'),\n",
       "   ('schema_version', '0.485')]},\n",
       " {'Candidate column': 'JAK1_Mutation',\n",
       "  'Top k columns': [('ajcc_pathologic_m', '0.5554'),\n",
       "   ('mutation_codon', '0.5268'),\n",
       "   ('chipseq_antibody', '0.4718'),\n",
       "   ('ajcc_pathologic_t', '0.4691'),\n",
       "   ('somatic_mutation_calling_workflows', '0.4638'),\n",
       "   ('schema_version', '0.4555'),\n",
       "   ('ajcc_pathologic_n', '0.4537'),\n",
       "   ('bmi', '0.4519'),\n",
       "   ('base_caller_version', '0.4452'),\n",
       "   ('fab_morphology_code', '0.4397')]},\n",
       " {'Candidate column': 'Log2_variant_per_Mbp',\n",
       "  'Top k columns': [('simple_germline_variations', '0.6519'),\n",
       "   ('enneking_msts_grade', '0.6161'),\n",
       "   ('proportion_base_mismatch', '0.5951'),\n",
       "   ('enneking_msts_metastasis', '0.5729'),\n",
       "   ('enneking_msts_stage', '0.5669'),\n",
       "   ('variant_type', '0.5595'),\n",
       "   ('multiplex_barcode', '0.55'),\n",
       "   ('copy_number_variation_workflows', '0.5391'),\n",
       "   ('fab_morphology_code', '0.5359'),\n",
       "   ('somatic_mutation_indexes', '0.5275')]},\n",
       " {'Candidate column': 'Log2_SNP_per_Mbp',\n",
       "  'Top k columns': [('multiplex_barcode', '0.6614'),\n",
       "   ('target_capture_kit_catalog_number', '0.6212'),\n",
       "   ('dbgap_accession_number', '0.6143'),\n",
       "   ('target_capture_kit_version', '0.6051'),\n",
       "   ('library_strand', '0.5885'),\n",
       "   ('programs', '0.5857'),\n",
       "   ('read_groups', '0.5835'),\n",
       "   ('read_group_qcs', '0.5827'),\n",
       "   ('library_preparation_kit_catalog_number', '0.5811'),\n",
       "   ('undescended_testis_corrected_method', '0.5634')]},\n",
       " {'Candidate column': 'Log2_INDEL_per_Mbp',\n",
       "  'Top k columns': [('multiplex_barcode', '0.6087'),\n",
       "   ('proportion_base_mismatch', '0.5787'),\n",
       "   ('library_strand', '0.5689'),\n",
       "   ('enneking_msts_metastasis', '0.5683'),\n",
       "   ('proportion_reads_mapped', '0.5627'),\n",
       "   ('enneking_msts_stage', '0.5622'),\n",
       "   ('run_metadata_files', '0.5614'),\n",
       "   ('read_group_qcs', '0.5469'),\n",
       "   ('simple_germline_variations', '0.5444'),\n",
       "   ('enneking_msts_grade', '0.5427')]},\n",
       " {'Candidate column': 'Log2_variant_total',\n",
       "  'Top k columns': [('test_value_range', '0.5718'),\n",
       "   ('blood_test_normal_range_upper', '0.5317'),\n",
       "   ('dlco_ref_predictive_percent', '0.4962'),\n",
       "   ('no_matched_normal_wgs', '0.4886'),\n",
       "   ('loci_abnormal_count', '0.4721'),\n",
       "   ('measurement_type', '0.4633'),\n",
       "   ('course_number', '0.4603'),\n",
       "   ('blood_test_normal_range_lower', '0.4569'),\n",
       "   ('average_insert_size', '0.4551'),\n",
       "   ('no_matched_normal_wxs', '0.4489')]},\n",
       " {'Candidate column': 'Log2_SNP_total',\n",
       "  'Top k columns': [('karnofsky_performance_status', '0.4996'),\n",
       "   ('target_capture_kit_catalog_number', '0.4966'),\n",
       "   ('well_number', '0.4813'),\n",
       "   ('copy_number_liftover_workflows', '0.4778'),\n",
       "   ('no_matched_normal_targeted_sequencing', '0.4681'),\n",
       "   ('undescended_testis_corrected_method', '0.4645'),\n",
       "   ('per_base_sequence_quality', '0.4592'),\n",
       "   ('no_matched_normal_low_pass_wgs', '0.4519'),\n",
       "   ('dbgap_accession_number', '0.4515'),\n",
       "   ('target_capture_kit_name', '0.4474')]},\n",
       " {'Candidate column': 'Log2_INDEL_total',\n",
       "  'Top k columns': [('extranodal_extension', '0.51'),\n",
       "   ('recist_targeted_regions_sum', '0.4657'),\n",
       "   ('sequence_duplication_levels', '0.4609'),\n",
       "   ('method_of_sample_procurement', '0.4531'),\n",
       "   ('test_units', '0.4507'),\n",
       "   ('satellite_nodule_present', '0.4473'),\n",
       "   ('ecog_performance_status', '0.4465'),\n",
       "   ('recist_targeted_regions_number', '0.4438'),\n",
       "   ('well_number', '0.4438'),\n",
       "   ('tumor_depth_measurement', '0.4374')]},\n",
       " {'Candidate column': 'Genomics_subtype',\n",
       "  'Top k columns': [('submitted_genomic_profiles', '0.7362'),\n",
       "   ('submitted_genotyping_arrays', '0.5936'),\n",
       "   ('analyte_type', '0.507'),\n",
       "   ('analyte_type_id', '0.4817'),\n",
       "   ('gene_symbol', '0.468'),\n",
       "   ('ensat_pathologic_n', '0.4624'),\n",
       "   ('analytes', '0.4522'),\n",
       "   ('biospecimen_anatomic_site', '0.4489'),\n",
       "   ('tissue_microarray_coordinates', '0.4482'),\n",
       "   ('data_category', '0.4453')]},\n",
       " {'Candidate column': 'Mutation_signature_C>A',\n",
       "  'Top k columns': [('mutation_codon', '0.5782'),\n",
       "   ('somatic_mutation_calling_workflows', '0.5557'),\n",
       "   ('variant_type', '0.5336'),\n",
       "   ('submitted_tangent_copy_numbers', '0.5111'),\n",
       "   ('annotated_somatic_mutations', '0.5074'),\n",
       "   ('morphologic_architectural_pattern', '0.503'),\n",
       "   ('uicc_pathologic_m', '0.5001'),\n",
       "   ('structural_variant_calling_workflows', '0.4937'),\n",
       "   ('copy_number_variation_workflows', '0.4932'),\n",
       "   ('germline_mutation_calling_workflows', '0.485')]},\n",
       " {'Candidate column': 'Mutation_signature_C>G',\n",
       "  'Top k columns': [('annotated_somatic_mutations', '0.5813'),\n",
       "   ('simple_germline_variations', '0.5491'),\n",
       "   ('variant_type', '0.5381'),\n",
       "   ('fab_morphology_code', '0.5338'),\n",
       "   ('mutation_codon', '0.5272'),\n",
       "   ('submitted_tangent_copy_numbers', '0.5029'),\n",
       "   ('germline_mutation_calling_workflows', '0.4915'),\n",
       "   ('ajcc_pathologic_m', '0.4901'),\n",
       "   ('copy_number_variation_workflows', '0.4885'),\n",
       "   ('aggregated_somatic_mutations', '0.4781')]},\n",
       " {'Candidate column': 'Mutation_signature_C>T',\n",
       "  'Top k columns': [('somatic_mutation_calling_workflows', '0.5756'),\n",
       "   ('submitted_tangent_copy_numbers', '0.5752'),\n",
       "   ('mutation_codon', '0.5708'),\n",
       "   ('variant_type', '0.5567'),\n",
       "   ('annotated_somatic_mutations', '0.534'),\n",
       "   ('uicc_pathologic_m', '0.5217'),\n",
       "   ('morphologic_architectural_pattern', '0.4964'),\n",
       "   ('copy_number_variation_workflows', '0.4789'),\n",
       "   ('ajcc_pathologic_m', '0.4732'),\n",
       "   ('germline_mutation_calling_workflows', '0.4704')]},\n",
       " {'Candidate column': 'Mutation_signature_T>C',\n",
       "  'Top k columns': [('variant_type', '0.596'),\n",
       "   ('somatic_mutation_calling_workflows', '0.5839'),\n",
       "   ('submitted_tangent_copy_numbers', '0.5825'),\n",
       "   ('annotated_somatic_mutations', '0.5807'),\n",
       "   ('type', '0.5276'),\n",
       "   ('mutation_codon', '0.5275'),\n",
       "   ('submitter_id', '0.5115'),\n",
       "   ('data_subtypes', '0.5112'),\n",
       "   ('somatic_mutation_indexes', '0.5007'),\n",
       "   ('morphologic_architectural_pattern', '0.4969')]},\n",
       " {'Candidate column': 'Mutation_signature_T>A',\n",
       "  'Top k columns': [('somatic_mutation_calling_workflows', '0.5636'),\n",
       "   ('somatic_mutation_indexes', '0.5407'),\n",
       "   ('variant_type', '0.5393'),\n",
       "   ('annotated_somatic_mutations', '0.5313'),\n",
       "   ('submitter_id', '0.5245'),\n",
       "   ('data_formats', '0.5112'),\n",
       "   ('submitted_tangent_copy_numbers', '0.51'),\n",
       "   ('data_subtypes', '0.5008'),\n",
       "   ('type', '0.4873'),\n",
       "   ('data_format', '0.4818')]},\n",
       " {'Candidate column': 'Mutation_signature_T>G',\n",
       "  'Top k columns': [('annotated_somatic_mutations', '0.6099'),\n",
       "   ('variant_type', '0.5653'),\n",
       "   ('type', '0.5187'),\n",
       "   ('submitted_tangent_copy_numbers', '0.5174'),\n",
       "   ('somatic_mutation_indexes', '0.4944'),\n",
       "   ('somatic_mutation_calling_workflows', '0.4863'),\n",
       "   ('data_subtypes', '0.4853'),\n",
       "   ('aggregated_somatic_mutations', '0.4773'),\n",
       "   ('data_format', '0.4767'),\n",
       "   ('data_types', '0.473')]},\n",
       " {'Candidate column': 'WXS_normal_sample_type',\n",
       "  'Top k columns': [('selected_normal_wxs', '0.7374'),\n",
       "   ('no_matched_normal_wxs', '0.7281'),\n",
       "   ('no_matched_normal_wgs', '0.6316'),\n",
       "   ('selected_normal_wgs', '0.6273'),\n",
       "   ('selected_normal_targeted_sequencing', '0.6076'),\n",
       "   ('sample_type', '0.6003'),\n",
       "   ('no_matched_normal_targeted_sequencing', '0.5956'),\n",
       "   ('sample_type_id', '0.5837'),\n",
       "   ('no_matched_normal_low_pass_wgs', '0.5692'),\n",
       "   ('method_of_sample_procurement', '0.5595')]},\n",
       " {'Candidate column': 'WXS_normal_filename',\n",
       "  'Top k columns': [('aligned_reads_files', '0.6652'),\n",
       "   ('undescended_testis_corrected_method', '0.6566'),\n",
       "   ('dbgap_accession_number', '0.6451'),\n",
       "   ('submitted_unaligned_reads_files', '0.6405'),\n",
       "   ('submitted_aligned_reads_files', '0.6383'),\n",
       "   ('copy_number_auxiliary_files', '0.6337'),\n",
       "   ('selected_normal_wxs', '0.6299'),\n",
       "   ('file_name', '0.6246'),\n",
       "   ('library_strand', '0.6221'),\n",
       "   ('undescended_testis_corrected', '0.622')]},\n",
       " {'Candidate column': 'WXS_normal_UUID',\n",
       "  'Top k columns': [('no_matched_normal_wxs', '0.675'),\n",
       "   ('selected_normal_wxs', '0.6663'),\n",
       "   ('no_matched_normal_wgs', '0.567'),\n",
       "   ('no_matched_normal_low_pass_wgs', '0.5592'),\n",
       "   ('undescended_testis_corrected_method', '0.5519'),\n",
       "   ('no_matched_normal_targeted_sequencing', '0.5471'),\n",
       "   ('undescended_testis_corrected', '0.5456'),\n",
       "   ('undescended_testis_corrected_laterality', '0.5353'),\n",
       "   ('selected_normal_wgs', '0.5293'),\n",
       "   ('read_pair_number', '0.4777')]},\n",
       " {'Candidate column': 'WXS_tumor_sample_type',\n",
       "  'Top k columns': [('normal_tumor_genotype_snp_match', '0.615'),\n",
       "   ('no_matched_normal_targeted_sequencing', '0.5832'),\n",
       "   ('radiosensitizing_agent', '0.551'),\n",
       "   ('enneking_msts_tumor_site', '0.5469'),\n",
       "   ('selected_normal_targeted_sequencing', '0.5367'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.5317'),\n",
       "   ('residual_tumor_measurement', '0.5225'),\n",
       "   ('parent_samples', '0.5072'),\n",
       "   ('no_matched_normal_wxs', '0.507'),\n",
       "   ('cause_of_response', '0.5006')]},\n",
       " {'Candidate column': 'WXS_tumor_filename',\n",
       "  'Top k columns': [('submitted_unaligned_reads_files', '0.5951'),\n",
       "   ('normal_tumor_genotype_snp_match', '0.5828'),\n",
       "   ('file_name', '0.5804'),\n",
       "   ('aligned_reads_files', '0.5771'),\n",
       "   ('history_of_tumor_type', '0.5712'),\n",
       "   ('undescended_testis_corrected_method', '0.5711'),\n",
       "   ('submitted_aligned_reads_files', '0.5691'),\n",
       "   ('files', '0.5686'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.5672'),\n",
       "   ('experiment_metadata_files', '0.5657')]},\n",
       " {'Candidate column': 'WXS_tumor_UUID',\n",
       "  'Top k columns': [('no_matched_normal_targeted_sequencing', '0.5956'),\n",
       "   ('enneking_msts_tumor_site', '0.5915'),\n",
       "   ('undescended_testis_corrected_method', '0.5675'),\n",
       "   ('normal_tumor_genotype_snp_match', '0.5667'),\n",
       "   ('history_of_tumor_type', '0.5215'),\n",
       "   ('undescended_testis_corrected_laterality', '0.5128'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.5125'),\n",
       "   ('submitted_unaligned_reads_files', '0.509'),\n",
       "   ('undescended_testis_corrected', '0.4955'),\n",
       "   ('target_capture_kit_target_region', '0.4898')]},\n",
       " {'Candidate column': 'WGS_normal_sample_type',\n",
       "  'Top k columns': [('no_matched_normal_wgs', '0.7245'),\n",
       "   ('selected_normal_wgs', '0.7123'),\n",
       "   ('no_matched_normal_wxs', '0.647'),\n",
       "   ('selected_normal_wxs', '0.6303'),\n",
       "   ('blood_test_normal_range_upper', '0.6247'),\n",
       "   ('selected_normal_low_pass_wgs', '0.6132'),\n",
       "   ('blood_test_normal_range_lower', '0.5862'),\n",
       "   ('no_matched_normal_low_pass_wgs', '0.5612'),\n",
       "   ('sample_type', '0.5353'),\n",
       "   ('method_of_sample_procurement', '0.5248')]},\n",
       " {'Candidate column': 'WGS_normal_UUID',\n",
       "  'Top k columns': [('no_matched_normal_wgs', '0.7295'),\n",
       "   ('selected_normal_wgs', '0.6784'),\n",
       "   ('no_matched_normal_wxs', '0.646'),\n",
       "   ('no_matched_normal_low_pass_wgs', '0.6225'),\n",
       "   ('selected_normal_low_pass_wgs', '0.6164'),\n",
       "   ('selected_normal_wxs', '0.6101'),\n",
       "   ('masked_somatic_mutations', '0.4579'),\n",
       "   ('blood_test_normal_range_upper', '0.4475'),\n",
       "   ('undescended_testis_corrected_laterality', '0.4288'),\n",
       "   ('blood_test_normal_range_lower', '0.423')]},\n",
       " {'Candidate column': 'WGS_tumor_sample_type',\n",
       "  'Top k columns': [('residual_tumor_measurement', '0.606'),\n",
       "   ('normal_tumor_genotype_snp_match', '0.5703'),\n",
       "   ('classification_of_tumor', '0.533'),\n",
       "   ('samples', '0.5191'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.5062'),\n",
       "   ('percent_tumor_invasion', '0.4928'),\n",
       "   ('tissue_collection_type', '0.4912'),\n",
       "   ('best_overall_response', '0.491'),\n",
       "   ('enneking_msts_tumor_site', '0.4852'),\n",
       "   ('history_of_tumor_type', '0.4829')]},\n",
       " {'Candidate column': 'WGS_tumor_UUID',\n",
       "  'Top k columns': [('residual_tumor_measurement', '0.6047'),\n",
       "   ('enneking_msts_tumor_site', '0.5685'),\n",
       "   ('wilms_tumor_histologic_subtype', '0.5455'),\n",
       "   ('normal_tumor_genotype_snp_match', '0.5375'),\n",
       "   ('history_of_tumor_type', '0.5372'),\n",
       "   ('non_nodal_tumor_deposits', '0.5237'),\n",
       "   ('classification_of_tumor', '0.4996'),\n",
       "   ('residual_tumor', '0.4992'),\n",
       "   ('percent_tumor_invasion', '0.4961'),\n",
       "   ('greatest_tumor_dimension', '0.4774')]},\n",
       " {'Candidate column': 'RNAseq_R1_sample_type',\n",
       "  'Top k columns': [('sample_type', '0.6305'),\n",
       "   ('sample_type_id', '0.6265'),\n",
       "   ('test_analyte_type', '0.5477'),\n",
       "   ('samples', '0.5352'),\n",
       "   ('ribosomal_rna_28s_16s_ratio', '0.5301'),\n",
       "   ('method_of_sample_procurement', '0.526'),\n",
       "   ('sample_ordinal', '0.5126'),\n",
       "   ('analyte_type_id', '0.501'),\n",
       "   ('aligned_reads_indexes', '0.4951'),\n",
       "   ('test_value', '0.4857')]},\n",
       " {'Candidate column': 'RNAseq_R1_filename',\n",
       "  'Top k columns': [('library_strand', '0.667'),\n",
       "   ('aligned_reads_files', '0.6331'),\n",
       "   ('library_selection', '0.6231'),\n",
       "   ('read_length', '0.6147'),\n",
       "   ('aligned_reads_indexes', '0.6118'),\n",
       "   ('submitted_tangent_copy_numbers', '0.6095'),\n",
       "   ('library_strategy', '0.5898'),\n",
       "   ('submitted_aligned_reads_files', '0.5896'),\n",
       "   ('experiment_metadata_files', '0.5895'),\n",
       "   ('read_pair_number', '0.5895')]},\n",
       " {'Candidate column': 'RNAseq_R1_UUID',\n",
       "  'Top k columns': [('library_strand', '0.6123'),\n",
       "   ('read_pair_number', '0.5785'),\n",
       "   ('aligned_reads_files', '0.5749'),\n",
       "   ('library_selection', '0.5632'),\n",
       "   ('aligned_reads_indexes', '0.5594'),\n",
       "   ('library_strategy', '0.5496'),\n",
       "   ('proportion_reads_duplicated', '0.5495'),\n",
       "   ('sample_type_id', '0.5435'),\n",
       "   ('experiment_metadata_files', '0.5404'),\n",
       "   ('read_length', '0.5402')]},\n",
       " {'Candidate column': 'RNAseq_R2_sample_type',\n",
       "  'Top k columns': [('sample_type', '0.6121'),\n",
       "   ('sample_type_id', '0.6056'),\n",
       "   ('test_analyte_type', '0.5534'),\n",
       "   ('ribosomal_rna_28s_16s_ratio', '0.5451'),\n",
       "   ('method_of_sample_procurement', '0.5238'),\n",
       "   ('sample_ordinal', '0.5236'),\n",
       "   ('samples', '0.5104'),\n",
       "   ('analyte_type_id', '0.4963'),\n",
       "   ('aligned_reads_indexes', '0.4932'),\n",
       "   ('sequence_length_distribution', '0.4884')]},\n",
       " {'Candidate column': 'RNAseq_R2_filename',\n",
       "  'Top k columns': [('library_strand', '0.6794'),\n",
       "   ('aligned_reads_files', '0.6275'),\n",
       "   ('read_length', '0.6228'),\n",
       "   ('library_selection', '0.6171'),\n",
       "   ('submitted_tangent_copy_numbers', '0.6149'),\n",
       "   ('aligned_reads_indexes', '0.6099'),\n",
       "   ('library_strategy', '0.6035'),\n",
       "   ('encoding', '0.6022'),\n",
       "   ('library_name', '0.5955'),\n",
       "   ('sequence_length_distribution', '0.5946')]},\n",
       " {'Candidate column': 'RNAseq_R2_UUID',\n",
       "  'Top k columns': [('library_strand', '0.6343'),\n",
       "   ('read_pair_number', '0.5743'),\n",
       "   ('aligned_reads_files', '0.5734'),\n",
       "   ('library_strategy', '0.5721'),\n",
       "   ('proportion_reads_duplicated', '0.5721'),\n",
       "   ('library_selection', '0.5608'),\n",
       "   ('encoding', '0.56'),\n",
       "   ('aligned_reads_indexes', '0.5576'),\n",
       "   ('library_name', '0.5495'),\n",
       "   ('read_length', '0.5493')]},\n",
       " {'Candidate column': 'miRNAseq_sample_type',\n",
       "  'Top k columns': [('sample_type', '0.6058'),\n",
       "   ('sample_type_id', '0.5946'),\n",
       "   ('samples', '0.5392'),\n",
       "   ('sample_ordinal', '0.5278'),\n",
       "   ('sequence_length_distribution', '0.502'),\n",
       "   ('test_analyte_type', '0.4845'),\n",
       "   ('run_metadata_files', '0.4815'),\n",
       "   ('method_of_sample_procurement', '0.4759'),\n",
       "   ('ribosomal_rna_28s_16s_ratio', '0.4739'),\n",
       "   ('proportion_reads_mapped', '0.4698')]},\n",
       " {'Candidate column': 'miRNAseq_UUID',\n",
       "  'Top k columns': [('sequence_length_distribution', '0.5341'),\n",
       "   ('experiment_metadata_files', '0.5317'),\n",
       "   ('data_format', '0.5289'),\n",
       "   ('library_strand', '0.5135'),\n",
       "   ('run_metadata_files', '0.5134'),\n",
       "   ('encoding', '0.5063'),\n",
       "   ('data_subtypes', '0.5052'),\n",
       "   ('proportion_reads_mapped', '0.4992'),\n",
       "   ('data_types', '0.4944'),\n",
       "   ('library_selection', '0.4925')]},\n",
       " {'Candidate column': 'Methylation_available',\n",
       "  'Top k columns': [('masked_methylation_arrays', '0.6349'),\n",
       "   ('raw_methylation_arrays', '0.6135'),\n",
       "   ('methylation_array_harmonization_workflows', '0.5605'),\n",
       "   ('methylation_liftover_workflows', '0.5596'),\n",
       "   ('enneking_msts_metastasis', '0.5352'),\n",
       "   ('residual_tumor_measurement', '0.5205'),\n",
       "   ('analyte_volume', '0.4802'),\n",
       "   ('enneking_msts_stage', '0.4733'),\n",
       "   ('specialized_molecular_test', '0.4671'),\n",
       "   ('somatic_mutation_indexes', '0.463')]},\n",
       " {'Candidate column': 'Methylation_quality',\n",
       "  'Top k columns': [('masked_methylation_arrays', '0.537'),\n",
       "   ('specialized_molecular_test', '0.5259'),\n",
       "   ('enneking_msts_metastasis', '0.5132'),\n",
       "   ('residual_tumor_measurement', '0.5103'),\n",
       "   ('analyte_volume', '0.4955'),\n",
       "   ('enneking_msts_stage', '0.4932'),\n",
       "   ('raw_methylation_arrays', '0.4911'),\n",
       "   ('enneking_msts_grade', '0.4865'),\n",
       "   ('ensat_clinical_m', '0.4547'),\n",
       "   ('enneking_msts_tumor_site', '0.4469')]}]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "manager.reduce_scope()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea6487b7-33cd-4d23-8be2-75e012086695",
   "metadata": {},
   "source": [
    "Perform column mapping."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ecbdaa25-5731-45fe-bc3a-cd3b3a7b6080",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'idx': 'id',\n",
       " 'Proteomics_Participant_ID': 'histologic_progression_type',\n",
       " 'Case_excluded': 'procedures_performed',\n",
       " 'Proteomics_TMT_batch': 'percent_tumor_cells',\n",
       " 'Proteomics_TMT_plex': 'percent_tumor_cells',\n",
       " 'Proteomics_TMT_channel': 'tissue_microarray_coordinates',\n",
       " 'Proteomics_Parent_Sample_IDs': 'protein_expressions',\n",
       " 'Proteomics_Aliquot_ID': 'prostatic_chips_positive_count',\n",
       " 'Proteomics_Tumor_Normal': 'residual_tumor_measurement',\n",
       " 'Proteomics_OCT': 'spectrophotometer_method',\n",
       " 'Country': 'country_of_birth',\n",
       " 'Histologic_Grade_FIGO': 'histologic_progression_type',\n",
       " 'Myometrial_invasion_Specify': 'perineural_invasion_present',\n",
       " 'Histologic_type': 'histologic_progression_type',\n",
       " 'Treatment_naive': 'treatment_type',\n",
       " 'Tumor_purity': 'tumor_purity',\n",
       " 'Path_Stage_Primary_Tumor-pT': 'history_of_tumor',\n",
       " 'Path_Stage_Reg_Lymph_Nodes-pN': 'lymph_nodes_tested',\n",
       " 'Clin_Stage_Dist_Mets-cM': 'msi_status',\n",
       " 'Path_Stage_Dist_Mets-pM': 'somatic_mutation_calling_workflows',\n",
       " 'tumor_Stage-Pathological': 'tumor_grade_category',\n",
       " 'FIGO_stage': 'irs_stage',\n",
       " 'LVSI': 'includes_spike_ins',\n",
       " 'BMI': 'weight',\n",
       " 'Age': 'age_is_obfuscated',\n",
       " 'Diabetes': 'disease_type',\n",
       " 'Race': 'race',\n",
       " 'Ethnicity': 'ethnicity',\n",
       " 'Gender': 'gender',\n",
       " 'Tumor_Site': 'tumor_descriptor',\n",
       " 'Tumor_Site_Other': 'tumor_descriptor',\n",
       " 'Tumor_Focality': 'tumor_focality',\n",
       " 'Tumor_Size_cm': 'tumor_shape',\n",
       " 'Estrogen_Receptor': 'protein_expressions',\n",
       " 'Estrogen_Receptor_%': 'hormonal_contraceptive_use',\n",
       " 'Progesterone_Receptor': 'hormone_replacement_therapy_type',\n",
       " 'Progesterone_Receptor_%': 'hormonal_contraceptive_use',\n",
       " 'MLH1': 'uicc_pathologic_m',\n",
       " 'MLH2': 'uicc_pathologic_m',\n",
       " 'MSH6': 'ensat_clinical_m',\n",
       " 'PMS2': 'pmid',\n",
       " 'p53': 'protein_expressions',\n",
       " 'Other_IHC_specify': 'hiv_viral_load',\n",
       " 'MLH1_Promoter_Hypermethylation': 'methylation_beta_values',\n",
       " 'Num_full_term_pregnancies': 'pregnant_at_diagnosis',\n",
       " 'EPIC_Bcells': 'percent_normal_cells',\n",
       " 'EPIC_CAFs': 'is_ffpe',\n",
       " 'EPIC_CD4_Tcells': 'number_proliferating_cells',\n",
       " 'EPIC_CD8_Tcells': 'number_proliferating_cells',\n",
       " 'EPIC_Endothelial': 'tumor_depth',\n",
       " 'EPIC_Macrophages': 'size_extraocular_nodule',\n",
       " 'EPIC_NKcells': 'percent_normal_cells',\n",
       " 'EPIC_otherCells': 'number_proliferating_cells',\n",
       " 'CIBERSORT_B _cells _naive': 'base_caller_name',\n",
       " 'CIBERSORT_B _cells _memory': 'single_cell_library',\n",
       " 'CIBERSORT_Plasma _cells': 'percent_normal_cells',\n",
       " 'CIBERSORT_T _cells _CD8': 'base_caller_name',\n",
       " 'CIBERSORT_T _cells _CD4 _naive': 'single_cell_library',\n",
       " 'CIBERSORT_T _cells _CD4 _memory _resting': 'base_caller_version',\n",
       " 'CIBERSORT_T _cells _CD4 _memory _activated': 'single_cell_library',\n",
       " 'CIBERSORT_T _cells _follicular _helper': 'percent_follicular_component',\n",
       " 'CIBERSORT_T _cells _regulatory _(Tregs)': 'recist_targeted_regions_sum',\n",
       " 'CIBERSORT_T _cells _gamma _delta': 'tumor_length_measurement',\n",
       " 'CIBERSORT_NK _cells _resting': 'number_expect_cells',\n",
       " 'CIBERSORT_NK _cells _activated': 'percent_normal_cells',\n",
       " 'CIBERSORT_Monocytes': 'percent_monocyte_infiltration',\n",
       " 'CIBERSORT_Macrophages _M0': 'uicc_pathologic_m',\n",
       " 'CIBERSORT_Macrophages _M1': 'uicc_pathologic_m',\n",
       " 'CIBERSORT_Macrophages _M2': 'uicc_pathologic_m',\n",
       " 'CIBERSORT_Dendritic _cells _resting': 'tumor_length_measurement',\n",
       " 'CIBERSORT_Dendritic _cells _activated': 'percent_tumor_cells',\n",
       " 'CIBERSORT_Mast _cells _resting': 'proportion_base_mismatch',\n",
       " 'CIBERSORT_Mast _cells _activated': 'proportion_base_mismatch',\n",
       " 'CIBERSORT_Eosinophils': 'tumor_infiltrating_macrophages',\n",
       " 'CIBERSORT_Neutrophils': 'tumor_infiltrating_macrophages',\n",
       " 'CIBERSORT_Absolute _score': 'msi_score',\n",
       " 'ESTIMATE_StromalScore': 'percent_stromal_cells',\n",
       " 'ESTIMATE_ImmuneScore': 'msi_score',\n",
       " 'ESTIMATE_ESTIMATEScore': 'undescended_testis_history',\n",
       " 'Stemness_score': 'msi_score',\n",
       " 'ER_ESR1': 'rin',\n",
       " 'PR_PGR': 'in_review',\n",
       " 'Pathway_activity_EGFR': 'route_of_administration',\n",
       " 'Pathway_activity_Hypoxia': 'treatment_intent_type',\n",
       " 'Pathway_activity_JAK.STAT': 'ajcc_pathologic_t',\n",
       " 'Pathway_activity_MAPK': 'ensat_pathologic_t',\n",
       " 'Pathway_activity_NFkB': 'scan_tracer_used',\n",
       " 'Pathway_activity_PI3K': 'sites_of_involvement_count',\n",
       " 'Pathway_activity_TGFb': 'route_of_administration',\n",
       " 'Pathway_activity_TNFa': 'route_of_administration',\n",
       " 'Pathway_activity_Trail': 'undescended_testis_history_laterality',\n",
       " 'Pathway_activity_VEGF': 'vascular_invasion_type',\n",
       " 'Pathway_activity_p53': 'prostatic_chips_positive_count',\n",
       " 'TP53_ATM': 'prostatic_chips_total_count',\n",
       " 'TP53_CHEK2': 'lane_number',\n",
       " 'TP53_MDM4': 'ensat_clinical_m',\n",
       " 'TP53_RPS6KA3': 'includes_spike_ins',\n",
       " 'TP53_TP53': 'is_ffpe',\n",
       " 'TP53_pathway': 'ensat_pathologic_t',\n",
       " 'PI3K_AKT1': 'is_ffpe',\n",
       " 'PI3K_AKT2': 'is_ffpe',\n",
       " 'PI3K_AKT3': 'is_ffpe',\n",
       " 'PI3K_DEPDC5': 'is_ffpe',\n",
       " 'PI3K_DEPTOR': 'is_ffpe',\n",
       " 'PI3K_INPP4B': 'prostatic_chips_positive_count',\n",
       " 'PI3K_MAPKAP1': 'pmid',\n",
       " 'PI3K_MLST8': 'fab_morphology_code',\n",
       " 'PI3K_MTOR': 'pmid',\n",
       " 'PI3K_NPRL2': 'is_ffpe',\n",
       " 'PI3K_NPRL3': 'primary_disease',\n",
       " 'PI3K_PDK1': 'hpv_positive_type',\n",
       " 'PI3K_PIK3CA': 'pairs_on_diff_chr',\n",
       " 'PI3K_PIK3CB': 'cd4_count',\n",
       " 'PI3K_PIK3R1': 'rin',\n",
       " 'PI3K_PIK3R2': 'primary_disease',\n",
       " 'PI3K_PPP2R1A': 'rin',\n",
       " 'PI3K_PTEN': 'is_ffpe',\n",
       " 'PI3K_RHEB': 'rin',\n",
       " 'PI3K_RICTOR': 'percent_tumor_invasion',\n",
       " 'PI3K_RPS6': 'rin',\n",
       " 'PI3K_RPS6KB1': 'kmer_content',\n",
       " 'PI3K_RPTOR': 'rin',\n",
       " 'PI3K_STK11': 'icd_10',\n",
       " 'PI3K_TSC1': 'is_ffpe',\n",
       " 'PI3K_TSC2': 'is_ffpe',\n",
       " 'PI3K_pathway': 'ensat_pathologic_t',\n",
       " 'HRD_BRCA1': 'ribosomal_rna_28s_16s_ratio',\n",
       " 'HRD_BRCA2': 'ribosomal_rna_28s_16s_ratio',\n",
       " 'HRD_BRCA1_or_BRCA2': 'icd_10_code',\n",
       " 'CNV_DEL': 'dlco_ref_predictive_percent',\n",
       " 'CNV_AMP': 'library_name',\n",
       " 'CNV_class': 'base_caller_name',\n",
       " 'CNV_idx': 'tumor_code_id',\n",
       " 'CNV_1q_DEL': 'dlco_ref_predictive_percent',\n",
       " 'CNV_3q_DEL': 'fab_morphology_code',\n",
       " 'CNV_4q_DEL': 'dlco_ref_predictive_percent',\n",
       " 'CNV_1q_AMP': 'chipseq_target',\n",
       " 'CNV_3q_AMP': 'read_group_qcs',\n",
       " 'CNV_4q_AMP': 'chipseq_target',\n",
       " 'Purity_Immune': 'biospecimen_volume',\n",
       " 'Purity_Cancer': 'percent_tumor_nuclei',\n",
       " 'Purity_Stroma': 'lesions_treated_number',\n",
       " 'MSI_status': 'msi_status',\n",
       " 'POLE_subtype': 'zygosity',\n",
       " 'JAK1_MS_INDEL': 'enneking_msts_grade',\n",
       " 'JAK1_Mutation': 'chipseq_antibody',\n",
       " 'Log2_variant_per_Mbp': 'variant_type',\n",
       " 'Log2_SNP_per_Mbp': 'target_capture_kit_version',\n",
       " 'Log2_INDEL_per_Mbp': 'proportion_reads_mapped',\n",
       " 'Log2_variant_total': 'blood_test_normal_range_lower',\n",
       " 'Log2_SNP_total': 'karnofsky_performance_status',\n",
       " 'Log2_INDEL_total': 'recist_targeted_regions_sum',\n",
       " 'Genomics_subtype': 'analyte_type',\n",
       " 'Mutation_signature_C>A': 'mutation_codon',\n",
       " 'Mutation_signature_C>G': 'mutation_codon',\n",
       " 'Mutation_signature_C>T': 'mutation_codon',\n",
       " 'Mutation_signature_T>C': 'mutation_codon',\n",
       " 'Mutation_signature_T>A': 'submitted_tangent_copy_numbers',\n",
       " 'Mutation_signature_T>G': 'submitted_tangent_copy_numbers',\n",
       " 'WXS_normal_sample_type': 'sample_type',\n",
       " 'WXS_normal_filename': 'file_name',\n",
       " 'WXS_normal_UUID': 'selected_normal_wgs',\n",
       " 'WXS_tumor_sample_type': 'wilms_tumor_histologic_subtype',\n",
       " 'WXS_tumor_filename': 'file_name',\n",
       " 'WXS_tumor_UUID': 'history_of_tumor_type',\n",
       " 'WGS_normal_sample_type': 'sample_type',\n",
       " 'WGS_normal_UUID': 'selected_normal_wgs',\n",
       " 'WGS_tumor_sample_type': 'wilms_tumor_histologic_subtype',\n",
       " 'WGS_tumor_UUID': 'percent_tumor_invasion',\n",
       " 'RNAseq_R1_sample_type': 'sample_type',\n",
       " 'RNAseq_R1_filename': 'read_length',\n",
       " 'RNAseq_R1_UUID': 'aligned_reads_files',\n",
       " 'RNAseq_R2_sample_type': 'sample_type',\n",
       " 'RNAseq_R2_filename': 'read_length',\n",
       " 'RNAseq_R2_UUID': 'aligned_reads_files',\n",
       " 'miRNAseq_sample_type': 'sample_type',\n",
       " 'miRNAseq_UUID': 'proportion_reads_mapped',\n",
       " 'Methylation_available': 'raw_methylation_arrays',\n",
       " 'Methylation_quality': 'raw_methylation_arrays'}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "column_mappings = manager.map_columns()\n",
    "column_mappings"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "81d7a0fd-3077-43e3-9a91-69e6a280075a",
   "metadata": {},
   "source": [
    "## Value Mapping"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a4607e0-4f6a-4b9a-82b3-6f5ff6d184d5",
   "metadata": {},
   "source": [
    "Perform value mapping."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ed0ff201-3984-4b3b-8a0f-eacd91c56bac",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Column tumor_Stage-Pathological:\n",
      "| Current Value   | Target Value   | Similarity         |\n",
      "|-----------------+----------------+--------------------|\n",
      "| Stage I         | Three Tier     | 1.0                |\n",
      "| Stage IV        | Three Tier     | 0.9224806201550388 |\n",
      "| Stage II        | Three Tier     | 0.9224806201550388 |\n",
      "| Stage III       | Three Tier     | 0.8531211750305997 |\n",
      "| nan             | -              | -                  | \n",
      "\n",
      "Column Race:\n",
      "| Current Value             | Target Value              | Similarity   |\n",
      "|---------------------------+---------------------------+--------------|\n",
      "| White                     | white                     | 1.0          |\n",
      "| Asian                     | asian                     | 1.0          |\n",
      "| Not Reported              | not reported              | 1.0          |\n",
      "| Black or African American | black or african american | 1.0          |\n",
      "| nan                       | -                         | -            | \n",
      "\n",
      "Column Tumor_Focality:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Unifocal        | Unifocal       | 1.0          |\n",
      "| Multifocal      | Multifocal     | 1.0          |\n",
      "| nan             | -              | -            | \n",
      "\n",
      "Column Country:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| United States   | United States  | 1.0          |\n",
      "| Ukraine         | Ukraine        | 1.0          |\n",
      "| Poland          | Poland         | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| Other_specify   | -              | -            | \n",
      "\n",
      "Column Case_excluded:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| No              | Unknown        | 1.0          |\n",
      "| Yes             | -              | -            | \n",
      "\n",
      "Column Clin_Stage_Dist_Mets-cM:\n",
      "| Current Value      | Target Value   | Similarity   |\n",
      "|--------------------+----------------+--------------|\n",
      "| cM0                | MSI            | 1.0          |\n",
      "| cM1                | MSI            | 1.0          |\n",
      "| nan                | -              | -            |\n",
      "| Staging Incomplete | -              | -            | \n",
      "\n",
      "Column Ethnicity:\n",
      "| Current Value          | Target Value           | Similarity         |\n",
      "|------------------------+------------------------+--------------------|\n",
      "| Hispanic or Latino     | hispanic or latino     | 1.0                |\n",
      "| Not-Hispanic or Latino | not hispanic or latino | 0.9363636363636364 |\n",
      "| nan                    | -                      | -                  |\n",
      "| Not reported           | -                      | -                  | \n",
      "\n",
      "Column Gender:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Female          | female         | 1.0          |\n",
      "| nan             | -              | -            | \n",
      "\n",
      "Column WXS_normal_sample_type:\n",
      "| Current Value   | Target Value         | Similarity   |\n",
      "|-----------------+----------------------+--------------|\n",
      "| Blood_normal    | Blood Derived Normal | 1.0          |\n",
      "| nan             | -                    | -            | \n",
      "\n",
      "Column WXS_tumor_sample_type:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| nan             | Unknown        | 1.0          |\n",
      "| Tumor           | -              | -            | \n",
      "\n",
      "Column WGS_normal_sample_type:\n",
      "| Current Value   | Target Value         | Similarity   |\n",
      "|-----------------+----------------------+--------------|\n",
      "| Blood_normal    | Blood Derived Normal | 1.0          |\n",
      "| nan             | -                    | -            | \n",
      "\n",
      "Column WGS_tumor_sample_type:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| nan             | Unknown        | 1.0          |\n",
      "| Tumor           | -              | -            | \n",
      "\n",
      "Column Treatment_naive:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| NO              | Unknown        | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| YES             | -              | -            | \n",
      "\n",
      "Column FIGO_stage:\n",
      "| Current Value   | Target Value   | Similarity         |\n",
      "|-----------------+----------------+--------------------|\n",
      "| nan             | Unknown        | 1.0                |\n",
      "| IIIC2           | 2              | 0.8333333333333334 |\n",
      "| IIIC1           | 1              | 0.8333333333333334 |\n",
      "| IIIA            | -              | -                  |\n",
      "| IB              | -              | -                  |\n",
      "| IVB             | -              | -                  |\n",
      "| IA              | -              | -                  |\n",
      "| II              | -              | -                  |\n",
      "| IIIB            | -              | -                  | \n",
      "\n",
      "Column MSI_status:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| MSS             | MSS            | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| MSI-H           | -              | -            | \n",
      "\n",
      "Column POLE_subtype:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| No              | Unknown        | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| Yes             | -              | -            | \n",
      "\n",
      "Column JAK1_MS_INDEL:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| nan             | Unknown        | 1.0          |\n",
      "| WT              | -              | -            |\n",
      "| MS_indel        | -              | -            | \n",
      "\n",
      "Column RNAseq_R1_sample_type:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Tumor           | Tumor          | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| Adjacent_normal | -              | -            | \n",
      "\n",
      "Column RNAseq_R2_sample_type:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Tumor           | Tumor          | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| Adjacent_normal | -              | -            | \n",
      "\n",
      "Column miRNAseq_sample_type:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Tumor           | Tumor          | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| Adjacent_normal | -              | -            | \n",
      "\n",
      "Column Proteomics_Tumor_Normal:\n",
      "| Current Value     | Target Value           | Similarity   |\n",
      "|-------------------+------------------------+--------------|\n",
      "| Enriched_normal   | No macroscopic disease | 1.0          |\n",
      "| Tumor             | -                      | -            |\n",
      "| Adjacent_normal   | -                      | -            |\n",
      "| Myometrium_normal | -                      | -            | \n",
      "\n",
      "Column Histologic_Grade_FIGO:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| nan             | Unknown        | 1.0          |\n",
      "| FIGO grade 2    | -              | -            |\n",
      "| FIGO grade 1    | -              | -            |\n",
      "| FIGO grade 3    | -              | -            | \n",
      "\n",
      "Column Myometrial_invasion_Specify:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Not identified  | Not Reported   | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| 50 % or more    | -              | -            |\n",
      "| under 50 %      | -              | -            | \n",
      "\n",
      "Column Diabetes:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Unknown         | Unknown        | 1.0          |\n",
      "| nan             | -              | -            |\n",
      "| No              | -              | -            |\n",
      "| Yes             | -              | -            | \n",
      "\n",
      "Column Tumor_Site:\n",
      "| Current Value         | Target Value   | Similarity   |\n",
      "|-----------------------+----------------+--------------|\n",
      "| Anterior endometrium  | Not Reported   | 1.0          |\n",
      "| nan                   | -              | -            |\n",
      "| Other, specify        | -              | -            |\n",
      "| Posterior endometrium | -              | -            | \n",
      "\n",
      "Column Histologic_type:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Endometrioid    | Not Reported   | 1.0          |\n",
      "| Carcinosarcoma  | -              | -            |\n",
      "| nan             | -              | -            |\n",
      "| Clear cell      | -              | -            |\n",
      "| Serous          | -              | -            | \n",
      "\n",
      "Column Progesterone_Receptor:\n",
      "| Current Value        | Target Value   | Similarity   |\n",
      "|----------------------+----------------+--------------|\n",
      "| Unknown              | Unknown        | 1.0          |\n",
      "| Cannot be determined | -              | -            |\n",
      "| nan                  | -              | -            |\n",
      "| Negative             | -              | -            |\n",
      "| Positive             | -              | -            | \n",
      "\n",
      "Column MLH1:\n",
      "| Current Value              | Target Value   | Similarity   |\n",
      "|----------------------------+----------------+--------------|\n",
      "| Unknown                    | Unknown        | 1.0          |\n",
      "| nan                        | -              | -            |\n",
      "| Cannot be determined       | -              | -            |\n",
      "| Loss of nuclear expression | -              | -            |\n",
      "| Intact nuclear expression  | -              | -            | \n",
      "\n",
      "Column MLH2:\n",
      "| Current Value              | Target Value   | Similarity   |\n",
      "|----------------------------+----------------+--------------|\n",
      "| Unknown                    | Unknown        | 1.0          |\n",
      "| nan                        | -              | -            |\n",
      "| Cannot be determined       | -              | -            |\n",
      "| Loss of nuclear expression | -              | -            |\n",
      "| Intact nuclear expression  | -              | -            | \n",
      "\n",
      "Column MSH6:\n",
      "| Current Value              | Target Value   | Similarity   |\n",
      "|----------------------------+----------------+--------------|\n",
      "| Cannot be determined       | M0             | 1.0          |\n",
      "| nan                        | -              | -            |\n",
      "| Unknown                    | -              | -            |\n",
      "| Loss of nuclear expression | -              | -            |\n",
      "| Intact nuclear expression  | -              | -            | \n",
      "\n",
      "Column Genomics_subtype:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| nan             | DNA            | 1.0          |\n",
      "| POLE            | -              | -            |\n",
      "| CNV_high        | -              | -            |\n",
      "| MSI-H           | -              | -            |\n",
      "| CNV_low         | -              | -            | \n",
      "\n",
      "Column Num_full_term_pregnancies:\n",
      "| Current Value   | Target Value   | Similarity   |\n",
      "|-----------------+----------------+--------------|\n",
      "| Unknown         | Unknown        | 1.0          |\n",
      "| 4 or more       | -              | -            |\n",
      "| 1               | -              | -            |\n",
      "| nan             | -              | -            |\n",
      "| 2               | -              | -            |\n",
      "| 3               | -              | -            | \n",
      "\n",
      "Column Path_Stage_Primary_Tumor-pT:\n",
      "| Current Value    | Target Value   | Similarity   |\n",
      "|------------------+----------------+--------------|\n",
      "| nan              | No             | 1.0          |\n",
      "| pT3a (FIGO IIIA) | -              | -            |\n",
      "| pT2 (FIGO II)    | -              | -            |\n",
      "| pT1 (FIGO I)     | -              | -            |\n",
      "| pT3b (FIGO IIIB) | -              | -            |\n",
      "| pT1b (FIGO IB)   | -              | -            |\n",
      "| pT1a (FIGO IA)   | -              | -            | \n",
      "\n",
      "Column JAK1_Mutation:\n",
      "| Current Value                     | Target Value   | Similarity   |\n",
      "|-----------------------------------+----------------+--------------|\n",
      "| nan                               | Unknown        | 1.0          |\n",
      "| Frame_Shift_Del_Nonsense_Mutation | -              | -            |\n",
      "| Frame_Shift_Del                   | -              | -            |\n",
      "| Missense_Mutation                 | -              | -            |\n",
      "| WT                                | -              | -            |\n",
      "| Frame_Shift_Ins_Nonsense_Mutation | -              | -            |\n",
      "| Nonsense_Mutation                 | -              | -            |\n",
      "| Frame_Shift_Del_Frame_Shift_Ins   | -              | -            | \n",
      "\n"
     ]
    }
   ],
   "source": [
    "manager.map_values()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
