{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data Harmonization Between Two Datasets with `bdikit`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Data harmonization is the process of integrating and aligning data from different sources into a consistent format to ensure compatibility and interoperability across data analyses and systems. `bdikit` is a library the helps with key data harmonization steps:\n",
    "- *Schema Mapping*: In this step, data from various sources are mapped to a unified schema or model. This involves identifying equivalent table columns and establishing relationships between disparate datasets.\n",
    "- *Value Mapping (Data Standardization)*: This step involves converting data into a common format or structure, using consistent naming conventions, units, and coding systems to ensure uniformity.\n",
    "\n",
    "In this example, we describe how `bdikit` can be used to harmonize datasets from two papers:\n",
    "- Dou et al. 2020, Proteogenomic Characterization of Endometrial Carcinoma. https://pubmed.ncbi.nlm.nih.gov/32059776/ (CPTAC Discovery)\n",
    "- Dou et al. 2023, Proteogenomic insights suggest druggable pathways in endometrial carcinoma. https://pubmed.ncbi.nlm.nih.gov/37567170/ (CPTAC Confirmatory)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Loading the data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, import the `bdikit` library and other libraries."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<script type=\"esms-options\">{\"shimMode\": true}</script><style>*[data-root-id],\n",
       "*[data-root-id] > * {\n",
       "  box-sizing: border-box;\n",
       "  font-family: var(--jp-ui-font-family);\n",
       "  font-size: var(--jp-ui-font-size1);\n",
       "  color: var(--vscode-editor-foreground, var(--jp-ui-font-color1));\n",
       "}\n",
       "\n",
       "/* Override VSCode background color */\n",
       ".cell-output-ipywidget-background:has(\n",
       "    > .cell-output-ipywidget-background > .lm-Widget > *[data-root-id]\n",
       "  ),\n",
       ".cell-output-ipywidget-background:has(> .lm-Widget > *[data-root-id]) {\n",
       "  background-color: transparent !important;\n",
       "}\n",
       "</style>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": "(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n  const py_version = '3.6.2'.replace('rc', '-rc.').replace('.dev', '-dev.');\n  const reloading = false;\n  const Bokeh = root.Bokeh;\n\n  // Set a timeout for this load but only if we are not already initializing\n  if (typeof (root._bokeh_timeout) === \"undefined\" || (force || !root._bokeh_is_initializing)) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks;\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, js_modules, js_exports, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n    if (js_modules == null) js_modules = [];\n    if (js_exports == null) js_exports = {};\n\n    root._bokeh_onload_callbacks.push(callback);\n\n    if (root._bokeh_is_loading > 0) {\n      // Don't load bokeh if it is still initializing\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    } else if (js_urls.length === 0 && js_modules.length === 0 && Object.keys(js_exports).length === 0) {\n      // There is nothing to load\n      run_callbacks();\n      return null;\n    }\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n    window._bokeh_on_load = on_load\n\n    function on_error(e) {\n      const src_el = e.srcElement\n      console.error(\"failed to load \" + (src_el.href || src_el.src));\n    }\n\n    const skip = [];\n    if (window.requirejs) {\n      window.requirejs.config({'packages': {}, 'paths': {'tabulator': 'https://cdn.jsdelivr.net/npm/tabulator-tables@6.3.0/dist/js/tabulator.min', 'moment': 'https://cdn.jsdelivr.net/npm/luxon/build/global/luxon.min'}, 'shim': {}});\n      require([\"tabulator\"], function(Tabulator) {\n        window.Tabulator = Tabulator\n        on_load()\n      })\n      require([\"moment\"], function(moment) {\n        window.moment = moment\n        on_load()\n      })\n      root._bokeh_is_loading = css_urls.length + 2;\n    } else {\n      root._bokeh_is_loading = css_urls.length + js_urls.length + js_modules.length + Object.keys(js_exports).length;\n    }\n\n    const existing_stylesheets = []\n    const links = document.getElementsByTagName('link')\n    for (let i = 0; i < links.length; i++) {\n      const link = links[i]\n      if (link.href != null) {\n        existing_stylesheets.push(link.href)\n      }\n    }\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const escaped = encodeURI(url)\n      if (existing_stylesheets.indexOf(escaped) !== -1) {\n        on_load()\n        continue;\n      }\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }    if (((window.Tabulator !== undefined) && (!(window.Tabulator instanceof HTMLElement))) || window.requirejs) {\n      var urls = ['https://cdn.holoviz.org/panel/1.5.5/dist/bundled/datatabulator/tabulator-tables@6.3.0/dist/js/tabulator.min.js'];\n      for (var i = 0; i < urls.length; i++) {\n        skip.push(encodeURI(urls[i]))\n      }\n    }    if (((window.moment !== undefined) && (!(window.moment instanceof HTMLElement))) || window.requirejs) {\n      var urls = ['https://cdn.holoviz.org/panel/1.5.5/dist/bundled/datatabulator/luxon/build/global/luxon.min.js'];\n      for (var i = 0; i < urls.length; i++) {\n        skip.push(encodeURI(urls[i]))\n      }\n    }    var existing_scripts = []\n    const scripts = document.getElementsByTagName('script')\n    for (let i = 0; i < scripts.length; i++) {\n      var script = scripts[i]\n      if (script.src != null) {\n        existing_scripts.push(script.src)\n      }\n    }\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const escaped = encodeURI(url)\n      if (skip.indexOf(escaped) !== -1 || existing_scripts.indexOf(escaped) !== -1) {\n        if (!window.requirejs) {\n          on_load();\n        }\n        continue;\n      }\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n    for (let i = 0; i < js_modules.length; i++) {\n      const url = js_modules[i];\n      const escaped = encodeURI(url)\n      if (skip.indexOf(escaped) !== -1 || existing_scripts.indexOf(escaped) !== -1) {\n        if (!window.requirejs) {\n          on_load();\n        }\n        continue;\n      }\n      var element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error;\n      element.async = false;\n      element.src = url;\n      element.type = \"module\";\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n    for (const name in js_exports) {\n      const url = js_exports[name];\n      const escaped = encodeURI(url)\n      if (skip.indexOf(escaped) >= 0 || root[name] != null) {\n        if (!window.requirejs) {\n          on_load();\n        }\n        continue;\n      }\n      var element = document.createElement('script');\n      element.onerror = on_error;\n      element.async = false;\n      element.type = \"module\";\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      element.textContent = `\n      import ${name} from \"${url}\"\n      window.${name} = ${name}\n      window._bokeh_on_load()\n      `\n      document.head.appendChild(element);\n    }\n    if (!js_urls.length && !js_modules.length) {\n      on_load()\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.holoviz.org/panel/1.5.5/dist/bundled/reactiveesm/es-module-shims@^1.10.0/dist/es-module-shims.min.js\", \"https://cdn.holoviz.org/panel/1.5.5/dist/bundled/datatabulator/tabulator-tables@6.3.0/dist/js/tabulator.min.js\", \"https://cdn.holoviz.org/panel/1.5.5/dist/bundled/datatabulator/luxon/build/global/luxon.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-3.6.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.2.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.2.min.js\", \"https://cdn.holoviz.org/panel/1.5.5/dist/panel.min.js\"];\n  const js_modules = [];\n  const js_exports = {};\n  const css_urls = [\"https://cdn.holoviz.org/panel/1.5.5/dist/bundled/datatabulator/tabulator-tables@6.3.0/dist/css/tabulator_simple.min.css?v=1.5.5\"];\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {} // ensure no trailing comma for IE\n  ];\n\n  function run_inline_js() {\n    if ((root.Bokeh !== undefined) || (force === true)) {\n      for (let i = 0; i < inline_js.length; i++) {\n        try {\n          inline_js[i].call(root, root.Bokeh);\n        } catch(e) {\n          if (!reloading) {\n            throw e;\n          }\n        }\n      }\n      // Cache old bokeh versions\n      if (Bokeh != undefined && !reloading) {\n        var NewBokeh = root.Bokeh;\n        if (Bokeh.versions === undefined) {\n          Bokeh.versions = new Map();\n        }\n        if (NewBokeh.version !== Bokeh.version) {\n          Bokeh.versions.set(NewBokeh.version, NewBokeh)\n        }\n        root.Bokeh = Bokeh;\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    }\n    root._bokeh_is_initializing = false\n  }\n\n  function load_or_wait() {\n    // Implement a backoff loop that tries to ensure we do not load multiple\n    // versions of Bokeh and its dependencies at the same time.\n    // In recent versions we use the root._bokeh_is_initializing flag\n    // to determine whether there is an ongoing attempt to initialize\n    // bokeh, however for backward compatibility we also try to ensure\n    // that we do not start loading a newer (Panel>=1.0 and Bokeh>3) version\n    // before older versions are fully initialized.\n    if (root._bokeh_is_initializing && Date.now() > root._bokeh_timeout) {\n      // If the timeout and bokeh was not successfully loaded we reset\n      // everything and try loading again\n      root._bokeh_timeout = Date.now() + 5000;\n      root._bokeh_is_initializing = false;\n      root._bokeh_onload_callbacks = undefined;\n      root._bokeh_is_loading = 0\n      console.log(\"Bokeh: BokehJS was loaded multiple times but one version failed to initialize.\");\n      load_or_wait();\n    } else if (root._bokeh_is_initializing || (typeof root._bokeh_is_initializing === \"undefined\" && root._bokeh_onload_callbacks !== undefined)) {\n      setTimeout(load_or_wait, 100);\n    } else {\n      root._bokeh_is_initializing = true\n      root._bokeh_onload_callbacks = []\n      const bokeh_loaded = root.Bokeh != null && (root.Bokeh.version === py_version || (root.Bokeh.versions !== undefined && root.Bokeh.versions.has(py_version)));\n      if (!reloading && !bokeh_loaded) {\n        if (root.Bokeh) {\n          root.Bokeh = undefined;\n        }\n        console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n      }\n      load_libs(css_urls, js_urls, js_modules, js_exports, function() {\n        console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n        run_inline_js();\n      });\n    }\n  }\n  // Give older versions of the autoload script a head-start to ensure\n  // they initialize before we start loading newer version.\n  setTimeout(load_or_wait, 100)\n}(window));",
      "application/vnd.holoviews_load.v0+json": ""
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/javascript": "\nif ((window.PyViz === undefined) || (window.PyViz instanceof HTMLElement)) {\n  window.PyViz = {comms: {}, comm_status:{}, kernels:{}, receivers: {}, plot_index: []}\n}\n\n\n    function JupyterCommManager() {\n    }\n\n    JupyterCommManager.prototype.register_target = function(plot_id, comm_id, msg_handler) {\n      if (window.comm_manager || ((window.Jupyter !== undefined) && (Jupyter.notebook.kernel != null))) {\n        var comm_manager = window.comm_manager || Jupyter.notebook.kernel.comm_manager;\n        comm_manager.register_target(comm_id, function(comm) {\n          comm.on_msg(msg_handler);\n        });\n      } else if ((plot_id in window.PyViz.kernels) && (window.PyViz.kernels[plot_id])) {\n        window.PyViz.kernels[plot_id].registerCommTarget(comm_id, function(comm) {\n          comm.onMsg = msg_handler;\n        });\n      } else if (typeof google != 'undefined' && google.colab.kernel != null) {\n        google.colab.kernel.comms.registerTarget(comm_id, (comm) => {\n          var messages = comm.messages[Symbol.asyncIterator]();\n          function processIteratorResult(result) {\n            var message = result.value;\n            console.log(message)\n            var content = {data: message.data, comm_id};\n            var buffers = []\n            for (var buffer of message.buffers || []) {\n              buffers.push(new DataView(buffer))\n            }\n            var metadata = message.metadata || {};\n            var msg = {content, buffers, metadata}\n            msg_handler(msg);\n            return messages.next().then(processIteratorResult);\n          }\n          return messages.next().then(processIteratorResult);\n        })\n      }\n    }\n\n    JupyterCommManager.prototype.get_client_comm = function(plot_id, comm_id, msg_handler) {\n      if (comm_id in window.PyViz.comms) {\n        return window.PyViz.comms[comm_id];\n      } else if (window.comm_manager || ((window.Jupyter !== undefined) && (Jupyter.notebook.kernel != null))) {\n        var comm_manager = window.comm_manager || Jupyter.notebook.kernel.comm_manager;\n        var comm = comm_manager.new_comm(comm_id, {}, {}, {}, comm_id);\n        if (msg_handler) {\n          comm.on_msg(msg_handler);\n        }\n      } else if ((plot_id in window.PyViz.kernels) && (window.PyViz.kernels[plot_id])) {\n        var comm = window.PyViz.kernels[plot_id].connectToComm(comm_id);\n        comm.open();\n        if (msg_handler) {\n          comm.onMsg = msg_handler;\n        }\n      } else if (typeof google != 'undefined' && google.colab.kernel != null) {\n        var comm_promise = google.colab.kernel.comms.open(comm_id)\n        comm_promise.then((comm) => {\n          window.PyViz.comms[comm_id] = comm;\n          if (msg_handler) {\n            var messages = comm.messages[Symbol.asyncIterator]();\n            function processIteratorResult(result) {\n              var message = result.value;\n              var content = {data: message.data};\n              var metadata = message.metadata || {comm_id};\n              var msg = {content, metadata}\n              msg_handler(msg);\n              return messages.next().then(processIteratorResult);\n            }\n            return messages.next().then(processIteratorResult);\n          }\n        }) \n        var sendClosure = (data, metadata, buffers, disposeOnDone) => {\n          return comm_promise.then((comm) => {\n            comm.send(data, metadata, buffers, disposeOnDone);\n          });\n        };\n        var comm = {\n          send: sendClosure\n        };\n      }\n      window.PyViz.comms[comm_id] = comm;\n      return comm;\n    }\n    window.PyViz.comm_manager = new JupyterCommManager();\n    \n\n\nvar JS_MIME_TYPE = 'application/javascript';\nvar HTML_MIME_TYPE = 'text/html';\nvar EXEC_MIME_TYPE = 'application/vnd.holoviews_exec.v0+json';\nvar CLASS_NAME = 'output';\n\n/**\n * Render data to the DOM node\n */\nfunction render(props, node) {\n  var div = document.createElement(\"div\");\n  var script = document.createElement(\"script\");\n  node.appendChild(div);\n  node.appendChild(script);\n}\n\n/**\n * Handle when a new output is added\n */\nfunction handle_add_output(event, handle) {\n  var output_area = handle.output_area;\n  var output = handle.output;\n  if ((output.data == undefined) || (!output.data.hasOwnProperty(EXEC_MIME_TYPE))) {\n    return\n  }\n  var id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n  var toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n  if (id !== undefined) {\n    var nchildren = toinsert.length;\n    var html_node = toinsert[nchildren-1].children[0];\n    html_node.innerHTML = output.data[HTML_MIME_TYPE];\n    var scripts = [];\n    var nodelist = html_node.querySelectorAll(\"script\");\n    for (var i in nodelist) {\n      if (nodelist.hasOwnProperty(i)) {\n        scripts.push(nodelist[i])\n      }\n    }\n\n    scripts.forEach( function (oldScript) {\n      var newScript = document.createElement(\"script\");\n      var attrs = [];\n      var nodemap = oldScript.attributes;\n      for (var j in nodemap) {\n        if (nodemap.hasOwnProperty(j)) {\n          attrs.push(nodemap[j])\n        }\n      }\n      attrs.forEach(function(attr) { newScript.setAttribute(attr.name, attr.value) });\n      newScript.appendChild(document.createTextNode(oldScript.innerHTML));\n      oldScript.parentNode.replaceChild(newScript, oldScript);\n    });\n    if (JS_MIME_TYPE in output.data) {\n      toinsert[nchildren-1].children[1].textContent = output.data[JS_MIME_TYPE];\n    }\n    output_area._hv_plot_id = id;\n    if ((window.Bokeh !== undefined) && (id in Bokeh.index)) {\n      window.PyViz.plot_index[id] = Bokeh.index[id];\n    } else {\n      window.PyViz.plot_index[id] = null;\n    }\n  } else if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n    var bk_div = document.createElement(\"div\");\n    bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n    var script_attrs = bk_div.children[0].attributes;\n    for (var i = 0; i < script_attrs.length; i++) {\n      toinsert[toinsert.length - 1].childNodes[1].setAttribute(script_attrs[i].name, script_attrs[i].value);\n    }\n    // store reference to server id on output_area\n    output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n  }\n}\n\n/**\n * Handle when an output is cleared or removed\n */\nfunction handle_clear_output(event, handle) {\n  var id = handle.cell.output_area._hv_plot_id;\n  var server_id = handle.cell.output_area._bokeh_server_id;\n  if (((id === undefined) || !(id in PyViz.plot_index)) && (server_id !== undefined)) { return; }\n  var comm = window.PyViz.comm_manager.get_client_comm(\"hv-extension-comm\", \"hv-extension-comm\", function () {});\n  if (server_id !== null) {\n    comm.send({event_type: 'server_delete', 'id': server_id});\n    return;\n  } else if (comm !== null) {\n    comm.send({event_type: 'delete', 'id': id});\n  }\n  delete PyViz.plot_index[id];\n  if ((window.Bokeh !== undefined) & (id in window.Bokeh.index)) {\n    var doc = window.Bokeh.index[id].model.document\n    doc.clear();\n    const i = window.Bokeh.documents.indexOf(doc);\n    if (i > -1) {\n      window.Bokeh.documents.splice(i, 1);\n    }\n  }\n}\n\n/**\n * Handle kernel restart event\n */\nfunction handle_kernel_cleanup(event, handle) {\n  delete PyViz.comms[\"hv-extension-comm\"];\n  window.PyViz.plot_index = {}\n}\n\n/**\n * Handle update_display_data messages\n */\nfunction handle_update_output(event, handle) {\n  handle_clear_output(event, {cell: {output_area: handle.output_area}})\n  handle_add_output(event, handle)\n}\n\nfunction register_renderer(events, OutputArea) {\n  function append_mime(data, metadata, element) {\n    // create a DOM node to render to\n    var toinsert = this.create_output_subarea(\n    metadata,\n    CLASS_NAME,\n    EXEC_MIME_TYPE\n    );\n    this.keyboard_manager.register_events(toinsert);\n    // Render to node\n    var props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n    render(props, toinsert[0]);\n    element.append(toinsert);\n    return toinsert\n  }\n\n  events.on('output_added.OutputArea', handle_add_output);\n  events.on('output_updated.OutputArea', handle_update_output);\n  events.on('clear_output.CodeCell', handle_clear_output);\n  events.on('delete.Cell', handle_clear_output);\n  events.on('kernel_ready.Kernel', handle_kernel_cleanup);\n\n  OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n    safe: true,\n    index: 0\n  });\n}\n\nif (window.Jupyter !== undefined) {\n  try {\n    var events = require('base/js/events');\n    var OutputArea = require('notebook/js/outputarea').OutputArea;\n    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n      register_renderer(events, OutputArea);\n    }\n  } catch(err) {\n  }\n}\n",
      "application/vnd.holoviews_load.v0+json": ""
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.holoviews_exec.v0+json": "",
      "text/html": [
       "<div id='f84500c5-86f2-403c-a374-ccecf418165a'>\n",
       "  <div id=\"ad0fc754-728b-4f33-89af-a379e413ab30\" data-root-id=\"f84500c5-86f2-403c-a374-ccecf418165a\" style=\"display: contents;\"></div>\n",
       "</div>\n",
       "<script type=\"application/javascript\">(function(root) {\n",
       "  var docs_json = {\"fa5b8bcd-9d11-448a-8882-9f25247bd9c1\":{\"version\":\"3.6.2\",\"title\":\"Bokeh Application\",\"roots\":[{\"type\":\"object\",\"name\":\"panel.models.browser.BrowserInfo\",\"id\":\"f84500c5-86f2-403c-a374-ccecf418165a\"},{\"type\":\"object\",\"name\":\"panel.models.comm_manager.CommManager\",\"id\":\"2f512100-70f4-40b8-9a76-26901bf76a3e\",\"attributes\":{\"plot_id\":\"f84500c5-86f2-403c-a374-ccecf418165a\",\"comm_id\":\"2ce5b6ef3bea4bb98ef2ff90e857da2d\",\"client_comm_id\":\"74e2720c87314b2e90b23c19fa3eef65\"}}],\"defs\":[{\"type\":\"model\",\"name\":\"ReactiveHTML1\"},{\"type\":\"model\",\"name\":\"FlexBox1\",\"properties\":[{\"name\":\"align_content\",\"kind\":\"Any\",\"default\":\"flex-start\"},{\"name\":\"align_items\",\"kind\":\"Any\",\"default\":\"flex-start\"},{\"name\":\"flex_direction\",\"kind\":\"Any\",\"default\":\"row\"},{\"name\":\"flex_wrap\",\"kind\":\"Any\",\"default\":\"wrap\"},{\"name\":\"gap\",\"kind\":\"Any\",\"default\":\"\"},{\"name\":\"justify_content\",\"kind\":\"Any\",\"default\":\"flex-start\"}]},{\"type\":\"model\",\"name\":\"FloatPanel1\",\"properties\":[{\"name\":\"config\",\"kind\":\"Any\",\"default\":{\"type\":\"map\"}},{\"name\":\"contained\",\"kind\":\"Any\",\"default\":true},{\"name\":\"position\",\"kind\":\"Any\",\"default\":\"right-top\"},{\"name\":\"offsetx\",\"kind\":\"Any\",\"default\":null},{\"name\":\"offsety\",\"kind\":\"Any\",\"default\":null},{\"name\":\"theme\",\"kind\":\"Any\",\"default\":\"primary\"},{\"name\":\"status\",\"kind\":\"Any\",\"default\":\"normalized\"}]},{\"type\":\"model\",\"name\":\"GridStack1\",\"properties\":[{\"name\":\"mode\",\"kind\":\"Any\",\"default\":\"warn\"},{\"name\":\"ncols\",\"kind\":\"Any\",\"default\":null},{\"name\":\"nrows\",\"kind\":\"Any\",\"default\":null},{\"name\":\"allow_resize\",\"kind\":\"Any\",\"default\":true},{\"name\":\"allow_drag\",\"kind\":\"Any\",\"default\":true},{\"name\":\"state\",\"kind\":\"Any\",\"default\":[]}]},{\"type\":\"model\",\"name\":\"drag1\",\"properties\":[{\"name\":\"slider_width\",\"kind\":\"Any\",\"default\":5},{\"name\":\"slider_color\",\"kind\":\"Any\",\"default\":\"black\"},{\"name\":\"value\",\"kind\":\"Any\",\"default\":50}]},{\"type\":\"model\",\"name\":\"click1\",\"properties\":[{\"name\":\"terminal_output\",\"kind\":\"Any\",\"default\":\"\"},{\"name\":\"debug_name\",\"kind\":\"Any\",\"default\":\"\"},{\"name\":\"clears\",\"kind\":\"Any\",\"default\":0}]},{\"type\":\"model\",\"name\":\"FastWrapper1\",\"properties\":[{\"name\":\"object\",\"kind\":\"Any\",\"default\":null},{\"name\":\"style\",\"kind\":\"Any\",\"default\":null}]},{\"type\":\"model\",\"name\":\"NotificationAreaBase1\",\"properties\":[{\"name\":\"js_events\",\"kind\":\"Any\",\"default\":{\"type\":\"map\"}},{\"name\":\"position\",\"kind\":\"Any\",\"default\":\"bottom-right\"},{\"name\":\"_clear\",\"kind\":\"Any\",\"default\":0}]},{\"type\":\"model\",\"name\":\"NotificationArea1\",\"properties\":[{\"name\":\"js_events\",\"kind\":\"Any\",\"default\":{\"type\":\"map\"}},{\"name\":\"notifications\",\"kind\":\"Any\",\"default\":[]},{\"name\":\"position\",\"kind\":\"Any\",\"default\":\"bottom-right\"},{\"name\":\"_clear\",\"kind\":\"Any\",\"default\":0},{\"name\":\"types\",\"kind\":\"Any\",\"default\":[{\"type\":\"map\",\"entries\":[[\"type\",\"warning\"],[\"background\",\"#ffc107\"],[\"icon\",{\"type\":\"map\",\"entries\":[[\"className\",\"fas fa-exclamation-triangle\"],[\"tagName\",\"i\"],[\"color\",\"white\"]]}]]},{\"type\":\"map\",\"entries\":[[\"type\",\"info\"],[\"background\",\"#007bff\"],[\"icon\",{\"type\":\"map\",\"entries\":[[\"className\",\"fas fa-info-circle\"],[\"tagName\",\"i\"],[\"color\",\"white\"]]}]]}]}]},{\"type\":\"model\",\"name\":\"Notification\",\"properties\":[{\"name\":\"background\",\"kind\":\"Any\",\"default\":null},{\"name\":\"duration\",\"kind\":\"Any\",\"default\":3000},{\"name\":\"icon\",\"kind\":\"Any\",\"default\":null},{\"name\":\"message\",\"kind\":\"Any\",\"default\":\"\"},{\"name\":\"notification_type\",\"kind\":\"Any\",\"default\":null},{\"name\":\"_rendered\",\"kind\":\"Any\",\"default\":false},{\"name\":\"_destroyed\",\"kind\":\"Any\",\"default\":false}]},{\"type\":\"model\",\"name\":\"TemplateActions1\",\"properties\":[{\"name\":\"open_modal\",\"kind\":\"Any\",\"default\":0},{\"name\":\"close_modal\",\"kind\":\"Any\",\"default\":0}]},{\"type\":\"model\",\"name\":\"BootstrapTemplateActions1\",\"properties\":[{\"name\":\"open_modal\",\"kind\":\"Any\",\"default\":0},{\"name\":\"close_modal\",\"kind\":\"Any\",\"default\":0}]},{\"type\":\"model\",\"name\":\"TemplateEditor1\",\"properties\":[{\"name\":\"layout\",\"kind\":\"Any\",\"default\":[]}]},{\"type\":\"model\",\"name\":\"MaterialTemplateActions1\",\"properties\":[{\"name\":\"open_modal\",\"kind\":\"Any\",\"default\":0},{\"name\":\"close_modal\",\"kind\":\"Any\",\"default\":0}]},{\"type\":\"model\",\"name\":\"ReactiveESM1\",\"properties\":[{\"name\":\"esm_constants\",\"kind\":\"Any\",\"default\":{\"type\":\"map\"}}]},{\"type\":\"model\",\"name\":\"JSComponent1\",\"properties\":[{\"name\":\"esm_constants\",\"kind\":\"Any\",\"default\":{\"type\":\"map\"}}]},{\"type\":\"model\",\"name\":\"ReactComponent1\",\"properties\":[{\"name\":\"esm_constants\",\"kind\":\"Any\",\"default\":{\"type\":\"map\"}}]},{\"type\":\"model\",\"name\":\"AnyWidgetComponent1\",\"properties\":[{\"name\":\"esm_constants\",\"kind\":\"Any\",\"default\":{\"type\":\"map\"}}]},{\"type\":\"model\",\"name\":\"request_value1\",\"properties\":[{\"name\":\"fill\",\"kind\":\"Any\",\"default\":\"none\"},{\"name\":\"_synced\",\"kind\":\"Any\",\"default\":null},{\"name\":\"_request_sync\",\"kind\":\"Any\",\"default\":0}]}]}};\n",
       "  var render_items = [{\"docid\":\"fa5b8bcd-9d11-448a-8882-9f25247bd9c1\",\"roots\":{\"f84500c5-86f2-403c-a374-ccecf418165a\":\"ad0fc754-728b-4f33-89af-a379e413ab30\"},\"root_ids\":[\"f84500c5-86f2-403c-a374-ccecf418165a\"]}];\n",
       "  var docs = Object.values(docs_json)\n",
       "  if (!docs) {\n",
       "    return\n",
       "  }\n",
       "  const py_version = docs[0].version.replace('rc', '-rc.').replace('.dev', '-dev.')\n",
       "  async function embed_document(root) {\n",
       "    var Bokeh = get_bokeh(root)\n",
       "    await Bokeh.embed.embed_items_notebook(docs_json, render_items);\n",
       "    for (const render_item of render_items) {\n",
       "      for (const root_id of render_item.root_ids) {\n",
       "\tconst id_el = document.getElementById(root_id)\n",
       "\tif (id_el.children.length && id_el.children[0].hasAttribute('data-root-id')) {\n",
       "\t  const root_el = id_el.children[0]\n",
       "\t  root_el.id = root_el.id + '-rendered'\n",
       "\t  for (const child of root_el.children) {\n",
       "            // Ensure JupyterLab does not capture keyboard shortcuts\n",
       "            // see: https://jupyterlab.readthedocs.io/en/4.1.x/extension/notebook.html#keyboard-interaction-model\n",
       "\t    child.setAttribute('data-lm-suppress-shortcuts', 'true')\n",
       "\t  }\n",
       "\t}\n",
       "      }\n",
       "    }\n",
       "  }\n",
       "  function get_bokeh(root) {\n",
       "    if (root.Bokeh === undefined) {\n",
       "      return null\n",
       "    } else if (root.Bokeh.version !== py_version) {\n",
       "      if (root.Bokeh.versions === undefined || !root.Bokeh.versions.has(py_version)) {\n",
       "\treturn null\n",
       "      }\n",
       "      return root.Bokeh.versions.get(py_version);\n",
       "    } else if (root.Bokeh.version === py_version) {\n",
       "      return root.Bokeh\n",
       "    }\n",
       "    return null\n",
       "  }\n",
       "  function is_loaded(root) {\n",
       "    var Bokeh = get_bokeh(root)\n",
       "    return (Bokeh != null && Bokeh.Panel !== undefined && ( root.Tabulator !== undefined) && ( root.Tabulator !== undefined))\n",
       "  }\n",
       "  if (is_loaded(root)) {\n",
       "    embed_document(root);\n",
       "  } else {\n",
       "    var attempts = 0;\n",
       "    var timer = setInterval(function(root) {\n",
       "      if (is_loaded(root)) {\n",
       "        clearInterval(timer);\n",
       "        embed_document(root);\n",
       "      } else if (document.readyState == \"complete\") {\n",
       "        attempts++;\n",
       "        if (attempts > 200) {\n",
       "          clearInterval(timer);\n",
       "\t  var Bokeh = get_bokeh(root)\n",
       "\t  if (Bokeh == null || Bokeh.Panel == null) {\n",
       "            console.warn(\"Panel: ERROR: Unable to run Panel code because Bokeh or Panel library is missing\");\n",
       "\t  } else {\n",
       "\t    console.warn(\"Panel: WARNING: Attempting to render but not all required libraries could be resolved.\")\n",
       "\t    embed_document(root)\n",
       "\t  }\n",
       "        }\n",
       "      }\n",
       "    }, 25, root)\n",
       "  }\n",
       "})(window);</script>"
      ]
     },
     "metadata": {
      "application/vnd.holoviews_exec.v0+json": {
       "id": "f84500c5-86f2-403c-a374-ccecf418165a"
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import bdikit as bdi\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Next, we load our source data using Pandas and select some columns we are interested in."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>idx</th>\n",
       "      <th>Proteomics_Participant_ID</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>Proteomics_TMT_batch</th>\n",
       "      <th>Proteomics_TMT_plex</th>\n",
       "      <th>Proteomics_TMT_channel</th>\n",
       "      <th>Proteomics_Parent_Sample_IDs</th>\n",
       "      <th>Proteomics_Aliquot_ID</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>Proteomics_OCT</th>\n",
       "      <th>...</th>\n",
       "      <th>RNAseq_R1_sample_type</th>\n",
       "      <th>RNAseq_R1_filename</th>\n",
       "      <th>RNAseq_R1_UUID</th>\n",
       "      <th>RNAseq_R2_sample_type</th>\n",
       "      <th>RNAseq_R2_filename</th>\n",
       "      <th>RNAseq_R2_UUID</th>\n",
       "      <th>miRNAseq_sample_type</th>\n",
       "      <th>miRNAseq_UUID</th>\n",
       "      <th>Methylation_available</th>\n",
       "      <th>Methylation_quality</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S001</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>No</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>128N</td>\n",
       "      <td>C3L-00006-01</td>\n",
       "      <td>CPT0001460012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8a1efc47-1c29-417f-a425-cdbd09565dcb</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...</td>\n",
       "      <td>8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>37bcba98-1094-459e-83ae-c23a602416fb</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S002</td>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>No</td>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>130N</td>\n",
       "      <td>C3L-00008-01</td>\n",
       "      <td>CPT0001300009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>555725e8-cba5-4676-9b0a-80100cbf9f47</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...</td>\n",
       "      <td>15235b12-b67a-4678-acc4-ed03d642bd5e</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>492b50d8-ec35-46e7-a65d-06512aaee394</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S003</td>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>No</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>131</td>\n",
       "      <td>C3L-00032-01</td>\n",
       "      <td>CPT0001420009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>9ae968f3-691d-4db3-9977-1ab3e5af9085</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...</td>\n",
       "      <td>423b6b09-02aa-4f47-9241-f75c1dad1161</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>1794ff56-db2d-4d1a-8758-cab7fe3d98c1</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S004</td>\n",
       "      <td>C3L-00084</td>\n",
       "      <td>Yes</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>129N</td>\n",
       "      <td>C3L-00084-01</td>\n",
       "      <td>CPT0000820012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>b0a7cdf2-2ad8-4442-91b0-548ea4975554</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...</td>\n",
       "      <td>c83987a5-1c13-4af4-b46c-218fe5f60c34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S005</td>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>No</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>129C</td>\n",
       "      <td>C3L-00090-01</td>\n",
       "      <td>CPT0001140003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>8ce5618d-9ff6-40f9-aeea-8d8e1633ae38</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...</td>\n",
       "      <td>06d3fd4a-a623-4146-8500-4f1f17235253</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>a6524c2d-d7dd-4629-980e-b45dbdc92c49</td>\n",
       "      <td>YES</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    idx Proteomics_Participant_ID Case_excluded  Proteomics_TMT_batch  \\\n",
       "0  S001                 C3L-00006            No                     2   \n",
       "1  S002                 C3L-00008            No                     4   \n",
       "2  S003                 C3L-00032            No                     1   \n",
       "3  S004                 C3L-00084           Yes                     3   \n",
       "4  S005                 C3L-00090            No                     3   \n",
       "\n",
       "   Proteomics_TMT_plex Proteomics_TMT_channel Proteomics_Parent_Sample_IDs  \\\n",
       "0                    5                   128N                 C3L-00006-01   \n",
       "1                   16                   130N                 C3L-00008-01   \n",
       "2                    2                    131                 C3L-00032-01   \n",
       "3                   11                   129N                 C3L-00084-01   \n",
       "4                   12                   129C                 C3L-00090-01   \n",
       "\n",
       "  Proteomics_Aliquot_ID Proteomics_Tumor_Normal Proteomics_OCT  ...  \\\n",
       "0         CPT0001460012                   Tumor             No  ...   \n",
       "1         CPT0001300009                   Tumor             No  ...   \n",
       "2         CPT0001420009                   Tumor             No  ...   \n",
       "3         CPT0000820012                   Tumor             No  ...   \n",
       "4         CPT0001140003                   Tumor             No  ...   \n",
       "\n",
       "  RNAseq_R1_sample_type                                 RNAseq_R1_filename  \\\n",
       "0                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "1                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "2                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "3                 Tumor  170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "4                 Tumor  170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "\n",
       "                         RNAseq_R1_UUID RNAseq_R2_sample_type  \\\n",
       "0  8a1efc47-1c29-417f-a425-cdbd09565dcb                 Tumor   \n",
       "1  555725e8-cba5-4676-9b0a-80100cbf9f47                 Tumor   \n",
       "2  9ae968f3-691d-4db3-9977-1ab3e5af9085                 Tumor   \n",
       "3  b0a7cdf2-2ad8-4442-91b0-548ea4975554                 Tumor   \n",
       "4  8ce5618d-9ff6-40f9-aeea-8d8e1633ae38                 Tumor   \n",
       "\n",
       "                                  RNAseq_R2_filename  \\\n",
       "0  170802_UNC31-K00269_0072_AHK3GVBBXX_TAGCTT_S17...   \n",
       "1  170802_UNC31-K00269_0072_AHK3GVBBXX_GGCTAC_S22...   \n",
       "2  170802_UNC31-K00269_0072_AHK3GVBBXX_GTCCGC_S18...   \n",
       "3  170818_UNC32-K00270_0050_AHL2FHBBXX_ATCACG_S5_...   \n",
       "4  170802_UNC31-K00269_0072_AHK3GVBBXX_GAGTGG_S10...   \n",
       "\n",
       "                         RNAseq_R2_UUID miRNAseq_sample_type  \\\n",
       "0  8c3fe9b7-7acd-4867-8d9c-a8e5d1516eda                Tumor   \n",
       "1  15235b12-b67a-4678-acc4-ed03d642bd5e                Tumor   \n",
       "2  423b6b09-02aa-4f47-9241-f75c1dad1161                Tumor   \n",
       "3  c83987a5-1c13-4af4-b46c-218fe5f60c34                  NaN   \n",
       "4  06d3fd4a-a623-4146-8500-4f1f17235253                Tumor   \n",
       "\n",
       "                          miRNAseq_UUID Methylation_available  \\\n",
       "0  37bcba98-1094-459e-83ae-c23a602416fb                   YES   \n",
       "1  492b50d8-ec35-46e7-a65d-06512aaee394                   YES   \n",
       "2  1794ff56-db2d-4d1a-8758-cab7fe3d98c1                   YES   \n",
       "3                                   NaN                   YES   \n",
       "4  a6524c2d-d7dd-4629-980e-b45dbdc92c49                   YES   \n",
       "\n",
       "  Methylation_quality  \n",
       "0                PASS  \n",
       "1                PASS  \n",
       "2                PASS  \n",
       "3                PASS  \n",
       "4                PASS  \n",
       "\n",
       "[5 rows x 179 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_source = pd.read_csv(\"./datasets/Dou-ucec-discovery.csv\")\n",
    "df_source.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then, we load the target dataset:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Idx</th>\n",
       "      <th>Case_id</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>Batch</th>\n",
       "      <th>Plex</th>\n",
       "      <th>ReporterName</th>\n",
       "      <th>Aliquot_ID</th>\n",
       "      <th>Group</th>\n",
       "      <th>Discovery_study</th>\n",
       "      <th>Age</th>\n",
       "      <th>...</th>\n",
       "      <th>Follow-up_additional_surgery_for_new_tumor</th>\n",
       "      <th>Follow-up_additional_treatment_radiation_therapy_for_new_tumor</th>\n",
       "      <th>Follow-up_additional_treatment_pharmaceutical_therapy_for_new_tumor</th>\n",
       "      <th>Follow-up_additional_treatment_immuno_for_new_tumor</th>\n",
       "      <th>Follow-up_days_from_date_of_collection_to_date_of_last_contact</th>\n",
       "      <th>Follow-up_cause_of_death</th>\n",
       "      <th>Follow-up_days_from_date_of_initial_pathologic_diagnosis_to_date_of_death</th>\n",
       "      <th>Follow-up_days_from_date_of_initial_pathologic_diagnosis_to_date_of_additional_surgery_for_new_tumor</th>\n",
       "      <th>Follow-up_procedure_type_of_new_tumor</th>\n",
       "      <th>Follow-up_residual_tumor_after_surgery_for_new_tumor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>No</td>\n",
       "      <td>b4</td>\n",
       "      <td>16.0</td>\n",
       "      <td>128N</td>\n",
       "      <td>CPT0092460003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>56</td>\n",
       "      <td>...</td>\n",
       "      <td>n/a|No|No|No|No</td>\n",
       "      <td>n/a|Yes|Yes|Yes|Yes</td>\n",
       "      <td>n/a|Yes|Yes|Yes|Yes</td>\n",
       "      <td>n/a|No|No|No|No</td>\n",
       "      <td>330.0|701.0|1046.0|1436.0|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a|Breast Carcinoma</td>\n",
       "      <td>n/a|n/a|n/a|n/a|1578.0</td>\n",
       "      <td>n/a|n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a|n/a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00898</td>\n",
       "      <td>C3L-00898</td>\n",
       "      <td>No</td>\n",
       "      <td>b4</td>\n",
       "      <td>14.0</td>\n",
       "      <td>128C</td>\n",
       "      <td>CPT0172200008</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>54</td>\n",
       "      <td>...</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>396.0|746.0|982.0|1600.0</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00943</td>\n",
       "      <td>C3L-00943</td>\n",
       "      <td>No</td>\n",
       "      <td>b4</td>\n",
       "      <td>15.0</td>\n",
       "      <td>130C</td>\n",
       "      <td>CPT0086090003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>63</td>\n",
       "      <td>...</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>237.0|693.0|1039.0</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-01064</td>\n",
       "      <td>C3L-01064</td>\n",
       "      <td>No</td>\n",
       "      <td>b3</td>\n",
       "      <td>9.0</td>\n",
       "      <td>129N</td>\n",
       "      <td>CPT0113430004</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>54</td>\n",
       "      <td>...</td>\n",
       "      <td>No|No|No|No</td>\n",
       "      <td>No|Yes|No|No</td>\n",
       "      <td>Yes|Yes|Yes|Yes</td>\n",
       "      <td>No|No|No|No</td>\n",
       "      <td>453.0|726.0|1062.0|1447.0</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a|n/a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-01277</td>\n",
       "      <td>C3L-01277</td>\n",
       "      <td>No</td>\n",
       "      <td>b4</td>\n",
       "      <td>13.0</td>\n",
       "      <td>130N</td>\n",
       "      <td>CPT0093170003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>No</td>\n",
       "      <td>61</td>\n",
       "      <td>...</td>\n",
       "      <td>n/a|No|No</td>\n",
       "      <td>n/a|No|Yes</td>\n",
       "      <td>n/a|Yes|No</td>\n",
       "      <td>n/a|No|No</td>\n",
       "      <td>351.0|713.0|967.0</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "      <td>n/a|n/a|n/a</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 213 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Idx    Case_id Case_excluded Batch  Plex ReporterName     Aliquot_ID  \\\n",
       "0  C3L-00086  C3L-00086            No    b4  16.0         128N  CPT0092460003   \n",
       "1  C3L-00898  C3L-00898            No    b4  14.0         128C  CPT0172200008   \n",
       "2  C3L-00943  C3L-00943            No    b4  15.0         130C  CPT0086090003   \n",
       "3  C3L-01064  C3L-01064            No    b3   9.0         129N  CPT0113430004   \n",
       "4  C3L-01277  C3L-01277            No    b4  13.0         130N  CPT0093170003   \n",
       "\n",
       "   Group Discovery_study Age  ... Follow-up_additional_surgery_for_new_tumor  \\\n",
       "0  Tumor              No  56  ...                            n/a|No|No|No|No   \n",
       "1  Tumor              No  54  ...                            n/a|n/a|n/a|n/a   \n",
       "2  Tumor              No  63  ...                                n/a|n/a|n/a   \n",
       "3  Tumor              No  54  ...                                No|No|No|No   \n",
       "4  Tumor              No  61  ...                                  n/a|No|No   \n",
       "\n",
       "  Follow-up_additional_treatment_radiation_therapy_for_new_tumor  \\\n",
       "0                                n/a|Yes|Yes|Yes|Yes               \n",
       "1                                    n/a|n/a|n/a|n/a               \n",
       "2                                        n/a|n/a|n/a               \n",
       "3                                       No|Yes|No|No               \n",
       "4                                         n/a|No|Yes               \n",
       "\n",
       "  Follow-up_additional_treatment_pharmaceutical_therapy_for_new_tumor  \\\n",
       "0                                n/a|Yes|Yes|Yes|Yes                    \n",
       "1                                    n/a|n/a|n/a|n/a                    \n",
       "2                                        n/a|n/a|n/a                    \n",
       "3                                    Yes|Yes|Yes|Yes                    \n",
       "4                                         n/a|Yes|No                    \n",
       "\n",
       "   Follow-up_additional_treatment_immuno_for_new_tumor  \\\n",
       "0                                    n/a|No|No|No|No     \n",
       "1                                    n/a|n/a|n/a|n/a     \n",
       "2                                        n/a|n/a|n/a     \n",
       "3                                        No|No|No|No     \n",
       "4                                          n/a|No|No     \n",
       "\n",
       "   Follow-up_days_from_date_of_collection_to_date_of_last_contact  \\\n",
       "0                      330.0|701.0|1046.0|1436.0|n/a                \n",
       "1                           396.0|746.0|982.0|1600.0                \n",
       "2                                 237.0|693.0|1039.0                \n",
       "3                          453.0|726.0|1062.0|1447.0                \n",
       "4                                  351.0|713.0|967.0                \n",
       "\n",
       "           Follow-up_cause_of_death  \\\n",
       "0  n/a|n/a|n/a|n/a|Breast Carcinoma   \n",
       "1                   n/a|n/a|n/a|n/a   \n",
       "2                       n/a|n/a|n/a   \n",
       "3                   n/a|n/a|n/a|n/a   \n",
       "4                       n/a|n/a|n/a   \n",
       "\n",
       "   Follow-up_days_from_date_of_initial_pathologic_diagnosis_to_date_of_death  \\\n",
       "0                             n/a|n/a|n/a|n/a|1578.0                           \n",
       "1                                    n/a|n/a|n/a|n/a                           \n",
       "2                                        n/a|n/a|n/a                           \n",
       "3                                    n/a|n/a|n/a|n/a                           \n",
       "4                                        n/a|n/a|n/a                           \n",
       "\n",
       "  Follow-up_days_from_date_of_initial_pathologic_diagnosis_to_date_of_additional_surgery_for_new_tumor  \\\n",
       "0                                n/a|n/a|n/a|n/a|n/a                                                     \n",
       "1                                    n/a|n/a|n/a|n/a                                                     \n",
       "2                                        n/a|n/a|n/a                                                     \n",
       "3                                    n/a|n/a|n/a|n/a                                                     \n",
       "4                                        n/a|n/a|n/a                                                     \n",
       "\n",
       "  Follow-up_procedure_type_of_new_tumor  \\\n",
       "0                   n/a|n/a|n/a|n/a|n/a   \n",
       "1                       n/a|n/a|n/a|n/a   \n",
       "2                           n/a|n/a|n/a   \n",
       "3                       n/a|n/a|n/a|n/a   \n",
       "4                           n/a|n/a|n/a   \n",
       "\n",
       "  Follow-up_residual_tumor_after_surgery_for_new_tumor  \n",
       "0                                n/a|n/a|n/a|n/a|n/a    \n",
       "1                                    n/a|n/a|n/a|n/a    \n",
       "2                                        n/a|n/a|n/a    \n",
       "3                                    n/a|n/a|n/a|n/a    \n",
       "4                                        n/a|n/a|n/a    \n",
       "\n",
       "[5 rows x 213 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_target = pd.read_csv(\"./datasets/Dou-ucec-confirmatory.csv\")\n",
    "df_target.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For this demo, we are going to focus on a subset of the columns."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Proteomics_Participant_ID</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>Proteomics_TMT_channel</th>\n",
       "      <th>Proteomics_Aliquot_ID</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>Country</th>\n",
       "      <th>Histologic_type</th>\n",
       "      <th>Path_Stage_Primary_Tumor-pT</th>\n",
       "      <th>Path_Stage_Reg_Lymph_Nodes-pN</th>\n",
       "      <th>Clin_Stage_Dist_Mets-cM</th>\n",
       "      <th>...</th>\n",
       "      <th>PMS2</th>\n",
       "      <th>p53</th>\n",
       "      <th>MLH1_Promoter_Hypermethylation</th>\n",
       "      <th>Num_full_term_pregnancies</th>\n",
       "      <th>ESTIMATE_StromalScore</th>\n",
       "      <th>ESTIMATE_ESTIMATEScore</th>\n",
       "      <th>CNV_class</th>\n",
       "      <th>MSI_status</th>\n",
       "      <th>POLE_subtype</th>\n",
       "      <th>Genomics_subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>No</td>\n",
       "      <td>128N</td>\n",
       "      <td>CPT0001460012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pN0</td>\n",
       "      <td>cM0</td>\n",
       "      <td>...</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1</td>\n",
       "      <td>2821.088429</td>\n",
       "      <td>7706.697310</td>\n",
       "      <td>CNV_LOW</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>MSI-H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>No</td>\n",
       "      <td>130N</td>\n",
       "      <td>CPT0001300009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pNX</td>\n",
       "      <td>cM0</td>\n",
       "      <td>...</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1</td>\n",
       "      <td>2714.248104</td>\n",
       "      <td>6346.448091</td>\n",
       "      <td>CNV_LOW</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>MSI-H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>No</td>\n",
       "      <td>131</td>\n",
       "      <td>CPT0001420009</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pN0</td>\n",
       "      <td>cM0</td>\n",
       "      <td>...</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>3856.501277</td>\n",
       "      <td>10459.413600</td>\n",
       "      <td>CNV_LOW</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>CNV_low</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00084</td>\n",
       "      <td>Yes</td>\n",
       "      <td>129N</td>\n",
       "      <td>CPT0000820012</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Carcinosarcoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>No</td>\n",
       "      <td>129C</td>\n",
       "      <td>CPT0001140003</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>United States</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pNX</td>\n",
       "      <td>cM0</td>\n",
       "      <td>...</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>3012.556593</td>\n",
       "      <td>7475.466867</td>\n",
       "      <td>CNV_LOW</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>CNV_low</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Proteomics_Participant_ID Case_excluded Proteomics_TMT_channel  \\\n",
       "0                 C3L-00006            No                   128N   \n",
       "1                 C3L-00008            No                   130N   \n",
       "2                 C3L-00032            No                    131   \n",
       "3                 C3L-00084           Yes                   129N   \n",
       "4                 C3L-00090            No                   129C   \n",
       "\n",
       "  Proteomics_Aliquot_ID Proteomics_Tumor_Normal        Country  \\\n",
       "0         CPT0001460012                   Tumor  United States   \n",
       "1         CPT0001300009                   Tumor  United States   \n",
       "2         CPT0001420009                   Tumor  United States   \n",
       "3         CPT0000820012                   Tumor            NaN   \n",
       "4         CPT0001140003                   Tumor  United States   \n",
       "\n",
       "  Histologic_type Path_Stage_Primary_Tumor-pT Path_Stage_Reg_Lymph_Nodes-pN  \\\n",
       "0    Endometrioid              pT1a (FIGO IA)                           pN0   \n",
       "1    Endometrioid              pT1a (FIGO IA)                           pNX   \n",
       "2    Endometrioid              pT1a (FIGO IA)                           pN0   \n",
       "3  Carcinosarcoma                         NaN                           NaN   \n",
       "4    Endometrioid              pT1a (FIGO IA)                           pNX   \n",
       "\n",
       "  Clin_Stage_Dist_Mets-cM  ...                        PMS2  \\\n",
       "0                     cM0  ...   Intact nuclear expression   \n",
       "1                     cM0  ...  Loss of nuclear expression   \n",
       "2                     cM0  ...   Intact nuclear expression   \n",
       "3                     NaN  ...                         NaN   \n",
       "4                     cM0  ...   Intact nuclear expression   \n",
       "\n",
       "                    p53  MLH1_Promoter_Hypermethylation  \\\n",
       "0  Cannot be determined            Cannot be determined   \n",
       "1  Cannot be determined            Cannot be determined   \n",
       "2  Cannot be determined            Cannot be determined   \n",
       "3                   NaN                             NaN   \n",
       "4  Cannot be determined            Cannot be determined   \n",
       "\n",
       "  Num_full_term_pregnancies ESTIMATE_StromalScore ESTIMATE_ESTIMATEScore  \\\n",
       "0                         1           2821.088429            7706.697310   \n",
       "1                         1           2714.248104            6346.448091   \n",
       "2                 4 or more           3856.501277           10459.413600   \n",
       "3                       NaN                   NaN                    NaN   \n",
       "4                 4 or more           3012.556593            7475.466867   \n",
       "\n",
       "  CNV_class MSI_status POLE_subtype Genomics_subtype  \n",
       "0   CNV_LOW      MSI-H           No            MSI-H  \n",
       "1   CNV_LOW      MSI-H           No            MSI-H  \n",
       "2   CNV_LOW        MSS           No          CNV_low  \n",
       "3       NaN        MSS           No              NaN  \n",
       "4   CNV_LOW        MSS           No          CNV_low  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "column_names = [\n",
    "    \"Proteomics_Participant_ID\",\n",
    "    \"Case_excluded\",\n",
    "    \"Proteomics_TMT_channel\",\n",
    "    \"Proteomics_Aliquot_ID\",\n",
    "    \"Proteomics_Tumor_Normal\",\n",
    "    \"Country\",\n",
    "    \"Histologic_type\",\n",
    "    \"Path_Stage_Primary_Tumor-pT\",\n",
    "    \"Path_Stage_Reg_Lymph_Nodes-pN\",\n",
    "    \"Clin_Stage_Dist_Mets-cM\",\n",
    "    \"tumor_Stage-Pathological\",\n",
    "    \"BMI\",\n",
    "    \"Age\",\n",
    "    \"Diabetes\",\n",
    "    \"Race\",\n",
    "    \"Ethnicity\",\n",
    "    \"Gender\",\n",
    "    \"Tumor_Site\",\n",
    "    \"Tumor_Site_Other\",\n",
    "    \"Tumor_Focality\",\n",
    "    \"Tumor_Size_cm\",\n",
    "    \"Estrogen_Receptor\",\n",
    "    \"Progesterone_Receptor\",\n",
    "    \"MLH1\",\n",
    "    \"PMS2\",\n",
    "    \"p53\",\n",
    "    \"MLH1_Promoter_Hypermethylation\",\n",
    "    \"Num_full_term_pregnancies\",\n",
    "    \"ESTIMATE_StromalScore\",\n",
    "    \"ESTIMATE_ESTIMATEScore\",\n",
    "    \"CNV_class\",\n",
    "    \"MSI_status\",\n",
    "    \"POLE_subtype\",\n",
    "    \"Genomics_subtype\",\n",
    "]\n",
    "df_source = df_source[column_names]\n",
    "df_source.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Finding column matches between two tables"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`bdi-kit` offers a suite of functions to help with data harmonization tasks.\n",
    "\n",
    "For instance, it can help automatically discover one-to-one mappings between the source and target dataset columns.\n",
    "\n",
    "To do so using `bdi-kit`, we can use the `match_schema()` function to match columns of the two schemas as follows. The `method` parameter allows modifying the schema matching algorithm used to perform the matching.\n",
    "\n",
    "```python\n",
    "bdi.match_schema(df_source, df_target, method=\"jaccard_distance\")\n",
    "bdi.match_schema(df_source, df_target, method=\"coma\")\n",
    "bdi.match_schema(df_source, df_target, method=\"similarity_flooding\")\n",
    "```\n",
    "\n",
    "The method we use below is based on a pre-trained language model fine-tuned for the biomedical domain.\n",
    "\n",
    "For all methods available, see: [https://bdi-kit.readthedocs.io/en/latest/schema-matching.html](https://bdi-kit.readthedocs.io/en/latest/schema-matching.html)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/miniconda3/envs/bdf/lib/python3.10/site-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning\n",
      "  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>target</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Race</td>\n",
       "      <td>Race</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MSI_status</td>\n",
       "      <td>MSI_status</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Case_excluded</td>\n",
       "      <td>Case_excluded</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>tumor_Stage-Pathological</td>\n",
       "      <td>Tumor_stage_pathological</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ethnicity</td>\n",
       "      <td>Ethnicity</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Tumor_Focality</td>\n",
       "      <td>Tumor_focality</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Gender</td>\n",
       "      <td>Sex</td>\n",
       "      <td>0.995859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Num_full_term_pregnancies</td>\n",
       "      <td>Donor_information_number_of_full_term_pregnancies</td>\n",
       "      <td>0.984231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POLE_subtype</td>\n",
       "      <td>POLE</td>\n",
       "      <td>0.981701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MLH1_Promoter_Hypermethylation</td>\n",
       "      <td>Ancillary_studies_mlh1_promoter_hypermethylation</td>\n",
       "      <td>0.964056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Path_Stage_Primary_Tumor-pT</td>\n",
       "      <td>Pathologic_staging_primary_tumor_pt</td>\n",
       "      <td>0.945246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Tumor_Site_Other</td>\n",
       "      <td>Tumor_site_other</td>\n",
       "      <td>0.929722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Tumor_Site</td>\n",
       "      <td>Tumor_site</td>\n",
       "      <td>0.924333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Tumor_Size_cm</td>\n",
       "      <td>Tumor_size_cm</td>\n",
       "      <td>0.888654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Histologic_type</td>\n",
       "      <td>Histologic_Type</td>\n",
       "      <td>0.877000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            source  \\\n",
       "0                             Race   \n",
       "1                       MSI_status   \n",
       "2                    Case_excluded   \n",
       "3         tumor_Stage-Pathological   \n",
       "4                        Ethnicity   \n",
       "5                   Tumor_Focality   \n",
       "6                           Gender   \n",
       "7        Num_full_term_pregnancies   \n",
       "8                     POLE_subtype   \n",
       "9   MLH1_Promoter_Hypermethylation   \n",
       "10     Path_Stage_Primary_Tumor-pT   \n",
       "11                Tumor_Site_Other   \n",
       "12                      Tumor_Site   \n",
       "13                   Tumor_Size_cm   \n",
       "14                 Histologic_type   \n",
       "\n",
       "                                               target  similarity  \n",
       "0                                                Race    1.000000  \n",
       "1                                          MSI_status    1.000000  \n",
       "2                                       Case_excluded    1.000000  \n",
       "3                            Tumor_stage_pathological    1.000000  \n",
       "4                                           Ethnicity    1.000000  \n",
       "5                                      Tumor_focality    1.000000  \n",
       "6                                                 Sex    0.995859  \n",
       "7   Donor_information_number_of_full_term_pregnancies    0.984231  \n",
       "8                                                POLE    0.981701  \n",
       "9    Ancillary_studies_mlh1_promoter_hypermethylation    0.964056  \n",
       "10                Pathologic_staging_primary_tumor_pt    0.945246  \n",
       "11                                   Tumor_site_other    0.929722  \n",
       "12                                         Tumor_site    0.924333  \n",
       "13                                      Tumor_size_cm    0.888654  \n",
       "14                                    Histologic_Type    0.877000  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "schema_mapping = bdi.match_schema(df_source, df_target, method=\"max_val_sim\")\n",
    "schema_mapping.head(15)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Inspecting values of each column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00090</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  value_name\n",
       "0  C3L-00006\n",
       "1  C3L-00008\n",
       "2  C3L-00032\n",
       "3  C3L-00084\n",
       "4  C3L-00090"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bdi.preview_domain(df_source, 'Proteomics_Participant_ID').head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>value_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-01064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-01277</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  value_name\n",
       "0  C3L-00086\n",
       "1  C3L-00898\n",
       "2  C3L-00943\n",
       "3  C3L-01064\n",
       "4  C3L-01277"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bdi.preview_domain(df_target, 'Case_id').head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Finding value matches between two columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Once the matching columns are identified, we can standardize data to ensure that no duplicate values represent the same entity/meaning.\n",
    "\n",
    "To do that, `bdikit` provides the function `match_values()` to find values that should potentially be merged. The library supports multiple methods to perform this task, including syntactic and semantic matching algorithms. In this example, we use the `tfidf` method, which finds values based on the similarity of character n-grams. Please, refer to the [bdikit documentation](https://bdi-kit.readthedocs.io/) to learn more about the methods available."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Race<br>**Target column:** Race<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>White</td>\n",
       "      <td>White</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>White</td>\n",
       "      <td>White</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Asian</td>\n",
       "      <td>Asian</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Not Reported</td>\n",
       "      <td>Not Reported</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Black or African American</td>\n",
       "      <td>Black or African American</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                source_value               target_value  similarity\n",
       "0                      White                      White         1.0\n",
       "1                      White                      White         1.0\n",
       "2                      Asian                      Asian         1.0\n",
       "3               Not Reported               Not Reported         1.0\n",
       "4  Black or African American  Black or African American         1.0\n",
       "5                        NaN                        NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** MSI_status<br>**Target column:** MSI_status<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSI-H</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MSS</td>\n",
       "      <td>MSS</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0        MSI-H        MSI-H         1.0\n",
       "1          MSS          MSS         1.0\n",
       "2          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Case_excluded<br>**Target column:** Case_excluded<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0           No           No         1.0\n",
       "1          Yes          Yes         1.0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** tumor_Stage-Pathological<br>**Target column:** Tumor_stage_pathological<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Stage I</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Stage IV</td>\n",
       "      <td>Stage IV</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Stage III</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Stage II</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0      Stage I      Stage I         1.0\n",
       "1     Stage IV     Stage IV         1.0\n",
       "2    Stage III    Stage III         1.0\n",
       "3     Stage II     Stage II         1.0\n",
       "4          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Ethnicity<br>**Target column:** Ethnicity<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Hispanic or Latino</td>\n",
       "      <td>Hispanic or Latino</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Not reported</td>\n",
       "      <td>Not reported</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             source_value            target_value  similarity\n",
       "0  Not-Hispanic or Latino  Not-Hispanic or Latino         1.0\n",
       "1      Hispanic or Latino      Hispanic or Latino         1.0\n",
       "2            Not reported            Not reported         1.0\n",
       "3                     NaN                     NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Tumor_Focality<br>**Target column:** Tumor_focality<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Multifocal</td>\n",
       "      <td>Multifocal</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0     Unifocal     Unifocal         1.0\n",
       "1   Multifocal   Multifocal         1.0\n",
       "2          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Gender<br>**Target column:** Sex<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Female</td>\n",
       "      <td>Female</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0       Female       Female         1.0\n",
       "1          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Num_full_term_pregnancies<br>**Target column:** Donor_information_number_of_full_term_pregnancies<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4 or more</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0            1            1         1.0\n",
       "1    4 or more    4 or more         1.0\n",
       "2            2            2         1.0\n",
       "3            3            3         1.0\n",
       "4      Unknown      Unknown         1.0\n",
       "5          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** POLE_subtype<br>**Target column:** POLE<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0           No           No         1.0\n",
       "1          Yes          Yes         1.0\n",
       "2          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** MLH1_Promoter_Hypermethylation<br>**Target column:** Ancillary_studies_mlh1_promoter_hypermethylation<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Absent</td>\n",
       "      <td>Absent</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Present</td>\n",
       "      <td>Present</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           source_value          target_value  similarity\n",
       "0  Cannot be determined  Cannot be determined         1.0\n",
       "1                Absent                Absent         1.0\n",
       "2               Present               Present         1.0\n",
       "3                   NaN                   NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Path_Stage_Primary_Tumor-pT<br>**Target column:** Pathologic_staging_primary_tumor_pt<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>pT1a (FIGO IA)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pT3a (FIGO IIIA)</td>\n",
       "      <td>pT3a (FIGO IIIA)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pT1 (FIGO I)</td>\n",
       "      <td>pT1 (FIGO I)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>pT1b (FIGO IB)</td>\n",
       "      <td>pT1b (FIGO IB)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pT2 (FIGO II)</td>\n",
       "      <td>pT2 (FIGO II)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>pT3b (FIGO IIIB)</td>\n",
       "      <td>pT3b (FIGO IIIB)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       source_value      target_value  similarity\n",
       "0    pT1a (FIGO IA)    pT1a (FIGO IA)         1.0\n",
       "1  pT3a (FIGO IIIA)  pT3a (FIGO IIIA)         1.0\n",
       "2      pT1 (FIGO I)      pT1 (FIGO I)         1.0\n",
       "3    pT1b (FIGO IB)    pT1b (FIGO IB)         1.0\n",
       "4     pT2 (FIGO II)     pT2 (FIGO II)         1.0\n",
       "5  pT3b (FIGO IIIB)  pT3b (FIGO IIIB)         1.0\n",
       "6               NaN               NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Tumor_Site_Other<br>**Target column:** Tumor_site_other<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Anterior and Posterior endometrium</td>\n",
       "      <td>Anterior and posterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Anterior  and Posterior endometrium</td>\n",
       "      <td>Anterior and posterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Both anterior and posterior</td>\n",
       "      <td>Both anterior and posterior</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Anterior and Posterior Endometrium</td>\n",
       "      <td>Anterior and posterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Both anterior and posterior endometrium</td>\n",
       "      <td>Both anterior and posterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Anterior and posterior endometrium</td>\n",
       "      <td>Anterior and posterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Anterior and Posterior</td>\n",
       "      <td>anterior and posterior</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Posterior and Anterior Endometrium</td>\n",
       "      <td>Anterior and posterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>anterior and posterior</td>\n",
       "      <td>anterior and posterior</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>both anterior and posterior</td>\n",
       "      <td>Both anterior and posterior</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Entire uterine cavity</td>\n",
       "      <td>Entire uterine cavity</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Uterine cavity</td>\n",
       "      <td>Uterine cavity</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Entire Uterine Cavity</td>\n",
       "      <td>Entire uterine cavity</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Entire uterine cavity.</td>\n",
       "      <td>Entire uterine cavity</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Endometrium</td>\n",
       "      <td>Endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>entire uterine cavity</td>\n",
       "      <td>Entire uterine cavity</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>anterior and posterior endometrial cavity</td>\n",
       "      <td>Anterior and posterior endometrium</td>\n",
       "      <td>0.831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Involves fundus, anterior and posterior walls</td>\n",
       "      <td>Involves anterior and posterior</td>\n",
       "      <td>0.784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Anterior, posterior and fundus</td>\n",
       "      <td>anterior and posterior</td>\n",
       "      <td>0.782</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Tumor occupies 75% of endometrial surface</td>\n",
       "      <td>Tumor occupies 80% of endometrial cavity</td>\n",
       "      <td>0.730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Both anterior and posterior endometrium are re...</td>\n",
       "      <td>Both anterior and posterior endometrium</td>\n",
       "      <td>0.704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>along anterior and posterior surface</td>\n",
       "      <td>anterior and posterior</td>\n",
       "      <td>0.641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>If Other, specify: Tumor occupies 100% of endo...</td>\n",
       "      <td>Tumor occupies 80% of endometrial cavity</td>\n",
       "      <td>0.630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Tumor involves 75% of endometrial cavity per d...</td>\n",
       "      <td>Tumor involves the entire endometrial cavity</td>\n",
       "      <td>0.617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Bulky tumor involving both anterior and poster...</td>\n",
       "      <td>Involves anterior and posterior</td>\n",
       "      <td>0.603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED</td>\n",
       "      <td>Tumor involves the entire endometrial cavity</td>\n",
       "      <td>0.507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>left cornu</td>\n",
       "      <td>left side of uterine cavity</td>\n",
       "      <td>0.376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         source_value  \\\n",
       "0                  Anterior and Posterior endometrium   \n",
       "1                 Anterior  and Posterior endometrium   \n",
       "2                         Both anterior and posterior   \n",
       "3                  Anterior and Posterior Endometrium   \n",
       "4             Both anterior and posterior endometrium   \n",
       "5                  Anterior and posterior endometrium   \n",
       "6                              Anterior and Posterior   \n",
       "7                  Posterior and Anterior Endometrium   \n",
       "8                              anterior and posterior   \n",
       "9                         both anterior and posterior   \n",
       "10                              Entire uterine cavity   \n",
       "11                                     Uterine cavity   \n",
       "12                              Entire Uterine Cavity   \n",
       "13                             Entire uterine cavity.   \n",
       "14                                        Endometrium   \n",
       "15                              entire uterine cavity   \n",
       "16          anterior and posterior endometrial cavity   \n",
       "17      Involves fundus, anterior and posterior walls   \n",
       "18                     Anterior, posterior and fundus   \n",
       "19          Tumor occupies 75% of endometrial surface   \n",
       "20  Both anterior and posterior endometrium are re...   \n",
       "21               along anterior and posterior surface   \n",
       "22  If Other, specify: Tumor occupies 100% of endo...   \n",
       "23  Tumor involves 75% of endometrial cavity per d...   \n",
       "24  Bulky tumor involving both anterior and poster...   \n",
       "25        100 PERCENT OF ENDOMETRIAL SURFACE INVOLVED   \n",
       "26                                         left cornu   \n",
       "27                                                NaN   \n",
       "\n",
       "                                    target_value  similarity  \n",
       "0             Anterior and posterior endometrium       1.000  \n",
       "1             Anterior and posterior endometrium       1.000  \n",
       "2                    Both anterior and posterior       1.000  \n",
       "3             Anterior and posterior endometrium       1.000  \n",
       "4        Both anterior and posterior endometrium       1.000  \n",
       "5             Anterior and posterior endometrium       1.000  \n",
       "6                         anterior and posterior       1.000  \n",
       "7             Anterior and posterior endometrium       1.000  \n",
       "8                         anterior and posterior       1.000  \n",
       "9                    Both anterior and posterior       1.000  \n",
       "10                         Entire uterine cavity       1.000  \n",
       "11                                Uterine cavity       1.000  \n",
       "12                         Entire uterine cavity       1.000  \n",
       "13                         Entire uterine cavity       1.000  \n",
       "14                                   Endometrium       1.000  \n",
       "15                         Entire uterine cavity       1.000  \n",
       "16            Anterior and posterior endometrium       0.831  \n",
       "17               Involves anterior and posterior       0.784  \n",
       "18                        anterior and posterior       0.782  \n",
       "19      Tumor occupies 80% of endometrial cavity       0.730  \n",
       "20       Both anterior and posterior endometrium       0.704  \n",
       "21                        anterior and posterior       0.641  \n",
       "22      Tumor occupies 80% of endometrial cavity       0.630  \n",
       "23  Tumor involves the entire endometrial cavity       0.617  \n",
       "24               Involves anterior and posterior       0.603  \n",
       "25  Tumor involves the entire endometrial cavity       0.507  \n",
       "26                   left side of uterine cavity       0.376  \n",
       "27                                           NaN         NaN  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Tumor_Site<br>**Target column:** Tumor_site<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Anterior endometrium</td>\n",
       "      <td>Anterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Posterior endometrium</td>\n",
       "      <td>Posterior endometrium</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Other, specify</td>\n",
       "      <td>Other</td>\n",
       "      <td>0.546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            source_value           target_value  similarity\n",
       "0   Anterior endometrium   Anterior endometrium       1.000\n",
       "1  Posterior endometrium  Posterior endometrium       1.000\n",
       "2         Other, specify                  Other       0.546\n",
       "3                    NaN                    NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Progesterone_Receptor<br>**Target column:** Ancillary_studies_progesterone_receptor<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive : 5 %</td>\n",
       "      <td>0.938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           source_value          target_value  similarity\n",
       "0  Cannot be determined  Cannot be determined       1.000\n",
       "1              Negative              Negative       1.000\n",
       "2              Positive        Positive : 5 %       0.938\n",
       "3                   NaN                   NaN         NaN\n",
       "4               Unknown                   NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Estrogen_Receptor<br>**Target column:** Ancillary_studies_estrogen_receptor<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Negative</td>\n",
       "      <td>Negative</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Positive</td>\n",
       "      <td>Positive : 5 %</td>\n",
       "      <td>0.937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           source_value          target_value  similarity\n",
       "0  Cannot be determined  Cannot be determined       1.000\n",
       "1              Negative              Negative       1.000\n",
       "2              Positive        Positive : 5 %       0.937\n",
       "3                   NaN                   NaN         NaN\n",
       "4               Unknown                   NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** p53<br>**Target column:** Ancillary_studies_p53<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal</td>\n",
       "      <td>Normal</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Overexpression</td>\n",
       "      <td>Overexpression</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Loss of expression</td>\n",
       "      <td>Loss of expression</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           source_value          target_value  similarity\n",
       "0  Cannot be determined  Cannot be determined         1.0\n",
       "1                Normal                Normal         1.0\n",
       "2        Overexpression        Overexpression         1.0\n",
       "3    Loss of expression    Loss of expression         1.0\n",
       "4                   NaN                   NaN         NaN\n",
       "5               Unknown                   NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Country<br>**Target column:** Participant_country<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>United States</td>\n",
       "      <td>United States</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Ukraine</td>\n",
       "      <td>Ukraine</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Poland</td>\n",
       "      <td>Poland</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Other_specify</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    source_value   target_value  similarity\n",
       "0  United States  United States         1.0\n",
       "1        Ukraine        Ukraine         1.0\n",
       "2         Poland         Poland         1.0\n",
       "3  Other_specify            NaN         NaN\n",
       "4            NaN            NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Clin_Stage_Dist_Mets-cM<br>**Target column:** Clinical_staging_distant_metastasis_cm<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cM0</td>\n",
       "      <td>cM0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Staging Incomplete</td>\n",
       "      <td>Staging Incomplete</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cM1</td>\n",
       "      <td>cM1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         source_value        target_value  similarity\n",
       "0                 cM0                 cM0         1.0\n",
       "1  Staging Incomplete  Staging Incomplete         1.0\n",
       "2                 cM1                 cM1         1.0\n",
       "3                 NaN                 NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Genomics_subtype<br>**Target column:** Genomic_subtype<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MSI-H</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>POLE</td>\n",
       "      <td>POLE</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CNV_low</td>\n",
       "      <td>CNV_L</td>\n",
       "      <td>0.658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CNV_high</td>\n",
       "      <td>CNV_H</td>\n",
       "      <td>0.625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0        MSI-H        MSI-H       1.000\n",
       "1         POLE         POLE       1.000\n",
       "2      CNV_low        CNV_L       0.658\n",
       "3     CNV_high        CNV_H       0.625\n",
       "4          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Diabetes<br>**Target column:** Diabetes<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0          Yes          Yes         1.0\n",
       "1           No          NaN         NaN\n",
       "2      Unknown          NaN         NaN\n",
       "3          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** MLH1<br>**Target column:** Ancillary_studies_msh2<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 source_value                target_value  similarity\n",
       "0   Intact nuclear expression   Intact nuclear expression         1.0\n",
       "1  Loss of nuclear expression  Loss of nuclear expression         1.0\n",
       "2        Cannot be determined        Cannot be determined         1.0\n",
       "3                         NaN                         NaN         NaN\n",
       "4                     Unknown                         NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Path_Stage_Reg_Lymph_Nodes-pN<br>**Target column:** Pathologic_staging_regional_lymph_nodes_pn<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pN0</td>\n",
       "      <td>pN0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pNX</td>\n",
       "      <td>pNX</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pN2 (FIGO IIIC2)</td>\n",
       "      <td>pN2 (FIGO IIIC2)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>pN1 (FIGO IIIC1)</td>\n",
       "      <td>pN1 (FIGO IIIC1)</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       source_value      target_value  similarity\n",
       "0               pN0               pN0         1.0\n",
       "1               pNX               pNX         1.0\n",
       "2  pN2 (FIGO IIIC2)  pN2 (FIGO IIIC2)         1.0\n",
       "3  pN1 (FIGO IIIC1)  pN1 (FIGO IIIC1)         1.0\n",
       "4               NaN               NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Proteomics_TMT_channel<br>**Target column:** ReporterName<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>128N</td>\n",
       "      <td>128N</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>130N</td>\n",
       "      <td>130N</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>129N</td>\n",
       "      <td>129N</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>129C</td>\n",
       "      <td>129C</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>130C</td>\n",
       "      <td>130C</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>127C</td>\n",
       "      <td>127C</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>127N</td>\n",
       "      <td>127N</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>128C</td>\n",
       "      <td>128C</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>131</td>\n",
       "      <td>131N</td>\n",
       "      <td>0.709</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0         128N         128N       1.000\n",
       "1         130N         130N       1.000\n",
       "2         129N         129N       1.000\n",
       "3         129C         129C       1.000\n",
       "4         130C         130C       1.000\n",
       "5         127C         127C       1.000\n",
       "6         127N         127N       1.000\n",
       "7         128C         128C       1.000\n",
       "8          131         131N       0.709"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Proteomics_Tumor_Normal<br>**Target column:** Group<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tumor</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>Adjacent_normal</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Enriched_normal</td>\n",
       "      <td>Enriched_Normal</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Myometrium_normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        source_value     target_value  similarity\n",
       "0              Tumor            Tumor         1.0\n",
       "1    Adjacent_normal  Adjacent_normal         1.0\n",
       "2    Enriched_normal  Enriched_Normal         1.0\n",
       "3  Myometrium_normal              NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** PMS2<br>**Target column:** Ancillary_studies_pms2<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 source_value                target_value  similarity\n",
       "0   Intact nuclear expression   Intact nuclear expression         1.0\n",
       "1  Loss of nuclear expression  Loss of nuclear expression         1.0\n",
       "2        Cannot be determined        Cannot be determined         1.0\n",
       "3                         NaN                         NaN         NaN\n",
       "4                     Unknown                         NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Proteomics_Aliquot_ID<br>**Target column:** Aliquot_ID<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CPT0026530003</td>\n",
       "      <td>CPT0265300003</td>\n",
       "      <td>0.959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CPT0016940003</td>\n",
       "      <td>CPT0116940003</td>\n",
       "      <td>0.860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CPT0018960003</td>\n",
       "      <td>CPT0089600003</td>\n",
       "      <td>0.771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CPT0097730003</td>\n",
       "      <td>CPT0097300003</td>\n",
       "      <td>0.757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CPT0027520003</td>\n",
       "      <td>CPT0275270003</td>\n",
       "      <td>0.743</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>CPT0077660001</td>\n",
       "      <td>CPT0126000003</td>\n",
       "      <td>0.343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>CPT0063900001</td>\n",
       "      <td>CPT0100010003</td>\n",
       "      <td>0.343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>CPT0064240001</td>\n",
       "      <td>CPT0129640003</td>\n",
       "      <td>0.319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>CPT0076680003</td>\n",
       "      <td>CPT0230070003</td>\n",
       "      <td>0.318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>CPT0066620003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      source_value   target_value  similarity\n",
       "0    CPT0026530003  CPT0265300003       0.959\n",
       "1    CPT0016940003  CPT0116940003       0.860\n",
       "2    CPT0018960003  CPT0089600003       0.771\n",
       "3    CPT0097730003  CPT0097300003       0.757\n",
       "4    CPT0027520003  CPT0275270003       0.743\n",
       "..             ...            ...         ...\n",
       "148  CPT0077660001  CPT0126000003       0.343\n",
       "149  CPT0063900001  CPT0100010003       0.343\n",
       "150  CPT0064240001  CPT0129640003       0.319\n",
       "151  CPT0076680003  CPT0230070003       0.318\n",
       "152  CPT0066620003            NaN         NaN\n",
       "\n",
       "[153 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Histologic_type<br>**Target column:** Histologic_Type<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>0.854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Clear cell</td>\n",
       "      <td>Clear cell carcinoma</td>\n",
       "      <td>0.836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Serous</td>\n",
       "      <td>Serous carcinoma</td>\n",
       "      <td>0.715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Carcinosarcoma</td>\n",
       "      <td>Serous carcinoma</td>\n",
       "      <td>0.611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     source_value            target_value  similarity\n",
       "0    Endometrioid  Endometrioid carcinoma       0.854\n",
       "1      Clear cell    Clear cell carcinoma       0.836\n",
       "2          Serous        Serous carcinoma       0.715\n",
       "3  Carcinosarcoma        Serous carcinoma       0.611\n",
       "4             NaN                     NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** Proteomics_Participant_ID<br>**Target column:** Idx<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3N-01003</td>\n",
       "      <td>C3N-01008</td>\n",
       "      <td>0.809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3N-01001</td>\n",
       "      <td>C3N-01008</td>\n",
       "      <td>0.808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>0.781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00358</td>\n",
       "      <td>C3L-03589</td>\n",
       "      <td>0.718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3N-01520</td>\n",
       "      <td>C3N-01525</td>\n",
       "      <td>0.713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>C3L-00769</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>NX5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>NX3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>NX11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>NX18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>123 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    source_value target_value  similarity\n",
       "0      C3N-01003    C3N-01008       0.809\n",
       "1      C3N-01001    C3N-01008       0.808\n",
       "2      C3L-00008    C3L-00086       0.781\n",
       "3      C3L-00358    C3L-03589       0.718\n",
       "4      C3N-01520    C3N-01525       0.713\n",
       "..           ...          ...         ...\n",
       "118    C3L-00769          NaN         NaN\n",
       "119          NX5          NaN         NaN\n",
       "120          NX3          NaN         NaN\n",
       "121         NX11          NaN         NaN\n",
       "122         NX18          NaN         NaN\n",
       "\n",
       "[123 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/markdown": [
       "<br>**Source column:** CNV_class<br>**Target column:** CNV_status<br>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_value</th>\n",
       "      <th>target_value</th>\n",
       "      <th>similarity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CNV_LOW</td>\n",
       "      <td>CNV_L</td>\n",
       "      <td>0.607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CNV_HIGH</td>\n",
       "      <td>CNV_H</td>\n",
       "      <td>0.603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_value target_value  similarity\n",
       "0      CNV_LOW        CNV_L       0.607\n",
       "1     CNV_HIGH        CNV_H       0.603\n",
       "2          NaN          NaN         NaN"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Matches values from each pair of source-target columns\n",
    "value_matches = bdi.match_values(df_source, df_target, schema_mapping, method=\"tfidf\")\n",
    "\n",
    "# Print value matches\n",
    "bdi.view_value_matches(value_matches, edit=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Generating a harmonized table"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After verifying and fixing value mappings, we can proceed to generate a harmonized table.\n",
    "\n",
    "In bdikit, we can do this using the `bdi.materialize_mapping()` function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Race</th>\n",
       "      <th>MSI_status</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>Tumor_stage_pathological</th>\n",
       "      <th>Ethnicity</th>\n",
       "      <th>Tumor_focality</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Donor_information_number_of_full_term_pregnancies</th>\n",
       "      <th>POLE</th>\n",
       "      <th>Ancillary_studies_mlh1_promoter_hypermethylation</th>\n",
       "      <th>...</th>\n",
       "      <th>Diabetes</th>\n",
       "      <th>Ancillary_studies_msh2</th>\n",
       "      <th>Pathologic_staging_regional_lymph_nodes_pn</th>\n",
       "      <th>ReporterName</th>\n",
       "      <th>Group</th>\n",
       "      <th>Ancillary_studies_pms2</th>\n",
       "      <th>Aliquot_ID</th>\n",
       "      <th>Histologic_Type</th>\n",
       "      <th>Idx</th>\n",
       "      <th>CNV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>White</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pN0</td>\n",
       "      <td>128N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0126000003</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>White</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage IV</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>130N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>CPT0095130004</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pN0</td>\n",
       "      <td>131N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0313420004</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>129N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CPT0182020004</td>\n",
       "      <td>Serous carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>129C</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0114340003</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Race MSI_status Case_excluded Tumor_stage_pathological  \\\n",
       "0  White      MSI-H            No                  Stage I   \n",
       "1  White      MSI-H            No                 Stage IV   \n",
       "2  White        MSS            No                  Stage I   \n",
       "3    NaN        MSS           Yes                      NaN   \n",
       "4  White        MSS            No                  Stage I   \n",
       "\n",
       "                Ethnicity Tumor_focality     Sex  \\\n",
       "0  Not-Hispanic or Latino       Unifocal  Female   \n",
       "1  Not-Hispanic or Latino       Unifocal  Female   \n",
       "2  Not-Hispanic or Latino       Unifocal  Female   \n",
       "3                     NaN            NaN     NaN   \n",
       "4  Not-Hispanic or Latino       Unifocal  Female   \n",
       "\n",
       "  Donor_information_number_of_full_term_pregnancies POLE  \\\n",
       "0                                                 1   No   \n",
       "1                                                 1   No   \n",
       "2                                         4 or more   No   \n",
       "3                                               NaN   No   \n",
       "4                                         4 or more   No   \n",
       "\n",
       "  Ancillary_studies_mlh1_promoter_hypermethylation  ... Diabetes  \\\n",
       "0                             Cannot be determined  ...      NaN   \n",
       "1                             Cannot be determined  ...      NaN   \n",
       "2                             Cannot be determined  ...      Yes   \n",
       "3                                              NaN  ...      NaN   \n",
       "4                             Cannot be determined  ...      NaN   \n",
       "\n",
       "      Ancillary_studies_msh2 Pathologic_staging_regional_lymph_nodes_pn  \\\n",
       "0  Intact nuclear expression                                        pN0   \n",
       "1  Intact nuclear expression                                        pNX   \n",
       "2  Intact nuclear expression                                        pN0   \n",
       "3                        NaN                                        NaN   \n",
       "4  Intact nuclear expression                                        pNX   \n",
       "\n",
       "  ReporterName  Group      Ancillary_studies_pms2     Aliquot_ID  \\\n",
       "0         128N  Tumor   Intact nuclear expression  CPT0126000003   \n",
       "1         130N  Tumor  Loss of nuclear expression  CPT0095130004   \n",
       "2         131N  Tumor   Intact nuclear expression  CPT0313420004   \n",
       "3         129N  Tumor                         NaN  CPT0182020004   \n",
       "4         129C  Tumor   Intact nuclear expression  CPT0114340003   \n",
       "\n",
       "          Histologic_Type        Idx CNV_status  \n",
       "0  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "1  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "2  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "3        Serous carcinoma  C3L-00086        NaN  \n",
       "4  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mapped = bdi.materialize_mapping(df_source, value_matches)\n",
    "df_mapped.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For comparisson sake, here is how these columns looked like in the original table:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Race</th>\n",
       "      <th>MSI_status</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>tumor_Stage-Pathological</th>\n",
       "      <th>Ethnicity</th>\n",
       "      <th>Tumor_Focality</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Num_full_term_pregnancies</th>\n",
       "      <th>POLE_subtype</th>\n",
       "      <th>MLH1_Promoter_Hypermethylation</th>\n",
       "      <th>...</th>\n",
       "      <th>Diabetes</th>\n",
       "      <th>MLH1</th>\n",
       "      <th>Path_Stage_Reg_Lymph_Nodes-pN</th>\n",
       "      <th>Proteomics_TMT_channel</th>\n",
       "      <th>Proteomics_Tumor_Normal</th>\n",
       "      <th>PMS2</th>\n",
       "      <th>Proteomics_Aliquot_ID</th>\n",
       "      <th>Histologic_type</th>\n",
       "      <th>Proteomics_Participant_ID</th>\n",
       "      <th>CNV_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>White</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pN0</td>\n",
       "      <td>128N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0001460012</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>C3L-00006</td>\n",
       "      <td>CNV_LOW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>White</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage IV</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>130N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>CPT0001300009</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>C3L-00008</td>\n",
       "      <td>CNV_LOW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pN0</td>\n",
       "      <td>131</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0001420009</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>C3L-00032</td>\n",
       "      <td>CNV_LOW</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>129N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CPT0000820012</td>\n",
       "      <td>Carcinosarcoma</td>\n",
       "      <td>C3L-00084</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>129C</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0001140003</td>\n",
       "      <td>Endometrioid</td>\n",
       "      <td>C3L-00090</td>\n",
       "      <td>CNV_LOW</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Race MSI_status Case_excluded tumor_Stage-Pathological  \\\n",
       "0  White      MSI-H            No                  Stage I   \n",
       "1  White      MSI-H            No                 Stage IV   \n",
       "2  White        MSS            No                  Stage I   \n",
       "3    NaN        MSS           Yes                      NaN   \n",
       "4  White        MSS            No                  Stage I   \n",
       "\n",
       "                Ethnicity Tumor_Focality  Gender Num_full_term_pregnancies  \\\n",
       "0  Not-Hispanic or Latino       Unifocal  Female                         1   \n",
       "1  Not-Hispanic or Latino       Unifocal  Female                         1   \n",
       "2  Not-Hispanic or Latino       Unifocal  Female                 4 or more   \n",
       "3                     NaN            NaN     NaN                       NaN   \n",
       "4  Not-Hispanic or Latino       Unifocal  Female                 4 or more   \n",
       "\n",
       "  POLE_subtype MLH1_Promoter_Hypermethylation  ... Diabetes  \\\n",
       "0           No           Cannot be determined  ...       No   \n",
       "1           No           Cannot be determined  ...       No   \n",
       "2           No           Cannot be determined  ...      Yes   \n",
       "3           No                            NaN  ...      NaN   \n",
       "4           No           Cannot be determined  ...       No   \n",
       "\n",
       "                        MLH1 Path_Stage_Reg_Lymph_Nodes-pN  \\\n",
       "0  Intact nuclear expression                           pN0   \n",
       "1  Intact nuclear expression                           pNX   \n",
       "2  Intact nuclear expression                           pN0   \n",
       "3                        NaN                           NaN   \n",
       "4  Intact nuclear expression                           pNX   \n",
       "\n",
       "  Proteomics_TMT_channel Proteomics_Tumor_Normal                        PMS2  \\\n",
       "0                   128N                   Tumor   Intact nuclear expression   \n",
       "1                   130N                   Tumor  Loss of nuclear expression   \n",
       "2                    131                   Tumor   Intact nuclear expression   \n",
       "3                   129N                   Tumor                         NaN   \n",
       "4                   129C                   Tumor   Intact nuclear expression   \n",
       "\n",
       "  Proteomics_Aliquot_ID Histologic_type Proteomics_Participant_ID CNV_class  \n",
       "0         CPT0001460012    Endometrioid                 C3L-00006   CNV_LOW  \n",
       "1         CPT0001300009    Endometrioid                 C3L-00008   CNV_LOW  \n",
       "2         CPT0001420009    Endometrioid                 C3L-00032   CNV_LOW  \n",
       "3         CPT0000820012  Carcinosarcoma                 C3L-00084       NaN  \n",
       "4         CPT0001140003    Endometrioid                 C3L-00090   CNV_LOW  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "source_column_names = value_matches['source_attribute'].unique().tolist()\n",
    "target_column_names = value_matches['target_attribute'].unique().tolist()\n",
    "df_source[source_column_names].head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The final step is to merge the two tables in a single harmonized dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Race</th>\n",
       "      <th>MSI_status</th>\n",
       "      <th>Case_excluded</th>\n",
       "      <th>Tumor_stage_pathological</th>\n",
       "      <th>Ethnicity</th>\n",
       "      <th>Tumor_focality</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Donor_information_number_of_full_term_pregnancies</th>\n",
       "      <th>POLE</th>\n",
       "      <th>Ancillary_studies_mlh1_promoter_hypermethylation</th>\n",
       "      <th>...</th>\n",
       "      <th>Diabetes</th>\n",
       "      <th>Ancillary_studies_msh2</th>\n",
       "      <th>Pathologic_staging_regional_lymph_nodes_pn</th>\n",
       "      <th>ReporterName</th>\n",
       "      <th>Group</th>\n",
       "      <th>Ancillary_studies_pms2</th>\n",
       "      <th>Aliquot_ID</th>\n",
       "      <th>Histologic_Type</th>\n",
       "      <th>Idx</th>\n",
       "      <th>CNV_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>White</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pN0</td>\n",
       "      <td>128N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0126000003</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>White</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage IV</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>130N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>CPT0095130004</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pN0</td>\n",
       "      <td>131N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0313420004</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>MSS</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>129N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CPT0182020004</td>\n",
       "      <td>Serous carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>129C</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0114340003</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>129N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0127980003</td>\n",
       "      <td>Serous carcinoma</td>\n",
       "      <td>C3L-00086</td>\n",
       "      <td>CNV_H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>3</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>129C</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0117300003</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-01355</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>130N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0100100003</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-01355</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>White</td>\n",
       "      <td>MSS</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>4 or more</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>pNX</td>\n",
       "      <td>130N</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Intact nuclear expression</td>\n",
       "      <td>CPT0113850003</td>\n",
       "      <td>Serous carcinoma</td>\n",
       "      <td>C3L-01355</td>\n",
       "      <td>CNV_H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>White</td>\n",
       "      <td>MSI-H</td>\n",
       "      <td>No</td>\n",
       "      <td>Stage I</td>\n",
       "      <td>Not-Hispanic or Latino</td>\n",
       "      <td>Unifocal</td>\n",
       "      <td>Female</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>Cannot be determined</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>pN0</td>\n",
       "      <td>130C</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>Loss of nuclear expression</td>\n",
       "      <td>CPT0091910003</td>\n",
       "      <td>Endometrioid carcinoma</td>\n",
       "      <td>C3L-03143</td>\n",
       "      <td>CNV_L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Race MSI_status Case_excluded Tumor_stage_pathological  \\\n",
       "0  White      MSI-H            No                  Stage I   \n",
       "1  White      MSI-H            No                 Stage IV   \n",
       "2  White        MSS            No                  Stage I   \n",
       "3    NaN        MSS           Yes                      NaN   \n",
       "4  White        MSS            No                  Stage I   \n",
       "5  White        MSS            No                  Stage I   \n",
       "6  White        MSS            No                  Stage I   \n",
       "7  White        MSS            No                  Stage I   \n",
       "8  White        MSS            No                Stage III   \n",
       "9  White      MSI-H            No                  Stage I   \n",
       "\n",
       "                Ethnicity Tumor_focality     Sex  \\\n",
       "0  Not-Hispanic or Latino       Unifocal  Female   \n",
       "1  Not-Hispanic or Latino       Unifocal  Female   \n",
       "2  Not-Hispanic or Latino       Unifocal  Female   \n",
       "3                     NaN            NaN     NaN   \n",
       "4  Not-Hispanic or Latino       Unifocal  Female   \n",
       "5  Not-Hispanic or Latino       Unifocal  Female   \n",
       "6  Not-Hispanic or Latino       Unifocal  Female   \n",
       "7  Not-Hispanic or Latino       Unifocal  Female   \n",
       "8  Not-Hispanic or Latino       Unifocal  Female   \n",
       "9  Not-Hispanic or Latino       Unifocal  Female   \n",
       "\n",
       "  Donor_information_number_of_full_term_pregnancies POLE  \\\n",
       "0                                                 1   No   \n",
       "1                                                 1   No   \n",
       "2                                         4 or more   No   \n",
       "3                                               NaN   No   \n",
       "4                                         4 or more   No   \n",
       "5                                                 2   No   \n",
       "6                                                 3   No   \n",
       "7                                                 2   No   \n",
       "8                                         4 or more   No   \n",
       "9                                                 2   No   \n",
       "\n",
       "  Ancillary_studies_mlh1_promoter_hypermethylation  ... Diabetes  \\\n",
       "0                             Cannot be determined  ...      NaN   \n",
       "1                             Cannot be determined  ...      NaN   \n",
       "2                             Cannot be determined  ...      Yes   \n",
       "3                                              NaN  ...      NaN   \n",
       "4                             Cannot be determined  ...      NaN   \n",
       "5                             Cannot be determined  ...      NaN   \n",
       "6                             Cannot be determined  ...      NaN   \n",
       "7                             Cannot be determined  ...      NaN   \n",
       "8                             Cannot be determined  ...      NaN   \n",
       "9                             Cannot be determined  ...      Yes   \n",
       "\n",
       "       Ancillary_studies_msh2 Pathologic_staging_regional_lymph_nodes_pn  \\\n",
       "0   Intact nuclear expression                                        pN0   \n",
       "1   Intact nuclear expression                                        pNX   \n",
       "2   Intact nuclear expression                                        pN0   \n",
       "3                         NaN                                        NaN   \n",
       "4   Intact nuclear expression                                        pNX   \n",
       "5   Intact nuclear expression                                        pNX   \n",
       "6   Intact nuclear expression                                        pNX   \n",
       "7   Intact nuclear expression                                        pNX   \n",
       "8   Intact nuclear expression                                        pNX   \n",
       "9  Loss of nuclear expression                                        pN0   \n",
       "\n",
       "  ReporterName  Group      Ancillary_studies_pms2     Aliquot_ID  \\\n",
       "0         128N  Tumor   Intact nuclear expression  CPT0126000003   \n",
       "1         130N  Tumor  Loss of nuclear expression  CPT0095130004   \n",
       "2         131N  Tumor   Intact nuclear expression  CPT0313420004   \n",
       "3         129N  Tumor                         NaN  CPT0182020004   \n",
       "4         129C  Tumor   Intact nuclear expression  CPT0114340003   \n",
       "5         129N  Tumor   Intact nuclear expression  CPT0127980003   \n",
       "6         129C  Tumor   Intact nuclear expression  CPT0117300003   \n",
       "7         130N  Tumor   Intact nuclear expression  CPT0100100003   \n",
       "8         130N  Tumor   Intact nuclear expression  CPT0113850003   \n",
       "9         130C  Tumor  Loss of nuclear expression  CPT0091910003   \n",
       "\n",
       "          Histologic_Type        Idx CNV_status  \n",
       "0  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "1  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "2  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "3        Serous carcinoma  C3L-00086        NaN  \n",
       "4  Endometrioid carcinoma  C3L-00086      CNV_L  \n",
       "5        Serous carcinoma  C3L-00086      CNV_H  \n",
       "6  Endometrioid carcinoma  C3L-01355      CNV_L  \n",
       "7  Endometrioid carcinoma  C3L-01355      CNV_L  \n",
       "8        Serous carcinoma  C3L-01355      CNV_H  \n",
       "9  Endometrioid carcinoma  C3L-03143      CNV_L  \n",
       "\n",
       "[10 rows x 29 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_table = pd.concat([df_mapped[target_column_names], df_target[target_column_names]])\n",
    "merged_table.head(10)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bdf",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
